{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Genome-wide assessment identifies novel runs of homozygosity linked to Parkinson’s disease etiology across diverse ancestral populations**\n",
    "\n",
    "---\n",
    "\n",
    "*   **Project:** Runs of Homozygosity (ROH) Analysis in Parkinson's Disease -- ROH Mapping and Functional Enrichment in Parkinson's Disease\n",
    "*   **Version:** Python/3.9\n",
    "*   **Authors**: Kathryn Step, Carlos F. Hernández, Esraa Eltaraifee, Ana Jimena Hernández-Medrano, Pin-Jui Kung, Miriam Ostrožovičová, Alexandra Zirra, Zih-Hua Fang and Sara Bandres-Ciga\n",
    "*   **Estimated Computation and Runtime**:\n",
    "   *   **Estimated Specifications:**  1 CPU, 3.75 GB Memory, 50 GB Persistent Disk Size\n",
    "   *   **Estimated Runtime:** 90 min\n",
    "*   **Status:** COMPLETE\n",
    "*   **Started:** 26-AUG-2023\n",
    "*   **Last Updated:** 24-NOV-2024\n",
    "   *   **Update Description:** Incorporated functional annotations for ROH regions and prioritized candidate genes\n",
    "\n",
    "---\n",
    "\n",
    "**Notebook Overview**  \n",
    "*   Fine-mapping ROH regions with overlapping alleles\n",
    "*   Identifying homozygous variants using VCF files\n",
    "*   Extracting and annotating functional impacts of variants\n",
    "*   Performing prioritization based on clinical and conservation scores\n",
    "\n",
    "\n",
    "---\n",
    "\n",
    "**Note:**\n",
    "This notebook presents the initial analysis for the American Admixed (AMR) ancestry group. The same analysis pipeline was repeated for the following additional ancestry groups:  \n",
    "- **African Admixed (AAC)**  \n",
    "- **African (AFR)**  \n",
    "- **Ashkenazi Jewish (AJ)**  \n",
    "- **American Admixed (AMR)**  \n",
    "- **Central Asian (CAS)**  \n",
    "- **East Asian (EAS)**  \n",
    "- **European (EUR)**  \n",
    "- **South Asian (SAS)**  \n",
    "\n",
    "While only the AMR results are shown here for clarity and to minimize redundancy, the methodology and scripts are identical across all groups. Please refer to the project documentation for a comprehensive overview.  \n",
    "\n",
    "---\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<h1>Table of Contents<span class=\"tocSkip\"></span></h1>\n",
    "<div class=\"toc\"><ul class=\"toc-item\">\n",
    "    <li><span><a href=\"#Getting-Started\" data-toc-modified-id=\"Getting-Started-1\"><span class=\"toc-item-num\">1&nbsp;&nbsp;</span>Getting Started</a></span></li>\n",
    "    <li><span><a href=\"#Create-Working-Directory\" data-toc-modified-id=\"Create-Working-Directory-2\"><span class=\"toc-item-num\">2&nbsp;&nbsp;</span>Create Working Directory</a></span></li>\n",
    "    <li><span><a href=\"#Homozygosity-mapping\" data-toc-modified-id=\"Homozygosity-mapping-3\"><span class=\"toc-item-num\">3&nbsp;&nbsp;</span>Homozygosity Mapping</a></span>\n",
    "        <ul class=\"toc-item\">\n",
    "            <li><span><a href=\"#Perform-QC-before-relationship-inference-with-KING\" data-toc-modified-id=\"Perform-QC-before-relationship-inference-with-KING-3.1\"><span class=\"toc-item-num\">3.1&nbsp;&nbsp;</span>Perform QC before relationship inference with KING</a></span></li>\n",
    "            <li><span><a href=\"#Update-Family-ID-and-prune-samples\" data-toc-modified-id=\"Update-Family-ID-and-prune-samples-3.2\"><span class=\"toc-item-num\">3.2&nbsp;&nbsp;</span>Update Family ID (FID) as IID and prune samples with missing phenotype</a></span></li>\n",
    "            <li><span><a href=\"#Relationship-inference-using-KING\" data-toc-modified-id=\"Relationship-inference-using-KING-3.3\"><span class=\"toc-item-num\">3.3&nbsp;&nbsp;</span>Relationship inference using KING</a></span></li>\n",
    "            <li><span><a href=\"#Update-Family-ID-based-on-relationship-inference\" data-toc-modified-id=\"Update-Family-ID-based-on-relationship-inference-3.4\"><span class=\"toc-item-num\">3.4&nbsp;&nbsp;</span>Update Family ID based on relationship inference</a></span></li>\n",
    "            <li><span><a href=\"#Check-for-inbred-individuals\" data-toc-modified-id=\"Check-for-inbred-individuals-3.5\"><span class=\"toc-item-num\">3.5&nbsp;&nbsp;</span>Check for inbred individuals</a></span>\n",
    "                <ul class=\"toc-item\">\n",
    "                    <li><span><a href=\"#Optional:-retain-individuals-with-F>0.0884\" data-toc-modified-id=\"Optional:-retain-individuals-with-F>0.0884-3.5.1\"><span class=\"toc-item-num\">3.5.1&nbsp;&nbsp;</span>Optional: retain individuals with F>0.0884 to check overlapped ROH by KING</a></span></li>\n",
    "                </ul>\n",
    "            </li>\n",
    "            <li><span><a href=\"#Find-ROH-overlaps-that-likely-segregate-within-families\" data-toc-modified-id=\"Find-ROH-overlaps-that-likely-segregate-within-families-3.6\"><span class=\"toc-item-num\">3.6&nbsp;&nbsp;</span>Find ROH overlaps that likely segregate within families</a></span>\n",
    "                <ul class=\"toc-item\">\n",
    "                    <li><span><a href=\"#Check-for-overlapping-allele-matched-ROH-within-families\" data-toc-modified-id=\"Check-for-overlapping-allele-matched-ROH-within-families-3.6.1\"><span class=\"toc-item-num\">3.6.1&nbsp;&nbsp;</span>Check for overlapping allele-matched ROH within families</a></span></li>\n",
    "                    <li><span><a href=\"#Fine-mapping\" data-toc-modified-id=\"Fine-mapping-3.6.2\"><span class=\"toc-item-num\">3.6.2&nbsp;&nbsp;</span>Fine mapping (if high penetrance variants were found)</a></span></li>\n",
    "                </ul>\n",
    "            </li>\n",
    "            <li><span><a href=\"#Find-ROH-overlaps-between-inbred-individuals\" data-toc-modified-id=\"Find-ROH-overlaps-between-inbred-individuals-3.7\"><span class=\"toc-item-num\">3.7&nbsp;&nbsp;</span>Find ROH overlaps between inbred individuals</a></span>\n",
    "                <ul class=\"toc-item\">\n",
    "                    <li><span><a href=\"#Subset-samples\" data-toc-modified-id=\"Subset-samples-3.7.1\"><span class=\"toc-item-num\">3.7.1&nbsp;&nbsp;</span>Subset samples</a></span></li>\n",
    "                    <li><span><a href=\"#Check-consensus-ROH\" data-toc-modified-id=\"Check-consensus-ROH-3.7.2\"><span class=\"toc-item-num\">3.7.2&nbsp;&nbsp;</span>Check consensus ROH (cROH)</a></span></li>\n",
    "                    <li><span><a href=\"#Find-ROH-enriched-in-inbred-cases\" data-toc-modified-id=\"Find-ROH-enriched-in-inbred-cases-3.7.3\"><span class=\"toc-item-num\">3.7.3&nbsp;&nbsp;</span>Find ROH enriched in inbred cases (test the association using a logistic model)</a></span>\n",
    "                        <ul class=\"toc-item\">\n",
    "                            <li><span><a href=\"#Consensus-ROH-filtered\" data-toc-modified-id=\"Consensus-ROH-filtered-3.7.3.1\"><span class=\"toc-item-num\">3.7.3.1&nbsp;&nbsp;</span>Consensus ROH filtered</a></span></li>\n",
    "                            <li><span><a href=\"#In-PD-genes\" data-toc-modified-id=\"In-PD-genes-3.7.3.2\"><span class=\"toc-item-num\">3.7.3.2&nbsp;&nbsp;</span>In PD genes</a></span></li>\n",
    "                        </ul>\n",
    "                    </li>\n",
    "                </ul>\n",
    "            </li>\n",
    "            <li><span><a href=\"#Find-overlaps-between-EOPD\" data-toc-modified-id=\"Find-overlaps-between-EOPD-3.8\"><span class=\"toc-item-num\">3.8&nbsp;&nbsp;</span>Find overlaps between EOPD</a></span>\n",
    "                <ul class=\"toc-item\">\n",
    "                    <li><span><a href=\"#Subset-EOPD-+-Controls\" data-toc-modified-id=\"Subset-EOPD-+-Controls-3.8.1\"><span class=\"toc-item-num\">3.8.1&nbsp;&nbsp;</span>Subset EOPD + Controls</a></span></li>\n",
    "                    <li><span><a href=\"#Check-consensus-ROH\" data-toc-modified-id=\"Check-consensus-ROH-3.8.2\"><span class=\"toc-item-num\">3.8.2&nbsp;&nbsp;</span>Check consensus ROH (cROH)</a></span></li>\n",
    "                    <li><span><a href=\"#Find-ROH-enriched-in-inbred-cases\" data-toc-modified-id=\"Find-ROH-enriched-in-inbred-cases-3.8.3\"><span class=\"toc-item-num\">3.8.3&nbsp;&nbsp;</span>Find ROH enriched in inbred cases (test the association using logistic model)</a></span>\n",
    "                        <ul class=\"toc-item\">\n",
    "                            <li><span><a href=\"#Consensus-ROH-filtered\" data-toc-modified-id=\"Consensus-ROH-filtered-3.8.3.1\"><span class=\"toc-item-num\">3.8.3.1&nbsp;&nbsp;</span>Consensus ROH filtered</a></span></li>\n",
    "                            <li><span><a href=\"#In-PD-genes\" data-toc-modified-id=\"In-PD-genes-3.8.3.2\"><span class=\"toc-item-num\">3.8.3.2&nbsp;&nbsp;</span>In PD genes</a></span></li>\n",
    "                        </ul>\n",
    "                    </li>\n",
    "                </ul>\n",
    "            </li>\n",
    "        </ul>\n",
    "    </li>\n",
    "    <li><span><a href=\"#Fine-mapping-loop\" data-toc-modified-id=\"Fine-mapping-loop-4\"><span class=\"toc-item-num\">4&nbsp;&nbsp;</span>Fine mapping loop (all)</a></span>\n",
    "        <ul class=\"toc-item\">\n",
    "            <li><span><a href=\"#Download-pools\" data-toc-modified-id=\"Download-pools-4.1\"><span class=\"toc-item-num\">4.1&nbsp;&nbsp;</span>Download pools</a></span></li>\n",
    "            <li><span><a href=\"#Fine-mapping-in-shared-alleles-ROH\" data-toc-modified-id=\"Fine-mapping-in-shared-alleles-ROH-4.2\"><span class=\"toc-item-num\">4.2&nbsp;&nbsp;</span>Fine mapping in shared alleles ROH (subset ROH regions shared)</a></span>\n",
    "                <ul class=\"toc-item\">\n",
    "                     <li><span><a href=\"#Select-individuals-that-share-alleles\" data-toc-modified-id=\"Select-individuals-that-share-alleles-4.2.1\"><span class=\"toc-item-num\">4.2.1&nbsp;&nbsp;</span>Select individuals that share alleles</a></span></li>\n",
    "                    <li><span><a href=\"#Select-homozygous-variants-and-extract-CSQ-from-the-VCF-file\" data-toc-modified-id=\"Select-homozygous-variants-and-extract-CSQ-from-the-VCF-file-4.2.2\"><span class=\"toc-item-num\">4.2.2&nbsp;&nbsp;</span>Select homozygous variants and extract CSQ from the VCF file</a></span></li>\n",
    "                       <li><span><a href=\"#Prioritization-criteria\" data-toc-modified-id=\"Prioritization-criteria-4.2.2.1\"><span class=\"toc-item-num\">4.2.2.1&nbsp;&nbsp;</span>Prioritization criteria</a></span></li>\n",
    "                </ul>\n",
    "            </li>\n",
    "            <li><span><a href=\"#Select-homozygous-variants-and-extract-CSQ-from-the-VCF-file-with-unique-samples\" data-toc-modified-id=\"Select-homozygous-variants-and-extract-CSQ-from-the-VCF-file-with-unique-samples-4.2.3\"><span class=\"toc-item-num\">4.2.3&nbsp;&nbsp;</span>Select homozygous variants and extract CSQ from the VCF file with unique samples</a></span>\n",
    "                <ul class=\"toc-item\">\n",
    "                       <li><span><a href=\"#Prioritization-criteria-unique-samples\" data-toc-modified-id=\"Prioritization-criteria-unique-samples-4.2.3.1\"><span class=\"toc-item-num\">4.2.3.1&nbsp;&nbsp;</span>Prioritization criteria</a></span></li>\n",
    "                </ul>\n",
    "            </li>\n",
    "        </ul>\n",
    "    </li>\n",
    "</ul></div>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Getting Started"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "\u001b[1m[\u001b[0m\u001b[34;49mnotice\u001b[0m\u001b[1;39;49m]\u001b[0m\u001b[39;49m A new release of pip is available: \u001b[0m\u001b[31;49m24.0\u001b[0m\u001b[39;49m -> \u001b[0m\u001b[32;49m24.3.1\u001b[0m\r\n",
      "\u001b[1m[\u001b[0m\u001b[34;49mnotice\u001b[0m\u001b[1;39;49m]\u001b[0m\u001b[39;49m To update, run: \u001b[0m\u001b[32;49mpip install --upgrade pip\u001b[0m\r\n"
     ]
    }
   ],
   "source": [
    "# Install necessary Python packages\n",
    "!pip install openpyxl --quiet\n",
    "\n",
    "# Import required libraries\n",
    "from pathlib import Path\n",
    "import seaborn as sns\n",
    "import openpyxl\n",
    "import glob\n",
    "\n",
    "# Use the os package to interact with the environment\n",
    "import os\n",
    "\n",
    "# Bring in Pandas for Dataframe functionality\n",
    "import pandas as pd\n",
    "\n",
    "# Numpy for basics\n",
    "import numpy as np\n",
    "\n",
    "# Use StringIO for working with file contents\n",
    "from io import StringIO\n",
    "\n",
    "# Enable IPython to display matplotlib graphs\n",
    "import matplotlib.pyplot as plt\n",
    "%matplotlib inline\n",
    "\n",
    "# Enable interaction with the FireCloud API\n",
    "from firecloud import api as fapi\n",
    "\n",
    "# Import the iPython HTML rendering for displaying links to Google Cloud Console\n",
    "from IPython.core.display import display, HTML\n",
    "\n",
    "# Import urllib modules for building URLs to Google Cloud Console\n",
    "import urllib.parse\n",
    "\n",
    "# BigQuery for querying data\n",
    "from google.cloud import bigquery\n",
    "\n",
    "#Import Sys\n",
    "import sys as sys"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# Set up R support with rpy2\n",
    "!pip install --upgrade rpy2 --quiet\n",
    "%reload_ext rpy2.ipython"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"R magic cell is working\"\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "In addition: Warning message:\n",
       "In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
       "  libraries ‘/home/jupyter/packages’, ‘/usr/lib/R/site-library’ contain no packages\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%R\n",
    "\n",
    "# Verify that R magic is working\n",
    "print('R magic cell is working')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Set up billing project and data path variables\n",
    "BILLING_PROJECT_ID = os.environ['GOOGLE_PROJECT']\n",
    "WORKSPACE_NAMESPACE = os.environ['WORKSPACE_NAMESPACE']\n",
    "WORKSPACE_NAME = os.environ['WORKSPACE_NAME']\n",
    "WORKSPACE_BUCKET = os.environ['WORKSPACE_BUCKET']\n",
    "\n",
    "WORKSPACE_ATTRIBUTES = fapi.get_workspace(WORKSPACE_NAMESPACE, WORKSPACE_NAME).json().get('workspace',{}).get('attributes',{})\n",
    "\n",
    "## Print the information to check we are in the proper release and billing \n",
    "## This will be different for you, the user, depending on the billing project your workspace is on\n",
    "print('Billing and Workspace')\n",
    "print(f'Workspace Name: {WORKSPACE_NAME}')\n",
    "print(f'Billing Project: {BILLING_PROJECT_ID}')\n",
    "print(f'Workspace Bucket, where you can upload and download data: {WORKSPACE_BUCKET}')\n",
    "print('')\n",
    "\n",
    "\n",
    "## GP2 v6.0\n",
    "## Explicitly define release v6.0 path \n",
    "GP2_RELEASE_PATH = 'gs:/path/to/data'  \n",
    "GP2_CLINICAL_RELEASE_PATH = f'{GP2_RELEASE_PATH}/clinical_data'\n",
    "GP2_RAW_GENO_PATH = f'{GP2_RELEASE_PATH}/raw_genotypes'\n",
    "GP2_IMPUTED_GENO_PATH = f'{GP2_RELEASE_PATH}/imputed_genotypes'\n",
    "print('GP2 v6.0')\n",
    "print(f'Path to GP2 v6.0 Clinical Data: {GP2_CLINICAL_RELEASE_PATH}')\n",
    "print(f'Path to GP2 v6.0 Raw Genotype Data: {GP2_RAW_GENO_PATH}')\n",
    "print(f'Path to GP2 v6.0 Imputed Genotype Data: {GP2_IMPUTED_GENO_PATH}')\n",
    "\n",
    "#borra esto"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Utility routine for printing a shell command before executing it\n",
    "def shell_do(command):\n",
    "    print(f'Executing: {command}', file=sys.stderr)\n",
    "    !$command\n",
    "    \n",
    "# Utility routine for executing a shell command and returning its output\n",
    "def shell_return(command):\n",
    "    print(f'Executing: {command}', file=sys.stderr)\n",
    "    output = !$command\n",
    "    return '\\n'.join(output)\n",
    "\n",
    "# Utility routine for executing a BigQuery query\n",
    "def bq_query(query):\n",
    "    print(f'Executing: {query}', file=sys.stderr)\n",
    "    return pd.read_gbq(query, project_id=BILLING_PROJECT_ID, dialect='standard')\n",
    "\n",
    "# Utility routine for displaying a message and a link\n",
    "def display_html_link(description, link_text, url):\n",
    "    html = f'''\n",
    "    <p>\n",
    "    </p>\n",
    "    <p>\n",
    "    {description}\n",
    "    <a target=_blank href=\"{url}\">{link_text}</a>.\n",
    "    </p>\n",
    "    '''\n",
    "    display(HTML(html))\n",
    "\n",
    "# Utility routines for reading files from Google Cloud Storage\n",
    "def gcs_read_file(path):\n",
    "    \"\"\"Return the contents of a file in GCS\"\"\"\n",
    "    contents = !gsutil -u {BILLING_PROJECT_ID} cat {path}\n",
    "    return '\\n'.join(contents)\n",
    "\n",
    "def gcs_read_csv(path, sep=None):\n",
    "    \"\"\"Return a DataFrame from the contents of a delimited file in GCS\"\"\"\n",
    "    return pd.read_csv(StringIO(gcs_read_file(path)), sep=sep, engine='python')\n",
    "\n",
    "# Utility routine for displaying a message and link to Cloud Console\n",
    "def link_to_cloud_console_gcs(description, link_text, gcs_path):\n",
    "    url = '{}?{}'.format(\n",
    "        os.path.join('https://console.cloud.google.com/storage/browser',\n",
    "                     gcs_path.replace(\"gs://\",\"\")),\n",
    "        urllib.parse.urlencode({'userProject': BILLING_PROJECT_ID}))\n",
    "    display_html_link(description, link_text, url)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Plink1.9 is already installed in /home/jupyter/tools/\n",
      "Plink2 is already installed in /home/jupyter/tools/\n",
      "KING is already installed in /home/jupyter/tools/\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "# Set up tools (PLINK, King, and ANNOVAR)\n",
    "mkdir -p ~/tools\n",
    "cd ~/tools\n",
    "\n",
    "# Check and install PLINK 1.9\n",
    "if test -e /home/jupyter/tools/plink; then\n",
    "   echo \"Plink1.9 is already installed in /home/jupyter/tools/\"\n",
    "else\n",
    "   echo -e \"Downloading plink \\n    -------\"\n",
    "   wget -N http://s3.amazonaws.com/plink1-assets/plink_linux_x86_64_20190304.zip \n",
    "   unzip -o plink_linux_x86_64_20190304.zip\n",
    "   echo -e \"\\n plink downloaded and unzipped in /home/jupyter/tools \\n \"\n",
    "fi\n",
    "\n",
    "# Check and install PLINK 2.0 with a corrected download link\n",
    "if test -e /home/jupyter/tools/plink2; then\n",
    "   echo \"Plink2 is already installed in /home/jupyter/tools/\"\n",
    "else\n",
    "   echo -e \"Downloading plink2 \\n    -------\"\n",
    "   wget -N https://s3.amazonaws.com/plink2-assets/alpha3/plink2_linux_avx2_20240806.zip\n",
    "   unzip -o plink2_linux_avx2_20240806.zip -d /home/jupyter/tools/\n",
    "   mv /home/jupyter/tools/plink2_linux_avx2_20240806 /home/jupyter/tools/plink2\n",
    "   echo -e \"\\nPlink2 downloaded, unzipped, and saved as 'plink2' in /home/jupyter/tools \\n\"\n",
    "fi\n",
    "\n",
    "# Install King 2.3.0\n",
    "if test -e /home/jupyter/tools/king; then\n",
    "   echo \"KING is already installed in /home/jupyter/tools/\"\n",
    "else\n",
    "   echo -e \"Downloading KING \\n    -------\"\n",
    "   wget -N https://www.kingrelatedness.com/executables/Linux-king230.tar.gz\n",
    "   tar -xzvf Linux-king230.tar.gz\n",
    "   echo -e \"\\n KING downloaded and unzipped in /home/jupyter/tools \\n \"\n",
    "fi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "king\n",
      "LICENSE\n",
      "Linux-king230.tar.gz\n",
      "plink\n",
      "plink2\n",
      "plink2_linux_avx2_20240806.zip\n",
      "plink_linux_x86_64_20190304.zip\n",
      "prettify\n",
      "toy.map\n",
      "toy.ped\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "# List contents of the tools directory\n",
    "ls /home/jupyter/tools/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "# Make the downloaded tools executable\n",
    "chmod u+x /home/jupyter/tools/plink\n",
    "chmod u+x /home/jupyter/tools/plink2\n",
    "chmod u+x /home/jupyter/tools/king"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "data.table 1.14.8 using 48 threads (see ?getDTthreads).  Latest news: r-datatable.com\n",
       "\n",
       "Attaching package: ‘dplyr’\n",
       "\n",
       "The following objects are masked from ‘package:plyr’:\n",
       "\n",
       "    arrange, count, desc, failwith, id, mutate, rename, summarise,\n",
       "    summarize\n",
       "\n",
       "The following objects are masked from ‘package:data.table’:\n",
       "\n",
       "    between, first, last\n",
       "\n",
       "The following objects are masked from ‘package:stats’:\n",
       "\n",
       "    filter, lag\n",
       "\n",
       "The following objects are masked from ‘package:base’:\n",
       "\n",
       "    intersect, setdiff, setequal, union\n",
       "\n",
       "\n",
       "Attaching package: ‘tidyr’\n",
       "\n",
       "The following object is masked from ‘package:magrittr’:\n",
       "\n",
       "    extract\n",
       "\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%R\n",
    "\n",
    "# Load necessary R packages\n",
    "library(data.table)\n",
    "library(magrittr)\n",
    "library(stringr)\n",
    "library(ggplot2)\n",
    "library(plyr)\n",
    "library(dplyr)\n",
    "library(tidyr)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Create Working Directory"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Making a working directory\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: mkdir -p /home/jupyter/Release6_AMR\n"
     ]
    }
   ],
   "source": [
    "# Make a directory for use in python\n",
    "print(\"Making a working directory\")\n",
    "ANC = 'AMR'\n",
    "WORK_DIR = f'/home/jupyter/Release6_{ANC}'\n",
    "shell_do(f'mkdir -p {WORK_DIR}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "%%R\n",
    "\n",
    "# Set directory for use in R\n",
    "ANC <- 'AMR'\n",
    "setwd(paste0(\"/home/jupyter/Release6_\",ANC,\"/\") )"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Homozygosity mapping"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Perform QC before relationship inference with KING\n",
    "\n",
    "Quality control is crucial for accurate relationship inference. We recommend filtering high-quality markers with `--geno 0.05` and `--hwe 1E-10`. Note that LD pruning is not recommended for KING.\n",
    "\n",
    "KING accepts only PLINK bfiles."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: /home/jupyter/tools/plink2         --pfile /home/jupyter/Release6_AMR/AMR_release6          --geno 0.05 --hwe 1E-10 --make-bed         --out /home/jupyter/Release6_AMR/gp2_AMR_r6_qced\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/Release6_AMR/gp2_AMR_r6_qced.log.\n",
      "Options in effect:\n",
      "  --geno 0.05\n",
      "  --hwe 1E-10\n",
      "  --make-bed\n",
      "  --out /home/jupyter/Release6_AMR/gp2_AMR_r6_qced\n",
      "  --pfile /home/jupyter/Release6_AMR/AMR_release6\n",
      "\n",
      "Start time: Tue Nov 26 05:22:34 2024\n",
      "628867 MiB RAM detected, ~622031 available; reserving 314433 MiB for main\n",
      "workspace.\n",
      "Using up to 96 threads (change this with --threads).\n",
      "529 samples (241 females, 288 males; 529 founders) loaded from\n",
      "/home/jupyter/Release6_AMR/AMR_release6.psam.\n",
      "1185307 variants loaded from /home/jupyter/Release6_AMR/AMR_release6.pvar.\n",
      "1 binary phenotype loaded (367 cases, 139 controls).\n",
      "Calculating allele frequencies... 1116222733384449556066717782889399done.\n",
      "--geno: 377 variants removed due to missing genotype data.\n",
      "Computing chrX Hardy-Weinberg p-values... 10101111121213131414151516161717181819192020212122222323242425252626272728282929303031313232333334343535363637373838393940404141424243434444454546464747484849505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "--hwe: 0 variants removed due to Hardy-Weinberg exact test (founders only).\n",
      "1184930 variants remaining after main filters.\n",
      "Writing /home/jupyter/Release6_AMR/gp2_AMR_r6_qced.fam ... done.\n",
      "Writing /home/jupyter/Release6_AMR/gp2_AMR_r6_qced.bim ... done.\n",
      "Writing /home/jupyter/Release6_AMR/gp2_AMR_r6_qced.bed ... 1116222733384449556066717782889399done.\n",
      "End time: Tue Nov 26 05:22:36 2024\n"
     ]
    }
   ],
   "source": [
    "# Run plink2 for quality control of genetic data\n",
    "shell_do(f'/home/jupyter/tools/plink2 \\\n",
    "        --pfile {WORK_DIR}/{ANC}_release6  \\\n",
    "        --geno 0.05 --hwe 1E-10 --make-bed \\\n",
    "        --out {WORK_DIR}/gp2_{ANC}_r6_qced')   "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Update Family ID (FID) as IID and prune samples with missing phenotype\n",
    "\n",
    "GP2 PLINK files have all Family IDs set to 0. This causes all samples to be considered one family. Update FID to match IID and remove samples with missing phenotype."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: cat /home/jupyter/Release6_AMR/gp2_AMR_r6_qced.fam | awk '{print $1, $2, $2, $2}' > /home/jupyter/Release6_AMR/update_ids.txt\n",
      "Executing: /home/jupyter/tools/plink --bfile /home/jupyter/Release6_AMR/gp2_AMR_r6_qced --update-ids /home/jupyter/Release6_AMR/update_ids.txt --prune --make-bed --out /home/jupyter/Release6_AMR/gp2_AMR_r6_qced_updated_id\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/Release6_AMR/gp2_AMR_r6_qced_updated_id.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/Release6_AMR/gp2_AMR_r6_qced\n",
      "  --make-bed\n",
      "  --out /home/jupyter/Release6_AMR/gp2_AMR_r6_qced_updated_id\n",
      "  --prune\n",
      "  --update-ids /home/jupyter/Release6_AMR/update_ids.txt\n",
      "\n",
      "628867 MB RAM detected; reserving 314433 MB for main workspace.\n",
      "1184930 variants loaded from .bim file.\n",
      "529 people (288 males, 241 females) loaded from .fam.\n",
      "506 phenotype values loaded from .fam.\n",
      "--update-ids: 529 people updated.\n",
      "--prune: 506 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 506 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Warning: 1843 het. haploid genotypes present (see\n",
      "/home/jupyter/Release6_AMR/gp2_AMR_r6_qced_updated_id.hh ); many commands treat\n",
      "these as missing.\n",
      "Warning: Nonmissing nonmale Y chromosome genotype(s) present; many commands\n",
      "treat these as missing.\n",
      "Total genotyping rate in remaining samples is 0.997556.\n",
      "1184930 variants and 506 people pass filters and QC.\n",
      "Among remaining phenotypes, 367 are cases and 139 are controls.\n",
      "--make-bed to /home/jupyter/Release6_AMR/gp2_AMR_r6_qced_updated_id.bed +\n",
      "/home/jupyter/Release6_AMR/gp2_AMR_r6_qced_updated_id.bim +\n",
      "/home/jupyter/Release6_AMR/gp2_AMR_r6_qced_updated_id.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    }
   ],
   "source": [
    "# Update IDs in the .fam file and create update_ids.txt\n",
    "shell_do(f\"cat {WORK_DIR}/gp2_{ANC}_r6_qced.fam | awk '{{print $1, $2, $2, $2}}' > {WORK_DIR}/update_ids.txt\")\n",
    "\n",
    "# Run plink to update IDs, prune, and output updated files\n",
    "shell_do(f'/home/jupyter/tools/plink --bfile {WORK_DIR}/gp2_{ANC}_r6_qced '\n",
    "         f'--update-ids {WORK_DIR}/update_ids.txt '\n",
    "         f'--prune '\n",
    "         f'--make-bed --out {WORK_DIR}/gp2_{ANC}_r6_qced_updated_id')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##  Relationship inference using KING\n",
    "\n",
    "Use KING to infer relationships between individuals in the dataset. This step helps to accurately define family structures and identify related individuals."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: /home/jupyter/tools/king -b /home/jupyter/Release6_AMR/gp2_AMR_r6_qced_updated_id.bed --build --related --degree 2 --prefix /home/jupyter/Release6_AMR/gp2_AMR\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "KING 2.3.0 - (c) 2010-2022 Wei-Min Chen\n",
      "\n",
      "The following parameters are in effect:\n",
      "                   Binary File : /home/jupyter/Release6_AMR/gp2_AMR_r6_qced_updated_id.bed (-bname)\n",
      "\n",
      "Additional Options\n",
      "         Close Relative Inference : --related [ON], --duplicate\n",
      "   Pairwise Relatedness Inference : --kinship, --ibdseg, --ibs, --makeGRM\n",
      "              Inference Parameter : --degree [2], --seglength\n",
      "         Relationship Application : --unrelated, --cluster, --build [ON]\n",
      "                        QC Report : --bysample, --bySNP, --roh, --autoQC\n",
      "                     QC Parameter : --callrateN, --callrateM\n",
      "             Population Structure : --pca, --mds\n",
      "              Structure Parameter : --projection, --pcs\n",
      "              Disease Association : --tdt\n",
      "   Quantitative Trait Association : --lmm\n",
      "                Association Model : --trait [], --covariate []\n",
      "            Association Parameter : --invnorm\n",
      "               Genetic Risk Score : --risk, --model [], --prevalence, --noflip\n",
      "              Computing Parameter : --cpus\n",
      "                   Optional Input : --fam [], --bim [], --sexchr [23]\n",
      "                           Output : --rplot, --pngplot, --plink\n",
      "                 Output Parameter :\n",
      "                                    --prefix [/home/jupyter/Release6_AMR/gp2_AMR],\n",
      "                                    --rpath []\n",
      "\n",
      "The following analyses will run separately:  --related --build\n",
      "\n",
      "KING starts at Tue Nov 26 05:22:41 2024\n",
      "Loading genotype data in PLINK binary format...\n",
      "Read in PLINK fam file /home/jupyter/Release6_AMR/gp2_AMR_r6_qced_updated_id.fam...\n",
      "  PLINK pedigrees loaded: 506 samples\n",
      "Read in PLINK bim file /home/jupyter/Release6_AMR/gp2_AMR_r6_qced_updated_id.bim...\n",
      "  Genotype data consist of 1141927 autosome SNPs (including 4345 XY SNPs), 38259 X-chromosome SNPs, 2241 Y-chromosome SNPs, 478 mitochondrial SNPs\n",
      "  2025 other SNPs are removed.\n",
      "  PLINK maps loaded: 1182905 SNPs\n",
      "Read in PLINK bed file /home/jupyter/Release6_AMR/gp2_AMR_r6_qced_updated_id.bed...\n",
      "  PLINK binary genotypes loaded.\n",
      "  KING format genotype data successfully converted.\n",
      "Autosome genotypes stored in 17843 words for each of 506 individuals.\n",
      "\n",
      "Options in effect:\n",
      "\t--related\n",
      "\t--degree 2\n",
      "\t--prefix /home/jupyter/Release6_AMR/gp2_AMR\n",
      "\n",
      "Sorting autosomes...\n",
      "Total length of 39 chromosomal segments usable for IBD segment analysis is 2690.0 Mb.\n",
      "  In addition to autosomes, 2 segments of length 148.9 Mb on X-chr can be further used.\n",
      "  Information of these chromosomal segments can be found in file /home/jupyter/Release6_AMR/gp2_AMRallsegs.txt\n",
      "\n",
      "Each family consists of one individual.\n",
      "A subset of informative SNPs will be used to screen close relatives.\n",
      "Relationship inference across families starts at Tue Nov 26 05:22:45 2024\n",
      "48 CPU cores are used...\n",
      "  Stages 1&2 (with 32768 SNPs): 5 pairs of relatives are detected (with kinship > 0.0625)\n",
      "                               Screening ends at Tue Nov 26 05:22:45 2024\n",
      "  Final Stage (with 1141927 SNPs): 5 pairs of relatives (up to 2nd-degree) are confirmed\n",
      "                               Inference ends at Tue Nov 26 05:22:45 2024\n",
      "\n",
      "Relationship summary (total relatives: 0 by pedigree, 5 by inference)\n",
      "        \tMZ\tPO\tFS\t2nd\t3rd\t4th\n",
      "  =========================================================\n",
      "  Inference\t0\t1\t4\t0\t0\t0\n",
      "\n",
      "\n",
      "Between-family relatives (kinship >= 0.08839) saved in file /home/jupyter/Release6_AMR/gp2_AMR.kin0\n",
      "  X-Chr IBD-sharing inference saved in file /home/jupyter/Release6_AMR/gp2_AMRX.kin0\n",
      "\n",
      "Options in effect:\n",
      "\t--build\n",
      "\t--degree 2\n",
      "\t--prefix /home/jupyter/Release6_AMR/gp2_AMR\n",
      "\n",
      "Family clustering starts at Tue Nov 26 05:22:45 2024\n",
      "Autosome genotypes stored in 17843 words for each of 506 individuals.\n",
      "Sorting autosomes...\n",
      "Total length of 39 chromosomal segments usable for IBD segment analysis is 2690.0 Mb.\n",
      "  In addition to autosomes, 2 segments of length 148.9 Mb on X-chr can be further used.\n",
      "  Information of these chromosomal segments can be found in file /home/jupyter/Release6_AMR/gp2_AMRallsegs.txt\n",
      "\n",
      "48 CPU cores are used to compute the pairwise kinship coefficients...\n",
      "Clustering up to 2nd-degree relatives in families...\n",
      "Individual IDs are unique across all families.\n",
      "\n",
      "Relationship summary (total relatives: 0 by pedigree, 5 by inference)\n",
      "        \tMZ\tPO\tFS\t2nd\t3rd\t4th\n",
      "  =========================================================\n",
      "  Inference\t0\t1\t4\t0\t0\t0\n",
      "\n",
      "The following families are found to be connected\n",
      "  NewFamID  OriginalFamID                                     \n",
      "  KING1     CORIELL_002325_s1,CORIELL_004251_s1\n",
      "  KING2     BCM_000739_s1,CORIELL_000771_s1,CORIELL_000892_s1\n",
      "  KING3     PDGNRTN_001424_s1,PDGNRTN_001425_s1\n",
      "\n",
      "Update-ID information is saved in file /home/jupyter/Release6_AMR/gp2_AMRupdateids.txt\n",
      "\n",
      "Pedigree reconstruction starts at Tue Nov 26 05:22:45 2024\n",
      "Reconstructing pedigree...\n",
      "Age information not provided.\n",
      "Total length of 39 chromosomal segments usable for IBD segment analysis is 2690.0 Mb.\n",
      "  In addition to autosomes, 2 segments of length 148.9 Mb on X-chr can be further used.\n",
      "  Information of these chromosomal segments can be found in file /home/jupyter/Release6_AMR/gp2_AMRallsegs.txt\n",
      "\n",
      "Family KING2:\n",
      "  Family KING2 RULE FS0: Sibship (BCM_000739_s1 CORIELL_000771_s1)'s parents are (1 2)\n",
      "  Family KING2 RULE FS1: CORIELL_000892_s1 joins in sibship (BCM_000739_s1 CORIELL_000771_s1)\n",
      "\n",
      "Family KING3:\n",
      "  Family KING3 RULE FS0: Sibship (PDGNRTN_001424_s1 PDGNRTN_001425_s1)'s parents are (3 4)\n",
      "\n",
      "\n",
      "Details of pedigree reconstruction are available in log file /home/jupyter/Release6_AMR/gp2_AMRbuild.log\n",
      "Update-ID information is saved in file /home/jupyter/Release6_AMR/gp2_AMRupdateids.txt\n",
      "Update-parent information is saved in file /home/jupyter/Release6_AMR/gp2_AMRupdateparents.txt\n",
      "Pedigree reconstruction ends at Tue Nov 26 05:22:45 2024\n",
      "KING ends at Tue Nov 26 05:22:45 2024\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Run KING to identify related individuals (up to second-degree relatives)\n",
    "shell_do(f'/home/jupyter/tools/king -b {WORK_DIR}/gp2_{ANC}_r6_qced_updated_id.bed '\n",
    "         f'--build --related --degree 2 '\n",
    "         f'--prefix {WORK_DIR}/gp2_{ANC}')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Update Family ID based on relationship inference\n",
    "\n",
    "After running relationship inference, update the Family IDs to reflect the identified relationships."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: /home/jupyter/tools/plink --bfile /home/jupyter/Release6_AMR/gp2_AMR_r6_qced_updated_id --update-ids /home/jupyter/Release6_AMR/gp2_AMRupdateids.txt --prune --make-bed --out /home/jupyter/Release6_AMR/gp2_AMR_r6_qced_updated_fid\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/Release6_AMR/gp2_AMR_r6_qced_updated_fid.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/Release6_AMR/gp2_AMR_r6_qced_updated_id\n",
      "  --make-bed\n",
      "  --out /home/jupyter/Release6_AMR/gp2_AMR_r6_qced_updated_fid\n",
      "  --prune\n",
      "  --update-ids /home/jupyter/Release6_AMR/gp2_AMRupdateids.txt\n",
      "\n",
      "628867 MB RAM detected; reserving 314433 MB for main workspace.\n",
      "1184930 variants loaded from .bim file.\n",
      "506 people (278 males, 228 females) loaded from .fam.\n",
      "506 phenotype values loaded from .fam.\n",
      "--update-ids: 7 people updated.\n",
      "--prune: 506 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 506 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Warning: 1843 het. haploid genotypes present (see\n",
      "/home/jupyter/Release6_AMR/gp2_AMR_r6_qced_updated_fid.hh ); many commands\n",
      "treat these as missing.\n",
      "Warning: Nonmissing nonmale Y chromosome genotype(s) present; many commands\n",
      "treat these as missing.\n",
      "Total genotyping rate is 0.997556.\n",
      "1184930 variants and 506 people pass filters and QC.\n",
      "Among remaining phenotypes, 367 are cases and 139 are controls.\n",
      "--make-bed to /home/jupyter/Release6_AMR/gp2_AMR_r6_qced_updated_fid.bed +\n",
      "/home/jupyter/Release6_AMR/gp2_AMR_r6_qced_updated_fid.bim +\n",
      "/home/jupyter/Release6_AMR/gp2_AMR_r6_qced_updated_fid.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    }
   ],
   "source": [
    "# Run plink to update FIDs/IDs, prune, and output updated binary files\n",
    "shell_do(f'/home/jupyter/tools/plink --bfile {WORK_DIR}/gp2_{ANC}_r6_qced_updated_id '\n",
    "         f'--update-ids {WORK_DIR}/gp2_{ANC}updateids.txt '\n",
    "         f'--prune '\n",
    "         f'--make-bed --out {WORK_DIR}/gp2_{ANC}_r6_qced_updated_fid')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Check for inbred individuals\n",
    "\n",
    "This step identifies individuals with a higher than normal inbreeding coefficient. These individuals may have regions of homozygosity indicative of consanguinity."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: /home/jupyter/tools/king -b /home/jupyter/Release6_AMR/gp2_AMR_r6_qced_updated_fid.bed --roh --prefix /home/jupyter/Release6_AMR/gp2_AMR\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "KING 2.3.0 - (c) 2010-2022 Wei-Min Chen\n",
      "\n",
      "The following parameters are in effect:\n",
      "                   Binary File : /home/jupyter/Release6_AMR/gp2_AMR_r6_qced_updated_fid.bed (-bname)\n",
      "\n",
      "Additional Options\n",
      "         Close Relative Inference : --related, --duplicate\n",
      "   Pairwise Relatedness Inference : --kinship, --ibdseg, --ibs, --makeGRM\n",
      "              Inference Parameter : --degree, --seglength\n",
      "         Relationship Application : --unrelated, --cluster, --build\n",
      "                        QC Report : --bysample, --bySNP, --roh [ON], --autoQC\n",
      "                     QC Parameter : --callrateN, --callrateM\n",
      "             Population Structure : --pca, --mds\n",
      "              Structure Parameter : --projection, --pcs\n",
      "              Disease Association : --tdt\n",
      "   Quantitative Trait Association : --lmm\n",
      "                Association Model : --trait [], --covariate []\n",
      "            Association Parameter : --invnorm\n",
      "               Genetic Risk Score : --risk, --model [], --prevalence, --noflip\n",
      "              Computing Parameter : --cpus\n",
      "                   Optional Input : --fam [], --bim [], --sexchr [23]\n",
      "                           Output : --rplot, --pngplot, --plink\n",
      "                 Output Parameter :\n",
      "                                    --prefix [/home/jupyter/Release6_AMR/gp2_AMR],\n",
      "                                    --rpath []\n",
      "\n",
      "KING starts at Tue Nov 26 05:22:49 2024\n",
      "Loading genotype data in PLINK binary format...\n",
      "Read in PLINK fam file /home/jupyter/Release6_AMR/gp2_AMR_r6_qced_updated_fid.fam...\n",
      "  PLINK pedigrees loaded: 506 samples\n",
      "Read in PLINK bim file /home/jupyter/Release6_AMR/gp2_AMR_r6_qced_updated_fid.bim...\n",
      "  Genotype data consist of 1141927 autosome SNPs (including 4345 XY SNPs), 38259 X-chromosome SNPs, 2241 Y-chromosome SNPs, 478 mitochondrial SNPs\n",
      "  2025 other SNPs are removed.\n",
      "  PLINK maps loaded: 1182905 SNPs\n",
      "Read in PLINK bed file /home/jupyter/Release6_AMR/gp2_AMR_r6_qced_updated_fid.bed...\n",
      "  PLINK binary genotypes loaded.\n",
      "  KING format genotype data successfully converted.\n",
      "\n",
      "Options in effect:\n",
      "\t--roh\n",
      "\t--prefix /home/jupyter/Release6_AMR/gp2_AMR\n",
      "\n",
      "Total length of 39 chromosomal segments usable for IBD segment analysis is 2690.0 Mb.\n",
      "  In addition to autosomes, 2 segments of length 148.9 Mb on X-chr can be further used.\n",
      "  Information of these chromosomal segments can be found in file /home/jupyter/Release6_AMR/gp2_AMRallsegs.txt\n",
      "\n",
      "Run of homozygosity analysis starts at Tue Nov 26 05:22:52 2024\n",
      "Run of homozygosity analysis   ends at Tue Nov 26 05:22:52 2024\n",
      "  Short ROHs (<5Mb) are discarded.\n",
      "\n",
      "The following 1 persons have excessive run of homozygosity (F>0.0884):\n",
      "  Person (PDGNRTN_001799_s2, PDGNRTN_001799_s2) has inbreeding coefficient = 0.1172\n",
      "\n",
      "Run of homozygosity summary saved in file /home/jupyter/Release6_AMR/gp2_AMR.roh\n",
      "Run of homozygosity segments saved in file /home/jupyter/Release6_AMR/gp2_AMR.rohseg.gz\n",
      "\n",
      "KING ends at Tue Nov 26 05:22:52 2024\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Run KING to detect runs of homozygosity (ROH)\n",
    "shell_do(f'/home/jupyter/tools/king -b {WORK_DIR}/gp2_{ANC}_r6_qced_updated_fid.bed '\n",
    "         f'--roh --prefix {WORK_DIR}/gp2_{ANC}')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Optional: retain individuals with F>0.0884 to check overlapped ROH by KING\n",
    "\n",
    "For reference, retain individuals with inbreeding coefficient (F) greater than 0.0884 and check for overlapping ROH."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load the ROH data from the specified file\n",
    "inbred = pd.read_csv(f'{WORK_DIR}/gp2_{ANC}.roh', sep='\\t')\n",
    "\n",
    "# Filter the rows where F_ROH (inbreeding coefficient) is greater than 0.0884\n",
    "inbred_high_froh = inbred.loc[inbred['F_ROH'] > 0.0884]\n",
    "\n",
    "inbred_high_froh"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Find ROH overlaps that likely segregate within families\n",
    "\n",
    "Identify ROH that overlap within families, as these shared regions may harbor disease-causing mutations. Ensure no control individuals carry the same segments."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: /home/jupyter/tools/plink --bfile /home/jupyter/Release6_AMR/gp2_AMR_r6_qced_updated_fid --homozyg group --homozyg-density 50 --homozyg-gap 1000 --homozyg-kb 1500 --homozyg-snp 100 --homozyg-window-het 1 --homozyg-window-missing 5 --homozyg-window-snp 50 --homozyg-window-threshold 0.05 --homozyg-match 0.95 --out /home/jupyter/Release6_AMR/gp2_AMR_r6\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/Release6_AMR/gp2_AMR_r6.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/Release6_AMR/gp2_AMR_r6_qced_updated_fid\n",
      "  --homozyg group\n",
      "  --homozyg-density 50\n",
      "  --homozyg-gap 1000\n",
      "  --homozyg-kb 1500\n",
      "  --homozyg-match 0.95\n",
      "  --homozyg-snp 100\n",
      "  --homozyg-window-het 1\n",
      "  --homozyg-window-missing 5\n",
      "  --homozyg-window-snp 50\n",
      "  --homozyg-window-threshold 0.05\n",
      "  --out /home/jupyter/Release6_AMR/gp2_AMR_r6\n",
      "\n",
      "628867 MB RAM detected; reserving 314433 MB for main workspace.\n",
      "1184930 variants loaded from .bim file.\n",
      "506 people (278 males, 228 females) loaded from .fam.\n",
      "506 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 506 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Warning: 1843 het. haploid genotypes present (see\n",
      "/home/jupyter/Release6_AMR/gp2_AMR_r6.hh ); many commands treat these as\n",
      "missing.\n",
      "Warning: Nonmissing nonmale Y chromosome genotype(s) present; many commands\n",
      "treat these as missing.\n",
      "Total genotyping rate is 0.997556.\n",
      "1184930 variants and 506 people pass filters and QC.\n",
      "Among remaining phenotypes, 367 are cases and 139 are controls.\n",
      "--homozyg: Scan complete, found 5440 ROH.\n",
      "Results saved to /home/jupyter/Release6_AMR/gp2_AMR_r6.hom +\n",
      "/home/jupyter/Release6_AMR/gp2_AMR_r6.hom.indiv +\n",
      "/home/jupyter/Release6_AMR/gp2_AMR_r6.hom.summary .\n",
      "1089 pools of overlapping ROH present.\n",
      "Determining within-pool allelic similarity... [chromosome 1]3]4]5]6]7]8]101112131415161718192021222325              done.\n",
      "ROH pool report written to /home/jupyter/Release6_AMR/gp2_AMR_r6.hom.overlap .\n"
     ]
    }
   ],
   "source": [
    "# Run plink to detect runs of homozygosity (ROH) with specified parameters\n",
    "shell_do(f'/home/jupyter/tools/plink --bfile {WORK_DIR}/gp2_{ANC}_r6_qced_updated_fid '\n",
    "         f'--homozyg group '\n",
    "         f'--homozyg-density 50 '\n",
    "         f'--homozyg-gap 1000 '\n",
    "         f'--homozyg-kb 1500 '\n",
    "         f'--homozyg-snp 100 '\n",
    "         f'--homozyg-window-het 1 '\n",
    "         f'--homozyg-window-missing 5 '\n",
    "         f'--homozyg-window-snp 50 '\n",
    "         f'--homozyg-window-threshold 0.05 '\n",
    "         f'--homozyg-match 0.95 '\n",
    "         f'--out {WORK_DIR}/gp2_{ANC}_r6')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Check for overlapping allele-matched ROH within families\n",
    "\n",
    "Ensure that ROHs with overlapping alleles within families are identified, particularly those that segregate among affected individuals. These regions can be pooled for further analysis.\n",
    "\n",
    "Detect regions where individuals share the same alleles, which may indicate genetic relatedness or a shared ancestry.\n",
    "\n",
    "Focus on allele segregation within families or shared segments that potentially harbor disease-causing mutations across different individuals."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load the overlap file for the specific ancestry group\n",
    "df = pd.read_csv(f'{WORK_DIR}/gp2_{ANC}_r6.hom.overlap',sep='\\s+')\n",
    "\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Check for pools that were shared within families\n",
    "ROH_fam = pd.concat(g for _, g in df.groupby([\"POOL\", \"FID\"]) if len(g) > 1)\n",
    "\n",
    "# Display the result\n",
    "ROH_fam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Read family structures from the .fam file\n",
    "fams = pd.read_csv(f'{WORK_DIR}/gp2_{ANC}_r6_qced_updated_fid.fam', \n",
    "                   sep=' ', header=None, \n",
    "                   names=['FID', 'IID', 'pat_ID', 'mat_ID', 'sex', 'pheno'])\n",
    "\n",
    "# Sort by family ID (FID) and individual ID (IID)\n",
    "fams = fams.sort_values(['FID', 'IID'])\n",
    "\n",
    "# Keep only families with multiple individuals (same FID)\n",
    "fams = pd.concat(g for _, g in fams.groupby(\"FID\") if len(g) > 1)\n",
    "\n",
    "# Display the result\n",
    "fams"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Define a function to filter groups that contain all IDs in the list\n",
    "def filter_groups(group):\n",
    "    \"\"\"\n",
    "    Check if all IDs in 'id_list' are present in the 'IID' column of the group.\n",
    "    \n",
    "    Args:\n",
    "        group (DataFrame): A grouped DataFrame with 'IID' as one of its columns.\n",
    "\n",
    "    Returns:\n",
    "        bool: True if all IDs in 'id_list' are present in the group, False otherwise.\n",
    "    \"\"\"\n",
    "    return set(id_list).issubset(group['IID'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Keep pools that contain all affected family members with allele-matched ROH\n",
    "li = []\n",
    "\n",
    "for pool, dat in ROH_fam.groupby('POOL'):\n",
    "    for fam, fam_dat in fams.groupby('FID'):\n",
    "        # List of affected individuals in the family (pheno == 2)\n",
    "        id_list = fam_dat.loc[fam_dat['pheno'] == 2, 'IID'].to_list()\n",
    "        \n",
    "        # Keep pools that contain all affected family members\n",
    "        keep = dat.groupby('POOL').filter(filter_groups)\n",
    "        \n",
    "        # Remove '*' from 'GRP' to match alleles\n",
    "        keep['GRP'] = keep['GRP'].str.replace('*', '', regex=False)\n",
    "        \n",
    "        # Keep allele-matched ROHs that segregate within the same families\n",
    "        seg_keep = keep.groupby('GRP').filter(filter_groups)\n",
    "        \n",
    "        li.append(seg_keep)\n",
    "\n",
    "# Concatenate the results into a single DataFrame\n",
    "fam_pools = pd.concat(li, axis=0)\n",
    "\n",
    "# Display the result\n",
    "fam_pools"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Check the pools for 'CON' and 'UNION' families\n",
    "fam_pools_CON_UNION = df.loc[\n",
    "    (df['POOL'].isin(fam_pools['POOL'])) & (df['FID'].isin(['CON', 'UNION']))\n",
    "]\n",
    "\n",
    "# Display the result\n",
    "fam_pools_CON_UNION"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>POOL</th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PHE</th>\n",
       "      <th>CHR</th>\n",
       "      <th>SNP1</th>\n",
       "      <th>SNP2</th>\n",
       "      <th>BP1</th>\n",
       "      <th>BP2</th>\n",
       "      <th>KB</th>\n",
       "      <th>NSNP</th>\n",
       "      <th>NSIM</th>\n",
       "      <th>GRP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [POOL, FID, IID, PHE, CHR, SNP1, SNP2, BP1, BP2, KB, NSNP, NSIM, GRP]\n",
       "Index: []"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Filter for ROHs that no control (CON) carried, looking for high penetrance variants\n",
    "check_ROH = df.loc[\n",
    "    (df['POOL'].isin(fam_pools['POOL'])) & \n",
    "    (df['FID'].isin(['CON', 'UNION'])) & \n",
    "    (df['PHE'].str.contains(':0'))\n",
    "]\n",
    "\n",
    "# Display the result\n",
    "check_ROH"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Fine mapping (if high penetrance variants were found)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Filter to identify rows corresponding to the region where the most individuals carried the allele-matched ROH\n",
    "# Specifically, focus on the pool labeled 'S164'\n",
    "filtered_df = df.loc[df['POOL'] == 'S164']\n",
    "\n",
    "# Display the filtered DataFrame to examine the individuals and details for this region\n",
    "filtered_df\n",
    "\n",
    "# Observation: The filtered results show one family and one unrelated individual carrying the allele-matched ROH\n",
    "# This suggests the presence of a potentially shared or segregating genomic region of interest."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Find ROH overlaps between inbred individuals"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Subset samples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Read the inbreeding ROH file (*_ROH.hom.indiv) from Google Cloud Storage\n",
    "FROH = gcs_read_csv(f'gs://fc-1e9cbcf3-a691-4415-8a35-f19d546e50dc/{ANC}_ROH.hom.indiv', sep='\\s+')\n",
    "\n",
    "# Calculate the FROH inbreeding coefficient and round it to 5 decimal places\n",
    "FROH['FROH'] = (FROH['KB'] / 2886428).round(5)\n",
    "\n",
    "# Select cases with FROH > 0.0156 and controls (PHE == 1)\n",
    "keep_inbred = FROH.loc[((FROH['FROH'] > 0.0156) & (FROH['PHE'] == 2)) | (FROH['PHE'] == 1)]\n",
    "\n",
    "# Read the updated FID and IID information from the .fam file\n",
    "new_fid = pd.read_csv(f'{WORK_DIR}/chrAll_{ANC}_QC.fam', sep='\\s+', usecols=[0, 1], header=None, names=['FID', 'IID'])\n",
    "\n",
    "# Merge the inbred samples with the new FID information\n",
    "all_df = pd.merge(keep_inbred, new_fid, on='IID')\n",
    "\n",
    "# Write out FID and IID for subsetting into a list file\n",
    "all_df[['FID_y', 'IID']].to_csv(f'{WORK_DIR}/keep_{ANC}_inbred_sampleset.list', sep='\\t', index=False, header=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: /home/jupyter/tools/plink --bfile /home/jupyter/Release6_AMR/chrAll_AMR_QC --keep /home/jupyter/Release6_AMR/keep_AMR_inbred_sampleset.list --homozyg group --homozyg-density 50 --homozyg-gap 1000 --homozyg-kb 1500 --homozyg-snp 100 --homozyg-window-het 1 --homozyg-window-missing 5 --homozyg-window-snp 50 --homozyg-window-threshold 0.05 --homozyg-match 0.95 --out /home/jupyter/Release6_AMR/gp2_AMR_r6_inbreed\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/Release6_AMR/gp2_AMR_r6_inbreed.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/Release6_AMR/chrAll_AMR_QC\n",
      "  --homozyg group\n",
      "  --homozyg-density 50\n",
      "  --homozyg-gap 1000\n",
      "  --homozyg-kb 1500\n",
      "  --homozyg-match 0.95\n",
      "  --homozyg-snp 100\n",
      "  --homozyg-window-het 1\n",
      "  --homozyg-window-missing 5\n",
      "  --homozyg-window-snp 50\n",
      "  --homozyg-window-threshold 0.05\n",
      "  --keep /home/jupyter/Release6_AMR/keep_AMR_inbred_sampleset.list\n",
      "  --out /home/jupyter/Release6_AMR/gp2_AMR_r6_inbreed\n",
      "\n",
      "628867 MB RAM detected; reserving 314433 MB for main workspace.\n",
      "4094211 variants loaded from .bim file.\n",
      "524 people (286 males, 238 females) loaded from .fam.\n",
      "502 phenotype values loaded from .fam.\n",
      "--keep: 159 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 159 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.97762.\n",
      "4094211 variants and 159 people pass filters and QC.\n",
      "Among remaining phenotypes, 20 are cases and 139 are controls.\n",
      "--homozyg: Scan complete, found 1079 ROH.\n",
      "Results saved to /home/jupyter/Release6_AMR/gp2_AMR_r6_inbreed.hom +\n",
      "/home/jupyter/Release6_AMR/gp2_AMR_r6_inbreed.hom.indiv +\n",
      "/home/jupyter/Release6_AMR/gp2_AMR_r6_inbreed.hom.summary .\n",
      "450 pools of overlapping ROH present.\n",
      "Determining within-pool allelic similarity... [chromosome 2]4]6]8]10111213141516171819202122              done.\n",
      "ROH pool report written to\n",
      "/home/jupyter/Release6_AMR/gp2_AMR_r6_inbreed.hom.overlap .\n"
     ]
    }
   ],
   "source": [
    "# Subset the samples of interest and run homozygosity mapping again\n",
    "shell_do(f'/home/jupyter/tools/plink --bfile {WORK_DIR}/chrAll_{ANC}_QC '\n",
    "         f'--keep {WORK_DIR}/keep_{ANC}_inbred_sampleset.list '\n",
    "         f'--homozyg group '\n",
    "         f'--homozyg-density 50 '\n",
    "         f'--homozyg-gap 1000 '\n",
    "         f'--homozyg-kb 1500 '\n",
    "         f'--homozyg-snp 100 '\n",
    "         f'--homozyg-window-het 1 '\n",
    "         f'--homozyg-window-missing 5 '\n",
    "         f'--homozyg-window-snp 50 '\n",
    "         f'--homozyg-window-threshold 0.05 '\n",
    "         f'--homozyg-match 0.95 '\n",
    "         f'--out {WORK_DIR}/gp2_{ANC}_r6_inbreed')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Check consensus ROH (cROH)\n",
    "\n",
    "The consensus ROH represents regions of homozygosity shared by multiple individuals. Identifying cROH may indicate regions of interest for further analysis."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Read the inbreeding ROH overlap data\n",
    "inbred_roh = pd.read_csv(f'{WORK_DIR}/gp2_{ANC}_r6_inbreed.hom.overlap', sep='\\s+')\n",
    "\n",
    "# Filter for rows where the FID is 'CON' (consensus)\n",
    "inbred_roh_consense = inbred_roh.loc[inbred_roh['FID'] == 'CON']\n",
    "\n",
    "# Display the result\n",
    "inbred_roh_consense"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>POOL</th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PHE</th>\n",
       "      <th>CHR</th>\n",
       "      <th>SNP1</th>\n",
       "      <th>SNP2</th>\n",
       "      <th>BP1</th>\n",
       "      <th>BP2</th>\n",
       "      <th>KB</th>\n",
       "      <th>NSNP</th>\n",
       "      <th>NSIM</th>\n",
       "      <th>GRP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>S1</td>\n",
       "      <td>CON</td>\n",
       "      <td>20</td>\n",
       "      <td>4:16</td>\n",
       "      <td>7</td>\n",
       "      <td>chr7:65477373:C:A</td>\n",
       "      <td>chr7:66044271:T:C</td>\n",
       "      <td>65477373</td>\n",
       "      <td>66044271</td>\n",
       "      <td>566.899</td>\n",
       "      <td>365</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>72</th>\n",
       "      <td>S4</td>\n",
       "      <td>CON</td>\n",
       "      <td>14</td>\n",
       "      <td>3:11</td>\n",
       "      <td>6</td>\n",
       "      <td>chr6:61118955:C:T</td>\n",
       "      <td>chr6:61867241:T:C</td>\n",
       "      <td>61118955</td>\n",
       "      <td>61867241</td>\n",
       "      <td>748.287</td>\n",
       "      <td>1415</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87</th>\n",
       "      <td>S5</td>\n",
       "      <td>CON</td>\n",
       "      <td>13</td>\n",
       "      <td>4:9</td>\n",
       "      <td>8</td>\n",
       "      <td>chr8:46622494:A:C</td>\n",
       "      <td>chr8:47646564:A:G</td>\n",
       "      <td>46622494</td>\n",
       "      <td>47646564</td>\n",
       "      <td>1024.071</td>\n",
       "      <td>827</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>115</th>\n",
       "      <td>S7</td>\n",
       "      <td>CON</td>\n",
       "      <td>12</td>\n",
       "      <td>2:10</td>\n",
       "      <td>11</td>\n",
       "      <td>chr11:49155393:A:G</td>\n",
       "      <td>chr11:49489304:A:C</td>\n",
       "      <td>49155393</td>\n",
       "      <td>49489304</td>\n",
       "      <td>333.912</td>\n",
       "      <td>441</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>129</th>\n",
       "      <td>S8</td>\n",
       "      <td>CON</td>\n",
       "      <td>12</td>\n",
       "      <td>0:12</td>\n",
       "      <td>22</td>\n",
       "      <td>chr22:41154950:C:T</td>\n",
       "      <td>chr22:41854130:TTTG:T</td>\n",
       "      <td>41154950</td>\n",
       "      <td>41854130</td>\n",
       "      <td>699.181</td>\n",
       "      <td>862</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2489</th>\n",
       "      <td>S446</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>20</td>\n",
       "      <td>chr20:59448560:T:C</td>\n",
       "      <td>chr20:61999812:G:A</td>\n",
       "      <td>59448560</td>\n",
       "      <td>61999812</td>\n",
       "      <td>2551.253</td>\n",
       "      <td>4161</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2493</th>\n",
       "      <td>S447</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>1:1</td>\n",
       "      <td>21</td>\n",
       "      <td>chr21:34816191:C:T</td>\n",
       "      <td>chr21:36669561:A:C</td>\n",
       "      <td>34816191</td>\n",
       "      <td>36669561</td>\n",
       "      <td>1853.371</td>\n",
       "      <td>2944</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2497</th>\n",
       "      <td>S448</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>21</td>\n",
       "      <td>chr21:42781756:G:C</td>\n",
       "      <td>chr21:42954192:C:T</td>\n",
       "      <td>42781756</td>\n",
       "      <td>42954192</td>\n",
       "      <td>172.437</td>\n",
       "      <td>417</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2501</th>\n",
       "      <td>S449</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>21</td>\n",
       "      <td>chr21:42997454:A:G</td>\n",
       "      <td>chr21:46518326:T:C</td>\n",
       "      <td>42997454</td>\n",
       "      <td>46518326</td>\n",
       "      <td>3520.873</td>\n",
       "      <td>5391</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2505</th>\n",
       "      <td>S450</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>22</td>\n",
       "      <td>chr22:36281354:G:A</td>\n",
       "      <td>chr22:36597041:G:C</td>\n",
       "      <td>36281354</td>\n",
       "      <td>36597041</td>\n",
       "      <td>315.688</td>\n",
       "      <td>423</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>406 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      POOL  FID IID   PHE  CHR                SNP1                   SNP2  \\\n",
       "20      S1  CON  20  4:16    7   chr7:65477373:C:A      chr7:66044271:T:C   \n",
       "72      S4  CON  14  3:11    6   chr6:61118955:C:T      chr6:61867241:T:C   \n",
       "87      S5  CON  13   4:9    8   chr8:46622494:A:C      chr8:47646564:A:G   \n",
       "115     S7  CON  12  2:10   11  chr11:49155393:A:G     chr11:49489304:A:C   \n",
       "129     S8  CON  12  0:12   22  chr22:41154950:C:T  chr22:41854130:TTTG:T   \n",
       "...    ...  ...  ..   ...  ...                 ...                    ...   \n",
       "2489  S446  CON   2   2:0   20  chr20:59448560:T:C     chr20:61999812:G:A   \n",
       "2493  S447  CON   2   1:1   21  chr21:34816191:C:T     chr21:36669561:A:C   \n",
       "2497  S448  CON   2   2:0   21  chr21:42781756:G:C     chr21:42954192:C:T   \n",
       "2501  S449  CON   2   2:0   21  chr21:42997454:A:G     chr21:46518326:T:C   \n",
       "2505  S450  CON   2   2:0   22  chr22:36281354:G:A     chr22:36597041:G:C   \n",
       "\n",
       "           BP1       BP2        KB  NSNP  NSIM  GRP  \n",
       "20    65477373  66044271   566.899   365   NaN  NaN  \n",
       "72    61118955  61867241   748.287  1415   NaN  NaN  \n",
       "87    46622494  47646564  1024.071   827   NaN  NaN  \n",
       "115   49155393  49489304   333.912   441   NaN  NaN  \n",
       "129   41154950  41854130   699.181   862   NaN  NaN  \n",
       "...        ...       ...       ...   ...   ...  ...  \n",
       "2489  59448560  61999812  2551.253  4161   NaN  NaN  \n",
       "2493  34816191  36669561  1853.371  2944   NaN  NaN  \n",
       "2497  42781756  42954192   172.437   417   NaN  NaN  \n",
       "2501  42997454  46518326  3520.873  5391   NaN  NaN  \n",
       "2505  36281354  36597041   315.688   423   NaN  NaN  \n",
       "\n",
       "[406 rows x 13 columns]"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Filter for consensus ROH (cROH) with a length > 100 Kb and > 100 SNPs\n",
    "inbred_roh_consense_filtered = inbred_roh_consense.loc[\n",
    "    (inbred_roh_consense['NSNP'] >= 100) & (inbred_roh_consense['KB'] >= 100)\n",
    "]\n",
    "\n",
    "# Save the filtered results to a file\n",
    "inbred_roh_consense_filtered.to_csv(\n",
    "    f'{WORK_DIR}/gp2_{ANC}_r6_inbreed.hom.overlap.filtered', \n",
    "    sep='\\t', index=False, header=False\n",
    ")\n",
    "\n",
    "# Display the filtered result\n",
    "inbred_roh_consense_filtered"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>POOL</th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PHE</th>\n",
       "      <th>CHR</th>\n",
       "      <th>SNP1</th>\n",
       "      <th>SNP2</th>\n",
       "      <th>BP1</th>\n",
       "      <th>BP2</th>\n",
       "      <th>KB</th>\n",
       "      <th>NSNP</th>\n",
       "      <th>NSIM</th>\n",
       "      <th>GRP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>782</th>\n",
       "      <td>S82</td>\n",
       "      <td>CON</td>\n",
       "      <td>5</td>\n",
       "      <td>5:0</td>\n",
       "      <td>9</td>\n",
       "      <td>chr9:82135547:C:A</td>\n",
       "      <td>chr9:84169832:A:T</td>\n",
       "      <td>82135547</td>\n",
       "      <td>84169832</td>\n",
       "      <td>2034.286</td>\n",
       "      <td>2754</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1022</th>\n",
       "      <td>S121</td>\n",
       "      <td>CON</td>\n",
       "      <td>4</td>\n",
       "      <td>4:0</td>\n",
       "      <td>10</td>\n",
       "      <td>chr10:24128513:T:C</td>\n",
       "      <td>chr10:25123571:A:G</td>\n",
       "      <td>24128513</td>\n",
       "      <td>25123571</td>\n",
       "      <td>995.059</td>\n",
       "      <td>1379</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1028</th>\n",
       "      <td>S122</td>\n",
       "      <td>CON</td>\n",
       "      <td>4</td>\n",
       "      <td>4:0</td>\n",
       "      <td>10</td>\n",
       "      <td>chr10:25129805:G:C</td>\n",
       "      <td>chr10:25475212:G:A</td>\n",
       "      <td>25129805</td>\n",
       "      <td>25475212</td>\n",
       "      <td>345.408</td>\n",
       "      <td>486</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1112</th>\n",
       "      <td>S136</td>\n",
       "      <td>CON</td>\n",
       "      <td>4</td>\n",
       "      <td>4:0</td>\n",
       "      <td>17</td>\n",
       "      <td>chr17:9973671:G:C</td>\n",
       "      <td>chr17:10259230:T:C</td>\n",
       "      <td>9973671</td>\n",
       "      <td>10259230</td>\n",
       "      <td>285.560</td>\n",
       "      <td>658</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1420</th>\n",
       "      <td>S196</td>\n",
       "      <td>CON</td>\n",
       "      <td>3</td>\n",
       "      <td>3:0</td>\n",
       "      <td>6</td>\n",
       "      <td>chr6:164604500:C:T</td>\n",
       "      <td>chr6:165879923:CG:C</td>\n",
       "      <td>164604500</td>\n",
       "      <td>165879923</td>\n",
       "      <td>1275.424</td>\n",
       "      <td>2645</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2473</th>\n",
       "      <td>S442</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>20</td>\n",
       "      <td>chr20:796225:T:G</td>\n",
       "      <td>chr20:5046718:T:C</td>\n",
       "      <td>796225</td>\n",
       "      <td>5046718</td>\n",
       "      <td>4250.494</td>\n",
       "      <td>7484</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2489</th>\n",
       "      <td>S446</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>20</td>\n",
       "      <td>chr20:59448560:T:C</td>\n",
       "      <td>chr20:61999812:G:A</td>\n",
       "      <td>59448560</td>\n",
       "      <td>61999812</td>\n",
       "      <td>2551.253</td>\n",
       "      <td>4161</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2497</th>\n",
       "      <td>S448</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>21</td>\n",
       "      <td>chr21:42781756:G:C</td>\n",
       "      <td>chr21:42954192:C:T</td>\n",
       "      <td>42781756</td>\n",
       "      <td>42954192</td>\n",
       "      <td>172.437</td>\n",
       "      <td>417</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2501</th>\n",
       "      <td>S449</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>21</td>\n",
       "      <td>chr21:42997454:A:G</td>\n",
       "      <td>chr21:46518326:T:C</td>\n",
       "      <td>42997454</td>\n",
       "      <td>46518326</td>\n",
       "      <td>3520.873</td>\n",
       "      <td>5391</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2505</th>\n",
       "      <td>S450</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>22</td>\n",
       "      <td>chr22:36281354:G:A</td>\n",
       "      <td>chr22:36597041:G:C</td>\n",
       "      <td>36281354</td>\n",
       "      <td>36597041</td>\n",
       "      <td>315.688</td>\n",
       "      <td>423</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>86 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      POOL  FID IID  PHE  CHR                SNP1                 SNP2  \\\n",
       "782    S82  CON   5  5:0    9   chr9:82135547:C:A    chr9:84169832:A:T   \n",
       "1022  S121  CON   4  4:0   10  chr10:24128513:T:C   chr10:25123571:A:G   \n",
       "1028  S122  CON   4  4:0   10  chr10:25129805:G:C   chr10:25475212:G:A   \n",
       "1112  S136  CON   4  4:0   17   chr17:9973671:G:C   chr17:10259230:T:C   \n",
       "1420  S196  CON   3  3:0    6  chr6:164604500:C:T  chr6:165879923:CG:C   \n",
       "...    ...  ...  ..  ...  ...                 ...                  ...   \n",
       "2473  S442  CON   2  2:0   20    chr20:796225:T:G    chr20:5046718:T:C   \n",
       "2489  S446  CON   2  2:0   20  chr20:59448560:T:C   chr20:61999812:G:A   \n",
       "2497  S448  CON   2  2:0   21  chr21:42781756:G:C   chr21:42954192:C:T   \n",
       "2501  S449  CON   2  2:0   21  chr21:42997454:A:G   chr21:46518326:T:C   \n",
       "2505  S450  CON   2  2:0   22  chr22:36281354:G:A   chr22:36597041:G:C   \n",
       "\n",
       "            BP1        BP2        KB  NSNP  NSIM  GRP  \n",
       "782    82135547   84169832  2034.286  2754   NaN  NaN  \n",
       "1022   24128513   25123571   995.059  1379   NaN  NaN  \n",
       "1028   25129805   25475212   345.408   486   NaN  NaN  \n",
       "1112    9973671   10259230   285.560   658   NaN  NaN  \n",
       "1420  164604500  165879923  1275.424  2645   NaN  NaN  \n",
       "...         ...        ...       ...   ...   ...  ...  \n",
       "2473     796225    5046718  4250.494  7484   NaN  NaN  \n",
       "2489   59448560   61999812  2551.253  4161   NaN  NaN  \n",
       "2497   42781756   42954192   172.437   417   NaN  NaN  \n",
       "2501   42997454   46518326  3520.873  5391   NaN  NaN  \n",
       "2505   36281354   36597041   315.688   423   NaN  NaN  \n",
       "\n",
       "[86 rows x 13 columns]"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Filter for ROHs that no control (CON) carried, looking for high penetrance variants\n",
    "check_ROH = inbred_roh_consense_filtered.loc[\n",
    "    inbred_roh_consense_filtered['PHE'].str.contains(':0')\n",
    "]\n",
    "\n",
    "# Display the result\n",
    "check_ROH"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "S82 S121 S122 S136 S196 S197 S199 S200 S201 S202 S206 S207 S211 S212 S220 S221 S222 S226 S252 S256 S265 S268 S269 S270 S276 S291 S303 S304 S309 S317 S324 S325 S328 S335 S336 S337 S341 S342 S345 S360 S361 S362 S363 S364 S366 S373 S374 S376 S378 S381 S382 S383 S384 S385 S386 S387 S394 S395 S397 S399 S400 S401 S407 S408 S409 S411 S412 S413 S414 S415 S421 S426 S427 S428 S430 S431 S436 S437 S438 S440 S441 S442 S446 S448 S449 S450\n"
     ]
    }
   ],
   "source": [
    "# Extract 'POOL' column values and join them into a space-separated string\n",
    "columns = check_ROH['POOL'].tolist()\n",
    "column_spaces = ' '.join(map(str, columns))\n",
    "\n",
    "# Display the result\n",
    "print(column_spaces)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Find ROH enriched in inbred cases (test the association using a logistic model)\n",
    "\n",
    "Assess whether specific ROHs are enriched in inbred cases using logistic regression models to identify significant associations."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Consensus ROH Filtered\n",
    "\n",
    "Apply filters to consensus ROH based on specific criteria to refine the list of regions under investigation."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Classes ‘data.table’ and 'data.frame':\t406 obs. of  13 variables:\n",
      " $ POOL: chr  \"S1\" \"S4\" \"S5\" \"S7\" ...\n",
      " $ FID : chr  \"CON\" \"CON\" \"CON\" \"CON\" ...\n",
      " $ IID : int  20 14 13 12 12 11 11 11 11 10 ...\n",
      " $ PHE : chr  \"4:16\" \"3:11\" \"4:9\" \"2:10\" ...\n",
      " $ CHR : int  7 6 8 11 22 11 11 16 20 11 ...\n",
      " $ SNP1: chr  \"chr7:65477373:C:A\" \"chr6:61118955:C:T\" \"chr8:46622494:A:C\" \"chr11:49155393:A:G\" ...\n",
      " $ SNP2: chr  \"chr7:66044271:T:C\" \"chr6:61867241:T:C\" \"chr8:47646564:A:G\" \"chr11:49489304:A:C\" ...\n",
      " $ BP1 : int  65477373 61118955 46622494 49155393 41154950 48022207 48812285 71380456 34716170 55586177 ...\n",
      " $ BP2 : int  66044271 61867241 47646564 49489304 41854130 48310586 48950614 71604736 35257135 56058978 ...\n",
      " $ KB  : num  567 748 1024 334 699 ...\n",
      " $ NSNP: int  365 1415 827 441 862 363 135 356 643 866 ...\n",
      " $ NSIM: logi  NA NA NA NA NA NA ...\n",
      " $ GRP : logi  NA NA NA NA NA NA ...\n",
      " - attr(*, \".internal.selfref\")=<externalptr> \n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "\n",
    "# Define the ancestry code (ANC) and read the filtered ROH data\n",
    "ANC <- \"AMR\"\n",
    "ROH <- fread(paste0(\"gp2_\", ANC, \"_r6_inbreed.hom.overlap.filtered\"), header = FALSE)\n",
    "\n",
    "# Assign column names to the ROH data\n",
    "colnames(ROH) <- c(\"POOL\", \"FID\", \"IID\", \"PHE\", \"CHR\", \"SNP1\", \"SNP2\", \"BP1\", \"BP2\", \"KB\", \"NSNP\", \"NSIM\", \"GRP\")\n",
    "\n",
    "# Display the structure of the ROH data\n",
    "str(ROH)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: grep 'are cases and' /home/jupyter/Release6_AMR/gp2_AMR_r6_inbreed.log\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Among remaining phenotypes, 20 are cases and 139 are controls.\r\n"
     ]
    }
   ],
   "source": [
    "# Use this as input for the next two steps\n",
    "shell_do(f\"grep 'are cases and' {WORK_DIR}/gp2_{ANC}_r6_inbreed.log\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Classes ‘data.table’ and 'data.frame':\t406 obs. of  20 variables:\n",
      " $ POOL             : chr  \"S1\" \"S4\" \"S5\" \"S7\" ...\n",
      " $ FID              : chr  \"CON\" \"CON\" \"CON\" \"CON\" ...\n",
      " $ IID              : int  20 14 13 12 12 11 11 11 11 10 ...\n",
      " $ PHE              : chr  \"4:16\" \"3:11\" \"4:9\" \"2:10\" ...\n",
      " $ CHR              : int  7 6 8 11 22 11 11 16 20 11 ...\n",
      " $ SNP1             : chr  \"chr7:65477373:C:A\" \"chr6:61118955:C:T\" \"chr8:46622494:A:C\" \"chr11:49155393:A:G\" ...\n",
      " $ SNP2             : chr  \"chr7:66044271:T:C\" \"chr6:61867241:T:C\" \"chr8:47646564:A:G\" \"chr11:49489304:A:C\" ...\n",
      " $ BP1              : int  65477373 61118955 46622494 49155393 41154950 48022207 48812285 71380456 34716170 55586177 ...\n",
      " $ BP2              : int  66044271 61867241 47646564 49489304 41854130 48310586 48950614 71604736 35257135 56058978 ...\n",
      " $ KB               : num  567 748 1024 334 699 ...\n",
      " $ NSNP             : int  365 1415 827 441 862 363 135 356 643 866 ...\n",
      " $ NSIM             : logi  NA NA NA NA NA NA ...\n",
      " $ GRP              : logi  NA NA NA NA NA NA ...\n",
      " $ case             : chr  \"4\" \"3\" \"4\" \"2\" ...\n",
      " $ control          : chr  \"16\" \"11\" \"9\" \"10\" ...\n",
      " $ NcasesWithROHs   : num  4 3 4 2 0 2 2 4 2 1 ...\n",
      " $ NcontrolsWithROHs: num  16 11 9 10 12 9 9 7 9 9 ...\n",
      " $ combinedN        : num  20 14 13 12 12 11 11 11 11 10 ...\n",
      " $ P                : num  0.478 0.533 0.104 1 0.361 ...\n",
      " $ total_ROH_count  : num  20 14 13 12 12 11 11 11 11 10 ...\n",
      " - attr(*, \".internal.selfref\")=<externalptr> \n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "In addition: There were 50 or more warnings (use warnings() to see the first 50)\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%R\n",
    "\n",
    "# Split the 'PHE' column into case and control components\n",
    "ROH$case <- ldply(strsplit(as.character(ROH$PHE), split = \":\"))[[1]]\n",
    "ROH$control <- ldply(strsplit(as.character(ROH$PHE), split = \":\"))[[2]]\n",
    "\n",
    "# Convert case and control components to numeric\n",
    "ROH$NcasesWithROHs <- as.numeric(ROH$case)\n",
    "ROH$NcontrolsWithROHs <- as.numeric(ROH$control)\n",
    "\n",
    "# Calculate total counts for cases, controls, and combined\n",
    "ROH$combinedN <- ROH$NcasesWithROHs + ROH$NcontrolsWithROHs\n",
    "\n",
    "# Initialize the 'P' column for p-values\n",
    "ROH$P <- NA\n",
    "\n",
    "# Perform proportion tests for each row\n",
    "for (i in seq_len(nrow(ROH))) {\n",
    "  thisP <- prop.test(x = c(ROH$NcasesWithROHs[i], ROH$NcontrolsWithROHs[i]), \n",
    "                     n = c(20, 139))  # n = (cases, controls)\n",
    "  ROH$P[i] <- thisP$p.value\n",
    "}\n",
    "\n",
    "# Calculate the total ROH count for cases and controls\n",
    "ROH$total_ROH_count <- ROH$NcasesWithROHs + ROH$NcontrolsWithROHs\n",
    "\n",
    "# Display the structure of the ROH data\n",
    "str(ROH)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. \n",
      "   0.00    0.00    0.00    0.02    0.00    1.00 \n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "\n",
    "# Calculate proportions of cases and controls with ROH\n",
    "ROH$propCases <- ROH$NcasesWithROHs / 20  # Number of cases\n",
    "ROH$propControls <- ROH$NcontrolsWithROHs / 139  # Number of controls\n",
    "\n",
    "# Identify case-enriched ROHs\n",
    "ROH$caseEnriched <- ifelse(ROH$propCases > ROH$propControls, 1, 0)\n",
    "\n",
    "# Subset ROHs where total ROH count >= 1 and case-enriched\n",
    "ROH_subsetted <- subset(ROH, total_ROH_count >= 1 & caseEnriched == 1)\n",
    "\n",
    "# Apply Bonferroni correction\n",
    "Nbonf <- length(ROH_subsetted$POOL)\n",
    "ROH_subsetted$passMultiTest <- ifelse(ROH_subsetted$P <= (0.05 / Nbonf), 1, 0)\n",
    "ROH_subsetted$BONF <- ROH_subsetted$P / Nbonf\n",
    "\n",
    "# Subset the significant results after multiple testing correction\n",
    "significant_ROH <- subset(ROH_subsetted, passMultiTest == 1)\n",
    "\n",
    "# Apply Bonferroni correction to the full dataset\n",
    "ROH$BONF <- ROH$P / Nbonf\n",
    "\n",
    "# Display summary of Bonferroni-corrected significant results\n",
    "summary(ROH_subsetted$passMultiTest)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R\n",
    "\n",
    "# Write the subsetted ROH data to text files\n",
    "fwrite(ROH_subsetted, paste0(ANC, \"_ROH.inbreed_subsetted.txt\"))\n",
    "fwrite(ROH_subsetted, paste0(ANC, \"_ROH.inbreed_caseEnriched.txt\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>POOL</th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PHE</th>\n",
       "      <th>CHR</th>\n",
       "      <th>SNP1</th>\n",
       "      <th>SNP2</th>\n",
       "      <th>BP1</th>\n",
       "      <th>BP2</th>\n",
       "      <th>KB</th>\n",
       "      <th>...</th>\n",
       "      <th>NcasesWithROHs</th>\n",
       "      <th>NcontrolsWithROHs</th>\n",
       "      <th>combinedN</th>\n",
       "      <th>P</th>\n",
       "      <th>total_ROH_count</th>\n",
       "      <th>propCases</th>\n",
       "      <th>propControls</th>\n",
       "      <th>caseEnriched</th>\n",
       "      <th>passMultiTest</th>\n",
       "      <th>BONF</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>S1</td>\n",
       "      <td>CON</td>\n",
       "      <td>20</td>\n",
       "      <td>4:16</td>\n",
       "      <td>7</td>\n",
       "      <td>chr7:65477373:C:A</td>\n",
       "      <td>chr7:66044271:T:C</td>\n",
       "      <td>65477373</td>\n",
       "      <td>66044271</td>\n",
       "      <td>566.899</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>16</td>\n",
       "      <td>20</td>\n",
       "      <td>0.477796</td>\n",
       "      <td>20</td>\n",
       "      <td>0.20</td>\n",
       "      <td>0.115108</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.001365</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>S4</td>\n",
       "      <td>CON</td>\n",
       "      <td>14</td>\n",
       "      <td>3:11</td>\n",
       "      <td>6</td>\n",
       "      <td>chr6:61118955:C:T</td>\n",
       "      <td>chr6:61867241:T:C</td>\n",
       "      <td>61118955</td>\n",
       "      <td>61867241</td>\n",
       "      <td>748.287</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>11</td>\n",
       "      <td>14</td>\n",
       "      <td>0.532834</td>\n",
       "      <td>14</td>\n",
       "      <td>0.15</td>\n",
       "      <td>0.079137</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.001522</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>S5</td>\n",
       "      <td>CON</td>\n",
       "      <td>13</td>\n",
       "      <td>4:9</td>\n",
       "      <td>8</td>\n",
       "      <td>chr8:46622494:A:C</td>\n",
       "      <td>chr8:47646564:A:G</td>\n",
       "      <td>46622494</td>\n",
       "      <td>47646564</td>\n",
       "      <td>1024.071</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>9</td>\n",
       "      <td>13</td>\n",
       "      <td>0.103606</td>\n",
       "      <td>13</td>\n",
       "      <td>0.20</td>\n",
       "      <td>0.064748</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000296</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>S7</td>\n",
       "      <td>CON</td>\n",
       "      <td>12</td>\n",
       "      <td>2:10</td>\n",
       "      <td>11</td>\n",
       "      <td>chr11:49155393:A:G</td>\n",
       "      <td>chr11:49489304:A:C</td>\n",
       "      <td>49155393</td>\n",
       "      <td>49489304</td>\n",
       "      <td>333.912</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>10</td>\n",
       "      <td>12</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>12</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.071942</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.002857</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>S11</td>\n",
       "      <td>CON</td>\n",
       "      <td>11</td>\n",
       "      <td>2:9</td>\n",
       "      <td>11</td>\n",
       "      <td>chr11:48022207:TA:T</td>\n",
       "      <td>chr11:48310586:C:T</td>\n",
       "      <td>48022207</td>\n",
       "      <td>48310586</td>\n",
       "      <td>288.380</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>9</td>\n",
       "      <td>11</td>\n",
       "      <td>0.912685</td>\n",
       "      <td>11</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.064748</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.002608</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>345</th>\n",
       "      <td>S446</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>20</td>\n",
       "      <td>chr20:59448560:T:C</td>\n",
       "      <td>chr20:61999812:G:A</td>\n",
       "      <td>59448560</td>\n",
       "      <td>61999812</td>\n",
       "      <td>2551.253</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.007384</td>\n",
       "      <td>2</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000021</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>346</th>\n",
       "      <td>S447</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>1:1</td>\n",
       "      <td>21</td>\n",
       "      <td>chr21:34816191:C:T</td>\n",
       "      <td>chr21:36669561:A:C</td>\n",
       "      <td>34816191</td>\n",
       "      <td>36669561</td>\n",
       "      <td>1853.371</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0.593965</td>\n",
       "      <td>2</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.007194</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.001697</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>347</th>\n",
       "      <td>S448</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>21</td>\n",
       "      <td>chr21:42781756:G:C</td>\n",
       "      <td>chr21:42954192:C:T</td>\n",
       "      <td>42781756</td>\n",
       "      <td>42954192</td>\n",
       "      <td>172.437</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.007384</td>\n",
       "      <td>2</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000021</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>348</th>\n",
       "      <td>S449</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>21</td>\n",
       "      <td>chr21:42997454:A:G</td>\n",
       "      <td>chr21:46518326:T:C</td>\n",
       "      <td>42997454</td>\n",
       "      <td>46518326</td>\n",
       "      <td>3520.873</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.007384</td>\n",
       "      <td>2</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000021</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>349</th>\n",
       "      <td>S450</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>22</td>\n",
       "      <td>chr22:36281354:G:A</td>\n",
       "      <td>chr22:36597041:G:C</td>\n",
       "      <td>36281354</td>\n",
       "      <td>36597041</td>\n",
       "      <td>315.688</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.007384</td>\n",
       "      <td>2</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000021</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>350 rows × 25 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     POOL  FID  IID   PHE  CHR                 SNP1                SNP2  \\\n",
       "0      S1  CON   20  4:16    7    chr7:65477373:C:A   chr7:66044271:T:C   \n",
       "1      S4  CON   14  3:11    6    chr6:61118955:C:T   chr6:61867241:T:C   \n",
       "2      S5  CON   13   4:9    8    chr8:46622494:A:C   chr8:47646564:A:G   \n",
       "3      S7  CON   12  2:10   11   chr11:49155393:A:G  chr11:49489304:A:C   \n",
       "4     S11  CON   11   2:9   11  chr11:48022207:TA:T  chr11:48310586:C:T   \n",
       "..    ...  ...  ...   ...  ...                  ...                 ...   \n",
       "345  S446  CON    2   2:0   20   chr20:59448560:T:C  chr20:61999812:G:A   \n",
       "346  S447  CON    2   1:1   21   chr21:34816191:C:T  chr21:36669561:A:C   \n",
       "347  S448  CON    2   2:0   21   chr21:42781756:G:C  chr21:42954192:C:T   \n",
       "348  S449  CON    2   2:0   21   chr21:42997454:A:G  chr21:46518326:T:C   \n",
       "349  S450  CON    2   2:0   22   chr22:36281354:G:A  chr22:36597041:G:C   \n",
       "\n",
       "          BP1       BP2        KB  ...  NcasesWithROHs  NcontrolsWithROHs  \\\n",
       "0    65477373  66044271   566.899  ...               4                 16   \n",
       "1    61118955  61867241   748.287  ...               3                 11   \n",
       "2    46622494  47646564  1024.071  ...               4                  9   \n",
       "3    49155393  49489304   333.912  ...               2                 10   \n",
       "4    48022207  48310586   288.380  ...               2                  9   \n",
       "..        ...       ...       ...  ...             ...                ...   \n",
       "345  59448560  61999812  2551.253  ...               2                  0   \n",
       "346  34816191  36669561  1853.371  ...               1                  1   \n",
       "347  42781756  42954192   172.437  ...               2                  0   \n",
       "348  42997454  46518326  3520.873  ...               2                  0   \n",
       "349  36281354  36597041   315.688  ...               2                  0   \n",
       "\n",
       "     combinedN         P  total_ROH_count  propCases  propControls  \\\n",
       "0           20  0.477796               20       0.20      0.115108   \n",
       "1           14  0.532834               14       0.15      0.079137   \n",
       "2           13  0.103606               13       0.20      0.064748   \n",
       "3           12  1.000000               12       0.10      0.071942   \n",
       "4           11  0.912685               11       0.10      0.064748   \n",
       "..         ...       ...              ...        ...           ...   \n",
       "345          2  0.007384                2       0.10      0.000000   \n",
       "346          2  0.593965                2       0.05      0.007194   \n",
       "347          2  0.007384                2       0.10      0.000000   \n",
       "348          2  0.007384                2       0.10      0.000000   \n",
       "349          2  0.007384                2       0.10      0.000000   \n",
       "\n",
       "     caseEnriched  passMultiTest      BONF  \n",
       "0               1              0  0.001365  \n",
       "1               1              0  0.001522  \n",
       "2               1              0  0.000296  \n",
       "3               1              0  0.002857  \n",
       "4               1              0  0.002608  \n",
       "..            ...            ...       ...  \n",
       "345             1              0  0.000021  \n",
       "346             1              0  0.001697  \n",
       "347             1              0  0.000021  \n",
       "348             1              0  0.000021  \n",
       "349             1              0  0.000021  \n",
       "\n",
       "[350 rows x 25 columns]"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Read the inbreed case-enriched ROH data from the file\n",
    "ROH = pd.read_csv(f\"{WORK_DIR}/{ANC}_ROH.inbreed_caseEnriched.txt\")\n",
    "\n",
    "# Display the ROH data\n",
    "ROH"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>POOL</th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PHE</th>\n",
       "      <th>CHR</th>\n",
       "      <th>SNP1</th>\n",
       "      <th>SNP2</th>\n",
       "      <th>BP1</th>\n",
       "      <th>BP2</th>\n",
       "      <th>KB</th>\n",
       "      <th>...</th>\n",
       "      <th>NcasesWithROHs</th>\n",
       "      <th>NcontrolsWithROHs</th>\n",
       "      <th>combinedN</th>\n",
       "      <th>P</th>\n",
       "      <th>total_ROH_count</th>\n",
       "      <th>propCases</th>\n",
       "      <th>propControls</th>\n",
       "      <th>caseEnriched</th>\n",
       "      <th>passMultiTest</th>\n",
       "      <th>BONF</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>S77</td>\n",
       "      <td>CON</td>\n",
       "      <td>5</td>\n",
       "      <td>4:1</td>\n",
       "      <td>8</td>\n",
       "      <td>chr8:42230547:G:T</td>\n",
       "      <td>chr8:43554958:C:T</td>\n",
       "      <td>42230547</td>\n",
       "      <td>43554958</td>\n",
       "      <td>1324.412</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>5</td>\n",
       "      <td>8.342658e-05</td>\n",
       "      <td>5</td>\n",
       "      <td>0.20</td>\n",
       "      <td>0.007194</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2.383617e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>S81</td>\n",
       "      <td>CON</td>\n",
       "      <td>5</td>\n",
       "      <td>4:1</td>\n",
       "      <td>9</td>\n",
       "      <td>chr9:79266337:T:A</td>\n",
       "      <td>chr9:80842614:G:A</td>\n",
       "      <td>79266337</td>\n",
       "      <td>80842614</td>\n",
       "      <td>1576.278</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>5</td>\n",
       "      <td>8.342658e-05</td>\n",
       "      <td>5</td>\n",
       "      <td>0.20</td>\n",
       "      <td>0.007194</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2.383617e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>S82</td>\n",
       "      <td>CON</td>\n",
       "      <td>5</td>\n",
       "      <td>5:0</td>\n",
       "      <td>9</td>\n",
       "      <td>chr9:82135547:C:A</td>\n",
       "      <td>chr9:84169832:A:T</td>\n",
       "      <td>82135547</td>\n",
       "      <td>84169832</td>\n",
       "      <td>2034.286</td>\n",
       "      <td>...</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>1.128733e-07</td>\n",
       "      <td>5</td>\n",
       "      <td>0.25</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>3.224952e-10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>S88</td>\n",
       "      <td>CON</td>\n",
       "      <td>5</td>\n",
       "      <td>4:1</td>\n",
       "      <td>17</td>\n",
       "      <td>chr17:20558851:A:C</td>\n",
       "      <td>chr17:20936047:G:A</td>\n",
       "      <td>20558851</td>\n",
       "      <td>20936047</td>\n",
       "      <td>377.197</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>5</td>\n",
       "      <td>8.342658e-05</td>\n",
       "      <td>5</td>\n",
       "      <td>0.20</td>\n",
       "      <td>0.007194</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2.383617e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>81</th>\n",
       "      <td>S121</td>\n",
       "      <td>CON</td>\n",
       "      <td>4</td>\n",
       "      <td>4:0</td>\n",
       "      <td>10</td>\n",
       "      <td>chr10:24128513:T:C</td>\n",
       "      <td>chr10:25123571:A:G</td>\n",
       "      <td>24128513</td>\n",
       "      <td>25123571</td>\n",
       "      <td>995.059</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>4.725875e-06</td>\n",
       "      <td>4</td>\n",
       "      <td>0.20</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1.350250e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82</th>\n",
       "      <td>S122</td>\n",
       "      <td>CON</td>\n",
       "      <td>4</td>\n",
       "      <td>4:0</td>\n",
       "      <td>10</td>\n",
       "      <td>chr10:25129805:G:C</td>\n",
       "      <td>chr10:25475212:G:A</td>\n",
       "      <td>25129805</td>\n",
       "      <td>25475212</td>\n",
       "      <td>345.408</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>4.725875e-06</td>\n",
       "      <td>4</td>\n",
       "      <td>0.20</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1.350250e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>94</th>\n",
       "      <td>S136</td>\n",
       "      <td>CON</td>\n",
       "      <td>4</td>\n",
       "      <td>4:0</td>\n",
       "      <td>17</td>\n",
       "      <td>chr17:9973671:G:C</td>\n",
       "      <td>chr17:10259230:T:C</td>\n",
       "      <td>9973671</td>\n",
       "      <td>10259230</td>\n",
       "      <td>285.560</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>4.725875e-06</td>\n",
       "      <td>4</td>\n",
       "      <td>0.20</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1.350250e-08</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>7 rows × 25 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    POOL  FID  IID  PHE  CHR                SNP1                SNP2  \\\n",
       "46   S77  CON    5  4:1    8   chr8:42230547:G:T   chr8:43554958:C:T   \n",
       "50   S81  CON    5  4:1    9   chr9:79266337:T:A   chr9:80842614:G:A   \n",
       "51   S82  CON    5  5:0    9   chr9:82135547:C:A   chr9:84169832:A:T   \n",
       "57   S88  CON    5  4:1   17  chr17:20558851:A:C  chr17:20936047:G:A   \n",
       "81  S121  CON    4  4:0   10  chr10:24128513:T:C  chr10:25123571:A:G   \n",
       "82  S122  CON    4  4:0   10  chr10:25129805:G:C  chr10:25475212:G:A   \n",
       "94  S136  CON    4  4:0   17   chr17:9973671:G:C  chr17:10259230:T:C   \n",
       "\n",
       "         BP1       BP2        KB  ...  NcasesWithROHs  NcontrolsWithROHs  \\\n",
       "46  42230547  43554958  1324.412  ...               4                  1   \n",
       "50  79266337  80842614  1576.278  ...               4                  1   \n",
       "51  82135547  84169832  2034.286  ...               5                  0   \n",
       "57  20558851  20936047   377.197  ...               4                  1   \n",
       "81  24128513  25123571   995.059  ...               4                  0   \n",
       "82  25129805  25475212   345.408  ...               4                  0   \n",
       "94   9973671  10259230   285.560  ...               4                  0   \n",
       "\n",
       "    combinedN             P  total_ROH_count  propCases  propControls  \\\n",
       "46          5  8.342658e-05                5       0.20      0.007194   \n",
       "50          5  8.342658e-05                5       0.20      0.007194   \n",
       "51          5  1.128733e-07                5       0.25      0.000000   \n",
       "57          5  8.342658e-05                5       0.20      0.007194   \n",
       "81          4  4.725875e-06                4       0.20      0.000000   \n",
       "82          4  4.725875e-06                4       0.20      0.000000   \n",
       "94          4  4.725875e-06                4       0.20      0.000000   \n",
       "\n",
       "    caseEnriched  passMultiTest          BONF  \n",
       "46             1              1  2.383617e-07  \n",
       "50             1              1  2.383617e-07  \n",
       "51             1              1  3.224952e-10  \n",
       "57             1              1  2.383617e-07  \n",
       "81             1              1  1.350250e-08  \n",
       "82             1              1  1.350250e-08  \n",
       "94             1              1  1.350250e-08  \n",
       "\n",
       "[7 rows x 25 columns]"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Filter the ROH data for entries that passed the multiple testing correction\n",
    "filtered_ROH = ROH.loc[ROH['passMultiTest'] == 1]\n",
    "\n",
    "# Display the filtered result\n",
    "filtered_ROH"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "bash: line 4: cd: /home/jupyter/Release6_{ANC}: No such file or directory\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of ROH: 406\n",
      "Number of ROH enriched in Cases: 350\n",
      "Number of ROH that Pass Bonferroni: 0\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "# Set working directory and ancestry code\n",
    "WORK_DIR='/home/jupyter/Release6_{ANC}'\n",
    "cd $WORK_DIR\n",
    "ANC=AMR\n",
    "\n",
    "# Count and display various ROH statistics\n",
    "{\n",
    "    printf \"Number of ROH: \"\n",
    "    wc -l < gp2_${ANC}_r6_inbreed.hom.overlap.filtered\n",
    "\n",
    "    printf \"Number of ROH enriched in Cases: \"\n",
    "    sed 1d < ${ANC}_ROH.inbreed_caseEnriched.txt | wc -l\n",
    "\n",
    "    printf \"Number of ROH that Pass Bonferroni: \"\n",
    "    cut -d',' -f26 < ${ANC}_ROH.inbreed_caseEnriched.txt | sed 1d | { grep -c '1' || true; }\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "bash: line 4: cd: /home/jupyter/Release6_{ANC}: No such file or directory\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of ROH that Pass Bonferroni: 0\n",
      "Rows of ROH that Pass Bonferroni (including header):\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "# Set working directory and ancestry code\n",
    "WORK_DIR='/home/jupyter/Release6_{ANC}'\n",
    "cd $WORK_DIR\n",
    "ANC=AMR\n",
    "\n",
    "# Count and filter ROHs that pass Bonferroni correction\n",
    "{\n",
    "    # Count the number of ROHs that pass Bonferroni correction\n",
    "    printf \"Number of ROH that Pass Bonferroni: \"\n",
    "    cut -d',' -f26 < ${ANC}_ROH.inbreed_caseEnriched.txt | sed 1d | { grep -c '1' || true; }\n",
    "\n",
    "    # Output the rows of ROHs that pass Bonferroni correction into a new CSV file\n",
    "    printf \"Rows of ROH that Pass Bonferroni (including header):\\n\"\n",
    "    head -n 1 ${ANC}_ROH.inbreed_caseEnriched.txt > ${ANC}_ROH_Pass_Bonferroni.csv\n",
    "    awk -F',' '$26 == 1' < ${ANC}_ROH.inbreed_caseEnriched.txt >> ${ANC}_ROH_Pass_Bonferroni.csv\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>POOL</th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PHE</th>\n",
       "      <th>CHR</th>\n",
       "      <th>SNP1</th>\n",
       "      <th>SNP2</th>\n",
       "      <th>BP1</th>\n",
       "      <th>BP2</th>\n",
       "      <th>KB</th>\n",
       "      <th>...</th>\n",
       "      <th>NcasesWithROHs</th>\n",
       "      <th>NcontrolsWithROHs</th>\n",
       "      <th>combinedN</th>\n",
       "      <th>P</th>\n",
       "      <th>total_ROH_count</th>\n",
       "      <th>propCases</th>\n",
       "      <th>propControls</th>\n",
       "      <th>caseEnriched</th>\n",
       "      <th>passMultiTest</th>\n",
       "      <th>BONF</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>0 rows × 25 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [POOL, FID, IID, PHE, CHR, SNP1, SNP2, BP1, BP2, KB, NSNP, NSIM, GRP, case, control, NcasesWithROHs, NcontrolsWithROHs, combinedN, P, total_ROH_count, propCases, propControls, caseEnriched, passMultiTest, BONF]\n",
       "Index: []\n",
       "\n",
       "[0 rows x 25 columns]"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Read the ROH data that passed Bonferroni correction\n",
    "roh_pass_bonferroni = pd.read_csv(f\"{WORK_DIR}/{ANC}_ROH_Pass_Bonferroni.csv\")\n",
    "\n",
    "# Display the DataFrame\n",
    "roh_pass_bonferroni"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### In PD genes\n",
    "\n",
    "Focus on ROH that overlap with genes known to be involved in PD. This may help in identifying potential pathogenic regions."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "bash: line 4: cd: /home/jupyter/Release6_{ANC}: No such file or directory\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PD gene: RBX1\n",
      "PD gene: LOC100131289\n",
      "PD gene: RSL24D1P1\n",
      "PD gene: TMEM163\n",
      "PD gene: TMEM163\n",
      "PD gene: SPG11\n",
      "PD gene: TRIM40\n",
      "PD gene: CHCHD2\n",
      "PD gene: RNF141\n",
      "PD gene: RNF141\n",
      "PD gene: HRNR\n",
      "PD gene: ITPKB\n",
      "PD gene: ITPKB\n",
      "PD gene: ITPKB\n",
      "PD gene: MED12L\n",
      "PD gene: SPTSSB\n",
      "PD gene: SPTSSB\n",
      "PD gene: MCCC1\n",
      "PD gene: MCCC1\n",
      "PD gene: MCCC1\n",
      "PD gene: EIF4G1\n",
      "PD gene: ELOVL7\n",
      "PD gene: PARK2\n",
      "PD gene: GALNT17\n",
      "PD gene: ITGA8\n",
      "PD gene: ITGA8\n",
      "PD gene: FAM171A1\n",
      "PD gene: GXYLT1\n",
      "PD gene: SCAF11\n",
      "PD gene: SLC38A1\n",
      "PD gene: USP8\n",
      "PD gene: POLG1\n",
      "PD gene: CD19\n",
      "PD gene: SETD1A\n",
      "PD gene: SETD1A\n",
      "PD gene: ZNF646\n",
      "PD gene: KAT8\n",
      "PD gene: ATP13A2\n",
      "PD gene: PINK1\n",
      "PD gene: KCNS3\n",
      "PD gene: KCNS3\n",
      "PD gene: KPNA1\n",
      "PD gene: KPNA1\n",
      "PD gene: BST1\n",
      "PD gene: BST1\n",
      "PD gene: FAM200B\n",
      "PD gene: CD38\n",
      "PD gene: SNCA\n",
      "PD gene: AC097478.1\n",
      "PD gene: AC097478.1\n",
      "PD gene: SNCA\n",
      "PD gene: ANK2\n",
      "PD gene: CAMK2D\n",
      "PD gene: CAMK2D\n",
      "PD gene: CAMK2D\n",
      "PD gene: ELOVL7\n",
      "PD gene: ELOVL7\n",
      "PD gene: SLC44A4\n",
      "PD gene: HLA-DRB5\n",
      "PD gene:  AHR\n",
      "PD gene: HLA-DRB6\n",
      "PD gene: HLA-DQA1\n",
      "PD gene: RIMS1\n",
      "PD gene: RIMS1\n",
      "PD gene: RPS12\n",
      "PD gene: HMGB1P13\n",
      "PD gene: FAM49B\n",
      "PD gene: FAM49B\n",
      "PD gene: RNF141\n",
      "PD gene: HIP1R\n",
      "PD gene: HIP1R\n",
      "PD gene: OGFOD2\n",
      "PD gene: MBNL2\n",
      "PD gene: MBNL2\n",
      "PD gene: MBNL2\n",
      "PD gene: MIPOL1\n",
      "PD gene: MIPOL1\n",
      "PD gene: RPS6KL1\n",
      "PD gene: RPS6KL1\n",
      "PD gene: RPS6KL1\n",
      "PD gene: FA2H\n",
      "PD gene: NSF\n",
      "PD gene: WNT3\n",
      "PD gene: WNT3\n",
      "PD gene: ARHGAP27\n",
      "PD gene: CRHR1\n",
      "PD gene: SPPL2C\n",
      "PD gene: STH\n",
      "PD gene: KANSL1\n",
      "PD gene: SYT4\n",
      "PD gene: DDRGK1\n",
      "PD gene: DDRGK1\n",
      "PD gene: PANK2\n",
      "PD gene: CRLS1\n",
      "PD gene: PSMF1\n",
      "PD gene: DYRK1A\n",
      "PD gene: PDXK\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "# Set working directory and ancestry code\n",
    "WORK_DIR='/home/jupyter/Release6_{ANC}'\n",
    "cd $WORK_DIR\n",
    "ANC=AMR\n",
    "\n",
    "# Filter ROH data for 'CON' and clean non-CON/UNION rows\n",
    "grep 'CON' gp2_${ANC}_r6_inbreed.hom.overlap.filtered | awk -F' ' '{if($4 != \"-9\")print}' > ${ANC}_ROH.hom.overlap.inbreed.CON.txt\n",
    "grep -v 'CON' gp2_${ANC}_r6_inbreed.hom.overlap.filtered | grep -v 'UNION' | awk -F' ' '{if($4 != \"-9\")print}' | grep -v 'NA' > ${ANC}_ROH.hom.overlap.inbreed.clean.txt\n",
    "\n",
    "# Run the Perl script to map ROH data with gene list\n",
    "perl post_plink_ROH_mapping.pl ${ANC}_ROH.hom.overlap.inbreed.clean.txt ${ANC}_ROH.hom.overlap.inbreed.CON.txt geneList.txt ${ANC}_ROH.hom.overlap.inbreed.CON.PDgenes.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Classes ‘data.table’ and 'data.frame':\t46 obs. of  14 variables:\n",
      " $ POOL           : chr  \"S113\" \"S123\" \"S150\" \"S154\" ...\n",
      " $ num_subjects   : int  4 4 3 3 3 3 3 3 3 3 ...\n",
      " $ case_to_control: chr  \"2:2\" \"3:1\" \"2:1\" \"2:1\" ...\n",
      " $ ratio          : chr  \"1.0\" \"3.0\" \"2.0\" \"2.0\" ...\n",
      " $ CHR            : chr  \"chr7\" \"chr11\" \"chr1\" \"chr1\" ...\n",
      " $ SNP1           : chr  \"chr7:55356967:G:A\" \"chr11:7463094:A:C\" \"chr1:151860425:A:C\" \"chr1:225993955:A:G\" ...\n",
      " $ SNP2           : chr  \"chr7:56922707:C:T\" \"chr11:10351412:C:T\" \"chr1:152618871:G:A\" \"chr1:230304061:A:G\" ...\n",
      " $ BP1            : int  55356967 7463094 151860425 225993955 149500048 160234491 182701608 59539026 160712093 70645942 ...\n",
      " $ BP2            : int  56922707 10351412 152618871 230304061 150884172 161775573 184390899 60257915 161123166 73741147 ...\n",
      " $ KB             : num  1566 2888 758 4310 1384 ...\n",
      " $ NSNP           : int  2408 4921 792 6747 2094 2049 2262 859 649 4115 ...\n",
      " $ PD gene        : chr  \"CHCHD2\" \"RNF141 RNF141\" \"HRNR\" \"ITPKB ITPKB ITPKB\" ...\n",
      " $ cases_IDs      : int  0 0 0 0 0 0 0 0 0 0 ...\n",
      " $ control_IDs    : int  0 0 0 0 0 0 0 0 0 0 ...\n",
      " - attr(*, \".internal.selfref\")=<externalptr> \n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "\n",
    "# Load the ROH data from the CON PD genes file\n",
    "ROH <- fread(paste0(ANC, \"_ROH.hom.overlap.inbreed.CON.PDgenes.txt\"), header = TRUE)\n",
    "\n",
    "# Display the structure of the ROH data\n",
    "str(ROH)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: grep \"are cases and\" /home/jupyter/Release6_AMR/gp2_AMR_r6_inbreed.log\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Among remaining phenotypes, 20 are cases and 139 are controls.\r\n"
     ]
    }
   ],
   "source": [
    "# Use this log data as input for the next two steps\n",
    "shell_do(f'grep \"are cases and\" {WORK_DIR}/gp2_{ANC}_r6_inbreed.log')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Classes ‘data.table’ and 'data.frame':\t46 obs. of  21 variables:\n",
      " $ POOL             : chr  \"S113\" \"S123\" \"S150\" \"S154\" ...\n",
      " $ num_subjects     : int  4 4 3 3 3 3 3 3 3 3 ...\n",
      " $ case_to_control  : chr  \"2:2\" \"3:1\" \"2:1\" \"2:1\" ...\n",
      " $ ratio            : chr  \"1.0\" \"3.0\" \"2.0\" \"2.0\" ...\n",
      " $ CHR              : chr  \"chr7\" \"chr11\" \"chr1\" \"chr1\" ...\n",
      " $ SNP1             : chr  \"chr7:55356967:G:A\" \"chr11:7463094:A:C\" \"chr1:151860425:A:C\" \"chr1:225993955:A:G\" ...\n",
      " $ SNP2             : chr  \"chr7:56922707:C:T\" \"chr11:10351412:C:T\" \"chr1:152618871:G:A\" \"chr1:230304061:A:G\" ...\n",
      " $ BP1              : int  55356967 7463094 151860425 225993955 149500048 160234491 182701608 59539026 160712093 70645942 ...\n",
      " $ BP2              : int  56922707 10351412 152618871 230304061 150884172 161775573 184390899 60257915 161123166 73741147 ...\n",
      " $ KB               : num  1566 2888 758 4310 1384 ...\n",
      " $ NSNP             : int  2408 4921 792 6747 2094 2049 2262 859 649 4115 ...\n",
      " $ PD gene          : chr  \"CHCHD2\" \"RNF141 RNF141\" \"HRNR\" \"ITPKB ITPKB ITPKB\" ...\n",
      " $ cases_IDs        : int  0 0 0 0 0 0 0 0 0 0 ...\n",
      " $ control_IDs      : int  0 0 0 0 0 0 0 0 0 0 ...\n",
      " $ case             : chr  \"2\" \"3\" \"2\" \"2\" ...\n",
      " $ control          : chr  \"2\" \"1\" \"1\" \"1\" ...\n",
      " $ NcasesWithROHs   : num  2 3 2 2 1 0 2 2 2 3 ...\n",
      " $ NcontrolsWithROHs: num  2 1 1 1 2 3 1 1 1 0 ...\n",
      " $ combinedN        : num  4 4 3 3 3 3 3 3 3 3 ...\n",
      " $ P                : num  0.12793 0.00229 0.04846 0.04846 0.82932 ...\n",
      " $ total_ROH_count  : num  4 4 3 3 3 3 3 3 3 3 ...\n",
      " - attr(*, \".internal.selfref\")=<externalptr> \n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "In addition: There were 46 warnings (use warnings() to see them)\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%R\n",
    "\n",
    "# Split the 'case_to_control' column into case and control components\n",
    "ROH$case <- ldply(strsplit(as.character(ROH$case_to_control), split = \":\"))[[1]]\n",
    "ROH$control <- ldply(strsplit(as.character(ROH$case_to_control), split = \":\"))[[2]]\n",
    "\n",
    "# Convert the case and control components to numeric\n",
    "ROH$NcasesWithROHs <- as.numeric(ROH$case)\n",
    "ROH$NcontrolsWithROHs <- as.numeric(ROH$control)\n",
    "\n",
    "# Calculate total counts for cases, controls, and combined\n",
    "ROH$combinedN <- ROH$NcasesWithROHs + ROH$NcontrolsWithROHs\n",
    "\n",
    "# Initialize the 'P' column for p-values\n",
    "ROH$P <- NA\n",
    "\n",
    "# Perform proportion tests for each row\n",
    "for (i in seq_len(nrow(ROH))) {\n",
    "  thisP <- prop.test(x = c(ROH$NcasesWithROHs[i], ROH$NcontrolsWithROHs[i]), \n",
    "                     n = c(20, 139))  # n = (cases, controls)\n",
    "  ROH$P[i] <- thisP$p.value\n",
    "}\n",
    "\n",
    "# Calculate the total ROH count for cases and controls\n",
    "ROH$total_ROH_count <- ROH$NcasesWithROHs + ROH$NcontrolsWithROHs\n",
    "\n",
    "# Display the structure of the ROH data\n",
    "str(ROH)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. \n",
      "0.00000 0.00000 0.00000 0.05263 0.00000 1.00000 \n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "\n",
    "# Calculate proportions of cases and controls with ROH\n",
    "ROH$propCases <- ROH$NcasesWithROHs / 20  # Number of cases\n",
    "ROH$propControls <- ROH$NcontrolsWithROHs / 139  # Number of controls\n",
    "\n",
    "# Identify case-enriched ROHs\n",
    "ROH$caseEnriched <- ifelse(ROH$propCases > ROH$propControls, 1, 0)\n",
    "\n",
    "# Subset ROHs where total ROH count >= 1 and case-enriched\n",
    "ROH_subsetted <- subset(ROH, total_ROH_count >= 1 & caseEnriched == 1)\n",
    "\n",
    "# Apply Bonferroni correction\n",
    "Ngenes <- length(ROH_subsetted$PD)  # Number of genes in the subset\n",
    "ROH_subsetted$passMultiTest <- ifelse(ROH_subsetted$P <= (0.05 / Ngenes), 1, 0)\n",
    "ROH_subsetted$BONF <- ROH_subsetted$P / Ngenes\n",
    "\n",
    "# Subset and show the results that pass multiple testing correction\n",
    "significant_ROH <- subset(ROH_subsetted, passMultiTest == 1)\n",
    "\n",
    "# Apply Bonferroni correction to the full dataset\n",
    "ROH$BONF <- ROH$P / Ngenes\n",
    "\n",
    "# Display a summary of the multi-test results\n",
    "summary(ROH_subsetted$passMultiTest)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R\n",
    "\n",
    "# Write the subsetted ROH data to text files\n",
    "fwrite(ROH_subsetted, paste0(ANC, \"_ROH.inbreed_PDgenes_subsetted.txt\"))\n",
    "fwrite(ROH_subsetted, paste0(ANC, \"_ROH.inbreed_PDgenes_caseEnriched.txt\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>POOL</th>\n",
       "      <th>num_subjects</th>\n",
       "      <th>case_to_control</th>\n",
       "      <th>ratio</th>\n",
       "      <th>CHR</th>\n",
       "      <th>SNP1</th>\n",
       "      <th>SNP2</th>\n",
       "      <th>BP1</th>\n",
       "      <th>BP2</th>\n",
       "      <th>KB</th>\n",
       "      <th>...</th>\n",
       "      <th>NcasesWithROHs</th>\n",
       "      <th>NcontrolsWithROHs</th>\n",
       "      <th>combinedN</th>\n",
       "      <th>P</th>\n",
       "      <th>total_ROH_count</th>\n",
       "      <th>propCases</th>\n",
       "      <th>propControls</th>\n",
       "      <th>caseEnriched</th>\n",
       "      <th>passMultiTest</th>\n",
       "      <th>BONF</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>S113</td>\n",
       "      <td>4</td>\n",
       "      <td>2:2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr7</td>\n",
       "      <td>chr7:55356967:G:A</td>\n",
       "      <td>chr7:56922707:C:T</td>\n",
       "      <td>55356967</td>\n",
       "      <td>56922707</td>\n",
       "      <td>1565.741</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td>0.127926</td>\n",
       "      <td>4</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.014388</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.003366</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>S123</td>\n",
       "      <td>4</td>\n",
       "      <td>3:1</td>\n",
       "      <td>3.0</td>\n",
       "      <td>chr11</td>\n",
       "      <td>chr11:7463094:A:C</td>\n",
       "      <td>chr11:10351412:C:T</td>\n",
       "      <td>7463094</td>\n",
       "      <td>10351412</td>\n",
       "      <td>2888.319</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>0.002292</td>\n",
       "      <td>4</td>\n",
       "      <td>0.15</td>\n",
       "      <td>0.007194</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000060</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>S150</td>\n",
       "      <td>3</td>\n",
       "      <td>2:1</td>\n",
       "      <td>2.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>chr1:151860425:A:C</td>\n",
       "      <td>chr1:152618871:G:A</td>\n",
       "      <td>151860425</td>\n",
       "      <td>152618871</td>\n",
       "      <td>758.447</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>0.048462</td>\n",
       "      <td>3</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.007194</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.001275</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>S154</td>\n",
       "      <td>3</td>\n",
       "      <td>2:1</td>\n",
       "      <td>2.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>chr1:225993955:A:G</td>\n",
       "      <td>chr1:230304061:A:G</td>\n",
       "      <td>225993955</td>\n",
       "      <td>230304061</td>\n",
       "      <td>4310.107</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>0.048462</td>\n",
       "      <td>3</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.007194</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.001275</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>S174</td>\n",
       "      <td>3</td>\n",
       "      <td>1:2</td>\n",
       "      <td>0.5</td>\n",
       "      <td>chr3</td>\n",
       "      <td>chr3:149500048:A:G</td>\n",
       "      <td>chr3:150884172:G:C</td>\n",
       "      <td>149500048</td>\n",
       "      <td>150884172</td>\n",
       "      <td>1384.125</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>0.829323</td>\n",
       "      <td>3</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.014388</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.021824</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>S177</td>\n",
       "      <td>3</td>\n",
       "      <td>2:1</td>\n",
       "      <td>2.0</td>\n",
       "      <td>chr3</td>\n",
       "      <td>chr3:182701608:C:T</td>\n",
       "      <td>chr3:184390899:T:G</td>\n",
       "      <td>182701608</td>\n",
       "      <td>184390899</td>\n",
       "      <td>1689.292</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>0.048462</td>\n",
       "      <td>3</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.007194</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.001275</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>S184</td>\n",
       "      <td>3</td>\n",
       "      <td>2:1</td>\n",
       "      <td>2.0</td>\n",
       "      <td>chr5</td>\n",
       "      <td>chr5:59539026:G:A</td>\n",
       "      <td>chr5:60257915:T:G</td>\n",
       "      <td>59539026</td>\n",
       "      <td>60257915</td>\n",
       "      <td>718.890</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>0.048462</td>\n",
       "      <td>3</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.007194</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.001275</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>S195</td>\n",
       "      <td>3</td>\n",
       "      <td>2:1</td>\n",
       "      <td>2.0</td>\n",
       "      <td>chr6</td>\n",
       "      <td>chr6:160712093:G:C</td>\n",
       "      <td>chr6:161123166:A:C</td>\n",
       "      <td>160712093</td>\n",
       "      <td>161123166</td>\n",
       "      <td>411.074</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>0.048462</td>\n",
       "      <td>3</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.007194</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.001275</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>S200</td>\n",
       "      <td>3</td>\n",
       "      <td>3:0</td>\n",
       "      <td>3:0</td>\n",
       "      <td>chr7</td>\n",
       "      <td>chr7:70645942:CT:C</td>\n",
       "      <td>chr7:73741147:T:A</td>\n",
       "      <td>70645942</td>\n",
       "      <td>73741147</td>\n",
       "      <td>3095.206</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0.000191</td>\n",
       "      <td>3</td>\n",
       "      <td>0.15</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>S211</td>\n",
       "      <td>3</td>\n",
       "      <td>3:0</td>\n",
       "      <td>3:0</td>\n",
       "      <td>chr10</td>\n",
       "      <td>chr10:15180551:G:C</td>\n",
       "      <td>chr10:15923173:ACTTAT:A</td>\n",
       "      <td>15180551</td>\n",
       "      <td>15923173</td>\n",
       "      <td>742.623</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0.000191</td>\n",
       "      <td>3</td>\n",
       "      <td>0.15</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>S232</td>\n",
       "      <td>3</td>\n",
       "      <td>1:2</td>\n",
       "      <td>0.5</td>\n",
       "      <td>chr12</td>\n",
       "      <td>chr12:42690715:CA:C</td>\n",
       "      <td>chr12:43689145:T:G</td>\n",
       "      <td>42690715</td>\n",
       "      <td>43689145</td>\n",
       "      <td>998.431</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>0.829323</td>\n",
       "      <td>3</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.014388</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.021824</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>S233</td>\n",
       "      <td>3</td>\n",
       "      <td>2:1</td>\n",
       "      <td>2.0</td>\n",
       "      <td>chr12</td>\n",
       "      <td>chr12:45654445:T:C</td>\n",
       "      <td>chr12:46179193:G:T</td>\n",
       "      <td>45654445</td>\n",
       "      <td>46179193</td>\n",
       "      <td>524.749</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>0.048462</td>\n",
       "      <td>3</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.007194</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.001275</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>S247</td>\n",
       "      <td>3</td>\n",
       "      <td>1:2</td>\n",
       "      <td>0.5</td>\n",
       "      <td>chr15</td>\n",
       "      <td>chr15:50316658:C:G</td>\n",
       "      <td>chr15:51079222:C:G</td>\n",
       "      <td>50316658</td>\n",
       "      <td>51079222</td>\n",
       "      <td>762.565</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>0.829323</td>\n",
       "      <td>3</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.014388</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.021824</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>S249</td>\n",
       "      <td>3</td>\n",
       "      <td>2:1</td>\n",
       "      <td>2.0</td>\n",
       "      <td>chr15</td>\n",
       "      <td>chr15:88834478:C:T</td>\n",
       "      <td>chr15:89840813:C:T</td>\n",
       "      <td>88834478</td>\n",
       "      <td>89840813</td>\n",
       "      <td>1006.336</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>0.048462</td>\n",
       "      <td>3</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.007194</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.001275</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>S251</td>\n",
       "      <td>3</td>\n",
       "      <td>1:2</td>\n",
       "      <td>0.5</td>\n",
       "      <td>chr16</td>\n",
       "      <td>chr16:29570828:C:G</td>\n",
       "      <td>chr16:31132672:C:T</td>\n",
       "      <td>29570828</td>\n",
       "      <td>31132672</td>\n",
       "      <td>1561.845</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>0.829323</td>\n",
       "      <td>3</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.014388</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.021824</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>S269</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>2:0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>chr1:15217870:A:C</td>\n",
       "      <td>chr1:17354782:T:C</td>\n",
       "      <td>15217870</td>\n",
       "      <td>17354782</td>\n",
       "      <td>2136.913</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.007384</td>\n",
       "      <td>2</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000194</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>S270</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>2:0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>chr1:19722592:T:G</td>\n",
       "      <td>chr1:21390789:T:TAAGA</td>\n",
       "      <td>19722592</td>\n",
       "      <td>21390789</td>\n",
       "      <td>1668.198</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.007384</td>\n",
       "      <td>2</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000194</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>S309</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>2:0</td>\n",
       "      <td>chr3</td>\n",
       "      <td>chr3:119390456:CATT:C</td>\n",
       "      <td>chr3:122409672:G:C</td>\n",
       "      <td>119390456</td>\n",
       "      <td>122409672</td>\n",
       "      <td>3019.217</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.007384</td>\n",
       "      <td>2</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000194</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>S317</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>2:0</td>\n",
       "      <td>chr4</td>\n",
       "      <td>chr4:14317177:T:C</td>\n",
       "      <td>chr4:15857239:A:T</td>\n",
       "      <td>14317177</td>\n",
       "      <td>15857239</td>\n",
       "      <td>1540.063</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.007384</td>\n",
       "      <td>2</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000194</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>S320</td>\n",
       "      <td>2</td>\n",
       "      <td>1:1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr4</td>\n",
       "      <td>chr4:89451624:G:T</td>\n",
       "      <td>chr4:91375869:T:C</td>\n",
       "      <td>89451624</td>\n",
       "      <td>91375869</td>\n",
       "      <td>1924.246</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0.593965</td>\n",
       "      <td>2</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.007194</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.015631</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>S324</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>2:0</td>\n",
       "      <td>chr4</td>\n",
       "      <td>chr4:111598914:T:G</td>\n",
       "      <td>chr4:116520324:T:C</td>\n",
       "      <td>111598914</td>\n",
       "      <td>116520324</td>\n",
       "      <td>4921.411</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.007384</td>\n",
       "      <td>2</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000194</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>S335</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>2:0</td>\n",
       "      <td>chr5</td>\n",
       "      <td>chr5:61364574:G:A</td>\n",
       "      <td>chr5:64668940:C:A</td>\n",
       "      <td>61364574</td>\n",
       "      <td>64668940</td>\n",
       "      <td>3304.367</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.007384</td>\n",
       "      <td>2</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000194</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>S350</td>\n",
       "      <td>2</td>\n",
       "      <td>1:1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr6</td>\n",
       "      <td>chr6:66915289:G:A</td>\n",
       "      <td>chr6:72540461:A:G</td>\n",
       "      <td>66915289</td>\n",
       "      <td>72540461</td>\n",
       "      <td>5625.173</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0.593965</td>\n",
       "      <td>2</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.007194</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.015631</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>S374</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>2:0</td>\n",
       "      <td>chr8</td>\n",
       "      <td>chr8:127456670:A:G</td>\n",
       "      <td>chr8:130595401:C:T</td>\n",
       "      <td>127456670</td>\n",
       "      <td>130595401</td>\n",
       "      <td>3138.732</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.007384</td>\n",
       "      <td>2</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000194</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>S394</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>2:0</td>\n",
       "      <td>chr11</td>\n",
       "      <td>chr11:11275717:T:G</td>\n",
       "      <td>chr11:14751126:C:G</td>\n",
       "      <td>11275717</td>\n",
       "      <td>14751126</td>\n",
       "      <td>3475.410</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.007384</td>\n",
       "      <td>2</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000194</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>S407</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>2:0</td>\n",
       "      <td>chr12</td>\n",
       "      <td>chr12:117954153:G:A</td>\n",
       "      <td>chr12:122976415:T:C</td>\n",
       "      <td>117954153</td>\n",
       "      <td>122976415</td>\n",
       "      <td>5022.263</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.007384</td>\n",
       "      <td>2</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000194</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>S410</td>\n",
       "      <td>2</td>\n",
       "      <td>1:1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr13</td>\n",
       "      <td>chr13:95365837:C:T</td>\n",
       "      <td>chr13:96866861:C:T</td>\n",
       "      <td>95365837</td>\n",
       "      <td>96866861</td>\n",
       "      <td>1501.025</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0.593965</td>\n",
       "      <td>2</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.007194</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.015631</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>S411</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>2:0</td>\n",
       "      <td>chr13</td>\n",
       "      <td>chr13:97928612:T:C</td>\n",
       "      <td>chr13:103315700:CAATAA:C</td>\n",
       "      <td>97928612</td>\n",
       "      <td>103315700</td>\n",
       "      <td>5387.089</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.007384</td>\n",
       "      <td>2</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000194</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>S416</td>\n",
       "      <td>2</td>\n",
       "      <td>1:1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr14</td>\n",
       "      <td>chr14:36475264:C:T</td>\n",
       "      <td>chr14:38016982:A:G</td>\n",
       "      <td>36475264</td>\n",
       "      <td>38016982</td>\n",
       "      <td>1541.719</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0.593965</td>\n",
       "      <td>2</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.007194</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.015631</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>S419</td>\n",
       "      <td>2</td>\n",
       "      <td>1:1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr14</td>\n",
       "      <td>chr14:73752664:A:G</td>\n",
       "      <td>chr14:74873430:G:T</td>\n",
       "      <td>73752664</td>\n",
       "      <td>74873430</td>\n",
       "      <td>1120.767</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0.593965</td>\n",
       "      <td>2</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.007194</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.015631</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>S421</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>2:0</td>\n",
       "      <td>chr14</td>\n",
       "      <td>chr14:75410142:C:T</td>\n",
       "      <td>chr14:77919259:A:G</td>\n",
       "      <td>75410142</td>\n",
       "      <td>77919259</td>\n",
       "      <td>2509.118</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.007384</td>\n",
       "      <td>2</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000194</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>S430</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>2:0</td>\n",
       "      <td>chr16</td>\n",
       "      <td>chr16:73533650:C:A</td>\n",
       "      <td>chr16:74879646:G:C</td>\n",
       "      <td>73533650</td>\n",
       "      <td>74879646</td>\n",
       "      <td>1345.997</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.007384</td>\n",
       "      <td>2</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000194</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>S432</td>\n",
       "      <td>2</td>\n",
       "      <td>1:1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr17</td>\n",
       "      <td>chr17:47224732:G:A</td>\n",
       "      <td>chr17:49105435:AAC:A</td>\n",
       "      <td>47224732</td>\n",
       "      <td>49105435</td>\n",
       "      <td>1880.704</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0.593965</td>\n",
       "      <td>2</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.007194</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.015631</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>S438</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>2:0</td>\n",
       "      <td>chr18</td>\n",
       "      <td>chr18:45394185:A:C</td>\n",
       "      <td>chr18:48021976:C:T</td>\n",
       "      <td>45394185</td>\n",
       "      <td>48021976</td>\n",
       "      <td>2627.792</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.007384</td>\n",
       "      <td>2</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000194</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>S442</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>2:0</td>\n",
       "      <td>chr20</td>\n",
       "      <td>chr20:796225:T:G</td>\n",
       "      <td>chr20:5046718:T:C</td>\n",
       "      <td>796225</td>\n",
       "      <td>5046718</td>\n",
       "      <td>4250.494</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.007384</td>\n",
       "      <td>2</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000194</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>S447</td>\n",
       "      <td>2</td>\n",
       "      <td>1:1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr21</td>\n",
       "      <td>chr21:34816191:C:T</td>\n",
       "      <td>chr21:36669561:A:C</td>\n",
       "      <td>34816191</td>\n",
       "      <td>36669561</td>\n",
       "      <td>1853.371</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0.593965</td>\n",
       "      <td>2</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.007194</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.015631</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>S449</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>2:0</td>\n",
       "      <td>chr21</td>\n",
       "      <td>chr21:42997454:A:G</td>\n",
       "      <td>chr21:46518326:T:C</td>\n",
       "      <td>42997454</td>\n",
       "      <td>46518326</td>\n",
       "      <td>3520.873</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.007384</td>\n",
       "      <td>2</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000194</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>S46</td>\n",
       "      <td>6</td>\n",
       "      <td>2:4</td>\n",
       "      <td>0.5</td>\n",
       "      <td>chr2</td>\n",
       "      <td>chr2:134836651:T:C</td>\n",
       "      <td>chr2:136064653:T:C</td>\n",
       "      <td>134836651</td>\n",
       "      <td>136064653</td>\n",
       "      <td>1228.003</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td>6</td>\n",
       "      <td>0.349609</td>\n",
       "      <td>6</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.028777</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.009200</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>38 rows × 26 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    POOL  num_subjects case_to_control ratio    CHR                   SNP1  \\\n",
       "0   S113             4             2:2   1.0   chr7      chr7:55356967:G:A   \n",
       "1   S123             4             3:1   3.0  chr11      chr11:7463094:A:C   \n",
       "2   S150             3             2:1   2.0   chr1     chr1:151860425:A:C   \n",
       "3   S154             3             2:1   2.0   chr1     chr1:225993955:A:G   \n",
       "4   S174             3             1:2   0.5   chr3     chr3:149500048:A:G   \n",
       "5   S177             3             2:1   2.0   chr3     chr3:182701608:C:T   \n",
       "6   S184             3             2:1   2.0   chr5      chr5:59539026:G:A   \n",
       "7   S195             3             2:1   2.0   chr6     chr6:160712093:G:C   \n",
       "8   S200             3             3:0   3:0   chr7     chr7:70645942:CT:C   \n",
       "9   S211             3             3:0   3:0  chr10     chr10:15180551:G:C   \n",
       "10  S232             3             1:2   0.5  chr12    chr12:42690715:CA:C   \n",
       "11  S233             3             2:1   2.0  chr12     chr12:45654445:T:C   \n",
       "12  S247             3             1:2   0.5  chr15     chr15:50316658:C:G   \n",
       "13  S249             3             2:1   2.0  chr15     chr15:88834478:C:T   \n",
       "14  S251             3             1:2   0.5  chr16     chr16:29570828:C:G   \n",
       "15  S269             2             2:0   2:0   chr1      chr1:15217870:A:C   \n",
       "16  S270             2             2:0   2:0   chr1      chr1:19722592:T:G   \n",
       "17  S309             2             2:0   2:0   chr3  chr3:119390456:CATT:C   \n",
       "18  S317             2             2:0   2:0   chr4      chr4:14317177:T:C   \n",
       "19  S320             2             1:1   1.0   chr4      chr4:89451624:G:T   \n",
       "20  S324             2             2:0   2:0   chr4     chr4:111598914:T:G   \n",
       "21  S335             2             2:0   2:0   chr5      chr5:61364574:G:A   \n",
       "22  S350             2             1:1   1.0   chr6      chr6:66915289:G:A   \n",
       "23  S374             2             2:0   2:0   chr8     chr8:127456670:A:G   \n",
       "24  S394             2             2:0   2:0  chr11     chr11:11275717:T:G   \n",
       "25  S407             2             2:0   2:0  chr12    chr12:117954153:G:A   \n",
       "26  S410             2             1:1   1.0  chr13     chr13:95365837:C:T   \n",
       "27  S411             2             2:0   2:0  chr13     chr13:97928612:T:C   \n",
       "28  S416             2             1:1   1.0  chr14     chr14:36475264:C:T   \n",
       "29  S419             2             1:1   1.0  chr14     chr14:73752664:A:G   \n",
       "30  S421             2             2:0   2:0  chr14     chr14:75410142:C:T   \n",
       "31  S430             2             2:0   2:0  chr16     chr16:73533650:C:A   \n",
       "32  S432             2             1:1   1.0  chr17     chr17:47224732:G:A   \n",
       "33  S438             2             2:0   2:0  chr18     chr18:45394185:A:C   \n",
       "34  S442             2             2:0   2:0  chr20       chr20:796225:T:G   \n",
       "35  S447             2             1:1   1.0  chr21     chr21:34816191:C:T   \n",
       "36  S449             2             2:0   2:0  chr21     chr21:42997454:A:G   \n",
       "37   S46             6             2:4   0.5   chr2     chr2:134836651:T:C   \n",
       "\n",
       "                        SNP2        BP1        BP2        KB  ...  \\\n",
       "0          chr7:56922707:C:T   55356967   56922707  1565.741  ...   \n",
       "1         chr11:10351412:C:T    7463094   10351412  2888.319  ...   \n",
       "2         chr1:152618871:G:A  151860425  152618871   758.447  ...   \n",
       "3         chr1:230304061:A:G  225993955  230304061  4310.107  ...   \n",
       "4         chr3:150884172:G:C  149500048  150884172  1384.125  ...   \n",
       "5         chr3:184390899:T:G  182701608  184390899  1689.292  ...   \n",
       "6          chr5:60257915:T:G   59539026   60257915   718.890  ...   \n",
       "7         chr6:161123166:A:C  160712093  161123166   411.074  ...   \n",
       "8          chr7:73741147:T:A   70645942   73741147  3095.206  ...   \n",
       "9    chr10:15923173:ACTTAT:A   15180551   15923173   742.623  ...   \n",
       "10        chr12:43689145:T:G   42690715   43689145   998.431  ...   \n",
       "11        chr12:46179193:G:T   45654445   46179193   524.749  ...   \n",
       "12        chr15:51079222:C:G   50316658   51079222   762.565  ...   \n",
       "13        chr15:89840813:C:T   88834478   89840813  1006.336  ...   \n",
       "14        chr16:31132672:C:T   29570828   31132672  1561.845  ...   \n",
       "15         chr1:17354782:T:C   15217870   17354782  2136.913  ...   \n",
       "16     chr1:21390789:T:TAAGA   19722592   21390789  1668.198  ...   \n",
       "17        chr3:122409672:G:C  119390456  122409672  3019.217  ...   \n",
       "18         chr4:15857239:A:T   14317177   15857239  1540.063  ...   \n",
       "19         chr4:91375869:T:C   89451624   91375869  1924.246  ...   \n",
       "20        chr4:116520324:T:C  111598914  116520324  4921.411  ...   \n",
       "21         chr5:64668940:C:A   61364574   64668940  3304.367  ...   \n",
       "22         chr6:72540461:A:G   66915289   72540461  5625.173  ...   \n",
       "23        chr8:130595401:C:T  127456670  130595401  3138.732  ...   \n",
       "24        chr11:14751126:C:G   11275717   14751126  3475.410  ...   \n",
       "25       chr12:122976415:T:C  117954153  122976415  5022.263  ...   \n",
       "26        chr13:96866861:C:T   95365837   96866861  1501.025  ...   \n",
       "27  chr13:103315700:CAATAA:C   97928612  103315700  5387.089  ...   \n",
       "28        chr14:38016982:A:G   36475264   38016982  1541.719  ...   \n",
       "29        chr14:74873430:G:T   73752664   74873430  1120.767  ...   \n",
       "30        chr14:77919259:A:G   75410142   77919259  2509.118  ...   \n",
       "31        chr16:74879646:G:C   73533650   74879646  1345.997  ...   \n",
       "32      chr17:49105435:AAC:A   47224732   49105435  1880.704  ...   \n",
       "33        chr18:48021976:C:T   45394185   48021976  2627.792  ...   \n",
       "34         chr20:5046718:T:C     796225    5046718  4250.494  ...   \n",
       "35        chr21:36669561:A:C   34816191   36669561  1853.371  ...   \n",
       "36        chr21:46518326:T:C   42997454   46518326  3520.873  ...   \n",
       "37        chr2:136064653:T:C  134836651  136064653  1228.003  ...   \n",
       "\n",
       "    NcasesWithROHs NcontrolsWithROHs  combinedN         P  total_ROH_count  \\\n",
       "0                2                 2          4  0.127926                4   \n",
       "1                3                 1          4  0.002292                4   \n",
       "2                2                 1          3  0.048462                3   \n",
       "3                2                 1          3  0.048462                3   \n",
       "4                1                 2          3  0.829323                3   \n",
       "5                2                 1          3  0.048462                3   \n",
       "6                2                 1          3  0.048462                3   \n",
       "7                2                 1          3  0.048462                3   \n",
       "8                3                 0          3  0.000191                3   \n",
       "9                3                 0          3  0.000191                3   \n",
       "10               1                 2          3  0.829323                3   \n",
       "11               2                 1          3  0.048462                3   \n",
       "12               1                 2          3  0.829323                3   \n",
       "13               2                 1          3  0.048462                3   \n",
       "14               1                 2          3  0.829323                3   \n",
       "15               2                 0          2  0.007384                2   \n",
       "16               2                 0          2  0.007384                2   \n",
       "17               2                 0          2  0.007384                2   \n",
       "18               2                 0          2  0.007384                2   \n",
       "19               1                 1          2  0.593965                2   \n",
       "20               2                 0          2  0.007384                2   \n",
       "21               2                 0          2  0.007384                2   \n",
       "22               1                 1          2  0.593965                2   \n",
       "23               2                 0          2  0.007384                2   \n",
       "24               2                 0          2  0.007384                2   \n",
       "25               2                 0          2  0.007384                2   \n",
       "26               1                 1          2  0.593965                2   \n",
       "27               2                 0          2  0.007384                2   \n",
       "28               1                 1          2  0.593965                2   \n",
       "29               1                 1          2  0.593965                2   \n",
       "30               2                 0          2  0.007384                2   \n",
       "31               2                 0          2  0.007384                2   \n",
       "32               1                 1          2  0.593965                2   \n",
       "33               2                 0          2  0.007384                2   \n",
       "34               2                 0          2  0.007384                2   \n",
       "35               1                 1          2  0.593965                2   \n",
       "36               2                 0          2  0.007384                2   \n",
       "37               2                 4          6  0.349609                6   \n",
       "\n",
       "    propCases  propControls  caseEnriched  passMultiTest      BONF  \n",
       "0        0.10      0.014388             1              0  0.003366  \n",
       "1        0.15      0.007194             1              0  0.000060  \n",
       "2        0.10      0.007194             1              0  0.001275  \n",
       "3        0.10      0.007194             1              0  0.001275  \n",
       "4        0.05      0.014388             1              0  0.021824  \n",
       "5        0.10      0.007194             1              0  0.001275  \n",
       "6        0.10      0.007194             1              0  0.001275  \n",
       "7        0.10      0.007194             1              0  0.001275  \n",
       "8        0.15      0.000000             1              1  0.000005  \n",
       "9        0.15      0.000000             1              1  0.000005  \n",
       "10       0.05      0.014388             1              0  0.021824  \n",
       "11       0.10      0.007194             1              0  0.001275  \n",
       "12       0.05      0.014388             1              0  0.021824  \n",
       "13       0.10      0.007194             1              0  0.001275  \n",
       "14       0.05      0.014388             1              0  0.021824  \n",
       "15       0.10      0.000000             1              0  0.000194  \n",
       "16       0.10      0.000000             1              0  0.000194  \n",
       "17       0.10      0.000000             1              0  0.000194  \n",
       "18       0.10      0.000000             1              0  0.000194  \n",
       "19       0.05      0.007194             1              0  0.015631  \n",
       "20       0.10      0.000000             1              0  0.000194  \n",
       "21       0.10      0.000000             1              0  0.000194  \n",
       "22       0.05      0.007194             1              0  0.015631  \n",
       "23       0.10      0.000000             1              0  0.000194  \n",
       "24       0.10      0.000000             1              0  0.000194  \n",
       "25       0.10      0.000000             1              0  0.000194  \n",
       "26       0.05      0.007194             1              0  0.015631  \n",
       "27       0.10      0.000000             1              0  0.000194  \n",
       "28       0.05      0.007194             1              0  0.015631  \n",
       "29       0.05      0.007194             1              0  0.015631  \n",
       "30       0.10      0.000000             1              0  0.000194  \n",
       "31       0.10      0.000000             1              0  0.000194  \n",
       "32       0.05      0.007194             1              0  0.015631  \n",
       "33       0.10      0.000000             1              0  0.000194  \n",
       "34       0.10      0.000000             1              0  0.000194  \n",
       "35       0.05      0.007194             1              0  0.015631  \n",
       "36       0.10      0.000000             1              0  0.000194  \n",
       "37       0.10      0.028777             1              0  0.009200  \n",
       "\n",
       "[38 rows x 26 columns]"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Read the case-enriched ROH data from the file\n",
    "ROH = pd.read_csv(f'{WORK_DIR}/{ANC}_ROH.inbreed_PDgenes_caseEnriched.txt')\n",
    "\n",
    "# Display the DataFrame\n",
    "ROH"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "bash: line 4: cd: /home/jupyter/Release6_{ANC}: No such file or directory\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of ROH overlapping PD genes: 46\n",
      "Number of ROH enriched in Cases: 38\n",
      "Number of ROH that Pass Bonferroni: 0\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "# Set working directory and ancestry code\n",
    "WORK_DIR='/home/jupyter/Release6_{ANC}'\n",
    "cd $WORK_DIR\n",
    "ANC=AMR\n",
    "\n",
    "# Display various ROH statistics\n",
    "{\n",
    "    # Count the number of ROHs overlapping PD genes\n",
    "    printf \"Number of ROH overlapping PD genes: \"\n",
    "    sed 1d < ${ANC}_ROH.hom.overlap.inbreed.CON.PDgenes.txt | wc -l\n",
    "\n",
    "    # Count the number of ROHs enriched in cases\n",
    "    printf \"Number of ROH enriched in Cases: \"\n",
    "    sed 1d < ${ANC}_ROH.inbreed_PDgenes_caseEnriched.txt | wc -l\n",
    "\n",
    "    # Count the number of ROHs that pass Bonferroni correction\n",
    "    printf \"Number of ROH that Pass Bonferroni: \"\n",
    "    cut -d',' -f27 < ${ANC}_ROH.inbreed_PDgenes_caseEnriched.txt | sed 1d | { grep -c '1' || true; }\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "bash: line 4: cd: /home/jupyter/Release6_{ANC}: No such file or directory\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of ROH that Pass Bonferroni: 0\n",
      "Rows of ROH that Pass Bonferroni (including header):\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "# Set working directory and ancestry code\n",
    "WORK_DIR='/home/jupyter/Release6_{ANC}'\n",
    "cd $WORK_DIR\n",
    "ANC=AMR\n",
    "\n",
    "# Count and filter ROHs that pass Bonferroni correction\n",
    "{\n",
    "    # Count the number of ROHs that pass Bonferroni correction\n",
    "    printf \"Number of ROH that Pass Bonferroni: \"\n",
    "    cut -d',' -f27 < ${ANC}_ROH.inbreed_PDgenes_caseEnriched.txt | sed 1d | { grep -c '1' || true; }\n",
    "\n",
    "    # Extract rows of ROHs that pass Bonferroni correction, including header\n",
    "    printf \"Rows of ROH that Pass Bonferroni (including header):\\n\"\n",
    "    head -n 1 ${ANC}_ROH.inbreed_PDgenes_caseEnriched.txt > ${ANC}_ROH_Pass_Bonferroni_inbred_cases_PD_genes.csv\n",
    "    awk -F',' '$27 == 1' < ${ANC}_ROH.inbreed_PDgenes_caseEnriched.txt >> ${ANC}_ROH_Pass_Bonferroni_inbred_cases_PD_genes.csv\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>POOL</th>\n",
       "      <th>num_subjects</th>\n",
       "      <th>case_to_control</th>\n",
       "      <th>ratio</th>\n",
       "      <th>CHR</th>\n",
       "      <th>SNP1</th>\n",
       "      <th>SNP2</th>\n",
       "      <th>BP1</th>\n",
       "      <th>BP2</th>\n",
       "      <th>KB</th>\n",
       "      <th>...</th>\n",
       "      <th>NcasesWithROHs</th>\n",
       "      <th>NcontrolsWithROHs</th>\n",
       "      <th>combinedN</th>\n",
       "      <th>P</th>\n",
       "      <th>total_ROH_count</th>\n",
       "      <th>propCases</th>\n",
       "      <th>propControls</th>\n",
       "      <th>caseEnriched</th>\n",
       "      <th>passMultiTest</th>\n",
       "      <th>BONF</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>0 rows × 26 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [POOL, num_subjects, case_to_control, ratio, CHR, SNP1, SNP2, BP1, BP2, KB, NSNP, PD gene, cases_IDs, control_IDs, case, control, NcasesWithROHs, NcontrolsWithROHs, combinedN, P, total_ROH_count, propCases, propControls, caseEnriched, passMultiTest, BONF]\n",
       "Index: []\n",
       "\n",
       "[0 rows x 26 columns]"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Read the ROH data that passed Bonferroni correction\n",
    "roh_pass_bonferroni = pd.read_csv(f\"{WORK_DIR}/{ANC}_ROH_Pass_Bonferroni_inbred_cases_PD_genes.csv\")\n",
    "\n",
    "# Display the DataFrame\n",
    "roh_pass_bonferroni"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Find overlaps between EOPD\n",
    "\n",
    "Identify ROH overlaps specifically within EOPD cases to investigate potential shared genetic causes."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Subset EOPD + Controls\n",
    "\n",
    "Retrieve the list of EOPD cases and control samples, and use them to generate PLINK bfiles for further analysis."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R\n",
    "\n",
    "# Load ROH data and clinical data\n",
    "ROH <- fread(paste0(ANC, \"_ROH.hom.indiv\"))[, 2]\n",
    "clinical <- fread(\"master_key_release6_final.csv\")[, c(2, 5, 10)]\n",
    "\n",
    "# Rename clinical data columns\n",
    "colnames(clinical) <- c(\"IID\", \"PHENO\", \"AAO\")\n",
    "\n",
    "# Merge ROH and clinical data by IID\n",
    "merged <- merge(ROH, clinical, by = \"IID\")\n",
    "\n",
    "# Add FID column with a default value of 0\n",
    "merged$FID <- 0\n",
    "\n",
    "# Filter for Early Onset Parkinson's Disease (EOPD) or controls (PHENO == 1)\n",
    "filtered <- subset(merged, (AAO < 45 & PHENO == 2) | PHENO == 1)\n",
    "\n",
    "# Select FID and IID columns for output\n",
    "selected <- select(filtered, c(\"FID\", \"IID\"))\n",
    "\n",
    "# Write the selected samples to a file\n",
    "fwrite(selected, paste0(\"keep_\", ANC, \"_EOPD_sampleset.list\"), sep = '\\t')\n",
    "\n",
    "# Display the selected data\n",
    "selected"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Display the first few lines of the EOPD sampleset list\n",
    "shell_do(f'head {WORK_DIR}/keep_{ANC}_EOPD_sampleset.list')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: /home/jupyter/tools/plink --bfile /home/jupyter/Release6_AMR/chrAll_AMR_QC --keep /home/jupyter/Release6_AMR/keep_AMR_EOPD_sampleset.list --homozyg group --homozyg-density 50 --homozyg-gap 1000 --homozyg-kb 1500 --homozyg-snp 100 --homozyg-window-het 1 --homozyg-window-missing 5 --homozyg-window-snp 50 --homozyg-window-threshold 0.05 --homozyg-match 0.95 --out /home/jupyter/Release6_AMR/gp2_AMR_r6_EOPD\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/Release6_AMR/gp2_AMR_r6_EOPD.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/Release6_AMR/chrAll_AMR_QC\n",
      "  --homozyg group\n",
      "  --homozyg-density 50\n",
      "  --homozyg-gap 1000\n",
      "  --homozyg-kb 1500\n",
      "  --homozyg-match 0.95\n",
      "  --homozyg-snp 100\n",
      "  --homozyg-window-het 1\n",
      "  --homozyg-window-missing 5\n",
      "  --homozyg-window-snp 50\n",
      "  --homozyg-window-threshold 0.05\n",
      "  --keep /home/jupyter/Release6_AMR/keep_AMR_EOPD_sampleset.list\n",
      "  --out /home/jupyter/Release6_AMR/gp2_AMR_r6_EOPD\n",
      "\n",
      "628867 MB RAM detected; reserving 314433 MB for main workspace.\n",
      "4094211 variants loaded from .bim file.\n",
      "524 people (286 males, 238 females) loaded from .fam.\n",
      "502 phenotype values loaded from .fam.\n",
      "--keep: 230 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 230 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.975539.\n",
      "4094211 variants and 230 people pass filters and QC.\n",
      "Among remaining phenotypes, 91 are cases and 139 are controls.\n",
      "--homozyg: Scan complete, found 1185 ROH.\n",
      "Results saved to /home/jupyter/Release6_AMR/gp2_AMR_r6_EOPD.hom +\n",
      "/home/jupyter/Release6_AMR/gp2_AMR_r6_EOPD.hom.indiv +\n",
      "/home/jupyter/Release6_AMR/gp2_AMR_r6_EOPD.hom.summary .\n",
      "414 pools of overlapping ROH present.\n",
      "Determining within-pool allelic similarity... [chromosome 2]4]6]8]9]10111213141516171819202122              done.\n",
      "ROH pool report written to\n",
      "/home/jupyter/Release6_AMR/gp2_AMR_r6_EOPD.hom.overlap .\n"
     ]
    }
   ],
   "source": [
    "# Subset the samples of interest and run homozygosity mapping\n",
    "shell_do(f'/home/jupyter/tools/plink --bfile {WORK_DIR}/chrAll_{ANC}_QC '\n",
    "         f'--keep {WORK_DIR}/keep_{ANC}_EOPD_sampleset.list '\n",
    "         f'--homozyg group '\n",
    "         f'--homozyg-density 50 '\n",
    "         f'--homozyg-gap 1000 '\n",
    "         f'--homozyg-kb 1500 '\n",
    "         f'--homozyg-snp 100 '\n",
    "         f'--homozyg-window-het 1 '\n",
    "         f'--homozyg-window-missing 5 '\n",
    "         f'--homozyg-window-snp 50 '\n",
    "         f'--homozyg-window-threshold 0.05 '\n",
    "         f'--homozyg-match 0.95 '\n",
    "         f'--out {WORK_DIR}/gp2_{ANC}_r6_EOPD')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Check consensus ROH (cROH)\n",
    "\n",
    "Confirm the consensus ROH in the generated sample lists."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>POOL</th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PHE</th>\n",
       "      <th>CHR</th>\n",
       "      <th>SNP1</th>\n",
       "      <th>SNP2</th>\n",
       "      <th>BP1</th>\n",
       "      <th>BP2</th>\n",
       "      <th>KB</th>\n",
       "      <th>NSNP</th>\n",
       "      <th>NSIM</th>\n",
       "      <th>GRP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>S1</td>\n",
       "      <td>CON</td>\n",
       "      <td>23</td>\n",
       "      <td>7:16</td>\n",
       "      <td>7</td>\n",
       "      <td>chr7:65477373:C:A</td>\n",
       "      <td>chr7:65477373:C:A</td>\n",
       "      <td>65477373</td>\n",
       "      <td>65477373</td>\n",
       "      <td>0.001</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>S2</td>\n",
       "      <td>CON</td>\n",
       "      <td>20</td>\n",
       "      <td>6:14</td>\n",
       "      <td>3</td>\n",
       "      <td>chr3:49588749:T:G</td>\n",
       "      <td>chr3:49612962:G:A</td>\n",
       "      <td>49588749</td>\n",
       "      <td>49612962</td>\n",
       "      <td>24.214</td>\n",
       "      <td>23</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67</th>\n",
       "      <td>S3</td>\n",
       "      <td>CON</td>\n",
       "      <td>20</td>\n",
       "      <td>7:13</td>\n",
       "      <td>3</td>\n",
       "      <td>chr3:49855463:A:G</td>\n",
       "      <td>chr3:49870590:G:A</td>\n",
       "      <td>49855463</td>\n",
       "      <td>49870590</td>\n",
       "      <td>15.128</td>\n",
       "      <td>11</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88</th>\n",
       "      <td>S4</td>\n",
       "      <td>CON</td>\n",
       "      <td>19</td>\n",
       "      <td>6:13</td>\n",
       "      <td>3</td>\n",
       "      <td>chr3:49948685:T:C</td>\n",
       "      <td>chr3:50141389:G:A</td>\n",
       "      <td>49948685</td>\n",
       "      <td>50141389</td>\n",
       "      <td>192.705</td>\n",
       "      <td>140</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>108</th>\n",
       "      <td>S5</td>\n",
       "      <td>CON</td>\n",
       "      <td>18</td>\n",
       "      <td>6:12</td>\n",
       "      <td>3</td>\n",
       "      <td>chr3:48466268:T:C</td>\n",
       "      <td>chr3:48544415:A:G</td>\n",
       "      <td>48466268</td>\n",
       "      <td>48544415</td>\n",
       "      <td>78.148</td>\n",
       "      <td>31</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2614</th>\n",
       "      <td>S410</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>1:1</td>\n",
       "      <td>21</td>\n",
       "      <td>chr21:27018589:G:A</td>\n",
       "      <td>chr21:28152094:A:G</td>\n",
       "      <td>27018589</td>\n",
       "      <td>28152094</td>\n",
       "      <td>1133.506</td>\n",
       "      <td>2165</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2618</th>\n",
       "      <td>S411</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>1:1</td>\n",
       "      <td>22</td>\n",
       "      <td>chr22:27779753:G:A</td>\n",
       "      <td>chr22:30214581:C:T</td>\n",
       "      <td>27779753</td>\n",
       "      <td>30214581</td>\n",
       "      <td>2434.829</td>\n",
       "      <td>2994</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2622</th>\n",
       "      <td>S412</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>22</td>\n",
       "      <td>chr22:30216960:A:G</td>\n",
       "      <td>chr22:30244319:A:G</td>\n",
       "      <td>30216960</td>\n",
       "      <td>30244319</td>\n",
       "      <td>27.360</td>\n",
       "      <td>35</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2626</th>\n",
       "      <td>S413</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>1:1</td>\n",
       "      <td>22</td>\n",
       "      <td>chr22:30558084:G:A</td>\n",
       "      <td>chr22:32332418:T:TC</td>\n",
       "      <td>30558084</td>\n",
       "      <td>32332418</td>\n",
       "      <td>1774.335</td>\n",
       "      <td>2502</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2630</th>\n",
       "      <td>S414</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>22</td>\n",
       "      <td>chr22:48192677:C:G</td>\n",
       "      <td>chr22:49350260:T:G</td>\n",
       "      <td>48192677</td>\n",
       "      <td>49350260</td>\n",
       "      <td>1157.584</td>\n",
       "      <td>3492</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>414 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      POOL  FID IID   PHE  CHR                SNP1                 SNP2  \\\n",
       "23      S1  CON  23  7:16    7   chr7:65477373:C:A    chr7:65477373:C:A   \n",
       "45      S2  CON  20  6:14    3   chr3:49588749:T:G    chr3:49612962:G:A   \n",
       "67      S3  CON  20  7:13    3   chr3:49855463:A:G    chr3:49870590:G:A   \n",
       "88      S4  CON  19  6:13    3   chr3:49948685:T:C    chr3:50141389:G:A   \n",
       "108     S5  CON  18  6:12    3   chr3:48466268:T:C    chr3:48544415:A:G   \n",
       "...    ...  ...  ..   ...  ...                 ...                  ...   \n",
       "2614  S410  CON   2   1:1   21  chr21:27018589:G:A   chr21:28152094:A:G   \n",
       "2618  S411  CON   2   1:1   22  chr22:27779753:G:A   chr22:30214581:C:T   \n",
       "2622  S412  CON   2   2:0   22  chr22:30216960:A:G   chr22:30244319:A:G   \n",
       "2626  S413  CON   2   1:1   22  chr22:30558084:G:A  chr22:32332418:T:TC   \n",
       "2630  S414  CON   2   2:0   22  chr22:48192677:C:G   chr22:49350260:T:G   \n",
       "\n",
       "           BP1       BP2        KB  NSNP  NSIM  GRP  \n",
       "23    65477373  65477373     0.001     1   NaN  NaN  \n",
       "45    49588749  49612962    24.214    23   NaN  NaN  \n",
       "67    49855463  49870590    15.128    11   NaN  NaN  \n",
       "88    49948685  50141389   192.705   140   NaN  NaN  \n",
       "108   48466268  48544415    78.148    31   NaN  NaN  \n",
       "...        ...       ...       ...   ...   ...  ...  \n",
       "2614  27018589  28152094  1133.506  2165   NaN  NaN  \n",
       "2618  27779753  30214581  2434.829  2994   NaN  NaN  \n",
       "2622  30216960  30244319    27.360    35   NaN  NaN  \n",
       "2626  30558084  32332418  1774.335  2502   NaN  NaN  \n",
       "2630  48192677  49350260  1157.584  3492   NaN  NaN  \n",
       "\n",
       "[414 rows x 13 columns]"
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Read the inbreeding ROH overlap data\n",
    "inbred_roh = pd.read_csv(f'{WORK_DIR}/gp2_{ANC}_r6_EOPD.hom.overlap', sep='\\s+')\n",
    "\n",
    "# Filter for rows where the FID is 'CON' (consensus)\n",
    "inbred_roh_consense = inbred_roh.loc[inbred_roh['FID'] == 'CON']\n",
    "\n",
    "# Display the filtered data\n",
    "inbred_roh_consense"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>POOL</th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PHE</th>\n",
       "      <th>CHR</th>\n",
       "      <th>SNP1</th>\n",
       "      <th>SNP2</th>\n",
       "      <th>BP1</th>\n",
       "      <th>BP2</th>\n",
       "      <th>KB</th>\n",
       "      <th>NSNP</th>\n",
       "      <th>NSIM</th>\n",
       "      <th>GRP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>88</th>\n",
       "      <td>S4</td>\n",
       "      <td>CON</td>\n",
       "      <td>19</td>\n",
       "      <td>6:13</td>\n",
       "      <td>3</td>\n",
       "      <td>chr3:49948685:T:C</td>\n",
       "      <td>chr3:50141389:G:A</td>\n",
       "      <td>49948685</td>\n",
       "      <td>50141389</td>\n",
       "      <td>192.705</td>\n",
       "      <td>140</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>147</th>\n",
       "      <td>S7</td>\n",
       "      <td>CON</td>\n",
       "      <td>17</td>\n",
       "      <td>10:7</td>\n",
       "      <td>6</td>\n",
       "      <td>chr6:27899165:G:C</td>\n",
       "      <td>chr6:28207771:G:C</td>\n",
       "      <td>27899165</td>\n",
       "      <td>28207771</td>\n",
       "      <td>308.607</td>\n",
       "      <td>385</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>166</th>\n",
       "      <td>S8</td>\n",
       "      <td>CON</td>\n",
       "      <td>17</td>\n",
       "      <td>5:12</td>\n",
       "      <td>22</td>\n",
       "      <td>chr22:41154950:C:T</td>\n",
       "      <td>chr22:41580474:G:C</td>\n",
       "      <td>41154950</td>\n",
       "      <td>41580474</td>\n",
       "      <td>425.525</td>\n",
       "      <td>471</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>184</th>\n",
       "      <td>S9</td>\n",
       "      <td>CON</td>\n",
       "      <td>16</td>\n",
       "      <td>5:11</td>\n",
       "      <td>6</td>\n",
       "      <td>chr6:61317309:G:C</td>\n",
       "      <td>chr6:61867241:T:C</td>\n",
       "      <td>61317309</td>\n",
       "      <td>61867241</td>\n",
       "      <td>549.933</td>\n",
       "      <td>838</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>202</th>\n",
       "      <td>S10</td>\n",
       "      <td>CON</td>\n",
       "      <td>16</td>\n",
       "      <td>7:9</td>\n",
       "      <td>8</td>\n",
       "      <td>chr8:46907634:ACCG:A</td>\n",
       "      <td>chr8:47646564:A:G</td>\n",
       "      <td>46907634</td>\n",
       "      <td>47646564</td>\n",
       "      <td>738.931</td>\n",
       "      <td>575</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2610</th>\n",
       "      <td>S409</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>1:1</td>\n",
       "      <td>21</td>\n",
       "      <td>chr21:24522616:C:T</td>\n",
       "      <td>chr21:26423081:C:T</td>\n",
       "      <td>24522616</td>\n",
       "      <td>26423081</td>\n",
       "      <td>1900.466</td>\n",
       "      <td>3001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2614</th>\n",
       "      <td>S410</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>1:1</td>\n",
       "      <td>21</td>\n",
       "      <td>chr21:27018589:G:A</td>\n",
       "      <td>chr21:28152094:A:G</td>\n",
       "      <td>27018589</td>\n",
       "      <td>28152094</td>\n",
       "      <td>1133.506</td>\n",
       "      <td>2165</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2618</th>\n",
       "      <td>S411</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>1:1</td>\n",
       "      <td>22</td>\n",
       "      <td>chr22:27779753:G:A</td>\n",
       "      <td>chr22:30214581:C:T</td>\n",
       "      <td>27779753</td>\n",
       "      <td>30214581</td>\n",
       "      <td>2434.829</td>\n",
       "      <td>2994</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2626</th>\n",
       "      <td>S413</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>1:1</td>\n",
       "      <td>22</td>\n",
       "      <td>chr22:30558084:G:A</td>\n",
       "      <td>chr22:32332418:T:TC</td>\n",
       "      <td>30558084</td>\n",
       "      <td>32332418</td>\n",
       "      <td>1774.335</td>\n",
       "      <td>2502</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2630</th>\n",
       "      <td>S414</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>22</td>\n",
       "      <td>chr22:48192677:C:G</td>\n",
       "      <td>chr22:49350260:T:G</td>\n",
       "      <td>48192677</td>\n",
       "      <td>49350260</td>\n",
       "      <td>1157.584</td>\n",
       "      <td>3492</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>345 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      POOL  FID IID   PHE  CHR                  SNP1                 SNP2  \\\n",
       "88      S4  CON  19  6:13    3     chr3:49948685:T:C    chr3:50141389:G:A   \n",
       "147     S7  CON  17  10:7    6     chr6:27899165:G:C    chr6:28207771:G:C   \n",
       "166     S8  CON  17  5:12   22    chr22:41154950:C:T   chr22:41580474:G:C   \n",
       "184     S9  CON  16  5:11    6     chr6:61317309:G:C    chr6:61867241:T:C   \n",
       "202    S10  CON  16   7:9    8  chr8:46907634:ACCG:A    chr8:47646564:A:G   \n",
       "...    ...  ...  ..   ...  ...                   ...                  ...   \n",
       "2610  S409  CON   2   1:1   21    chr21:24522616:C:T   chr21:26423081:C:T   \n",
       "2614  S410  CON   2   1:1   21    chr21:27018589:G:A   chr21:28152094:A:G   \n",
       "2618  S411  CON   2   1:1   22    chr22:27779753:G:A   chr22:30214581:C:T   \n",
       "2626  S413  CON   2   1:1   22    chr22:30558084:G:A  chr22:32332418:T:TC   \n",
       "2630  S414  CON   2   2:0   22    chr22:48192677:C:G   chr22:49350260:T:G   \n",
       "\n",
       "           BP1       BP2        KB  NSNP  NSIM  GRP  \n",
       "88    49948685  50141389   192.705   140   NaN  NaN  \n",
       "147   27899165  28207771   308.607   385   NaN  NaN  \n",
       "166   41154950  41580474   425.525   471   NaN  NaN  \n",
       "184   61317309  61867241   549.933   838   NaN  NaN  \n",
       "202   46907634  47646564   738.931   575   NaN  NaN  \n",
       "...        ...       ...       ...   ...   ...  ...  \n",
       "2610  24522616  26423081  1900.466  3001   NaN  NaN  \n",
       "2614  27018589  28152094  1133.506  2165   NaN  NaN  \n",
       "2618  27779753  30214581  2434.829  2994   NaN  NaN  \n",
       "2626  30558084  32332418  1774.335  2502   NaN  NaN  \n",
       "2630  48192677  49350260  1157.584  3492   NaN  NaN  \n",
       "\n",
       "[345 rows x 13 columns]"
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Filter for consensus ROHs (cROH) with length > 100 Kb and > 100 SNPs\n",
    "inbred_roh_consense_filtered = inbred_roh_consense.loc[\n",
    "    (inbred_roh_consense['NSNP'] >= 100) & (inbred_roh_consense['KB'] >= 100)\n",
    "]\n",
    "\n",
    "# Save the filtered results to a file\n",
    "inbred_roh_consense_filtered.to_csv(\n",
    "    f'{WORK_DIR}/gp2_{ANC}_r6_EOPD.hom.overlap.filtered', sep='\\t', index=False, header=False\n",
    ")\n",
    "\n",
    "# Display the filtered data\n",
    "inbred_roh_consense_filtered"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>POOL</th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PHE</th>\n",
       "      <th>CHR</th>\n",
       "      <th>SNP1</th>\n",
       "      <th>SNP2</th>\n",
       "      <th>BP1</th>\n",
       "      <th>BP2</th>\n",
       "      <th>KB</th>\n",
       "      <th>NSNP</th>\n",
       "      <th>NSIM</th>\n",
       "      <th>GRP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1109</th>\n",
       "      <td>S97</td>\n",
       "      <td>CON</td>\n",
       "      <td>5</td>\n",
       "      <td>5:0</td>\n",
       "      <td>4</td>\n",
       "      <td>chr4:121656615:T:G</td>\n",
       "      <td>chr4:121970619:G:C</td>\n",
       "      <td>121656615</td>\n",
       "      <td>121970619</td>\n",
       "      <td>314.005</td>\n",
       "      <td>416</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1303</th>\n",
       "      <td>S127</td>\n",
       "      <td>CON</td>\n",
       "      <td>4</td>\n",
       "      <td>4:0</td>\n",
       "      <td>4</td>\n",
       "      <td>chr4:114942670:T:A</td>\n",
       "      <td>chr4:116313712:G:A</td>\n",
       "      <td>114942670</td>\n",
       "      <td>116313712</td>\n",
       "      <td>1371.043</td>\n",
       "      <td>1915</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1309</th>\n",
       "      <td>S128</td>\n",
       "      <td>CON</td>\n",
       "      <td>4</td>\n",
       "      <td>4:0</td>\n",
       "      <td>4</td>\n",
       "      <td>chr4:130700347:C:G</td>\n",
       "      <td>chr4:131682533:A:C</td>\n",
       "      <td>130700347</td>\n",
       "      <td>131682533</td>\n",
       "      <td>982.187</td>\n",
       "      <td>1570</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1315</th>\n",
       "      <td>S129</td>\n",
       "      <td>CON</td>\n",
       "      <td>4</td>\n",
       "      <td>4:0</td>\n",
       "      <td>4</td>\n",
       "      <td>chr4:131985505:A:G</td>\n",
       "      <td>chr4:132300720:G:C</td>\n",
       "      <td>131985505</td>\n",
       "      <td>132300720</td>\n",
       "      <td>315.216</td>\n",
       "      <td>581</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1542</th>\n",
       "      <td>S167</td>\n",
       "      <td>CON</td>\n",
       "      <td>3</td>\n",
       "      <td>3:0</td>\n",
       "      <td>1</td>\n",
       "      <td>chr1:15673611:C:T</td>\n",
       "      <td>chr1:17215454:A:G</td>\n",
       "      <td>15673611</td>\n",
       "      <td>17215454</td>\n",
       "      <td>1541.844</td>\n",
       "      <td>1569</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1547</th>\n",
       "      <td>S168</td>\n",
       "      <td>CON</td>\n",
       "      <td>3</td>\n",
       "      <td>3:0</td>\n",
       "      <td>1</td>\n",
       "      <td>chr1:31767341:A:G</td>\n",
       "      <td>chr1:33234313:G:A</td>\n",
       "      <td>31767341</td>\n",
       "      <td>33234313</td>\n",
       "      <td>1466.973</td>\n",
       "      <td>1089</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1657</th>\n",
       "      <td>S190</td>\n",
       "      <td>CON</td>\n",
       "      <td>3</td>\n",
       "      <td>3:0</td>\n",
       "      <td>3</td>\n",
       "      <td>chr3:120504534:G:C</td>\n",
       "      <td>chr3:122063104:T:A</td>\n",
       "      <td>120504534</td>\n",
       "      <td>122063104</td>\n",
       "      <td>1558.571</td>\n",
       "      <td>2698</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1662</th>\n",
       "      <td>S191</td>\n",
       "      <td>CON</td>\n",
       "      <td>3</td>\n",
       "      <td>3:0</td>\n",
       "      <td>3</td>\n",
       "      <td>chr3:135906881:A:T</td>\n",
       "      <td>chr3:137453392:T:A</td>\n",
       "      <td>135906881</td>\n",
       "      <td>137453392</td>\n",
       "      <td>1546.512</td>\n",
       "      <td>1695</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1697</th>\n",
       "      <td>S198</td>\n",
       "      <td>CON</td>\n",
       "      <td>3</td>\n",
       "      <td>3:0</td>\n",
       "      <td>4</td>\n",
       "      <td>chr4:12655552:T:A</td>\n",
       "      <td>chr4:13756738:G:A</td>\n",
       "      <td>12655552</td>\n",
       "      <td>13756738</td>\n",
       "      <td>1101.187</td>\n",
       "      <td>1681</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1702</th>\n",
       "      <td>S199</td>\n",
       "      <td>CON</td>\n",
       "      <td>3</td>\n",
       "      <td>3:0</td>\n",
       "      <td>4</td>\n",
       "      <td>chr4:71774726:T:G</td>\n",
       "      <td>chr4:73412553:C:T</td>\n",
       "      <td>71774726</td>\n",
       "      <td>73412553</td>\n",
       "      <td>1637.828</td>\n",
       "      <td>2053</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1717</th>\n",
       "      <td>S202</td>\n",
       "      <td>CON</td>\n",
       "      <td>3</td>\n",
       "      <td>3:0</td>\n",
       "      <td>4</td>\n",
       "      <td>chr4:112860471:G:C</td>\n",
       "      <td>chr4:114748490:G:T</td>\n",
       "      <td>112860471</td>\n",
       "      <td>114748490</td>\n",
       "      <td>1888.020</td>\n",
       "      <td>2729</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1722</th>\n",
       "      <td>S203</td>\n",
       "      <td>CON</td>\n",
       "      <td>3</td>\n",
       "      <td>3:0</td>\n",
       "      <td>4</td>\n",
       "      <td>chr4:116523818:C:T</td>\n",
       "      <td>chr4:116681540:G:A</td>\n",
       "      <td>116523818</td>\n",
       "      <td>116681540</td>\n",
       "      <td>157.723</td>\n",
       "      <td>382</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1737</th>\n",
       "      <td>S206</td>\n",
       "      <td>CON</td>\n",
       "      <td>3</td>\n",
       "      <td>3:0</td>\n",
       "      <td>5</td>\n",
       "      <td>chr5:61627648:G:A</td>\n",
       "      <td>chr5:63237065:A:G</td>\n",
       "      <td>61627648</td>\n",
       "      <td>63237065</td>\n",
       "      <td>1609.418</td>\n",
       "      <td>2351</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1777</th>\n",
       "      <td>S214</td>\n",
       "      <td>CON</td>\n",
       "      <td>3</td>\n",
       "      <td>3:0</td>\n",
       "      <td>6</td>\n",
       "      <td>chr6:161389511:A:C</td>\n",
       "      <td>chr6:161815720:A:G</td>\n",
       "      <td>161389511</td>\n",
       "      <td>161815720</td>\n",
       "      <td>426.210</td>\n",
       "      <td>681</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1782</th>\n",
       "      <td>S215</td>\n",
       "      <td>CON</td>\n",
       "      <td>3</td>\n",
       "      <td>3:0</td>\n",
       "      <td>6</td>\n",
       "      <td>chr6:162375099:C:G</td>\n",
       "      <td>chr6:163939035:A:T</td>\n",
       "      <td>162375099</td>\n",
       "      <td>163939035</td>\n",
       "      <td>1563.937</td>\n",
       "      <td>2832</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1787</th>\n",
       "      <td>S216</td>\n",
       "      <td>CON</td>\n",
       "      <td>3</td>\n",
       "      <td>3:0</td>\n",
       "      <td>6</td>\n",
       "      <td>chr6:164604500:C:T</td>\n",
       "      <td>chr6:165879923:CG:C</td>\n",
       "      <td>164604500</td>\n",
       "      <td>165879923</td>\n",
       "      <td>1275.424</td>\n",
       "      <td>2645</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1792</th>\n",
       "      <td>S217</td>\n",
       "      <td>CON</td>\n",
       "      <td>3</td>\n",
       "      <td>3:0</td>\n",
       "      <td>6</td>\n",
       "      <td>chr6:166257344:C:A</td>\n",
       "      <td>chr6:166408820:A:C</td>\n",
       "      <td>166257344</td>\n",
       "      <td>166408820</td>\n",
       "      <td>151.477</td>\n",
       "      <td>251</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1797</th>\n",
       "      <td>S218</td>\n",
       "      <td>CON</td>\n",
       "      <td>3</td>\n",
       "      <td>3:0</td>\n",
       "      <td>6</td>\n",
       "      <td>chr6:166471905:G:C</td>\n",
       "      <td>chr6:168176780:C:T</td>\n",
       "      <td>166471905</td>\n",
       "      <td>168176780</td>\n",
       "      <td>1704.876</td>\n",
       "      <td>3320</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1817</th>\n",
       "      <td>S222</td>\n",
       "      <td>CON</td>\n",
       "      <td>3</td>\n",
       "      <td>3:0</td>\n",
       "      <td>9</td>\n",
       "      <td>chr9:81361754:G:C</td>\n",
       "      <td>chr9:83336556:T:C</td>\n",
       "      <td>81361754</td>\n",
       "      <td>83336556</td>\n",
       "      <td>1974.803</td>\n",
       "      <td>2305</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1837</th>\n",
       "      <td>S226</td>\n",
       "      <td>CON</td>\n",
       "      <td>3</td>\n",
       "      <td>3:0</td>\n",
       "      <td>10</td>\n",
       "      <td>chr10:24128513:T:C</td>\n",
       "      <td>chr10:25123571:A:G</td>\n",
       "      <td>24128513</td>\n",
       "      <td>25123571</td>\n",
       "      <td>995.059</td>\n",
       "      <td>1379</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1842</th>\n",
       "      <td>S227</td>\n",
       "      <td>CON</td>\n",
       "      <td>3</td>\n",
       "      <td>3:0</td>\n",
       "      <td>10</td>\n",
       "      <td>chr10:25129805:G:C</td>\n",
       "      <td>chr10:25475212:G:A</td>\n",
       "      <td>25129805</td>\n",
       "      <td>25475212</td>\n",
       "      <td>345.408</td>\n",
       "      <td>486</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1977</th>\n",
       "      <td>S254</td>\n",
       "      <td>CON</td>\n",
       "      <td>3</td>\n",
       "      <td>3:0</td>\n",
       "      <td>17</td>\n",
       "      <td>chr17:44995295:C:T</td>\n",
       "      <td>chr17:45483692:C:T</td>\n",
       "      <td>44995295</td>\n",
       "      <td>45483692</td>\n",
       "      <td>488.398</td>\n",
       "      <td>745</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1982</th>\n",
       "      <td>S255</td>\n",
       "      <td>CON</td>\n",
       "      <td>3</td>\n",
       "      <td>3:0</td>\n",
       "      <td>17</td>\n",
       "      <td>chr17:45869341:T:C</td>\n",
       "      <td>chr17:46788635:A:C</td>\n",
       "      <td>45869341</td>\n",
       "      <td>46788635</td>\n",
       "      <td>919.295</td>\n",
       "      <td>685</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2002</th>\n",
       "      <td>S259</td>\n",
       "      <td>CON</td>\n",
       "      <td>3</td>\n",
       "      <td>3:0</td>\n",
       "      <td>18</td>\n",
       "      <td>chr18:56098540:T:C</td>\n",
       "      <td>chr18:56242791:G:A</td>\n",
       "      <td>56098540</td>\n",
       "      <td>56242791</td>\n",
       "      <td>144.252</td>\n",
       "      <td>204</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2054</th>\n",
       "      <td>S270</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>1</td>\n",
       "      <td>chr1:12206555:A:G</td>\n",
       "      <td>chr1:13793755:C:G</td>\n",
       "      <td>12206555</td>\n",
       "      <td>13793755</td>\n",
       "      <td>1587.201</td>\n",
       "      <td>1414</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2062</th>\n",
       "      <td>S272</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>1</td>\n",
       "      <td>chr1:27325641:A:G</td>\n",
       "      <td>chr1:29547214:C:T</td>\n",
       "      <td>27325641</td>\n",
       "      <td>29547214</td>\n",
       "      <td>2221.574</td>\n",
       "      <td>2327</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2066</th>\n",
       "      <td>S273</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>1</td>\n",
       "      <td>chr1:29689974:G:T</td>\n",
       "      <td>chr1:30027088:A:G</td>\n",
       "      <td>29689974</td>\n",
       "      <td>30027088</td>\n",
       "      <td>337.115</td>\n",
       "      <td>614</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2098</th>\n",
       "      <td>S281</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>1</td>\n",
       "      <td>chr1:113312602:T:C</td>\n",
       "      <td>chr1:116481574:C:A</td>\n",
       "      <td>113312602</td>\n",
       "      <td>116481574</td>\n",
       "      <td>3168.973</td>\n",
       "      <td>4524</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2146</th>\n",
       "      <td>S293</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>2</td>\n",
       "      <td>chr2:109485238:TG:T</td>\n",
       "      <td>chr2:111196746:T:C</td>\n",
       "      <td>109485238</td>\n",
       "      <td>111196746</td>\n",
       "      <td>1711.509</td>\n",
       "      <td>1268</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2186</th>\n",
       "      <td>S303</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>3</td>\n",
       "      <td>chr3:69864088:C:T</td>\n",
       "      <td>chr3:70323580:A:C</td>\n",
       "      <td>69864088</td>\n",
       "      <td>70323580</td>\n",
       "      <td>459.493</td>\n",
       "      <td>553</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2210</th>\n",
       "      <td>S309</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>3</td>\n",
       "      <td>chr3:133428876:C:G</td>\n",
       "      <td>chr3:135199680:C:T</td>\n",
       "      <td>133428876</td>\n",
       "      <td>135199680</td>\n",
       "      <td>1770.805</td>\n",
       "      <td>2789</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2234</th>\n",
       "      <td>S315</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>4</td>\n",
       "      <td>chr4:78888378:G:T</td>\n",
       "      <td>chr4:79617847:C:T</td>\n",
       "      <td>78888378</td>\n",
       "      <td>79617847</td>\n",
       "      <td>729.470</td>\n",
       "      <td>795</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2254</th>\n",
       "      <td>S320</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>4</td>\n",
       "      <td>chr4:134974414:C:G</td>\n",
       "      <td>chr4:136728215:GA:G</td>\n",
       "      <td>134974414</td>\n",
       "      <td>136728215</td>\n",
       "      <td>1753.802</td>\n",
       "      <td>2690</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2258</th>\n",
       "      <td>S321</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>4</td>\n",
       "      <td>chr4:137045526:T:C</td>\n",
       "      <td>chr4:139802325:T:A</td>\n",
       "      <td>137045526</td>\n",
       "      <td>139802325</td>\n",
       "      <td>2756.800</td>\n",
       "      <td>4543</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2278</th>\n",
       "      <td>S326</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>5</td>\n",
       "      <td>chr5:91353481:C:T</td>\n",
       "      <td>chr5:93308527:C:T</td>\n",
       "      <td>91353481</td>\n",
       "      <td>93308527</td>\n",
       "      <td>1955.047</td>\n",
       "      <td>2150</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2282</th>\n",
       "      <td>S327</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>5</td>\n",
       "      <td>chr5:113201365:G:A</td>\n",
       "      <td>chr5:115316816:G:C</td>\n",
       "      <td>113201365</td>\n",
       "      <td>115316816</td>\n",
       "      <td>2115.452</td>\n",
       "      <td>4067</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2350</th>\n",
       "      <td>S344</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>7</td>\n",
       "      <td>chr7:21546333:G:C</td>\n",
       "      <td>chr7:24060359:C:G</td>\n",
       "      <td>21546333</td>\n",
       "      <td>24060359</td>\n",
       "      <td>2514.027</td>\n",
       "      <td>4937</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2354</th>\n",
       "      <td>S345</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>7</td>\n",
       "      <td>chr7:28265918:G:A</td>\n",
       "      <td>chr7:29111400:G:C</td>\n",
       "      <td>28265918</td>\n",
       "      <td>29111400</td>\n",
       "      <td>845.483</td>\n",
       "      <td>1670</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2358</th>\n",
       "      <td>S346</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>7</td>\n",
       "      <td>chr7:30915502:A:G</td>\n",
       "      <td>chr7:32277201:A:C</td>\n",
       "      <td>30915502</td>\n",
       "      <td>32277201</td>\n",
       "      <td>1361.700</td>\n",
       "      <td>2484</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2362</th>\n",
       "      <td>S347</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>7</td>\n",
       "      <td>chr7:32409956:C:T</td>\n",
       "      <td>chr7:34217918:G:A</td>\n",
       "      <td>32409956</td>\n",
       "      <td>34217918</td>\n",
       "      <td>1807.963</td>\n",
       "      <td>2854</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2370</th>\n",
       "      <td>S349</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>7</td>\n",
       "      <td>chr7:113884394:G:A</td>\n",
       "      <td>chr7:115481028:G:A</td>\n",
       "      <td>113884394</td>\n",
       "      <td>115481028</td>\n",
       "      <td>1596.635</td>\n",
       "      <td>1793</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2390</th>\n",
       "      <td>S354</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>8</td>\n",
       "      <td>chr8:63605264:G:A</td>\n",
       "      <td>chr8:67767035:T:G</td>\n",
       "      <td>63605264</td>\n",
       "      <td>67767035</td>\n",
       "      <td>4161.772</td>\n",
       "      <td>4766</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2398</th>\n",
       "      <td>S356</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>8</td>\n",
       "      <td>chr8:118092082:G:A</td>\n",
       "      <td>chr8:119982154:A:G</td>\n",
       "      <td>118092082</td>\n",
       "      <td>119982154</td>\n",
       "      <td>1890.073</td>\n",
       "      <td>2638</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2406</th>\n",
       "      <td>S358</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>9</td>\n",
       "      <td>chr9:10783920:C:T</td>\n",
       "      <td>chr9:12375732:A:G</td>\n",
       "      <td>10783920</td>\n",
       "      <td>12375732</td>\n",
       "      <td>1591.813</td>\n",
       "      <td>3314</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2442</th>\n",
       "      <td>S367</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>11</td>\n",
       "      <td>chr11:27900385:T:C</td>\n",
       "      <td>chr11:29955694:C:A</td>\n",
       "      <td>27900385</td>\n",
       "      <td>29955694</td>\n",
       "      <td>2055.310</td>\n",
       "      <td>2569</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2454</th>\n",
       "      <td>S370</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>11</td>\n",
       "      <td>chr11:97985078:G:C</td>\n",
       "      <td>chr11:98760670:T:C</td>\n",
       "      <td>97985078</td>\n",
       "      <td>98760670</td>\n",
       "      <td>775.593</td>\n",
       "      <td>1694</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2458</th>\n",
       "      <td>S371</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>11</td>\n",
       "      <td>chr11:99170300:A:G</td>\n",
       "      <td>chr11:99733091:C:T</td>\n",
       "      <td>99170300</td>\n",
       "      <td>99733091</td>\n",
       "      <td>562.792</td>\n",
       "      <td>1285</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2462</th>\n",
       "      <td>S372</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>11</td>\n",
       "      <td>chr11:101446584:A:G</td>\n",
       "      <td>chr11:101637914:C:T</td>\n",
       "      <td>101446584</td>\n",
       "      <td>101637914</td>\n",
       "      <td>191.331</td>\n",
       "      <td>433</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2474</th>\n",
       "      <td>S375</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>12</td>\n",
       "      <td>chr12:32943085:C:T</td>\n",
       "      <td>chr12:34717351:T:G</td>\n",
       "      <td>32943085</td>\n",
       "      <td>34717351</td>\n",
       "      <td>1774.267</td>\n",
       "      <td>3250</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2502</th>\n",
       "      <td>S382</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>12</td>\n",
       "      <td>chr12:117954153:G:A</td>\n",
       "      <td>chr12:118808222:G:A</td>\n",
       "      <td>117954153</td>\n",
       "      <td>118808222</td>\n",
       "      <td>854.070</td>\n",
       "      <td>1500</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2526</th>\n",
       "      <td>S388</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>14</td>\n",
       "      <td>chr14:70881249:T:C</td>\n",
       "      <td>chr14:72425523:A:T</td>\n",
       "      <td>70881249</td>\n",
       "      <td>72425523</td>\n",
       "      <td>1544.275</td>\n",
       "      <td>2302</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2550</th>\n",
       "      <td>S394</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>15</td>\n",
       "      <td>chr15:56031825:C:T</td>\n",
       "      <td>chr15:56521535:A:T</td>\n",
       "      <td>56031825</td>\n",
       "      <td>56521535</td>\n",
       "      <td>489.711</td>\n",
       "      <td>907</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2554</th>\n",
       "      <td>S395</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>15</td>\n",
       "      <td>chr15:82099391:A:G</td>\n",
       "      <td>chr15:84843059:G:A</td>\n",
       "      <td>82099391</td>\n",
       "      <td>84843059</td>\n",
       "      <td>2743.669</td>\n",
       "      <td>3215</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2570</th>\n",
       "      <td>S399</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>17</td>\n",
       "      <td>chr17:12661342:C:A</td>\n",
       "      <td>chr17:13723171:C:T</td>\n",
       "      <td>12661342</td>\n",
       "      <td>13723171</td>\n",
       "      <td>1061.830</td>\n",
       "      <td>2192</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2582</th>\n",
       "      <td>S402</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>17</td>\n",
       "      <td>chr17:36843262:T:G</td>\n",
       "      <td>chr17:40599413:C:T</td>\n",
       "      <td>36843262</td>\n",
       "      <td>40599413</td>\n",
       "      <td>3756.152</td>\n",
       "      <td>3658</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2586</th>\n",
       "      <td>S403</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>17</td>\n",
       "      <td>chr17:41816525:G:A</td>\n",
       "      <td>chr17:43433482:A:G</td>\n",
       "      <td>41816525</td>\n",
       "      <td>43433482</td>\n",
       "      <td>1616.958</td>\n",
       "      <td>1258</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2590</th>\n",
       "      <td>S404</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>18</td>\n",
       "      <td>chr18:28108827:C:T</td>\n",
       "      <td>chr18:29174653:A:C</td>\n",
       "      <td>28108827</td>\n",
       "      <td>29174653</td>\n",
       "      <td>1065.827</td>\n",
       "      <td>1372</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2594</th>\n",
       "      <td>S405</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>18</td>\n",
       "      <td>chr18:60006901:T:A</td>\n",
       "      <td>chr18:61518053:C:A</td>\n",
       "      <td>60006901</td>\n",
       "      <td>61518053</td>\n",
       "      <td>1511.153</td>\n",
       "      <td>2088</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2606</th>\n",
       "      <td>S408</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>20</td>\n",
       "      <td>chr20:62091537:A:G</td>\n",
       "      <td>chr20:62411193:T:C</td>\n",
       "      <td>62091537</td>\n",
       "      <td>62411193</td>\n",
       "      <td>319.657</td>\n",
       "      <td>573</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2630</th>\n",
       "      <td>S414</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>22</td>\n",
       "      <td>chr22:48192677:C:G</td>\n",
       "      <td>chr22:49350260:T:G</td>\n",
       "      <td>48192677</td>\n",
       "      <td>49350260</td>\n",
       "      <td>1157.584</td>\n",
       "      <td>3492</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      POOL  FID IID  PHE  CHR                 SNP1                 SNP2  \\\n",
       "1109   S97  CON   5  5:0    4   chr4:121656615:T:G   chr4:121970619:G:C   \n",
       "1303  S127  CON   4  4:0    4   chr4:114942670:T:A   chr4:116313712:G:A   \n",
       "1309  S128  CON   4  4:0    4   chr4:130700347:C:G   chr4:131682533:A:C   \n",
       "1315  S129  CON   4  4:0    4   chr4:131985505:A:G   chr4:132300720:G:C   \n",
       "1542  S167  CON   3  3:0    1    chr1:15673611:C:T    chr1:17215454:A:G   \n",
       "1547  S168  CON   3  3:0    1    chr1:31767341:A:G    chr1:33234313:G:A   \n",
       "1657  S190  CON   3  3:0    3   chr3:120504534:G:C   chr3:122063104:T:A   \n",
       "1662  S191  CON   3  3:0    3   chr3:135906881:A:T   chr3:137453392:T:A   \n",
       "1697  S198  CON   3  3:0    4    chr4:12655552:T:A    chr4:13756738:G:A   \n",
       "1702  S199  CON   3  3:0    4    chr4:71774726:T:G    chr4:73412553:C:T   \n",
       "1717  S202  CON   3  3:0    4   chr4:112860471:G:C   chr4:114748490:G:T   \n",
       "1722  S203  CON   3  3:0    4   chr4:116523818:C:T   chr4:116681540:G:A   \n",
       "1737  S206  CON   3  3:0    5    chr5:61627648:G:A    chr5:63237065:A:G   \n",
       "1777  S214  CON   3  3:0    6   chr6:161389511:A:C   chr6:161815720:A:G   \n",
       "1782  S215  CON   3  3:0    6   chr6:162375099:C:G   chr6:163939035:A:T   \n",
       "1787  S216  CON   3  3:0    6   chr6:164604500:C:T  chr6:165879923:CG:C   \n",
       "1792  S217  CON   3  3:0    6   chr6:166257344:C:A   chr6:166408820:A:C   \n",
       "1797  S218  CON   3  3:0    6   chr6:166471905:G:C   chr6:168176780:C:T   \n",
       "1817  S222  CON   3  3:0    9    chr9:81361754:G:C    chr9:83336556:T:C   \n",
       "1837  S226  CON   3  3:0   10   chr10:24128513:T:C   chr10:25123571:A:G   \n",
       "1842  S227  CON   3  3:0   10   chr10:25129805:G:C   chr10:25475212:G:A   \n",
       "1977  S254  CON   3  3:0   17   chr17:44995295:C:T   chr17:45483692:C:T   \n",
       "1982  S255  CON   3  3:0   17   chr17:45869341:T:C   chr17:46788635:A:C   \n",
       "2002  S259  CON   3  3:0   18   chr18:56098540:T:C   chr18:56242791:G:A   \n",
       "2054  S270  CON   2  2:0    1    chr1:12206555:A:G    chr1:13793755:C:G   \n",
       "2062  S272  CON   2  2:0    1    chr1:27325641:A:G    chr1:29547214:C:T   \n",
       "2066  S273  CON   2  2:0    1    chr1:29689974:G:T    chr1:30027088:A:G   \n",
       "2098  S281  CON   2  2:0    1   chr1:113312602:T:C   chr1:116481574:C:A   \n",
       "2146  S293  CON   2  2:0    2  chr2:109485238:TG:T   chr2:111196746:T:C   \n",
       "2186  S303  CON   2  2:0    3    chr3:69864088:C:T    chr3:70323580:A:C   \n",
       "2210  S309  CON   2  2:0    3   chr3:133428876:C:G   chr3:135199680:C:T   \n",
       "2234  S315  CON   2  2:0    4    chr4:78888378:G:T    chr4:79617847:C:T   \n",
       "2254  S320  CON   2  2:0    4   chr4:134974414:C:G  chr4:136728215:GA:G   \n",
       "2258  S321  CON   2  2:0    4   chr4:137045526:T:C   chr4:139802325:T:A   \n",
       "2278  S326  CON   2  2:0    5    chr5:91353481:C:T    chr5:93308527:C:T   \n",
       "2282  S327  CON   2  2:0    5   chr5:113201365:G:A   chr5:115316816:G:C   \n",
       "2350  S344  CON   2  2:0    7    chr7:21546333:G:C    chr7:24060359:C:G   \n",
       "2354  S345  CON   2  2:0    7    chr7:28265918:G:A    chr7:29111400:G:C   \n",
       "2358  S346  CON   2  2:0    7    chr7:30915502:A:G    chr7:32277201:A:C   \n",
       "2362  S347  CON   2  2:0    7    chr7:32409956:C:T    chr7:34217918:G:A   \n",
       "2370  S349  CON   2  2:0    7   chr7:113884394:G:A   chr7:115481028:G:A   \n",
       "2390  S354  CON   2  2:0    8    chr8:63605264:G:A    chr8:67767035:T:G   \n",
       "2398  S356  CON   2  2:0    8   chr8:118092082:G:A   chr8:119982154:A:G   \n",
       "2406  S358  CON   2  2:0    9    chr9:10783920:C:T    chr9:12375732:A:G   \n",
       "2442  S367  CON   2  2:0   11   chr11:27900385:T:C   chr11:29955694:C:A   \n",
       "2454  S370  CON   2  2:0   11   chr11:97985078:G:C   chr11:98760670:T:C   \n",
       "2458  S371  CON   2  2:0   11   chr11:99170300:A:G   chr11:99733091:C:T   \n",
       "2462  S372  CON   2  2:0   11  chr11:101446584:A:G  chr11:101637914:C:T   \n",
       "2474  S375  CON   2  2:0   12   chr12:32943085:C:T   chr12:34717351:T:G   \n",
       "2502  S382  CON   2  2:0   12  chr12:117954153:G:A  chr12:118808222:G:A   \n",
       "2526  S388  CON   2  2:0   14   chr14:70881249:T:C   chr14:72425523:A:T   \n",
       "2550  S394  CON   2  2:0   15   chr15:56031825:C:T   chr15:56521535:A:T   \n",
       "2554  S395  CON   2  2:0   15   chr15:82099391:A:G   chr15:84843059:G:A   \n",
       "2570  S399  CON   2  2:0   17   chr17:12661342:C:A   chr17:13723171:C:T   \n",
       "2582  S402  CON   2  2:0   17   chr17:36843262:T:G   chr17:40599413:C:T   \n",
       "2586  S403  CON   2  2:0   17   chr17:41816525:G:A   chr17:43433482:A:G   \n",
       "2590  S404  CON   2  2:0   18   chr18:28108827:C:T   chr18:29174653:A:C   \n",
       "2594  S405  CON   2  2:0   18   chr18:60006901:T:A   chr18:61518053:C:A   \n",
       "2606  S408  CON   2  2:0   20   chr20:62091537:A:G   chr20:62411193:T:C   \n",
       "2630  S414  CON   2  2:0   22   chr22:48192677:C:G   chr22:49350260:T:G   \n",
       "\n",
       "            BP1        BP2        KB  NSNP  NSIM  GRP  \n",
       "1109  121656615  121970619   314.005   416   NaN  NaN  \n",
       "1303  114942670  116313712  1371.043  1915   NaN  NaN  \n",
       "1309  130700347  131682533   982.187  1570   NaN  NaN  \n",
       "1315  131985505  132300720   315.216   581   NaN  NaN  \n",
       "1542   15673611   17215454  1541.844  1569   NaN  NaN  \n",
       "1547   31767341   33234313  1466.973  1089   NaN  NaN  \n",
       "1657  120504534  122063104  1558.571  2698   NaN  NaN  \n",
       "1662  135906881  137453392  1546.512  1695   NaN  NaN  \n",
       "1697   12655552   13756738  1101.187  1681   NaN  NaN  \n",
       "1702   71774726   73412553  1637.828  2053   NaN  NaN  \n",
       "1717  112860471  114748490  1888.020  2729   NaN  NaN  \n",
       "1722  116523818  116681540   157.723   382   NaN  NaN  \n",
       "1737   61627648   63237065  1609.418  2351   NaN  NaN  \n",
       "1777  161389511  161815720   426.210   681   NaN  NaN  \n",
       "1782  162375099  163939035  1563.937  2832   NaN  NaN  \n",
       "1787  164604500  165879923  1275.424  2645   NaN  NaN  \n",
       "1792  166257344  166408820   151.477   251   NaN  NaN  \n",
       "1797  166471905  168176780  1704.876  3320   NaN  NaN  \n",
       "1817   81361754   83336556  1974.803  2305   NaN  NaN  \n",
       "1837   24128513   25123571   995.059  1379   NaN  NaN  \n",
       "1842   25129805   25475212   345.408   486   NaN  NaN  \n",
       "1977   44995295   45483692   488.398   745   NaN  NaN  \n",
       "1982   45869341   46788635   919.295   685   NaN  NaN  \n",
       "2002   56098540   56242791   144.252   204   NaN  NaN  \n",
       "2054   12206555   13793755  1587.201  1414   NaN  NaN  \n",
       "2062   27325641   29547214  2221.574  2327   NaN  NaN  \n",
       "2066   29689974   30027088   337.115   614   NaN  NaN  \n",
       "2098  113312602  116481574  3168.973  4524   NaN  NaN  \n",
       "2146  109485238  111196746  1711.509  1268   NaN  NaN  \n",
       "2186   69864088   70323580   459.493   553   NaN  NaN  \n",
       "2210  133428876  135199680  1770.805  2789   NaN  NaN  \n",
       "2234   78888378   79617847   729.470   795   NaN  NaN  \n",
       "2254  134974414  136728215  1753.802  2690   NaN  NaN  \n",
       "2258  137045526  139802325  2756.800  4543   NaN  NaN  \n",
       "2278   91353481   93308527  1955.047  2150   NaN  NaN  \n",
       "2282  113201365  115316816  2115.452  4067   NaN  NaN  \n",
       "2350   21546333   24060359  2514.027  4937   NaN  NaN  \n",
       "2354   28265918   29111400   845.483  1670   NaN  NaN  \n",
       "2358   30915502   32277201  1361.700  2484   NaN  NaN  \n",
       "2362   32409956   34217918  1807.963  2854   NaN  NaN  \n",
       "2370  113884394  115481028  1596.635  1793   NaN  NaN  \n",
       "2390   63605264   67767035  4161.772  4766   NaN  NaN  \n",
       "2398  118092082  119982154  1890.073  2638   NaN  NaN  \n",
       "2406   10783920   12375732  1591.813  3314   NaN  NaN  \n",
       "2442   27900385   29955694  2055.310  2569   NaN  NaN  \n",
       "2454   97985078   98760670   775.593  1694   NaN  NaN  \n",
       "2458   99170300   99733091   562.792  1285   NaN  NaN  \n",
       "2462  101446584  101637914   191.331   433   NaN  NaN  \n",
       "2474   32943085   34717351  1774.267  3250   NaN  NaN  \n",
       "2502  117954153  118808222   854.070  1500   NaN  NaN  \n",
       "2526   70881249   72425523  1544.275  2302   NaN  NaN  \n",
       "2550   56031825   56521535   489.711   907   NaN  NaN  \n",
       "2554   82099391   84843059  2743.669  3215   NaN  NaN  \n",
       "2570   12661342   13723171  1061.830  2192   NaN  NaN  \n",
       "2582   36843262   40599413  3756.152  3658   NaN  NaN  \n",
       "2586   41816525   43433482  1616.958  1258   NaN  NaN  \n",
       "2590   28108827   29174653  1065.827  1372   NaN  NaN  \n",
       "2594   60006901   61518053  1511.153  2088   NaN  NaN  \n",
       "2606   62091537   62411193   319.657   573   NaN  NaN  \n",
       "2630   48192677   49350260  1157.584  3492   NaN  NaN  "
      ]
     },
     "execution_count": 61,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Filter for ROHs that no control (CON) carried, looking for high penetrance variants\n",
    "check_ROH = inbred_roh_consense_filtered.loc[\n",
    "    inbred_roh_consense_filtered['PHE'].str.contains(':0')\n",
    "]\n",
    "\n",
    "# Display the filtered data\n",
    "check_ROH"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "S97 S127 S128 S129 S167 S168 S190 S191 S198 S199 S202 S203 S206 S214 S215 S216 S217 S218 S222 S226 S227 S254 S255 S259 S270 S272 S273 S281 S293 S303 S309 S315 S320 S321 S326 S327 S344 S345 S346 S347 S349 S354 S356 S358 S367 S370 S371 S372 S375 S382 S388 S394 S395 S399 S402 S403 S404 S405 S408 S414\n"
     ]
    }
   ],
   "source": [
    "# Extract 'POOL' column values and join them into a space-separated string\n",
    "columns = check_ROH['POOL'].tolist()\n",
    "column_spaces = ' '.join(map(str, columns))\n",
    "\n",
    "# Display the result\n",
    "print(column_spaces)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Find ROH enriched in inbred cases (test the association using logistic model)\n",
    "\n",
    "Run logistic regression models to test for associations between ROHs and inbred status."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Concensus ROH filtered"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Classes ‘data.table’ and 'data.frame':\t345 obs. of  13 variables:\n",
      " $ POOL: chr  \"S4\" \"S7\" \"S8\" \"S9\" ...\n",
      " $ FID : chr  \"CON\" \"CON\" \"CON\" \"CON\" ...\n",
      " $ IID : int  19 17 17 16 16 14 14 13 13 12 ...\n",
      " $ PHE : chr  \"6:13\" \"10:7\" \"5:12\" \"5:11\" ...\n",
      " $ CHR : int  3 6 22 6 8 8 20 12 16 10 ...\n",
      " $ SNP1: chr  \"chr3:49948685:T:C\" \"chr6:27899165:G:C\" \"chr22:41154950:C:T\" \"chr6:61317309:G:C\" ...\n",
      " $ SNP2: chr  \"chr3:50141389:G:A\" \"chr6:28207771:G:C\" \"chr22:41580474:G:C\" \"chr6:61867241:T:C\" ...\n",
      " $ BP1 : int  49948685 27899165 41154950 61317309 46907634 99012219 34787094 111278914 47050042 37413507 ...\n",
      " $ BP2 : int  50141389 28207771 41580474 61867241 47646564 99904724 35257135 112549081 47942933 38306784 ...\n",
      " $ KB  : num  193 309 426 550 739 ...\n",
      " $ NSNP: int  140 385 471 838 575 571 582 1129 224 1446 ...\n",
      " $ NSIM: logi  NA NA NA NA NA NA ...\n",
      " $ GRP : logi  NA NA NA NA NA NA ...\n",
      " - attr(*, \".internal.selfref\")=<externalptr> \n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "\n",
    "# Load the filtered ROH data\n",
    "ROH <- fread(paste0(\"gp2_\", ANC, \"_r6_EOPD.hom.overlap.filtered\"), header = FALSE)\n",
    "\n",
    "# Assign column names to the ROH data\n",
    "colnames(ROH) <- c(\"POOL\", \"FID\", \"IID\", \"PHE\", \"CHR\", \"SNP1\", \"SNP2\", \"BP1\", \"BP2\", \"KB\", \"NSNP\", \"NSIM\", \"GRP\")\n",
    "\n",
    "# Display the structure of the ROH data\n",
    "str(ROH)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: grep \"are cases and\" /home/jupyter/Release6_AMR/gp2_AMR_r6_EOPD.log\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Among remaining phenotypes, 91 are cases and 139 are controls.\r\n"
     ]
    }
   ],
   "source": [
    "# Use this log data as input for the next two steps\n",
    "shell_do(f'grep \"are cases and\" {WORK_DIR}/gp2_{ANC}_r6_EOPD.log')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Classes ‘data.table’ and 'data.frame':\t345 obs. of  20 variables:\n",
      " $ POOL             : chr  \"S4\" \"S7\" \"S8\" \"S9\" ...\n",
      " $ FID              : chr  \"CON\" \"CON\" \"CON\" \"CON\" ...\n",
      " $ IID              : int  19 17 17 16 16 14 14 13 13 12 ...\n",
      " $ PHE              : chr  \"6:13\" \"10:7\" \"5:12\" \"5:11\" ...\n",
      " $ CHR              : int  3 6 22 6 8 8 20 12 16 10 ...\n",
      " $ SNP1             : chr  \"chr3:49948685:T:C\" \"chr6:27899165:G:C\" \"chr22:41154950:C:T\" \"chr6:61317309:G:C\" ...\n",
      " $ SNP2             : chr  \"chr3:50141389:G:A\" \"chr6:28207771:G:C\" \"chr22:41580474:G:C\" \"chr6:61867241:T:C\" ...\n",
      " $ BP1              : int  49948685 27899165 41154950 61317309 46907634 99012219 34787094 111278914 47050042 37413507 ...\n",
      " $ BP2              : int  50141389 28207771 41580474 61867241 47646564 99904724 35257135 112549081 47942933 38306784 ...\n",
      " $ KB               : num  193 309 426 550 739 ...\n",
      " $ NSNP             : int  140 385 471 838 575 571 582 1129 224 1446 ...\n",
      " $ NSIM             : logi  NA NA NA NA NA NA ...\n",
      " $ GRP              : logi  NA NA NA NA NA NA ...\n",
      " $ case             : chr  \"6\" \"10\" \"5\" \"5\" ...\n",
      " $ control          : chr  \"13\" \"7\" \"12\" \"11\" ...\n",
      " $ NcasesWithROHs   : num  6 10 5 5 7 8 5 7 5 7 ...\n",
      " $ NcontrolsWithROHs: num  13 7 12 11 9 6 9 6 8 5 ...\n",
      " $ combinedN        : num  19 17 17 16 16 14 14 13 13 12 ...\n",
      " $ P                : num  0.618 0.153 0.527 0.66 0.928 ...\n",
      " $ total_ROH_count  : num  19 17 17 16 16 14 14 13 13 12 ...\n",
      " - attr(*, \".internal.selfref\")=<externalptr> \n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "In addition: There were 50 or more warnings (use warnings() to see the first 50)\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%R\n",
    "\n",
    "# Split the 'PHE' column into case and control components\n",
    "ROH$case <- ldply(strsplit(as.character(ROH$PHE), split = \":\"))[[1]]\n",
    "ROH$control <- ldply(strsplit(as.character(ROH$PHE), split = \":\"))[[2]]\n",
    "\n",
    "# Convert case and control components to numeric\n",
    "ROH$NcasesWithROHs <- as.numeric(ROH$case)\n",
    "ROH$NcontrolsWithROHs <- as.numeric(ROH$control)\n",
    "\n",
    "# Calculate the total counts for cases, controls, and combined\n",
    "ROH$combinedN <- ROH$NcasesWithROHs + ROH$NcontrolsWithROHs\n",
    "\n",
    "# Initialize the 'P' column for p-values\n",
    "ROH$P <- NA\n",
    "\n",
    "# Perform proportion tests for each row\n",
    "for (i in seq_len(nrow(ROH))) {\n",
    "  thisP <- prop.test(x = c(ROH$NcasesWithROHs[i], ROH$NcontrolsWithROHs[i]), \n",
    "                     n = c(91, 139))  # n = (cases, controls)\n",
    "  ROH$P[i] <- thisP$p.value\n",
    "}\n",
    "\n",
    "# Calculate the total ROH count for cases and controls\n",
    "ROH$total_ROH_count <- ROH$NcasesWithROHs + ROH$NcontrolsWithROHs\n",
    "\n",
    "# Display the structure of the ROH data\n",
    "str(ROH)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. \n",
      "      0       0       0       0       0       0 \n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "\n",
    "# Calculate proportions of cases and controls with ROH\n",
    "ROH$propCases <- ROH$NcasesWithROHs / 91  # Number of cases\n",
    "ROH$propControls <- ROH$NcontrolsWithROHs / 139  # Number of controls\n",
    "\n",
    "# Identify case-enriched ROHs\n",
    "ROH$caseEnriched <- ifelse(ROH$propCases > ROH$propControls, 1, 0)\n",
    "\n",
    "# Subset ROHs where total ROH count >= 1 and case-enriched\n",
    "ROH_subsetted <- subset(ROH, total_ROH_count >= 1 & caseEnriched == 1)\n",
    "\n",
    "# Apply Bonferroni correction\n",
    "Nbonf <- length(ROH_subsetted$POOL)  # Number of tests (unique POOLs)\n",
    "ROH_subsetted$passMultiTest <- ifelse(ROH_subsetted$P <= (0.05 / Nbonf), 1, 0)\n",
    "ROH_subsetted$BONF <- ROH_subsetted$P / Nbonf\n",
    "\n",
    "# Subset the significant results after multiple testing correction\n",
    "significant_ROH <- subset(ROH_subsetted, passMultiTest == 1)\n",
    "\n",
    "# Apply Bonferroni correction to the full dataset\n",
    "ROH$BONF <- ROH$P / Nbonf\n",
    "\n",
    "# Display summary of Bonferroni-corrected results\n",
    "summary(ROH_subsetted$passMultiTest)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R\n",
    "\n",
    "# Write the subsetted ROH data to text files\n",
    "fwrite(ROH_subsetted, paste0(ANC, \"_ROH.EOPD_subsetted.txt\"))\n",
    "fwrite(ROH_subsetted, paste0(ANC, \"_ROH.EOPD_caseEnriched.txt\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>POOL</th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PHE</th>\n",
       "      <th>CHR</th>\n",
       "      <th>SNP1</th>\n",
       "      <th>SNP2</th>\n",
       "      <th>BP1</th>\n",
       "      <th>BP2</th>\n",
       "      <th>KB</th>\n",
       "      <th>...</th>\n",
       "      <th>NcasesWithROHs</th>\n",
       "      <th>NcontrolsWithROHs</th>\n",
       "      <th>combinedN</th>\n",
       "      <th>P</th>\n",
       "      <th>total_ROH_count</th>\n",
       "      <th>propCases</th>\n",
       "      <th>propControls</th>\n",
       "      <th>caseEnriched</th>\n",
       "      <th>passMultiTest</th>\n",
       "      <th>BONF</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>S7</td>\n",
       "      <td>CON</td>\n",
       "      <td>17</td>\n",
       "      <td>10:7</td>\n",
       "      <td>6</td>\n",
       "      <td>chr6:27899165:G:C</td>\n",
       "      <td>chr6:28207771:G:C</td>\n",
       "      <td>27899165</td>\n",
       "      <td>28207771</td>\n",
       "      <td>308.607</td>\n",
       "      <td>...</td>\n",
       "      <td>10</td>\n",
       "      <td>7</td>\n",
       "      <td>17</td>\n",
       "      <td>0.152806</td>\n",
       "      <td>17</td>\n",
       "      <td>0.109890</td>\n",
       "      <td>0.050360</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000670</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>S10</td>\n",
       "      <td>CON</td>\n",
       "      <td>16</td>\n",
       "      <td>7:9</td>\n",
       "      <td>8</td>\n",
       "      <td>chr8:46907634:ACCG:A</td>\n",
       "      <td>chr8:47646564:A:G</td>\n",
       "      <td>46907634</td>\n",
       "      <td>47646564</td>\n",
       "      <td>738.931</td>\n",
       "      <td>...</td>\n",
       "      <td>7</td>\n",
       "      <td>9</td>\n",
       "      <td>16</td>\n",
       "      <td>0.928387</td>\n",
       "      <td>16</td>\n",
       "      <td>0.076923</td>\n",
       "      <td>0.064748</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.004072</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>S13</td>\n",
       "      <td>CON</td>\n",
       "      <td>14</td>\n",
       "      <td>8:6</td>\n",
       "      <td>8</td>\n",
       "      <td>chr8:99012219:T:G</td>\n",
       "      <td>chr8:99904724:T:C</td>\n",
       "      <td>99012219</td>\n",
       "      <td>99904724</td>\n",
       "      <td>892.506</td>\n",
       "      <td>...</td>\n",
       "      <td>8</td>\n",
       "      <td>6</td>\n",
       "      <td>14</td>\n",
       "      <td>0.268764</td>\n",
       "      <td>14</td>\n",
       "      <td>0.087912</td>\n",
       "      <td>0.043165</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.001179</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>S18</td>\n",
       "      <td>CON</td>\n",
       "      <td>13</td>\n",
       "      <td>7:6</td>\n",
       "      <td>12</td>\n",
       "      <td>chr12:111278914:G:A</td>\n",
       "      <td>chr12:112549081:T:C</td>\n",
       "      <td>111278914</td>\n",
       "      <td>112549081</td>\n",
       "      <td>1270.168</td>\n",
       "      <td>...</td>\n",
       "      <td>7</td>\n",
       "      <td>6</td>\n",
       "      <td>13</td>\n",
       "      <td>0.428293</td>\n",
       "      <td>13</td>\n",
       "      <td>0.076923</td>\n",
       "      <td>0.043165</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.001878</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>S23</td>\n",
       "      <td>CON</td>\n",
       "      <td>12</td>\n",
       "      <td>7:5</td>\n",
       "      <td>10</td>\n",
       "      <td>chr10:37413507:T:A</td>\n",
       "      <td>chr10:38306784:T:C</td>\n",
       "      <td>37413507</td>\n",
       "      <td>38306784</td>\n",
       "      <td>893.278</td>\n",
       "      <td>...</td>\n",
       "      <td>7</td>\n",
       "      <td>5</td>\n",
       "      <td>12</td>\n",
       "      <td>0.288016</td>\n",
       "      <td>12</td>\n",
       "      <td>0.076923</td>\n",
       "      <td>0.035971</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.001263</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>223</th>\n",
       "      <td>S409</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>1:1</td>\n",
       "      <td>21</td>\n",
       "      <td>chr21:24522616:C:T</td>\n",
       "      <td>chr21:26423081:C:T</td>\n",
       "      <td>24522616</td>\n",
       "      <td>26423081</td>\n",
       "      <td>1900.466</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>2</td>\n",
       "      <td>0.010989</td>\n",
       "      <td>0.007194</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.004386</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>224</th>\n",
       "      <td>S410</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>1:1</td>\n",
       "      <td>21</td>\n",
       "      <td>chr21:27018589:G:A</td>\n",
       "      <td>chr21:28152094:A:G</td>\n",
       "      <td>27018589</td>\n",
       "      <td>28152094</td>\n",
       "      <td>1133.506</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>2</td>\n",
       "      <td>0.010989</td>\n",
       "      <td>0.007194</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.004386</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>225</th>\n",
       "      <td>S411</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>1:1</td>\n",
       "      <td>22</td>\n",
       "      <td>chr22:27779753:G:A</td>\n",
       "      <td>chr22:30214581:C:T</td>\n",
       "      <td>27779753</td>\n",
       "      <td>30214581</td>\n",
       "      <td>2434.829</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>2</td>\n",
       "      <td>0.010989</td>\n",
       "      <td>0.007194</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.004386</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>226</th>\n",
       "      <td>S413</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>1:1</td>\n",
       "      <td>22</td>\n",
       "      <td>chr22:30558084:G:A</td>\n",
       "      <td>chr22:32332418:T:TC</td>\n",
       "      <td>30558084</td>\n",
       "      <td>32332418</td>\n",
       "      <td>1774.335</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>2</td>\n",
       "      <td>0.010989</td>\n",
       "      <td>0.007194</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.004386</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>227</th>\n",
       "      <td>S414</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>22</td>\n",
       "      <td>chr22:48192677:C:G</td>\n",
       "      <td>chr22:49350260:T:G</td>\n",
       "      <td>48192677</td>\n",
       "      <td>49350260</td>\n",
       "      <td>1157.584</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.303340</td>\n",
       "      <td>2</td>\n",
       "      <td>0.021978</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.001330</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>228 rows × 25 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     POOL  FID  IID   PHE  CHR                  SNP1                 SNP2  \\\n",
       "0      S7  CON   17  10:7    6     chr6:27899165:G:C    chr6:28207771:G:C   \n",
       "1     S10  CON   16   7:9    8  chr8:46907634:ACCG:A    chr8:47646564:A:G   \n",
       "2     S13  CON   14   8:6    8     chr8:99012219:T:G    chr8:99904724:T:C   \n",
       "3     S18  CON   13   7:6   12   chr12:111278914:G:A  chr12:112549081:T:C   \n",
       "4     S23  CON   12   7:5   10    chr10:37413507:T:A   chr10:38306784:T:C   \n",
       "..    ...  ...  ...   ...  ...                   ...                  ...   \n",
       "223  S409  CON    2   1:1   21    chr21:24522616:C:T   chr21:26423081:C:T   \n",
       "224  S410  CON    2   1:1   21    chr21:27018589:G:A   chr21:28152094:A:G   \n",
       "225  S411  CON    2   1:1   22    chr22:27779753:G:A   chr22:30214581:C:T   \n",
       "226  S413  CON    2   1:1   22    chr22:30558084:G:A  chr22:32332418:T:TC   \n",
       "227  S414  CON    2   2:0   22    chr22:48192677:C:G   chr22:49350260:T:G   \n",
       "\n",
       "           BP1        BP2        KB  ...  NcasesWithROHs  NcontrolsWithROHs  \\\n",
       "0     27899165   28207771   308.607  ...              10                  7   \n",
       "1     46907634   47646564   738.931  ...               7                  9   \n",
       "2     99012219   99904724   892.506  ...               8                  6   \n",
       "3    111278914  112549081  1270.168  ...               7                  6   \n",
       "4     37413507   38306784   893.278  ...               7                  5   \n",
       "..         ...        ...       ...  ...             ...                ...   \n",
       "223   24522616   26423081  1900.466  ...               1                  1   \n",
       "224   27018589   28152094  1133.506  ...               1                  1   \n",
       "225   27779753   30214581  2434.829  ...               1                  1   \n",
       "226   30558084   32332418  1774.335  ...               1                  1   \n",
       "227   48192677   49350260  1157.584  ...               2                  0   \n",
       "\n",
       "     combinedN         P  total_ROH_count  propCases  propControls  \\\n",
       "0           17  0.152806               17   0.109890      0.050360   \n",
       "1           16  0.928387               16   0.076923      0.064748   \n",
       "2           14  0.268764               14   0.087912      0.043165   \n",
       "3           13  0.428293               13   0.076923      0.043165   \n",
       "4           12  0.288016               12   0.076923      0.035971   \n",
       "..         ...       ...              ...        ...           ...   \n",
       "223          2  1.000000                2   0.010989      0.007194   \n",
       "224          2  1.000000                2   0.010989      0.007194   \n",
       "225          2  1.000000                2   0.010989      0.007194   \n",
       "226          2  1.000000                2   0.010989      0.007194   \n",
       "227          2  0.303340                2   0.021978      0.000000   \n",
       "\n",
       "     caseEnriched  passMultiTest      BONF  \n",
       "0               1              0  0.000670  \n",
       "1               1              0  0.004072  \n",
       "2               1              0  0.001179  \n",
       "3               1              0  0.001878  \n",
       "4               1              0  0.001263  \n",
       "..            ...            ...       ...  \n",
       "223             1              0  0.004386  \n",
       "224             1              0  0.004386  \n",
       "225             1              0  0.004386  \n",
       "226             1              0  0.004386  \n",
       "227             1              0  0.001330  \n",
       "\n",
       "[228 rows x 25 columns]"
      ]
     },
     "execution_count": 69,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Read the case-enriched ROH data from the file\n",
    "ROH = pd.read_csv(f'{WORK_DIR}/{ANC}_ROH.EOPD_caseEnriched.txt')\n",
    "\n",
    "# Display the DataFrame\n",
    "ROH"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>POOL</th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PHE</th>\n",
       "      <th>CHR</th>\n",
       "      <th>SNP1</th>\n",
       "      <th>SNP2</th>\n",
       "      <th>BP1</th>\n",
       "      <th>BP2</th>\n",
       "      <th>KB</th>\n",
       "      <th>...</th>\n",
       "      <th>NcasesWithROHs</th>\n",
       "      <th>NcontrolsWithROHs</th>\n",
       "      <th>combinedN</th>\n",
       "      <th>P</th>\n",
       "      <th>total_ROH_count</th>\n",
       "      <th>propCases</th>\n",
       "      <th>propControls</th>\n",
       "      <th>caseEnriched</th>\n",
       "      <th>passMultiTest</th>\n",
       "      <th>BONF</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>S97</td>\n",
       "      <td>CON</td>\n",
       "      <td>5</td>\n",
       "      <td>5:0</td>\n",
       "      <td>4</td>\n",
       "      <td>chr4:121656615:T:G</td>\n",
       "      <td>chr4:121970619:G:C</td>\n",
       "      <td>121656615</td>\n",
       "      <td>121970619</td>\n",
       "      <td>314.005</td>\n",
       "      <td>...</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>0.019712</td>\n",
       "      <td>5</td>\n",
       "      <td>0.054945</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000086</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>S127</td>\n",
       "      <td>CON</td>\n",
       "      <td>4</td>\n",
       "      <td>4:0</td>\n",
       "      <td>4</td>\n",
       "      <td>chr4:114942670:T:A</td>\n",
       "      <td>chr4:116313712:G:A</td>\n",
       "      <td>114942670</td>\n",
       "      <td>116313712</td>\n",
       "      <td>1371.043</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>0.047947</td>\n",
       "      <td>4</td>\n",
       "      <td>0.043956</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000210</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>S128</td>\n",
       "      <td>CON</td>\n",
       "      <td>4</td>\n",
       "      <td>4:0</td>\n",
       "      <td>4</td>\n",
       "      <td>chr4:130700347:C:G</td>\n",
       "      <td>chr4:131682533:A:C</td>\n",
       "      <td>130700347</td>\n",
       "      <td>131682533</td>\n",
       "      <td>982.187</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>0.047947</td>\n",
       "      <td>4</td>\n",
       "      <td>0.043956</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000210</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>S129</td>\n",
       "      <td>CON</td>\n",
       "      <td>4</td>\n",
       "      <td>4:0</td>\n",
       "      <td>4</td>\n",
       "      <td>chr4:131985505:A:G</td>\n",
       "      <td>chr4:132300720:G:C</td>\n",
       "      <td>131985505</td>\n",
       "      <td>132300720</td>\n",
       "      <td>315.216</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>0.047947</td>\n",
       "      <td>4</td>\n",
       "      <td>0.043956</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000210</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>77</th>\n",
       "      <td>S167</td>\n",
       "      <td>CON</td>\n",
       "      <td>3</td>\n",
       "      <td>3:0</td>\n",
       "      <td>1</td>\n",
       "      <td>chr1:15673611:C:T</td>\n",
       "      <td>chr1:17215454:A:G</td>\n",
       "      <td>15673611</td>\n",
       "      <td>17215454</td>\n",
       "      <td>1541.844</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0.118637</td>\n",
       "      <td>3</td>\n",
       "      <td>0.032967</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000520</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78</th>\n",
       "      <td>S168</td>\n",
       "      <td>CON</td>\n",
       "      <td>3</td>\n",
       "      <td>3:0</td>\n",
       "      <td>1</td>\n",
       "      <td>chr1:31767341:A:G</td>\n",
       "      <td>chr1:33234313:G:A</td>\n",
       "      <td>31767341</td>\n",
       "      <td>33234313</td>\n",
       "      <td>1466.973</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0.118637</td>\n",
       "      <td>3</td>\n",
       "      <td>0.032967</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000520</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86</th>\n",
       "      <td>S190</td>\n",
       "      <td>CON</td>\n",
       "      <td>3</td>\n",
       "      <td>3:0</td>\n",
       "      <td>3</td>\n",
       "      <td>chr3:120504534:G:C</td>\n",
       "      <td>chr3:122063104:T:A</td>\n",
       "      <td>120504534</td>\n",
       "      <td>122063104</td>\n",
       "      <td>1558.571</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0.118637</td>\n",
       "      <td>3</td>\n",
       "      <td>0.032967</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000520</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87</th>\n",
       "      <td>S191</td>\n",
       "      <td>CON</td>\n",
       "      <td>3</td>\n",
       "      <td>3:0</td>\n",
       "      <td>3</td>\n",
       "      <td>chr3:135906881:A:T</td>\n",
       "      <td>chr3:137453392:T:A</td>\n",
       "      <td>135906881</td>\n",
       "      <td>137453392</td>\n",
       "      <td>1546.512</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0.118637</td>\n",
       "      <td>3</td>\n",
       "      <td>0.032967</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000520</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>90</th>\n",
       "      <td>S198</td>\n",
       "      <td>CON</td>\n",
       "      <td>3</td>\n",
       "      <td>3:0</td>\n",
       "      <td>4</td>\n",
       "      <td>chr4:12655552:T:A</td>\n",
       "      <td>chr4:13756738:G:A</td>\n",
       "      <td>12655552</td>\n",
       "      <td>13756738</td>\n",
       "      <td>1101.187</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0.118637</td>\n",
       "      <td>3</td>\n",
       "      <td>0.032967</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000520</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91</th>\n",
       "      <td>S199</td>\n",
       "      <td>CON</td>\n",
       "      <td>3</td>\n",
       "      <td>3:0</td>\n",
       "      <td>4</td>\n",
       "      <td>chr4:71774726:T:G</td>\n",
       "      <td>chr4:73412553:C:T</td>\n",
       "      <td>71774726</td>\n",
       "      <td>73412553</td>\n",
       "      <td>1637.828</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0.118637</td>\n",
       "      <td>3</td>\n",
       "      <td>0.032967</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000520</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>94</th>\n",
       "      <td>S202</td>\n",
       "      <td>CON</td>\n",
       "      <td>3</td>\n",
       "      <td>3:0</td>\n",
       "      <td>4</td>\n",
       "      <td>chr4:112860471:G:C</td>\n",
       "      <td>chr4:114748490:G:T</td>\n",
       "      <td>112860471</td>\n",
       "      <td>114748490</td>\n",
       "      <td>1888.020</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0.118637</td>\n",
       "      <td>3</td>\n",
       "      <td>0.032967</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000520</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>95</th>\n",
       "      <td>S203</td>\n",
       "      <td>CON</td>\n",
       "      <td>3</td>\n",
       "      <td>3:0</td>\n",
       "      <td>4</td>\n",
       "      <td>chr4:116523818:C:T</td>\n",
       "      <td>chr4:116681540:G:A</td>\n",
       "      <td>116523818</td>\n",
       "      <td>116681540</td>\n",
       "      <td>157.723</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0.118637</td>\n",
       "      <td>3</td>\n",
       "      <td>0.032967</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000520</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>S206</td>\n",
       "      <td>CON</td>\n",
       "      <td>3</td>\n",
       "      <td>3:0</td>\n",
       "      <td>5</td>\n",
       "      <td>chr5:61627648:G:A</td>\n",
       "      <td>chr5:63237065:A:G</td>\n",
       "      <td>61627648</td>\n",
       "      <td>63237065</td>\n",
       "      <td>1609.418</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0.118637</td>\n",
       "      <td>3</td>\n",
       "      <td>0.032967</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000520</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>100</th>\n",
       "      <td>S214</td>\n",
       "      <td>CON</td>\n",
       "      <td>3</td>\n",
       "      <td>3:0</td>\n",
       "      <td>6</td>\n",
       "      <td>chr6:161389511:A:C</td>\n",
       "      <td>chr6:161815720:A:G</td>\n",
       "      <td>161389511</td>\n",
       "      <td>161815720</td>\n",
       "      <td>426.210</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0.118637</td>\n",
       "      <td>3</td>\n",
       "      <td>0.032967</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000520</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101</th>\n",
       "      <td>S215</td>\n",
       "      <td>CON</td>\n",
       "      <td>3</td>\n",
       "      <td>3:0</td>\n",
       "      <td>6</td>\n",
       "      <td>chr6:162375099:C:G</td>\n",
       "      <td>chr6:163939035:A:T</td>\n",
       "      <td>162375099</td>\n",
       "      <td>163939035</td>\n",
       "      <td>1563.937</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0.118637</td>\n",
       "      <td>3</td>\n",
       "      <td>0.032967</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000520</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102</th>\n",
       "      <td>S216</td>\n",
       "      <td>CON</td>\n",
       "      <td>3</td>\n",
       "      <td>3:0</td>\n",
       "      <td>6</td>\n",
       "      <td>chr6:164604500:C:T</td>\n",
       "      <td>chr6:165879923:CG:C</td>\n",
       "      <td>164604500</td>\n",
       "      <td>165879923</td>\n",
       "      <td>1275.424</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0.118637</td>\n",
       "      <td>3</td>\n",
       "      <td>0.032967</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000520</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>103</th>\n",
       "      <td>S217</td>\n",
       "      <td>CON</td>\n",
       "      <td>3</td>\n",
       "      <td>3:0</td>\n",
       "      <td>6</td>\n",
       "      <td>chr6:166257344:C:A</td>\n",
       "      <td>chr6:166408820:A:C</td>\n",
       "      <td>166257344</td>\n",
       "      <td>166408820</td>\n",
       "      <td>151.477</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0.118637</td>\n",
       "      <td>3</td>\n",
       "      <td>0.032967</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000520</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>104</th>\n",
       "      <td>S218</td>\n",
       "      <td>CON</td>\n",
       "      <td>3</td>\n",
       "      <td>3:0</td>\n",
       "      <td>6</td>\n",
       "      <td>chr6:166471905:G:C</td>\n",
       "      <td>chr6:168176780:C:T</td>\n",
       "      <td>166471905</td>\n",
       "      <td>168176780</td>\n",
       "      <td>1704.876</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0.118637</td>\n",
       "      <td>3</td>\n",
       "      <td>0.032967</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000520</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>106</th>\n",
       "      <td>S222</td>\n",
       "      <td>CON</td>\n",
       "      <td>3</td>\n",
       "      <td>3:0</td>\n",
       "      <td>9</td>\n",
       "      <td>chr9:81361754:G:C</td>\n",
       "      <td>chr9:83336556:T:C</td>\n",
       "      <td>81361754</td>\n",
       "      <td>83336556</td>\n",
       "      <td>1974.803</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0.118637</td>\n",
       "      <td>3</td>\n",
       "      <td>0.032967</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000520</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>107</th>\n",
       "      <td>S226</td>\n",
       "      <td>CON</td>\n",
       "      <td>3</td>\n",
       "      <td>3:0</td>\n",
       "      <td>10</td>\n",
       "      <td>chr10:24128513:T:C</td>\n",
       "      <td>chr10:25123571:A:G</td>\n",
       "      <td>24128513</td>\n",
       "      <td>25123571</td>\n",
       "      <td>995.059</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0.118637</td>\n",
       "      <td>3</td>\n",
       "      <td>0.032967</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000520</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>108</th>\n",
       "      <td>S227</td>\n",
       "      <td>CON</td>\n",
       "      <td>3</td>\n",
       "      <td>3:0</td>\n",
       "      <td>10</td>\n",
       "      <td>chr10:25129805:G:C</td>\n",
       "      <td>chr10:25475212:G:A</td>\n",
       "      <td>25129805</td>\n",
       "      <td>25475212</td>\n",
       "      <td>345.408</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0.118637</td>\n",
       "      <td>3</td>\n",
       "      <td>0.032967</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000520</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>118</th>\n",
       "      <td>S254</td>\n",
       "      <td>CON</td>\n",
       "      <td>3</td>\n",
       "      <td>3:0</td>\n",
       "      <td>17</td>\n",
       "      <td>chr17:44995295:C:T</td>\n",
       "      <td>chr17:45483692:C:T</td>\n",
       "      <td>44995295</td>\n",
       "      <td>45483692</td>\n",
       "      <td>488.398</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0.118637</td>\n",
       "      <td>3</td>\n",
       "      <td>0.032967</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000520</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>119</th>\n",
       "      <td>S255</td>\n",
       "      <td>CON</td>\n",
       "      <td>3</td>\n",
       "      <td>3:0</td>\n",
       "      <td>17</td>\n",
       "      <td>chr17:45869341:T:C</td>\n",
       "      <td>chr17:46788635:A:C</td>\n",
       "      <td>45869341</td>\n",
       "      <td>46788635</td>\n",
       "      <td>919.295</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0.118637</td>\n",
       "      <td>3</td>\n",
       "      <td>0.032967</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000520</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>122</th>\n",
       "      <td>S259</td>\n",
       "      <td>CON</td>\n",
       "      <td>3</td>\n",
       "      <td>3:0</td>\n",
       "      <td>18</td>\n",
       "      <td>chr18:56098540:T:C</td>\n",
       "      <td>chr18:56242791:G:A</td>\n",
       "      <td>56098540</td>\n",
       "      <td>56242791</td>\n",
       "      <td>144.252</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0.118637</td>\n",
       "      <td>3</td>\n",
       "      <td>0.032967</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000520</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>127</th>\n",
       "      <td>S270</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>1</td>\n",
       "      <td>chr1:12206555:A:G</td>\n",
       "      <td>chr1:13793755:C:G</td>\n",
       "      <td>12206555</td>\n",
       "      <td>13793755</td>\n",
       "      <td>1587.201</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.303340</td>\n",
       "      <td>2</td>\n",
       "      <td>0.021978</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.001330</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>129</th>\n",
       "      <td>S272</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>1</td>\n",
       "      <td>chr1:27325641:A:G</td>\n",
       "      <td>chr1:29547214:C:T</td>\n",
       "      <td>27325641</td>\n",
       "      <td>29547214</td>\n",
       "      <td>2221.574</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.303340</td>\n",
       "      <td>2</td>\n",
       "      <td>0.021978</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.001330</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>130</th>\n",
       "      <td>S273</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>1</td>\n",
       "      <td>chr1:29689974:G:T</td>\n",
       "      <td>chr1:30027088:A:G</td>\n",
       "      <td>29689974</td>\n",
       "      <td>30027088</td>\n",
       "      <td>337.115</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.303340</td>\n",
       "      <td>2</td>\n",
       "      <td>0.021978</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.001330</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>134</th>\n",
       "      <td>S281</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>1</td>\n",
       "      <td>chr1:113312602:T:C</td>\n",
       "      <td>chr1:116481574:C:A</td>\n",
       "      <td>113312602</td>\n",
       "      <td>116481574</td>\n",
       "      <td>3168.973</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.303340</td>\n",
       "      <td>2</td>\n",
       "      <td>0.021978</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.001330</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139</th>\n",
       "      <td>S293</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>2</td>\n",
       "      <td>chr2:109485238:TG:T</td>\n",
       "      <td>chr2:111196746:T:C</td>\n",
       "      <td>109485238</td>\n",
       "      <td>111196746</td>\n",
       "      <td>1711.509</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.303340</td>\n",
       "      <td>2</td>\n",
       "      <td>0.021978</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.001330</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>145</th>\n",
       "      <td>S303</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>3</td>\n",
       "      <td>chr3:69864088:C:T</td>\n",
       "      <td>chr3:70323580:A:C</td>\n",
       "      <td>69864088</td>\n",
       "      <td>70323580</td>\n",
       "      <td>459.493</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.303340</td>\n",
       "      <td>2</td>\n",
       "      <td>0.021978</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.001330</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>150</th>\n",
       "      <td>S309</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>3</td>\n",
       "      <td>chr3:133428876:C:G</td>\n",
       "      <td>chr3:135199680:C:T</td>\n",
       "      <td>133428876</td>\n",
       "      <td>135199680</td>\n",
       "      <td>1770.805</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.303340</td>\n",
       "      <td>2</td>\n",
       "      <td>0.021978</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.001330</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>153</th>\n",
       "      <td>S315</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>4</td>\n",
       "      <td>chr4:78888378:G:T</td>\n",
       "      <td>chr4:79617847:C:T</td>\n",
       "      <td>78888378</td>\n",
       "      <td>79617847</td>\n",
       "      <td>729.470</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.303340</td>\n",
       "      <td>2</td>\n",
       "      <td>0.021978</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.001330</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>157</th>\n",
       "      <td>S320</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>4</td>\n",
       "      <td>chr4:134974414:C:G</td>\n",
       "      <td>chr4:136728215:GA:G</td>\n",
       "      <td>134974414</td>\n",
       "      <td>136728215</td>\n",
       "      <td>1753.802</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.303340</td>\n",
       "      <td>2</td>\n",
       "      <td>0.021978</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.001330</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>158</th>\n",
       "      <td>S321</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>4</td>\n",
       "      <td>chr4:137045526:T:C</td>\n",
       "      <td>chr4:139802325:T:A</td>\n",
       "      <td>137045526</td>\n",
       "      <td>139802325</td>\n",
       "      <td>2756.800</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.303340</td>\n",
       "      <td>2</td>\n",
       "      <td>0.021978</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.001330</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>161</th>\n",
       "      <td>S326</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>5</td>\n",
       "      <td>chr5:91353481:C:T</td>\n",
       "      <td>chr5:93308527:C:T</td>\n",
       "      <td>91353481</td>\n",
       "      <td>93308527</td>\n",
       "      <td>1955.047</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.303340</td>\n",
       "      <td>2</td>\n",
       "      <td>0.021978</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.001330</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>162</th>\n",
       "      <td>S327</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>5</td>\n",
       "      <td>chr5:113201365:G:A</td>\n",
       "      <td>chr5:115316816:G:C</td>\n",
       "      <td>113201365</td>\n",
       "      <td>115316816</td>\n",
       "      <td>2115.452</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.303340</td>\n",
       "      <td>2</td>\n",
       "      <td>0.021978</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.001330</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>169</th>\n",
       "      <td>S344</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>7</td>\n",
       "      <td>chr7:21546333:G:C</td>\n",
       "      <td>chr7:24060359:C:G</td>\n",
       "      <td>21546333</td>\n",
       "      <td>24060359</td>\n",
       "      <td>2514.027</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.303340</td>\n",
       "      <td>2</td>\n",
       "      <td>0.021978</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.001330</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170</th>\n",
       "      <td>S345</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>7</td>\n",
       "      <td>chr7:28265918:G:A</td>\n",
       "      <td>chr7:29111400:G:C</td>\n",
       "      <td>28265918</td>\n",
       "      <td>29111400</td>\n",
       "      <td>845.483</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.303340</td>\n",
       "      <td>2</td>\n",
       "      <td>0.021978</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.001330</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>171</th>\n",
       "      <td>S346</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>7</td>\n",
       "      <td>chr7:30915502:A:G</td>\n",
       "      <td>chr7:32277201:A:C</td>\n",
       "      <td>30915502</td>\n",
       "      <td>32277201</td>\n",
       "      <td>1361.700</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.303340</td>\n",
       "      <td>2</td>\n",
       "      <td>0.021978</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.001330</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>172</th>\n",
       "      <td>S347</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>7</td>\n",
       "      <td>chr7:32409956:C:T</td>\n",
       "      <td>chr7:34217918:G:A</td>\n",
       "      <td>32409956</td>\n",
       "      <td>34217918</td>\n",
       "      <td>1807.963</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.303340</td>\n",
       "      <td>2</td>\n",
       "      <td>0.021978</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.001330</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>173</th>\n",
       "      <td>S349</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>7</td>\n",
       "      <td>chr7:113884394:G:A</td>\n",
       "      <td>chr7:115481028:G:A</td>\n",
       "      <td>113884394</td>\n",
       "      <td>115481028</td>\n",
       "      <td>1596.635</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.303340</td>\n",
       "      <td>2</td>\n",
       "      <td>0.021978</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.001330</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>177</th>\n",
       "      <td>S354</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>8</td>\n",
       "      <td>chr8:63605264:G:A</td>\n",
       "      <td>chr8:67767035:T:G</td>\n",
       "      <td>63605264</td>\n",
       "      <td>67767035</td>\n",
       "      <td>4161.772</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.303340</td>\n",
       "      <td>2</td>\n",
       "      <td>0.021978</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.001330</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>179</th>\n",
       "      <td>S356</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>8</td>\n",
       "      <td>chr8:118092082:G:A</td>\n",
       "      <td>chr8:119982154:A:G</td>\n",
       "      <td>118092082</td>\n",
       "      <td>119982154</td>\n",
       "      <td>1890.073</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.303340</td>\n",
       "      <td>2</td>\n",
       "      <td>0.021978</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.001330</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>181</th>\n",
       "      <td>S358</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>9</td>\n",
       "      <td>chr9:10783920:C:T</td>\n",
       "      <td>chr9:12375732:A:G</td>\n",
       "      <td>10783920</td>\n",
       "      <td>12375732</td>\n",
       "      <td>1591.813</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.303340</td>\n",
       "      <td>2</td>\n",
       "      <td>0.021978</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.001330</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>188</th>\n",
       "      <td>S367</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>11</td>\n",
       "      <td>chr11:27900385:T:C</td>\n",
       "      <td>chr11:29955694:C:A</td>\n",
       "      <td>27900385</td>\n",
       "      <td>29955694</td>\n",
       "      <td>2055.310</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.303340</td>\n",
       "      <td>2</td>\n",
       "      <td>0.021978</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.001330</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>190</th>\n",
       "      <td>S370</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>11</td>\n",
       "      <td>chr11:97985078:G:C</td>\n",
       "      <td>chr11:98760670:T:C</td>\n",
       "      <td>97985078</td>\n",
       "      <td>98760670</td>\n",
       "      <td>775.593</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.303340</td>\n",
       "      <td>2</td>\n",
       "      <td>0.021978</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.001330</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>191</th>\n",
       "      <td>S371</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>11</td>\n",
       "      <td>chr11:99170300:A:G</td>\n",
       "      <td>chr11:99733091:C:T</td>\n",
       "      <td>99170300</td>\n",
       "      <td>99733091</td>\n",
       "      <td>562.792</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.303340</td>\n",
       "      <td>2</td>\n",
       "      <td>0.021978</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.001330</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>192</th>\n",
       "      <td>S372</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>11</td>\n",
       "      <td>chr11:101446584:A:G</td>\n",
       "      <td>chr11:101637914:C:T</td>\n",
       "      <td>101446584</td>\n",
       "      <td>101637914</td>\n",
       "      <td>191.331</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.303340</td>\n",
       "      <td>2</td>\n",
       "      <td>0.021978</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.001330</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>195</th>\n",
       "      <td>S375</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>12</td>\n",
       "      <td>chr12:32943085:C:T</td>\n",
       "      <td>chr12:34717351:T:G</td>\n",
       "      <td>32943085</td>\n",
       "      <td>34717351</td>\n",
       "      <td>1774.267</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.303340</td>\n",
       "      <td>2</td>\n",
       "      <td>0.021978</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.001330</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>199</th>\n",
       "      <td>S382</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>12</td>\n",
       "      <td>chr12:117954153:G:A</td>\n",
       "      <td>chr12:118808222:G:A</td>\n",
       "      <td>117954153</td>\n",
       "      <td>118808222</td>\n",
       "      <td>854.070</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.303340</td>\n",
       "      <td>2</td>\n",
       "      <td>0.021978</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.001330</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>204</th>\n",
       "      <td>S388</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>14</td>\n",
       "      <td>chr14:70881249:T:C</td>\n",
       "      <td>chr14:72425523:A:T</td>\n",
       "      <td>70881249</td>\n",
       "      <td>72425523</td>\n",
       "      <td>1544.275</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.303340</td>\n",
       "      <td>2</td>\n",
       "      <td>0.021978</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.001330</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208</th>\n",
       "      <td>S394</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>15</td>\n",
       "      <td>chr15:56031825:C:T</td>\n",
       "      <td>chr15:56521535:A:T</td>\n",
       "      <td>56031825</td>\n",
       "      <td>56521535</td>\n",
       "      <td>489.711</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.303340</td>\n",
       "      <td>2</td>\n",
       "      <td>0.021978</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.001330</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>209</th>\n",
       "      <td>S395</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>15</td>\n",
       "      <td>chr15:82099391:A:G</td>\n",
       "      <td>chr15:84843059:G:A</td>\n",
       "      <td>82099391</td>\n",
       "      <td>84843059</td>\n",
       "      <td>2743.669</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.303340</td>\n",
       "      <td>2</td>\n",
       "      <td>0.021978</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.001330</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>213</th>\n",
       "      <td>S399</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>17</td>\n",
       "      <td>chr17:12661342:C:A</td>\n",
       "      <td>chr17:13723171:C:T</td>\n",
       "      <td>12661342</td>\n",
       "      <td>13723171</td>\n",
       "      <td>1061.830</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.303340</td>\n",
       "      <td>2</td>\n",
       "      <td>0.021978</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.001330</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>216</th>\n",
       "      <td>S402</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>17</td>\n",
       "      <td>chr17:36843262:T:G</td>\n",
       "      <td>chr17:40599413:C:T</td>\n",
       "      <td>36843262</td>\n",
       "      <td>40599413</td>\n",
       "      <td>3756.152</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.303340</td>\n",
       "      <td>2</td>\n",
       "      <td>0.021978</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.001330</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>217</th>\n",
       "      <td>S403</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>17</td>\n",
       "      <td>chr17:41816525:G:A</td>\n",
       "      <td>chr17:43433482:A:G</td>\n",
       "      <td>41816525</td>\n",
       "      <td>43433482</td>\n",
       "      <td>1616.958</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.303340</td>\n",
       "      <td>2</td>\n",
       "      <td>0.021978</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.001330</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>218</th>\n",
       "      <td>S404</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>18</td>\n",
       "      <td>chr18:28108827:C:T</td>\n",
       "      <td>chr18:29174653:A:C</td>\n",
       "      <td>28108827</td>\n",
       "      <td>29174653</td>\n",
       "      <td>1065.827</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.303340</td>\n",
       "      <td>2</td>\n",
       "      <td>0.021978</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.001330</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>219</th>\n",
       "      <td>S405</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>18</td>\n",
       "      <td>chr18:60006901:T:A</td>\n",
       "      <td>chr18:61518053:C:A</td>\n",
       "      <td>60006901</td>\n",
       "      <td>61518053</td>\n",
       "      <td>1511.153</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.303340</td>\n",
       "      <td>2</td>\n",
       "      <td>0.021978</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.001330</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>222</th>\n",
       "      <td>S408</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>20</td>\n",
       "      <td>chr20:62091537:A:G</td>\n",
       "      <td>chr20:62411193:T:C</td>\n",
       "      <td>62091537</td>\n",
       "      <td>62411193</td>\n",
       "      <td>319.657</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.303340</td>\n",
       "      <td>2</td>\n",
       "      <td>0.021978</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.001330</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>227</th>\n",
       "      <td>S414</td>\n",
       "      <td>CON</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>22</td>\n",
       "      <td>chr22:48192677:C:G</td>\n",
       "      <td>chr22:49350260:T:G</td>\n",
       "      <td>48192677</td>\n",
       "      <td>49350260</td>\n",
       "      <td>1157.584</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.303340</td>\n",
       "      <td>2</td>\n",
       "      <td>0.021978</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.001330</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>60 rows × 25 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     POOL  FID  IID  PHE  CHR                 SNP1                 SNP2  \\\n",
       "31    S97  CON    5  5:0    4   chr4:121656615:T:G   chr4:121970619:G:C   \n",
       "53   S127  CON    4  4:0    4   chr4:114942670:T:A   chr4:116313712:G:A   \n",
       "54   S128  CON    4  4:0    4   chr4:130700347:C:G   chr4:131682533:A:C   \n",
       "55   S129  CON    4  4:0    4   chr4:131985505:A:G   chr4:132300720:G:C   \n",
       "77   S167  CON    3  3:0    1    chr1:15673611:C:T    chr1:17215454:A:G   \n",
       "78   S168  CON    3  3:0    1    chr1:31767341:A:G    chr1:33234313:G:A   \n",
       "86   S190  CON    3  3:0    3   chr3:120504534:G:C   chr3:122063104:T:A   \n",
       "87   S191  CON    3  3:0    3   chr3:135906881:A:T   chr3:137453392:T:A   \n",
       "90   S198  CON    3  3:0    4    chr4:12655552:T:A    chr4:13756738:G:A   \n",
       "91   S199  CON    3  3:0    4    chr4:71774726:T:G    chr4:73412553:C:T   \n",
       "94   S202  CON    3  3:0    4   chr4:112860471:G:C   chr4:114748490:G:T   \n",
       "95   S203  CON    3  3:0    4   chr4:116523818:C:T   chr4:116681540:G:A   \n",
       "97   S206  CON    3  3:0    5    chr5:61627648:G:A    chr5:63237065:A:G   \n",
       "100  S214  CON    3  3:0    6   chr6:161389511:A:C   chr6:161815720:A:G   \n",
       "101  S215  CON    3  3:0    6   chr6:162375099:C:G   chr6:163939035:A:T   \n",
       "102  S216  CON    3  3:0    6   chr6:164604500:C:T  chr6:165879923:CG:C   \n",
       "103  S217  CON    3  3:0    6   chr6:166257344:C:A   chr6:166408820:A:C   \n",
       "104  S218  CON    3  3:0    6   chr6:166471905:G:C   chr6:168176780:C:T   \n",
       "106  S222  CON    3  3:0    9    chr9:81361754:G:C    chr9:83336556:T:C   \n",
       "107  S226  CON    3  3:0   10   chr10:24128513:T:C   chr10:25123571:A:G   \n",
       "108  S227  CON    3  3:0   10   chr10:25129805:G:C   chr10:25475212:G:A   \n",
       "118  S254  CON    3  3:0   17   chr17:44995295:C:T   chr17:45483692:C:T   \n",
       "119  S255  CON    3  3:0   17   chr17:45869341:T:C   chr17:46788635:A:C   \n",
       "122  S259  CON    3  3:0   18   chr18:56098540:T:C   chr18:56242791:G:A   \n",
       "127  S270  CON    2  2:0    1    chr1:12206555:A:G    chr1:13793755:C:G   \n",
       "129  S272  CON    2  2:0    1    chr1:27325641:A:G    chr1:29547214:C:T   \n",
       "130  S273  CON    2  2:0    1    chr1:29689974:G:T    chr1:30027088:A:G   \n",
       "134  S281  CON    2  2:0    1   chr1:113312602:T:C   chr1:116481574:C:A   \n",
       "139  S293  CON    2  2:0    2  chr2:109485238:TG:T   chr2:111196746:T:C   \n",
       "145  S303  CON    2  2:0    3    chr3:69864088:C:T    chr3:70323580:A:C   \n",
       "150  S309  CON    2  2:0    3   chr3:133428876:C:G   chr3:135199680:C:T   \n",
       "153  S315  CON    2  2:0    4    chr4:78888378:G:T    chr4:79617847:C:T   \n",
       "157  S320  CON    2  2:0    4   chr4:134974414:C:G  chr4:136728215:GA:G   \n",
       "158  S321  CON    2  2:0    4   chr4:137045526:T:C   chr4:139802325:T:A   \n",
       "161  S326  CON    2  2:0    5    chr5:91353481:C:T    chr5:93308527:C:T   \n",
       "162  S327  CON    2  2:0    5   chr5:113201365:G:A   chr5:115316816:G:C   \n",
       "169  S344  CON    2  2:0    7    chr7:21546333:G:C    chr7:24060359:C:G   \n",
       "170  S345  CON    2  2:0    7    chr7:28265918:G:A    chr7:29111400:G:C   \n",
       "171  S346  CON    2  2:0    7    chr7:30915502:A:G    chr7:32277201:A:C   \n",
       "172  S347  CON    2  2:0    7    chr7:32409956:C:T    chr7:34217918:G:A   \n",
       "173  S349  CON    2  2:0    7   chr7:113884394:G:A   chr7:115481028:G:A   \n",
       "177  S354  CON    2  2:0    8    chr8:63605264:G:A    chr8:67767035:T:G   \n",
       "179  S356  CON    2  2:0    8   chr8:118092082:G:A   chr8:119982154:A:G   \n",
       "181  S358  CON    2  2:0    9    chr9:10783920:C:T    chr9:12375732:A:G   \n",
       "188  S367  CON    2  2:0   11   chr11:27900385:T:C   chr11:29955694:C:A   \n",
       "190  S370  CON    2  2:0   11   chr11:97985078:G:C   chr11:98760670:T:C   \n",
       "191  S371  CON    2  2:0   11   chr11:99170300:A:G   chr11:99733091:C:T   \n",
       "192  S372  CON    2  2:0   11  chr11:101446584:A:G  chr11:101637914:C:T   \n",
       "195  S375  CON    2  2:0   12   chr12:32943085:C:T   chr12:34717351:T:G   \n",
       "199  S382  CON    2  2:0   12  chr12:117954153:G:A  chr12:118808222:G:A   \n",
       "204  S388  CON    2  2:0   14   chr14:70881249:T:C   chr14:72425523:A:T   \n",
       "208  S394  CON    2  2:0   15   chr15:56031825:C:T   chr15:56521535:A:T   \n",
       "209  S395  CON    2  2:0   15   chr15:82099391:A:G   chr15:84843059:G:A   \n",
       "213  S399  CON    2  2:0   17   chr17:12661342:C:A   chr17:13723171:C:T   \n",
       "216  S402  CON    2  2:0   17   chr17:36843262:T:G   chr17:40599413:C:T   \n",
       "217  S403  CON    2  2:0   17   chr17:41816525:G:A   chr17:43433482:A:G   \n",
       "218  S404  CON    2  2:0   18   chr18:28108827:C:T   chr18:29174653:A:C   \n",
       "219  S405  CON    2  2:0   18   chr18:60006901:T:A   chr18:61518053:C:A   \n",
       "222  S408  CON    2  2:0   20   chr20:62091537:A:G   chr20:62411193:T:C   \n",
       "227  S414  CON    2  2:0   22   chr22:48192677:C:G   chr22:49350260:T:G   \n",
       "\n",
       "           BP1        BP2        KB  ...  NcasesWithROHs  NcontrolsWithROHs  \\\n",
       "31   121656615  121970619   314.005  ...               5                  0   \n",
       "53   114942670  116313712  1371.043  ...               4                  0   \n",
       "54   130700347  131682533   982.187  ...               4                  0   \n",
       "55   131985505  132300720   315.216  ...               4                  0   \n",
       "77    15673611   17215454  1541.844  ...               3                  0   \n",
       "78    31767341   33234313  1466.973  ...               3                  0   \n",
       "86   120504534  122063104  1558.571  ...               3                  0   \n",
       "87   135906881  137453392  1546.512  ...               3                  0   \n",
       "90    12655552   13756738  1101.187  ...               3                  0   \n",
       "91    71774726   73412553  1637.828  ...               3                  0   \n",
       "94   112860471  114748490  1888.020  ...               3                  0   \n",
       "95   116523818  116681540   157.723  ...               3                  0   \n",
       "97    61627648   63237065  1609.418  ...               3                  0   \n",
       "100  161389511  161815720   426.210  ...               3                  0   \n",
       "101  162375099  163939035  1563.937  ...               3                  0   \n",
       "102  164604500  165879923  1275.424  ...               3                  0   \n",
       "103  166257344  166408820   151.477  ...               3                  0   \n",
       "104  166471905  168176780  1704.876  ...               3                  0   \n",
       "106   81361754   83336556  1974.803  ...               3                  0   \n",
       "107   24128513   25123571   995.059  ...               3                  0   \n",
       "108   25129805   25475212   345.408  ...               3                  0   \n",
       "118   44995295   45483692   488.398  ...               3                  0   \n",
       "119   45869341   46788635   919.295  ...               3                  0   \n",
       "122   56098540   56242791   144.252  ...               3                  0   \n",
       "127   12206555   13793755  1587.201  ...               2                  0   \n",
       "129   27325641   29547214  2221.574  ...               2                  0   \n",
       "130   29689974   30027088   337.115  ...               2                  0   \n",
       "134  113312602  116481574  3168.973  ...               2                  0   \n",
       "139  109485238  111196746  1711.509  ...               2                  0   \n",
       "145   69864088   70323580   459.493  ...               2                  0   \n",
       "150  133428876  135199680  1770.805  ...               2                  0   \n",
       "153   78888378   79617847   729.470  ...               2                  0   \n",
       "157  134974414  136728215  1753.802  ...               2                  0   \n",
       "158  137045526  139802325  2756.800  ...               2                  0   \n",
       "161   91353481   93308527  1955.047  ...               2                  0   \n",
       "162  113201365  115316816  2115.452  ...               2                  0   \n",
       "169   21546333   24060359  2514.027  ...               2                  0   \n",
       "170   28265918   29111400   845.483  ...               2                  0   \n",
       "171   30915502   32277201  1361.700  ...               2                  0   \n",
       "172   32409956   34217918  1807.963  ...               2                  0   \n",
       "173  113884394  115481028  1596.635  ...               2                  0   \n",
       "177   63605264   67767035  4161.772  ...               2                  0   \n",
       "179  118092082  119982154  1890.073  ...               2                  0   \n",
       "181   10783920   12375732  1591.813  ...               2                  0   \n",
       "188   27900385   29955694  2055.310  ...               2                  0   \n",
       "190   97985078   98760670   775.593  ...               2                  0   \n",
       "191   99170300   99733091   562.792  ...               2                  0   \n",
       "192  101446584  101637914   191.331  ...               2                  0   \n",
       "195   32943085   34717351  1774.267  ...               2                  0   \n",
       "199  117954153  118808222   854.070  ...               2                  0   \n",
       "204   70881249   72425523  1544.275  ...               2                  0   \n",
       "208   56031825   56521535   489.711  ...               2                  0   \n",
       "209   82099391   84843059  2743.669  ...               2                  0   \n",
       "213   12661342   13723171  1061.830  ...               2                  0   \n",
       "216   36843262   40599413  3756.152  ...               2                  0   \n",
       "217   41816525   43433482  1616.958  ...               2                  0   \n",
       "218   28108827   29174653  1065.827  ...               2                  0   \n",
       "219   60006901   61518053  1511.153  ...               2                  0   \n",
       "222   62091537   62411193   319.657  ...               2                  0   \n",
       "227   48192677   49350260  1157.584  ...               2                  0   \n",
       "\n",
       "     combinedN         P  total_ROH_count  propCases  propControls  \\\n",
       "31           5  0.019712                5   0.054945           0.0   \n",
       "53           4  0.047947                4   0.043956           0.0   \n",
       "54           4  0.047947                4   0.043956           0.0   \n",
       "55           4  0.047947                4   0.043956           0.0   \n",
       "77           3  0.118637                3   0.032967           0.0   \n",
       "78           3  0.118637                3   0.032967           0.0   \n",
       "86           3  0.118637                3   0.032967           0.0   \n",
       "87           3  0.118637                3   0.032967           0.0   \n",
       "90           3  0.118637                3   0.032967           0.0   \n",
       "91           3  0.118637                3   0.032967           0.0   \n",
       "94           3  0.118637                3   0.032967           0.0   \n",
       "95           3  0.118637                3   0.032967           0.0   \n",
       "97           3  0.118637                3   0.032967           0.0   \n",
       "100          3  0.118637                3   0.032967           0.0   \n",
       "101          3  0.118637                3   0.032967           0.0   \n",
       "102          3  0.118637                3   0.032967           0.0   \n",
       "103          3  0.118637                3   0.032967           0.0   \n",
       "104          3  0.118637                3   0.032967           0.0   \n",
       "106          3  0.118637                3   0.032967           0.0   \n",
       "107          3  0.118637                3   0.032967           0.0   \n",
       "108          3  0.118637                3   0.032967           0.0   \n",
       "118          3  0.118637                3   0.032967           0.0   \n",
       "119          3  0.118637                3   0.032967           0.0   \n",
       "122          3  0.118637                3   0.032967           0.0   \n",
       "127          2  0.303340                2   0.021978           0.0   \n",
       "129          2  0.303340                2   0.021978           0.0   \n",
       "130          2  0.303340                2   0.021978           0.0   \n",
       "134          2  0.303340                2   0.021978           0.0   \n",
       "139          2  0.303340                2   0.021978           0.0   \n",
       "145          2  0.303340                2   0.021978           0.0   \n",
       "150          2  0.303340                2   0.021978           0.0   \n",
       "153          2  0.303340                2   0.021978           0.0   \n",
       "157          2  0.303340                2   0.021978           0.0   \n",
       "158          2  0.303340                2   0.021978           0.0   \n",
       "161          2  0.303340                2   0.021978           0.0   \n",
       "162          2  0.303340                2   0.021978           0.0   \n",
       "169          2  0.303340                2   0.021978           0.0   \n",
       "170          2  0.303340                2   0.021978           0.0   \n",
       "171          2  0.303340                2   0.021978           0.0   \n",
       "172          2  0.303340                2   0.021978           0.0   \n",
       "173          2  0.303340                2   0.021978           0.0   \n",
       "177          2  0.303340                2   0.021978           0.0   \n",
       "179          2  0.303340                2   0.021978           0.0   \n",
       "181          2  0.303340                2   0.021978           0.0   \n",
       "188          2  0.303340                2   0.021978           0.0   \n",
       "190          2  0.303340                2   0.021978           0.0   \n",
       "191          2  0.303340                2   0.021978           0.0   \n",
       "192          2  0.303340                2   0.021978           0.0   \n",
       "195          2  0.303340                2   0.021978           0.0   \n",
       "199          2  0.303340                2   0.021978           0.0   \n",
       "204          2  0.303340                2   0.021978           0.0   \n",
       "208          2  0.303340                2   0.021978           0.0   \n",
       "209          2  0.303340                2   0.021978           0.0   \n",
       "213          2  0.303340                2   0.021978           0.0   \n",
       "216          2  0.303340                2   0.021978           0.0   \n",
       "217          2  0.303340                2   0.021978           0.0   \n",
       "218          2  0.303340                2   0.021978           0.0   \n",
       "219          2  0.303340                2   0.021978           0.0   \n",
       "222          2  0.303340                2   0.021978           0.0   \n",
       "227          2  0.303340                2   0.021978           0.0   \n",
       "\n",
       "     caseEnriched  passMultiTest      BONF  \n",
       "31              1              0  0.000086  \n",
       "53              1              0  0.000210  \n",
       "54              1              0  0.000210  \n",
       "55              1              0  0.000210  \n",
       "77              1              0  0.000520  \n",
       "78              1              0  0.000520  \n",
       "86              1              0  0.000520  \n",
       "87              1              0  0.000520  \n",
       "90              1              0  0.000520  \n",
       "91              1              0  0.000520  \n",
       "94              1              0  0.000520  \n",
       "95              1              0  0.000520  \n",
       "97              1              0  0.000520  \n",
       "100             1              0  0.000520  \n",
       "101             1              0  0.000520  \n",
       "102             1              0  0.000520  \n",
       "103             1              0  0.000520  \n",
       "104             1              0  0.000520  \n",
       "106             1              0  0.000520  \n",
       "107             1              0  0.000520  \n",
       "108             1              0  0.000520  \n",
       "118             1              0  0.000520  \n",
       "119             1              0  0.000520  \n",
       "122             1              0  0.000520  \n",
       "127             1              0  0.001330  \n",
       "129             1              0  0.001330  \n",
       "130             1              0  0.001330  \n",
       "134             1              0  0.001330  \n",
       "139             1              0  0.001330  \n",
       "145             1              0  0.001330  \n",
       "150             1              0  0.001330  \n",
       "153             1              0  0.001330  \n",
       "157             1              0  0.001330  \n",
       "158             1              0  0.001330  \n",
       "161             1              0  0.001330  \n",
       "162             1              0  0.001330  \n",
       "169             1              0  0.001330  \n",
       "170             1              0  0.001330  \n",
       "171             1              0  0.001330  \n",
       "172             1              0  0.001330  \n",
       "173             1              0  0.001330  \n",
       "177             1              0  0.001330  \n",
       "179             1              0  0.001330  \n",
       "181             1              0  0.001330  \n",
       "188             1              0  0.001330  \n",
       "190             1              0  0.001330  \n",
       "191             1              0  0.001330  \n",
       "192             1              0  0.001330  \n",
       "195             1              0  0.001330  \n",
       "199             1              0  0.001330  \n",
       "204             1              0  0.001330  \n",
       "208             1              0  0.001330  \n",
       "209             1              0  0.001330  \n",
       "213             1              0  0.001330  \n",
       "216             1              0  0.001330  \n",
       "217             1              0  0.001330  \n",
       "218             1              0  0.001330  \n",
       "219             1              0  0.001330  \n",
       "222             1              0  0.001330  \n",
       "227             1              0  0.001330  \n",
       "\n",
       "[60 rows x 25 columns]"
      ]
     },
     "execution_count": 70,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Filter the ROH data for rows where the 'PHE' column contains ':0'\n",
    "filtered_ROH = ROH.loc[ROH['PHE'].str.contains(':0')]\n",
    "\n",
    "# Display the filtered data\n",
    "filtered_ROH"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "bash: line 4: cd: /home/jupyter/Release6_{ANC}: No such file or directory\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of ROH: 345\n",
      "Number of ROH enriched in Cases: 228\n",
      "Number of ROH that Pass Bonferroni: 0\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "# Set working directory and ancestry code\n",
    "WORK_DIR='/home/jupyter/Release6_{ANC}'\n",
    "cd $WORK_DIR\n",
    "ANC=AMR\n",
    "\n",
    "# Count and display various ROH statistics\n",
    "{\n",
    "    printf \"Number of ROH: \"\n",
    "    wc -l < gp2_${ANC}_r6_EOPD.hom.overlap.filtered\n",
    "\n",
    "    printf \"Number of ROH enriched in Cases: \"\n",
    "    sed 1d < ${ANC}_ROH.EOPD_caseEnriched.txt | wc -l\n",
    "\n",
    "    printf \"Number of ROH that Pass Bonferroni: \"\n",
    "    cut -d',' -f26 < ${ANC}_ROH.EOPD_caseEnriched.txt | sed 1d | { grep -c '1' || true; }\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### In PD genes\n",
    "\n",
    "Examine the ROHs that overlap with genes associated with PD."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "bash: line 4: cd: /home/jupyter/Release6_{ANC}: No such file or directory\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PD gene: LOC100131289\n",
      "PD gene: RSL24D1P1\n",
      "PD gene: RBX1\n",
      "PD gene: SPG11\n",
      "PD gene: TMEM229B\n",
      "PD gene: TMEM163\n",
      "PD gene: TMEM163\n",
      "PD gene: ELOVL7\n",
      "PD gene: ELOVL7\n",
      "PD gene: SLC44A4\n",
      "PD gene: HLA-DRB5\n",
      "PD gene: CD19\n",
      "PD gene: SETD1A\n",
      "PD gene: SETD1A\n",
      "PD gene: ZNF646\n",
      "PD gene: KAT8\n",
      "PD gene: ASXL3\n",
      "PD gene: ASXL3\n",
      "PD gene: ATP13A2\n",
      "PD gene: KPNA1\n",
      "PD gene: KPNA1\n",
      "PD gene: SPTSSB\n",
      "PD gene: SPTSSB\n",
      "PD gene: CAMK2D\n",
      "PD gene: CAMK2D\n",
      "PD gene: ELOVL7\n",
      "PD gene: PARK2\n",
      "PD gene: PARK2\n",
      "PD gene: PARK2\n",
      "PD gene: USP8\n",
      "PD gene: UBTF\n",
      "PD gene: FAM171A2\n",
      "PD gene: MAP3K14\n",
      "PD gene: CRHR1\n",
      "PD gene: MAPT-AS1\n",
      "PD gene: KANSL1\n",
      "PD gene: MAPT\n",
      "PD gene: MAPT\n",
      "PD gene: MAP3K14\n",
      "PD gene: CRHR1\n",
      "PD gene: MAPT-AS1\n",
      "PD gene: KANSL1\n",
      "PD gene: MAPT\n",
      "PD gene: MAPT\n",
      "PD gene: NSF\n",
      "PD gene: WNT3\n",
      "PD gene: WNT3\n",
      "PD gene: NSF\n",
      "PD gene: WNT3\n",
      "PD gene: WNT3\n",
      "PD gene: ARHGAP27\n",
      "PD gene: MYLK2\n",
      "PD gene: HRNR\n",
      "PD gene: KCNS3\n",
      "PD gene: KCNS3\n",
      "PD gene: SCN3A\n",
      "PD gene: STK39\n",
      "PD gene: STK39\n",
      "PD gene: MED12L\n",
      "PD gene: MED12L\n",
      "PD gene: UCHL1\n",
      "PD gene: SNCA\n",
      "PD gene: AC097478.1\n",
      "PD gene: AC097478.1\n",
      "PD gene: SNCA\n",
      "PD gene: FYN\n",
      "PD gene: WISP3\n",
      "PD gene: RPS12\n",
      "PD gene: HMGB1P13\n",
      "PD gene: RAB32\n",
      "PD gene: GPNMB\n",
      "PD gene: GPNMB\n",
      "PD gene: KLHL7\n",
      "PD gene: NUPL2\n",
      "PD gene: GPNMB\n",
      "PD gene: CHCHD2\n",
      "PD gene: BAG3\n",
      "PD gene: INPP5F\n",
      "PD gene: BAG3\n",
      "PD gene: INPP5F\n",
      "PD gene: FGD4\n",
      "PD gene: PKP2\n",
      "PD gene: ALG10\n",
      "PD gene: GXYLT1\n",
      "PD gene: SCAF11\n",
      "PD gene: SLC38A1\n",
      "PD gene: MBNL2\n",
      "PD gene: MBNL2\n",
      "PD gene: POLG1\n",
      "PD gene: SYT17\n",
      "PD gene: SYT17\n",
      "PD gene: COQ7\n",
      "PD gene: RETREG3\n",
      "PD gene: RETREG3\n",
      "PD gene: UBTF\n",
      "PD gene: FAM171A2\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "# Set working directory and ancestry code\n",
    "WORK_DIR='/home/jupyter/Release6_{ANC}'\n",
    "cd $WORK_DIR\n",
    "ANC=AMR\n",
    "\n",
    "# Filter ROH data for 'CON' and create clean file\n",
    "grep 'CON' gp2_${ANC}_r6_EOPD.hom.overlap.filtered | awk -F' ' '{if($4 != \"-9\")print}' > ${ANC}_ROH.hom.overlap.EOPD.CON.txt\n",
    "\n",
    "# Filter non-CON/UNION ROHs and clean the file\n",
    "grep -v 'CON' gp2_${ANC}_r6_EOPD.hom.overlap.filtered | grep -v 'UNION' | awk -F' ' '{if($4 != \"-9\")print}' | grep -v 'NA' > ${ANC}_ROH.hom.overlap.EOPD.clean.txt\n",
    "\n",
    "# Run the Perl script to map ROH data with gene list\n",
    "perl post_plink_ROH_mapping.pl ${ANC}_ROH.hom.overlap.EOPD.clean.txt ${ANC}_ROH.hom.overlap.EOPD.CON.txt geneList.txt ${ANC}_ROH.hom.overlap.EOPD.CON.PDgenes.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Classes ‘data.table’ and 'data.frame':\t42 obs. of  14 variables:\n",
      " $ POOL           : chr  \"S117\" \"S133\" \"S138\" \"S165\" ...\n",
      " $ num_subjects   : int  4 4 4 4 4 3 3 3 3 3 ...\n",
      " $ case_to_control: chr  \"0:4\" \"3:1\" \"2:2\" \"2:2\" ...\n",
      " $ ratio          : chr  \"0.0\" \"3.0\" \"1.0\" \"1.0\" ...\n",
      " $ CHR            : chr  \"chr2\" \"chr5\" \"chr6\" \"chr16\" ...\n",
      " $ SNP1           : chr  \"chr2:134725300:C:T\" \"chr5:60274169:TG:T\" \"chr6:31353056:C:T\" \"chr16:29651317:G:T\" ...\n",
      " $ SNP2           : chr  \"chr2:136064653:T:C\" \"chr5:61320199:T:G\" \"chr6:31949174:T:C\" \"chr16:31132672:C:T\" ...\n",
      " $ BP1            : int  134725300 60274169 31353056 29651317 32982212 15673611 120504534 160234491 112860471 61627648 ...\n",
      " $ BP2            : int  136064653 61320199 31949174 31132672 34497055 17215454 122063104 161775573 114748490 63237065 ...\n",
      " $ KB             : num  1339 1046 596 1481 1515 ...\n",
      " $ NSNP           : int  1272 1277 1055 923 1597 1569 2698 2049 2729 2351 ...\n",
      " $ PD gene        : chr  \"TMEM163 TMEM163\" \"ELOVL7 ELOVL7\" \"SLC44A4 HLA-DRB5\" \"CD19 SETD1A SETD1A ZNF646 KAT8\" ...\n",
      " $ cases_IDs      : int  0 0 0 0 0 0 0 0 0 0 ...\n",
      " $ control_IDs    : int  0 0 0 0 0 0 0 0 0 0 ...\n",
      " - attr(*, \".internal.selfref\")=<externalptr> \n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "\n",
    "# Load the ROH data from the CON PD genes file\n",
    "ROH <- fread(paste0(ANC, \"_ROH.hom.overlap.EOPD.CON.PDgenes.txt\"), header = TRUE)\n",
    "\n",
    "# Display the structure of the ROH data\n",
    "str(ROH)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: grep \"are cases and\" /home/jupyter/Release6_AMR/gp2_AMR_r6_EOPD.log\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Among remaining phenotypes, 91 are cases and 139 are controls.\r\n"
     ]
    }
   ],
   "source": [
    "# Use this log data as input for the next two steps\n",
    "shell_do(f'grep \"are cases and\" {WORK_DIR}/gp2_{ANC}_r6_EOPD.log')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Classes ‘data.table’ and 'data.frame':\t42 obs. of  21 variables:\n",
      " $ POOL             : chr  \"S117\" \"S133\" \"S138\" \"S165\" ...\n",
      " $ num_subjects     : int  4 4 4 4 4 3 3 3 3 3 ...\n",
      " $ case_to_control  : chr  \"0:4\" \"3:1\" \"2:2\" \"2:2\" ...\n",
      " $ ratio            : chr  \"0.0\" \"3.0\" \"1.0\" \"1.0\" ...\n",
      " $ CHR              : chr  \"chr2\" \"chr5\" \"chr6\" \"chr16\" ...\n",
      " $ SNP1             : chr  \"chr2:134725300:C:T\" \"chr5:60274169:TG:T\" \"chr6:31353056:C:T\" \"chr16:29651317:G:T\" ...\n",
      " $ SNP2             : chr  \"chr2:136064653:T:C\" \"chr5:61320199:T:G\" \"chr6:31949174:T:C\" \"chr16:31132672:C:T\" ...\n",
      " $ BP1              : int  134725300 60274169 31353056 29651317 32982212 15673611 120504534 160234491 112860471 61627648 ...\n",
      " $ BP2              : int  136064653 61320199 31949174 31132672 34497055 17215454 122063104 161775573 114748490 63237065 ...\n",
      " $ KB               : num  1339 1046 596 1481 1515 ...\n",
      " $ NSNP             : int  1272 1277 1055 923 1597 1569 2698 2049 2729 2351 ...\n",
      " $ PD gene          : chr  \"TMEM163 TMEM163\" \"ELOVL7 ELOVL7\" \"SLC44A4 HLA-DRB5\" \"CD19 SETD1A SETD1A ZNF646 KAT8\" ...\n",
      " $ cases_IDs        : int  0 0 0 0 0 0 0 0 0 0 ...\n",
      " $ control_IDs      : int  0 0 0 0 0 0 0 0 0 0 ...\n",
      " $ case             : chr  \"0\" \"3\" \"2\" \"2\" ...\n",
      " $ control          : chr  \"4\" \"1\" \"2\" \"2\" ...\n",
      " $ NcasesWithROHs   : num  0 3 2 2 3 3 3 0 3 3 ...\n",
      " $ NcontrolsWithROHs: num  4 1 2 2 1 0 0 3 0 0 ...\n",
      " $ combinedN        : num  4 4 4 4 4 3 3 3 3 3 ...\n",
      " $ P                : num  0.264 0.344 1 1 0.344 ...\n",
      " $ total_ROH_count  : num  4 4 4 4 4 3 3 3 3 3 ...\n",
      " - attr(*, \".internal.selfref\")=<externalptr> \n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "In addition: There were 40 warnings (use warnings() to see them)\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%R\n",
    "\n",
    "# Split the 'case_to_control' column into case and control components\n",
    "ROH$case <- ldply(strsplit(as.character(ROH$case_to_control), split = \":\"))[[1]]\n",
    "ROH$control <- ldply(strsplit(as.character(ROH$case_to_control), split = \":\"))[[2]]\n",
    "\n",
    "# Convert case and control components to numeric\n",
    "ROH$NcasesWithROHs <- as.numeric(ROH$case)\n",
    "ROH$NcontrolsWithROHs <- as.numeric(ROH$control)\n",
    "\n",
    "# Calculate total counts for cases, controls, and combined\n",
    "ROH$combinedN <- ROH$NcasesWithROHs + ROH$NcontrolsWithROHs\n",
    "\n",
    "# Initialize the 'P' column for p-values\n",
    "ROH$P <- NA\n",
    "\n",
    "# Perform proportion tests for each row\n",
    "for (i in seq_len(nrow(ROH))) {\n",
    "  thisP <- prop.test(x = c(ROH$NcasesWithROHs[i], ROH$NcontrolsWithROHs[i]), \n",
    "                     n = c(91, 139))  # n = (cases, controls)\n",
    "  ROH$P[i] <- thisP$p.value\n",
    "}\n",
    "\n",
    "# Calculate the total ROH count for cases and controls\n",
    "ROH$total_ROH_count <- ROH$NcasesWithROHs + ROH$NcontrolsWithROHs\n",
    "\n",
    "# Display the structure of the ROH data\n",
    "str(ROH)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. \n",
      "      0       0       0       0       0       0 \n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "\n",
    "# Calculate proportions of cases and controls with ROH\n",
    "ROH$propCases <- ROH$NcasesWithROHs / 91  # Number of cases\n",
    "ROH$propControls <- ROH$NcontrolsWithROHs / 139  # Number of controls\n",
    "\n",
    "# Identify case-enriched ROHs\n",
    "ROH$caseEnriched <- ifelse(ROH$propCases > ROH$propControls, 1, 0)\n",
    "\n",
    "# Subset ROHs where total ROH count >= 1 and case-enriched\n",
    "ROH_subsetted <- subset(ROH, total_ROH_count >= 1 & caseEnriched == 1)\n",
    "\n",
    "# Apply Bonferroni correction\n",
    "Ngenes <- length(ROH_subsetted$PD)  # Number of genes\n",
    "ROH_subsetted$passMultiTest <- ifelse(ROH_subsetted$P <= (0.05 / Ngenes), 1, 0)\n",
    "ROH_subsetted$BONF <- ROH_subsetted$P / Ngenes\n",
    "\n",
    "# Subset the significant results after multiple testing correction\n",
    "significant_ROH <- subset(ROH_subsetted, passMultiTest == 1)\n",
    "\n",
    "# Apply Bonferroni correction to the full dataset\n",
    "ROH$BONF <- ROH$P / Ngenes\n",
    "\n",
    "# Display summary of Bonferroni-corrected results\n",
    "summary(ROH_subsetted$passMultiTest)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R\n",
    "\n",
    "# Write the subsetted ROH data to text files\n",
    "fwrite(ROH_subsetted, paste0(ANC, \"_ROH.EOPD_PDgenes_subsetted.txt\"))\n",
    "fwrite(ROH_subsetted, paste0(ANC, \"_ROH.EOPD_PDgenes_caseEnriched.txt\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>POOL</th>\n",
       "      <th>num_subjects</th>\n",
       "      <th>case_to_control</th>\n",
       "      <th>ratio</th>\n",
       "      <th>CHR</th>\n",
       "      <th>SNP1</th>\n",
       "      <th>SNP2</th>\n",
       "      <th>BP1</th>\n",
       "      <th>BP2</th>\n",
       "      <th>KB</th>\n",
       "      <th>...</th>\n",
       "      <th>NcasesWithROHs</th>\n",
       "      <th>NcontrolsWithROHs</th>\n",
       "      <th>combinedN</th>\n",
       "      <th>P</th>\n",
       "      <th>total_ROH_count</th>\n",
       "      <th>propCases</th>\n",
       "      <th>propControls</th>\n",
       "      <th>caseEnriched</th>\n",
       "      <th>passMultiTest</th>\n",
       "      <th>BONF</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>S133</td>\n",
       "      <td>4</td>\n",
       "      <td>3:1</td>\n",
       "      <td>3.0</td>\n",
       "      <td>chr5</td>\n",
       "      <td>chr5:60274169:TG:T</td>\n",
       "      <td>chr5:61320199:T:G</td>\n",
       "      <td>60274169</td>\n",
       "      <td>61320199</td>\n",
       "      <td>1046.031</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>0.343990</td>\n",
       "      <td>4</td>\n",
       "      <td>0.032967</td>\n",
       "      <td>0.007194</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.010424</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>S138</td>\n",
       "      <td>4</td>\n",
       "      <td>2:2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr6</td>\n",
       "      <td>chr6:31353056:C:T</td>\n",
       "      <td>chr6:31949174:T:C</td>\n",
       "      <td>31353056</td>\n",
       "      <td>31949174</td>\n",
       "      <td>596.119</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>4</td>\n",
       "      <td>0.021978</td>\n",
       "      <td>0.014388</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.030303</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>S165</td>\n",
       "      <td>4</td>\n",
       "      <td>2:2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr16</td>\n",
       "      <td>chr16:29651317:G:T</td>\n",
       "      <td>chr16:31132672:C:T</td>\n",
       "      <td>29651317</td>\n",
       "      <td>31132672</td>\n",
       "      <td>1481.356</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>4</td>\n",
       "      <td>0.021978</td>\n",
       "      <td>0.014388</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.030303</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>S166</td>\n",
       "      <td>4</td>\n",
       "      <td>3:1</td>\n",
       "      <td>3.0</td>\n",
       "      <td>chr18</td>\n",
       "      <td>chr18:32982212:C:T</td>\n",
       "      <td>chr18:34497055:T:C</td>\n",
       "      <td>32982212</td>\n",
       "      <td>34497055</td>\n",
       "      <td>1514.844</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>0.343990</td>\n",
       "      <td>4</td>\n",
       "      <td>0.032967</td>\n",
       "      <td>0.007194</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.010424</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>S167</td>\n",
       "      <td>3</td>\n",
       "      <td>3:0</td>\n",
       "      <td>3:0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>chr1:15673611:C:T</td>\n",
       "      <td>chr1:17215454:A:G</td>\n",
       "      <td>15673611</td>\n",
       "      <td>17215454</td>\n",
       "      <td>1541.844</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0.118637</td>\n",
       "      <td>3</td>\n",
       "      <td>0.032967</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.003595</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>S190</td>\n",
       "      <td>3</td>\n",
       "      <td>3:0</td>\n",
       "      <td>3:0</td>\n",
       "      <td>chr3</td>\n",
       "      <td>chr3:120504534:G:C</td>\n",
       "      <td>chr3:122063104:T:A</td>\n",
       "      <td>120504534</td>\n",
       "      <td>122063104</td>\n",
       "      <td>1558.571</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0.118637</td>\n",
       "      <td>3</td>\n",
       "      <td>0.032967</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.003595</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>S202</td>\n",
       "      <td>3</td>\n",
       "      <td>3:0</td>\n",
       "      <td>3:0</td>\n",
       "      <td>chr4</td>\n",
       "      <td>chr4:112860471:G:C</td>\n",
       "      <td>chr4:114748490:G:T</td>\n",
       "      <td>112860471</td>\n",
       "      <td>114748490</td>\n",
       "      <td>1888.020</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0.118637</td>\n",
       "      <td>3</td>\n",
       "      <td>0.032967</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.003595</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>S206</td>\n",
       "      <td>3</td>\n",
       "      <td>3:0</td>\n",
       "      <td>3:0</td>\n",
       "      <td>chr5</td>\n",
       "      <td>chr5:61627648:G:A</td>\n",
       "      <td>chr5:63237065:A:G</td>\n",
       "      <td>61627648</td>\n",
       "      <td>63237065</td>\n",
       "      <td>1609.418</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0.118637</td>\n",
       "      <td>3</td>\n",
       "      <td>0.032967</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.003595</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>S213</td>\n",
       "      <td>3</td>\n",
       "      <td>2:1</td>\n",
       "      <td>2.0</td>\n",
       "      <td>chr6</td>\n",
       "      <td>chr6:160712093:G:C</td>\n",
       "      <td>chr6:161123166:A:C</td>\n",
       "      <td>160712093</td>\n",
       "      <td>161123166</td>\n",
       "      <td>411.074</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>0.709860</td>\n",
       "      <td>3</td>\n",
       "      <td>0.021978</td>\n",
       "      <td>0.007194</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.021511</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>S214</td>\n",
       "      <td>3</td>\n",
       "      <td>3:0</td>\n",
       "      <td>3:0</td>\n",
       "      <td>chr6</td>\n",
       "      <td>chr6:161389511:A:C</td>\n",
       "      <td>chr6:161815720:A:G</td>\n",
       "      <td>161389511</td>\n",
       "      <td>161815720</td>\n",
       "      <td>426.210</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0.118637</td>\n",
       "      <td>3</td>\n",
       "      <td>0.032967</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.003595</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>S215</td>\n",
       "      <td>3</td>\n",
       "      <td>3:0</td>\n",
       "      <td>3:0</td>\n",
       "      <td>chr6</td>\n",
       "      <td>chr6:162375099:C:G</td>\n",
       "      <td>chr6:163939035:A:T</td>\n",
       "      <td>162375099</td>\n",
       "      <td>163939035</td>\n",
       "      <td>1563.937</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0.118637</td>\n",
       "      <td>3</td>\n",
       "      <td>0.032967</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.003595</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>S254</td>\n",
       "      <td>3</td>\n",
       "      <td>3:0</td>\n",
       "      <td>3:0</td>\n",
       "      <td>chr17</td>\n",
       "      <td>chr17:44995295:C:T</td>\n",
       "      <td>chr17:45483692:C:T</td>\n",
       "      <td>44995295</td>\n",
       "      <td>45483692</td>\n",
       "      <td>488.398</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0.118637</td>\n",
       "      <td>3</td>\n",
       "      <td>0.032967</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.003595</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>S255</td>\n",
       "      <td>3</td>\n",
       "      <td>3:0</td>\n",
       "      <td>3:0</td>\n",
       "      <td>chr17</td>\n",
       "      <td>chr17:45869341:T:C</td>\n",
       "      <td>chr17:46788635:A:C</td>\n",
       "      <td>45869341</td>\n",
       "      <td>46788635</td>\n",
       "      <td>919.295</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0.118637</td>\n",
       "      <td>3</td>\n",
       "      <td>0.032967</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.003595</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>S256</td>\n",
       "      <td>3</td>\n",
       "      <td>2:1</td>\n",
       "      <td>2.0</td>\n",
       "      <td>chr17</td>\n",
       "      <td>chr17:47224732:G:A</td>\n",
       "      <td>chr17:47687326:T:G</td>\n",
       "      <td>47224732</td>\n",
       "      <td>47687326</td>\n",
       "      <td>462.595</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>0.709860</td>\n",
       "      <td>3</td>\n",
       "      <td>0.021978</td>\n",
       "      <td>0.007194</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.021511</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>S267</td>\n",
       "      <td>3</td>\n",
       "      <td>2:1</td>\n",
       "      <td>2.0</td>\n",
       "      <td>chr20</td>\n",
       "      <td>chr20:32232311:AAAAG:A</td>\n",
       "      <td>chr20:32619906:T:C</td>\n",
       "      <td>32232311</td>\n",
       "      <td>32619906</td>\n",
       "      <td>387.596</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>0.709860</td>\n",
       "      <td>3</td>\n",
       "      <td>0.021978</td>\n",
       "      <td>0.007194</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.021511</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>S282</td>\n",
       "      <td>2</td>\n",
       "      <td>1:1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>chr1:151860425:A:C</td>\n",
       "      <td>chr1:153368348:C:G</td>\n",
       "      <td>151860425</td>\n",
       "      <td>153368348</td>\n",
       "      <td>1507.924</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>2</td>\n",
       "      <td>0.010989</td>\n",
       "      <td>0.007194</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.030303</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>S296</td>\n",
       "      <td>2</td>\n",
       "      <td>1:1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr2</td>\n",
       "      <td>chr2:162970618:C:A</td>\n",
       "      <td>chr2:166628367:T:G</td>\n",
       "      <td>162970618</td>\n",
       "      <td>166628367</td>\n",
       "      <td>3657.750</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>2</td>\n",
       "      <td>0.010989</td>\n",
       "      <td>0.007194</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.030303</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>S297</td>\n",
       "      <td>2</td>\n",
       "      <td>1:1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr2</td>\n",
       "      <td>chr2:167662447:C:T</td>\n",
       "      <td>chr2:170102278:A:G</td>\n",
       "      <td>167662447</td>\n",
       "      <td>170102278</td>\n",
       "      <td>2439.832</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>2</td>\n",
       "      <td>0.010989</td>\n",
       "      <td>0.007194</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.030303</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>S313</td>\n",
       "      <td>2</td>\n",
       "      <td>1:1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr4</td>\n",
       "      <td>chr4:40525869:TG:T</td>\n",
       "      <td>chr4:45848903:T:A</td>\n",
       "      <td>40525869</td>\n",
       "      <td>45848903</td>\n",
       "      <td>5323.035</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>2</td>\n",
       "      <td>0.010989</td>\n",
       "      <td>0.007194</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.030303</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>S317</td>\n",
       "      <td>2</td>\n",
       "      <td>1:1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr4</td>\n",
       "      <td>chr4:89451624:G:T</td>\n",
       "      <td>chr4:91375869:T:C</td>\n",
       "      <td>89451624</td>\n",
       "      <td>91375869</td>\n",
       "      <td>1924.246</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>2</td>\n",
       "      <td>0.010989</td>\n",
       "      <td>0.007194</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.030303</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>S339</td>\n",
       "      <td>2</td>\n",
       "      <td>1:1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr6</td>\n",
       "      <td>chr6:110472496:A:G</td>\n",
       "      <td>chr6:112035038:G:C</td>\n",
       "      <td>110472496</td>\n",
       "      <td>112035038</td>\n",
       "      <td>1562.543</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>2</td>\n",
       "      <td>0.010989</td>\n",
       "      <td>0.007194</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.030303</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>S342</td>\n",
       "      <td>2</td>\n",
       "      <td>1:1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr6</td>\n",
       "      <td>chr6:144712479:T:C</td>\n",
       "      <td>chr6:146408080:A:G</td>\n",
       "      <td>144712479</td>\n",
       "      <td>146408080</td>\n",
       "      <td>1695.602</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>2</td>\n",
       "      <td>0.010989</td>\n",
       "      <td>0.007194</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.030303</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>S344</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>2:0</td>\n",
       "      <td>chr7</td>\n",
       "      <td>chr7:21546333:G:C</td>\n",
       "      <td>chr7:24060359:C:G</td>\n",
       "      <td>21546333</td>\n",
       "      <td>24060359</td>\n",
       "      <td>2514.027</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.303340</td>\n",
       "      <td>2</td>\n",
       "      <td>0.021978</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.009192</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>S365</td>\n",
       "      <td>2</td>\n",
       "      <td>1:1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr10</td>\n",
       "      <td>chr10:118903708:T:C</td>\n",
       "      <td>chr10:120480281:C:G</td>\n",
       "      <td>118903708</td>\n",
       "      <td>120480281</td>\n",
       "      <td>1576.574</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>2</td>\n",
       "      <td>0.010989</td>\n",
       "      <td>0.007194</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.030303</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>S375</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>2:0</td>\n",
       "      <td>chr12</td>\n",
       "      <td>chr12:32943085:C:T</td>\n",
       "      <td>chr12:34717351:T:G</td>\n",
       "      <td>32943085</td>\n",
       "      <td>34717351</td>\n",
       "      <td>1774.267</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.303340</td>\n",
       "      <td>2</td>\n",
       "      <td>0.021978</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.009192</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>S378</td>\n",
       "      <td>2</td>\n",
       "      <td>1:1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr12</td>\n",
       "      <td>chr12:45598371:G:A</td>\n",
       "      <td>chr12:46179193:G:T</td>\n",
       "      <td>45598371</td>\n",
       "      <td>46179193</td>\n",
       "      <td>580.823</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>2</td>\n",
       "      <td>0.010989</td>\n",
       "      <td>0.007194</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.030303</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>S38</td>\n",
       "      <td>10</td>\n",
       "      <td>4:6</td>\n",
       "      <td>0.7</td>\n",
       "      <td>chr15</td>\n",
       "      <td>chr15:43296286:G:A</td>\n",
       "      <td>chr15:44126835:TA:T</td>\n",
       "      <td>43296286</td>\n",
       "      <td>44126835</td>\n",
       "      <td>830.550</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>6</td>\n",
       "      <td>10</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>10</td>\n",
       "      <td>0.043956</td>\n",
       "      <td>0.043165</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.030303</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>S385</td>\n",
       "      <td>2</td>\n",
       "      <td>1:1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr13</td>\n",
       "      <td>chr13:95365837:C:T</td>\n",
       "      <td>chr13:96866861:C:T</td>\n",
       "      <td>95365837</td>\n",
       "      <td>96866861</td>\n",
       "      <td>1501.025</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>2</td>\n",
       "      <td>0.010989</td>\n",
       "      <td>0.007194</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.030303</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>S396</td>\n",
       "      <td>2</td>\n",
       "      <td>1:1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr15</td>\n",
       "      <td>chr15:88291048:A:C</td>\n",
       "      <td>chr15:89840813:C:T</td>\n",
       "      <td>88291048</td>\n",
       "      <td>89840813</td>\n",
       "      <td>1549.766</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>2</td>\n",
       "      <td>0.010989</td>\n",
       "      <td>0.007194</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.030303</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>S397</td>\n",
       "      <td>2</td>\n",
       "      <td>1:1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr16</td>\n",
       "      <td>chr16:17615011:T:C</td>\n",
       "      <td>chr16:19160981:TTGAAAAGAAAAGAAAA:T</td>\n",
       "      <td>17615011</td>\n",
       "      <td>19160981</td>\n",
       "      <td>1545.971</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>2</td>\n",
       "      <td>0.010989</td>\n",
       "      <td>0.007194</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.030303</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>S403</td>\n",
       "      <td>2</td>\n",
       "      <td>2:0</td>\n",
       "      <td>2:0</td>\n",
       "      <td>chr17</td>\n",
       "      <td>chr17:41816525:G:A</td>\n",
       "      <td>chr17:43433482:A:G</td>\n",
       "      <td>41816525</td>\n",
       "      <td>43433482</td>\n",
       "      <td>1616.958</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.303340</td>\n",
       "      <td>2</td>\n",
       "      <td>0.021978</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.009192</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>S7</td>\n",
       "      <td>17</td>\n",
       "      <td>10:7</td>\n",
       "      <td>1.4</td>\n",
       "      <td>chr6</td>\n",
       "      <td>chr6:27899165:G:C</td>\n",
       "      <td>chr6:28207771:G:C</td>\n",
       "      <td>27899165</td>\n",
       "      <td>28207771</td>\n",
       "      <td>308.607</td>\n",
       "      <td>...</td>\n",
       "      <td>10</td>\n",
       "      <td>7</td>\n",
       "      <td>17</td>\n",
       "      <td>0.152806</td>\n",
       "      <td>17</td>\n",
       "      <td>0.109890</td>\n",
       "      <td>0.050360</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.004630</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>S82</td>\n",
       "      <td>6</td>\n",
       "      <td>5:1</td>\n",
       "      <td>5.0</td>\n",
       "      <td>chr14</td>\n",
       "      <td>chr14:65967520:T:A</td>\n",
       "      <td>chr14:67465921:A:C</td>\n",
       "      <td>65967520</td>\n",
       "      <td>67465921</td>\n",
       "      <td>1498.402</td>\n",
       "      <td>...</td>\n",
       "      <td>5</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>0.072075</td>\n",
       "      <td>6</td>\n",
       "      <td>0.054945</td>\n",
       "      <td>0.007194</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.002184</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>33 rows × 26 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    POOL  num_subjects case_to_control ratio    CHR                    SNP1  \\\n",
       "0   S133             4             3:1   3.0   chr5      chr5:60274169:TG:T   \n",
       "1   S138             4             2:2   1.0   chr6       chr6:31353056:C:T   \n",
       "2   S165             4             2:2   1.0  chr16      chr16:29651317:G:T   \n",
       "3   S166             4             3:1   3.0  chr18      chr18:32982212:C:T   \n",
       "4   S167             3             3:0   3:0   chr1       chr1:15673611:C:T   \n",
       "5   S190             3             3:0   3:0   chr3      chr3:120504534:G:C   \n",
       "6   S202             3             3:0   3:0   chr4      chr4:112860471:G:C   \n",
       "7   S206             3             3:0   3:0   chr5       chr5:61627648:G:A   \n",
       "8   S213             3             2:1   2.0   chr6      chr6:160712093:G:C   \n",
       "9   S214             3             3:0   3:0   chr6      chr6:161389511:A:C   \n",
       "10  S215             3             3:0   3:0   chr6      chr6:162375099:C:G   \n",
       "11  S254             3             3:0   3:0  chr17      chr17:44995295:C:T   \n",
       "12  S255             3             3:0   3:0  chr17      chr17:45869341:T:C   \n",
       "13  S256             3             2:1   2.0  chr17      chr17:47224732:G:A   \n",
       "14  S267             3             2:1   2.0  chr20  chr20:32232311:AAAAG:A   \n",
       "15  S282             2             1:1   1.0   chr1      chr1:151860425:A:C   \n",
       "16  S296             2             1:1   1.0   chr2      chr2:162970618:C:A   \n",
       "17  S297             2             1:1   1.0   chr2      chr2:167662447:C:T   \n",
       "18  S313             2             1:1   1.0   chr4      chr4:40525869:TG:T   \n",
       "19  S317             2             1:1   1.0   chr4       chr4:89451624:G:T   \n",
       "20  S339             2             1:1   1.0   chr6      chr6:110472496:A:G   \n",
       "21  S342             2             1:1   1.0   chr6      chr6:144712479:T:C   \n",
       "22  S344             2             2:0   2:0   chr7       chr7:21546333:G:C   \n",
       "23  S365             2             1:1   1.0  chr10     chr10:118903708:T:C   \n",
       "24  S375             2             2:0   2:0  chr12      chr12:32943085:C:T   \n",
       "25  S378             2             1:1   1.0  chr12      chr12:45598371:G:A   \n",
       "26   S38            10             4:6   0.7  chr15      chr15:43296286:G:A   \n",
       "27  S385             2             1:1   1.0  chr13      chr13:95365837:C:T   \n",
       "28  S396             2             1:1   1.0  chr15      chr15:88291048:A:C   \n",
       "29  S397             2             1:1   1.0  chr16      chr16:17615011:T:C   \n",
       "30  S403             2             2:0   2:0  chr17      chr17:41816525:G:A   \n",
       "31    S7            17            10:7   1.4   chr6       chr6:27899165:G:C   \n",
       "32   S82             6             5:1   5.0  chr14      chr14:65967520:T:A   \n",
       "\n",
       "                                  SNP2        BP1        BP2        KB  ...  \\\n",
       "0                    chr5:61320199:T:G   60274169   61320199  1046.031  ...   \n",
       "1                    chr6:31949174:T:C   31353056   31949174   596.119  ...   \n",
       "2                   chr16:31132672:C:T   29651317   31132672  1481.356  ...   \n",
       "3                   chr18:34497055:T:C   32982212   34497055  1514.844  ...   \n",
       "4                    chr1:17215454:A:G   15673611   17215454  1541.844  ...   \n",
       "5                   chr3:122063104:T:A  120504534  122063104  1558.571  ...   \n",
       "6                   chr4:114748490:G:T  112860471  114748490  1888.020  ...   \n",
       "7                    chr5:63237065:A:G   61627648   63237065  1609.418  ...   \n",
       "8                   chr6:161123166:A:C  160712093  161123166   411.074  ...   \n",
       "9                   chr6:161815720:A:G  161389511  161815720   426.210  ...   \n",
       "10                  chr6:163939035:A:T  162375099  163939035  1563.937  ...   \n",
       "11                  chr17:45483692:C:T   44995295   45483692   488.398  ...   \n",
       "12                  chr17:46788635:A:C   45869341   46788635   919.295  ...   \n",
       "13                  chr17:47687326:T:G   47224732   47687326   462.595  ...   \n",
       "14                  chr20:32619906:T:C   32232311   32619906   387.596  ...   \n",
       "15                  chr1:153368348:C:G  151860425  153368348  1507.924  ...   \n",
       "16                  chr2:166628367:T:G  162970618  166628367  3657.750  ...   \n",
       "17                  chr2:170102278:A:G  167662447  170102278  2439.832  ...   \n",
       "18                   chr4:45848903:T:A   40525869   45848903  5323.035  ...   \n",
       "19                   chr4:91375869:T:C   89451624   91375869  1924.246  ...   \n",
       "20                  chr6:112035038:G:C  110472496  112035038  1562.543  ...   \n",
       "21                  chr6:146408080:A:G  144712479  146408080  1695.602  ...   \n",
       "22                   chr7:24060359:C:G   21546333   24060359  2514.027  ...   \n",
       "23                 chr10:120480281:C:G  118903708  120480281  1576.574  ...   \n",
       "24                  chr12:34717351:T:G   32943085   34717351  1774.267  ...   \n",
       "25                  chr12:46179193:G:T   45598371   46179193   580.823  ...   \n",
       "26                 chr15:44126835:TA:T   43296286   44126835   830.550  ...   \n",
       "27                  chr13:96866861:C:T   95365837   96866861  1501.025  ...   \n",
       "28                  chr15:89840813:C:T   88291048   89840813  1549.766  ...   \n",
       "29  chr16:19160981:TTGAAAAGAAAAGAAAA:T   17615011   19160981  1545.971  ...   \n",
       "30                  chr17:43433482:A:G   41816525   43433482  1616.958  ...   \n",
       "31                   chr6:28207771:G:C   27899165   28207771   308.607  ...   \n",
       "32                  chr14:67465921:A:C   65967520   67465921  1498.402  ...   \n",
       "\n",
       "    NcasesWithROHs NcontrolsWithROHs  combinedN         P  total_ROH_count  \\\n",
       "0                3                 1          4  0.343990                4   \n",
       "1                2                 2          4  1.000000                4   \n",
       "2                2                 2          4  1.000000                4   \n",
       "3                3                 1          4  0.343990                4   \n",
       "4                3                 0          3  0.118637                3   \n",
       "5                3                 0          3  0.118637                3   \n",
       "6                3                 0          3  0.118637                3   \n",
       "7                3                 0          3  0.118637                3   \n",
       "8                2                 1          3  0.709860                3   \n",
       "9                3                 0          3  0.118637                3   \n",
       "10               3                 0          3  0.118637                3   \n",
       "11               3                 0          3  0.118637                3   \n",
       "12               3                 0          3  0.118637                3   \n",
       "13               2                 1          3  0.709860                3   \n",
       "14               2                 1          3  0.709860                3   \n",
       "15               1                 1          2  1.000000                2   \n",
       "16               1                 1          2  1.000000                2   \n",
       "17               1                 1          2  1.000000                2   \n",
       "18               1                 1          2  1.000000                2   \n",
       "19               1                 1          2  1.000000                2   \n",
       "20               1                 1          2  1.000000                2   \n",
       "21               1                 1          2  1.000000                2   \n",
       "22               2                 0          2  0.303340                2   \n",
       "23               1                 1          2  1.000000                2   \n",
       "24               2                 0          2  0.303340                2   \n",
       "25               1                 1          2  1.000000                2   \n",
       "26               4                 6         10  1.000000               10   \n",
       "27               1                 1          2  1.000000                2   \n",
       "28               1                 1          2  1.000000                2   \n",
       "29               1                 1          2  1.000000                2   \n",
       "30               2                 0          2  0.303340                2   \n",
       "31              10                 7         17  0.152806               17   \n",
       "32               5                 1          6  0.072075                6   \n",
       "\n",
       "    propCases  propControls  caseEnriched  passMultiTest      BONF  \n",
       "0    0.032967      0.007194             1              0  0.010424  \n",
       "1    0.021978      0.014388             1              0  0.030303  \n",
       "2    0.021978      0.014388             1              0  0.030303  \n",
       "3    0.032967      0.007194             1              0  0.010424  \n",
       "4    0.032967      0.000000             1              0  0.003595  \n",
       "5    0.032967      0.000000             1              0  0.003595  \n",
       "6    0.032967      0.000000             1              0  0.003595  \n",
       "7    0.032967      0.000000             1              0  0.003595  \n",
       "8    0.021978      0.007194             1              0  0.021511  \n",
       "9    0.032967      0.000000             1              0  0.003595  \n",
       "10   0.032967      0.000000             1              0  0.003595  \n",
       "11   0.032967      0.000000             1              0  0.003595  \n",
       "12   0.032967      0.000000             1              0  0.003595  \n",
       "13   0.021978      0.007194             1              0  0.021511  \n",
       "14   0.021978      0.007194             1              0  0.021511  \n",
       "15   0.010989      0.007194             1              0  0.030303  \n",
       "16   0.010989      0.007194             1              0  0.030303  \n",
       "17   0.010989      0.007194             1              0  0.030303  \n",
       "18   0.010989      0.007194             1              0  0.030303  \n",
       "19   0.010989      0.007194             1              0  0.030303  \n",
       "20   0.010989      0.007194             1              0  0.030303  \n",
       "21   0.010989      0.007194             1              0  0.030303  \n",
       "22   0.021978      0.000000             1              0  0.009192  \n",
       "23   0.010989      0.007194             1              0  0.030303  \n",
       "24   0.021978      0.000000             1              0  0.009192  \n",
       "25   0.010989      0.007194             1              0  0.030303  \n",
       "26   0.043956      0.043165             1              0  0.030303  \n",
       "27   0.010989      0.007194             1              0  0.030303  \n",
       "28   0.010989      0.007194             1              0  0.030303  \n",
       "29   0.010989      0.007194             1              0  0.030303  \n",
       "30   0.021978      0.000000             1              0  0.009192  \n",
       "31   0.109890      0.050360             1              0  0.004630  \n",
       "32   0.054945      0.007194             1              0  0.002184  \n",
       "\n",
       "[33 rows x 26 columns]"
      ]
     },
     "execution_count": 78,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Read the case-enriched ROH data from the file\n",
    "ROH = pd.read_csv(f'{WORK_DIR}/{ANC}_ROH.EOPD_PDgenes_caseEnriched.txt')\n",
    "\n",
    "# Display the DataFrame\n",
    "ROH"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "bash: line 4: cd: /home/jupyter/Release6_{ANC}: No such file or directory\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of ROH overlapping PD genes: 42\n",
      "Number of ROH enriched in Cases: 33\n",
      "Number of ROH that Pass Bonferroni: 0\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "# Set working directory and ancestry code\n",
    "WORK_DIR='/home/jupyter/Release6_{ANC}'\n",
    "cd $WORK_DIR\n",
    "ANC=AMR\n",
    "\n",
    "# Count and display various ROH statistics\n",
    "{\n",
    "    printf \"Number of ROH overlapping PD genes: \"\n",
    "    sed 1d < ${ANC}_ROH.hom.overlap.EOPD.CON.PDgenes.txt | wc -l\n",
    "\n",
    "    printf \"Number of ROH enriched in Cases: \"\n",
    "    sed 1d < ${ANC}_ROH.EOPD_PDgenes_caseEnriched.txt | wc -l\n",
    "\n",
    "    printf \"Number of ROH that Pass Bonferroni: \"\n",
    "    cut -d',' -f27 < ${ANC}_ROH.EOPD_PDgenes_caseEnriched.txt | sed 1d | { grep -c '1' || true; }\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Fine mapping loop (all)\n",
    "\n",
    "Iterate over all identified regions to perform fine mapping, isolating the regions most likely to contain disease-causing mutations."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Making a working directory\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: mkdir -p /home/jupyter/Mapping\n"
     ]
    }
   ],
   "source": [
    "# Create a working directory for use in Python\n",
    "print(\"Making a working directory\")\n",
    "WORK_DIR = '/home/jupyter/Mapping'\n",
    "shell_do(f'mkdir -p {WORK_DIR}')  # Create the directory if it doesn't exist"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Download pools\n",
    "\n",
    "Download the pools of samples for further analysis of shared alleles"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: ls /home/jupyter/Mapping/*.vcf.gz | sed 's/.vcf.gz//g' | sort -V | grep -v 'chr1_shared_roh' > /home/jupyter/Mapping/list_pools.txt\n",
      "Executing: grep 'S[0-9]\\+_' /home/jupyter/Mapping/list_pools.txt > /home/jupyter/Mapping/to_rename_pools.txt\n"
     ]
    }
   ],
   "source": [
    "# Identify and list different names in pools, excluding certain files\n",
    "shell_do(f\"ls {WORK_DIR}/*.vcf.gz | sed 's/.vcf.gz//g' | sort -V | grep -v 'chr1_shared_roh' > {WORK_DIR}/list_pools.txt\")\n",
    "\n",
    "# Filter and identify pools that need to be renamed\n",
    "shell_do(f\"grep 'S[0-9]\\\\+_' {WORK_DIR}/list_pools.txt > {WORK_DIR}/to_rename_pools.txt\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: mv /home/jupyter/Mapping/S7_MDE.vcf.gz /home/jupyter/Mapping/MDE_S7.vcf.gz\n",
      "Executing: mv /home/jupyter/Mapping/S7_MDE.vcf.gz.tbi /home/jupyter/Mapping/MDE_S7.vcf.gz.tbi\n",
      "Executing: mv /home/jupyter/Mapping/S14_MDE.vcf.gz /home/jupyter/Mapping/MDE_S14.vcf.gz\n",
      "Executing: mv /home/jupyter/Mapping/S14_MDE.vcf.gz.tbi /home/jupyter/Mapping/MDE_S14.vcf.gz.tbi\n",
      "Executing: mv /home/jupyter/Mapping/S15_MDE.vcf.gz /home/jupyter/Mapping/MDE_S15.vcf.gz\n",
      "Executing: mv /home/jupyter/Mapping/S15_MDE.vcf.gz.tbi /home/jupyter/Mapping/MDE_S15.vcf.gz.tbi\n",
      "Executing: mv /home/jupyter/Mapping/S16_MDE.vcf.gz /home/jupyter/Mapping/MDE_S16.vcf.gz\n",
      "Executing: mv /home/jupyter/Mapping/S16_MDE.vcf.gz.tbi /home/jupyter/Mapping/MDE_S16.vcf.gz.tbi\n",
      "Executing: mv /home/jupyter/Mapping/S17_MDE.vcf.gz /home/jupyter/Mapping/MDE_S17.vcf.gz\n",
      "Executing: mv /home/jupyter/Mapping/S17_MDE.vcf.gz.tbi /home/jupyter/Mapping/MDE_S17.vcf.gz.tbi\n",
      "Executing: mv /home/jupyter/Mapping/S51_MDE.vcf.gz /home/jupyter/Mapping/MDE_S51.vcf.gz\n",
      "Executing: mv /home/jupyter/Mapping/S51_MDE.vcf.gz.tbi /home/jupyter/Mapping/MDE_S51.vcf.gz.tbi\n",
      "Executing: mv /home/jupyter/Mapping/S65_MDE.vcf.gz /home/jupyter/Mapping/MDE_S65.vcf.gz\n",
      "Executing: mv /home/jupyter/Mapping/S65_MDE.vcf.gz.tbi /home/jupyter/Mapping/MDE_S65.vcf.gz.tbi\n",
      "Executing: mv /home/jupyter/Mapping/S83_MDE.vcf.gz /home/jupyter/Mapping/MDE_S83.vcf.gz\n",
      "Executing: mv /home/jupyter/Mapping/S83_MDE.vcf.gz.tbi /home/jupyter/Mapping/MDE_S83.vcf.gz.tbi\n",
      "Executing: mv /home/jupyter/Mapping/S87_MDE.vcf.gz /home/jupyter/Mapping/MDE_S87.vcf.gz\n",
      "Executing: mv /home/jupyter/Mapping/S87_MDE.vcf.gz.tbi /home/jupyter/Mapping/MDE_S87.vcf.gz.tbi\n",
      "Executing: mv /home/jupyter/Mapping/S90_MDE.vcf.gz /home/jupyter/Mapping/MDE_S90.vcf.gz\n",
      "Executing: mv /home/jupyter/Mapping/S90_MDE.vcf.gz.tbi /home/jupyter/Mapping/MDE_S90.vcf.gz.tbi\n",
      "Executing: mv /home/jupyter/Mapping/S164_MDE.vcf.gz /home/jupyter/Mapping/MDE_S164.vcf.gz\n",
      "Executing: mv /home/jupyter/Mapping/S164_MDE.vcf.gz.tbi /home/jupyter/Mapping/MDE_S164.vcf.gz.tbi\n",
      "Executing: mv /home/jupyter/Mapping/S186_MDE.vcf.gz /home/jupyter/Mapping/MDE_S186.vcf.gz\n",
      "Executing: mv /home/jupyter/Mapping/S186_MDE.vcf.gz.tbi /home/jupyter/Mapping/MDE_S186.vcf.gz.tbi\n",
      "Executing: mv /home/jupyter/Mapping/S235_MDE.vcf.gz /home/jupyter/Mapping/MDE_S235.vcf.gz\n",
      "Executing: mv /home/jupyter/Mapping/S235_MDE.vcf.gz.tbi /home/jupyter/Mapping/MDE_S235.vcf.gz.tbi\n",
      "Executing: mv /home/jupyter/Mapping/S259_MDE.vcf.gz /home/jupyter/Mapping/MDE_S259.vcf.gz\n",
      "Executing: mv /home/jupyter/Mapping/S259_MDE.vcf.gz.tbi /home/jupyter/Mapping/MDE_S259.vcf.gz.tbi\n",
      "Executing: mv /home/jupyter/Mapping/S288_MDE.vcf.gz /home/jupyter/Mapping/MDE_S288.vcf.gz\n",
      "Executing: mv /home/jupyter/Mapping/S288_MDE.vcf.gz.tbi /home/jupyter/Mapping/MDE_S288.vcf.gz.tbi\n",
      "Executing: mv /home/jupyter/Mapping/S289_MDE.vcf.gz /home/jupyter/Mapping/MDE_S289.vcf.gz\n",
      "Executing: mv /home/jupyter/Mapping/S289_MDE.vcf.gz.tbi /home/jupyter/Mapping/MDE_S289.vcf.gz.tbi\n",
      "Executing: mv /home/jupyter/Mapping/S406_MDE.vcf.gz /home/jupyter/Mapping/MDE_S406.vcf.gz\n",
      "Executing: mv /home/jupyter/Mapping/S406_MDE.vcf.gz.tbi /home/jupyter/Mapping/MDE_S406.vcf.gz.tbi\n",
      "Executing: mv /home/jupyter/Mapping/S419_MDE.vcf.gz /home/jupyter/Mapping/MDE_S419.vcf.gz\n",
      "Executing: mv /home/jupyter/Mapping/S419_MDE.vcf.gz.tbi /home/jupyter/Mapping/MDE_S419.vcf.gz.tbi\n",
      "Executing: mv /home/jupyter/Mapping/S606_MDE.vcf.gz /home/jupyter/Mapping/MDE_S606.vcf.gz\n",
      "Executing: mv /home/jupyter/Mapping/S606_MDE.vcf.gz.tbi /home/jupyter/Mapping/MDE_S606.vcf.gz.tbi\n",
      "Executing: mv /home/jupyter/Mapping/S618_MDE.vcf.gz /home/jupyter/Mapping/MDE_S618.vcf.gz\n",
      "Executing: mv /home/jupyter/Mapping/S618_MDE.vcf.gz.tbi /home/jupyter/Mapping/MDE_S618.vcf.gz.tbi\n",
      "Executing: mv /home/jupyter/Mapping/S892_SAS.vcf.gz /home/jupyter/Mapping/SAS_S892.vcf.gz\n",
      "Executing: mv /home/jupyter/Mapping/S892_SAS.vcf.gz.tbi /home/jupyter/Mapping/SAS_S892.vcf.gz.tbi\n"
     ]
    }
   ],
   "source": [
    "# Rename pools to match the desired nomenclature\n",
    "with open(f'{WORK_DIR}/to_rename_pools.txt', 'r') as file:\n",
    "    # Iterate over each line in the file\n",
    "    for line in file:\n",
    "        # Remove the newline character and get the pool name\n",
    "        pool = line.strip()\n",
    "        name = pool.split('/')[-1]  # Extract the file name from the path\n",
    "        pool_id, ancestry = name.split('_')  # Split the name into pool ID and ancestry\n",
    "\n",
    "        # Rename the VCF and index files to match the new nomenclature\n",
    "        shell_do(f\"mv {WORK_DIR}/{name}.vcf.gz {WORK_DIR}/{ancestry}_{pool_id}.vcf.gz\")\n",
    "        shell_do(f\"mv {WORK_DIR}/{name}.vcf.gz.tbi {WORK_DIR}/{ancestry}_{pool_id}.vcf.gz.tbi\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: ls /home/jupyter/Mapping/*.vcf.gz | sed 's/.vcf.gz//g' | sort -V | grep -v 'chr1_shared_roh' > /home/jupyter/Mapping/list_renamed_pools.txt\n"
     ]
    }
   ],
   "source": [
    "# Generate an updated list of renamed pools, excluding certain files\n",
    "shell_do(f\"ls {WORK_DIR}/*.vcf.gz | sed 's/.vcf.gz//g' | sort -V | grep -v 'chr1_shared_roh' > {WORK_DIR}/list_renamed_pools.txt\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Fine mapping in shared alleles ROH (subset ROH regions shared)\n",
    "\n",
    "Perform fine mapping on the ROH regions shared between individuals. This step narrows down potential causal variants."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: tail /home/jupyter/Mapping/pools_origin.tsv\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "\r\n",
      "\r\n",
      "\r\n",
      "\r\n",
      "\r\n",
      "\r\n",
      "\r\n",
      "\r\n",
      "\r\n"
     ]
    }
   ],
   "source": [
    "# Display the last few lines of the pool origin file\n",
    "shell_do(f'tail {WORK_DIR}/pools_origin.tsv')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Select individuals that share alleles\n",
    "\n",
    "Identify individuals who share alleles in the same ROH to investigate potential genetic similarities."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Initialize an empty DataFrame to store allele-matched ROH data\n",
    "allele_match_ROH = pd.DataFrame()\n",
    "\n",
    "with open(f'{WORK_DIR}/pools_origin.tsv', 'r') as file:\n",
    "    # Iterate over each line in the file\n",
    "    for line in file:\n",
    "        # Generate variables from the line\n",
    "        pool = line.strip()\n",
    "        ancestry, type_pool, pool_id, path = pool.split('\\t')\n",
    "\n",
    "        # Load the corresponding .hom.overlap file\n",
    "        hom_overlap = pd.read_csv(f\"{WORK_DIR}/{path}\", sep='\\s+')\n",
    "\n",
    "        # Clean the GRP column by removing asterisks (*)\n",
    "        hom_overlap['GRP'] = hom_overlap['GRP'].astype(str).str.replace('*', '', regex=False)\n",
    "\n",
    "        # Filter the pool for the current pool ID\n",
    "        work_pool = hom_overlap[hom_overlap['POOL'] == pool_id]\n",
    "\n",
    "        # Select ROH pools that shared alleles\n",
    "        result = work_pool[work_pool['NSIM'] > 0]\n",
    "        shared1 = result[result['GRP'] == '1']\n",
    "\n",
    "        # Insert the ancestry as the first column\n",
    "        shared1.insert(0, 'Ancestry', ancestry)\n",
    "\n",
    "        # Append the shared ROH data to the main DataFrame\n",
    "        allele_match_ROH = pd.concat([allele_match_ROH, shared1], ignore_index=True)\n",
    "\n",
    "        # If there are shared ROHs, write to a TSV file\n",
    "        if len(shared1) > 0:\n",
    "            shared1['Total'] = len(work_pool) - 2\n",
    "            shared1[['IID', 'CHR', 'BP1', 'BP2', 'Total']].to_csv(\n",
    "                f'{WORK_DIR}/PHENO_{ancestry}_{pool_id}.tsv', sep='\\t', index=False\n",
    "            )\n",
    "\n",
    "# Display the final allele-matched ROH DataFrame\n",
    "allele_match_ROH"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: ls /home/jupyter/Mapping/PHENO* | sed 's|/home/jupyter/Mapping/PHENO_||g' | sed 's/.tsv//g' > /home/jupyter/Mapping/shared_alleles_pools.txt\n"
     ]
    }
   ],
   "source": [
    "# List PHENO files, clean the filenames, and save the result to shared_alleles_pools.txt\n",
    "shell_do(f\"ls {WORK_DIR}/PHENO* | sed 's|{WORK_DIR}/PHENO_||g' | sed 's/.tsv//g' > {WORK_DIR}/shared_alleles_pools.txt\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: grep -f /home/jupyter/Mapping/shared_alleles_pools.txt /home/jupyter/Mapping/list_renamed_pools.txt > /home/jupyter/Mapping/list_shared_alleles_pools.txt\n"
     ]
    }
   ],
   "source": [
    "# Filter list_renamed_pools.txt using shared_alleles_pools.txt and save the result\n",
    "shell_do(f\"grep -f {WORK_DIR}/shared_alleles_pools.txt {WORK_DIR}/list_renamed_pools.txt > {WORK_DIR}/list_shared_alleles_pools.txt\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Select homozygous variants and extract CSQ from the VCF file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing pool: /home/jupyter/Mapping/AMR_S97\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: cat /home/jupyter/Mapping/PHENO_AMR_S97.tsv | cut -f1 | sed 1d > /home/jupyter/Mapping/PHENO_AMR_S97.IID\n",
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/AMR_S97.vcf.gz --make-just-psam --out /home/jupyter/Mapping/AMR_S97\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/Mapping/AMR_S97.log.\n",
      "Options in effect:\n",
      "  --make-just-psam\n",
      "  --out /home/jupyter/Mapping/AMR_S97\n",
      "  --vcf /home/jupyter/Mapping/AMR_S97.vcf.gz\n",
      "\n",
      "Start time: Tue Nov 26 05:25:49 2024\n",
      "628867 MiB RAM detected, ~621586 available; reserving 314433 MiB for main\n",
      "workspace.\n",
      "Using up to 96 threads (change this with --threads).\n",
      "--vcf: 57669 variants scanned.\n",
      "--vcf: /home/jupyter/Mapping/AMR_S97-temporary.pgen +\n",
      "/home/jupyter/Mapping/AMR_S97-temporary.pvar.zst +\n",
      "/home/jupyter/Mapping/AMR_S97-temporary.psam written.\n",
      "2 samples (0 females, 0 males, 2 ambiguous; 2 founders) loaded from\n",
      "/home/jupyter/Mapping/AMR_S97-temporary.psam.\n",
      "Note: No phenotype data present.\n",
      "Writing /home/jupyter/Mapping/AMR_S97.psam ... done.\n",
      "End time: Tue Nov 26 05:25:49 2024\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/AMR_S97.vcf.gz                            --set-all-var-ids @:#:\\$r:\\$a                            --new-id-max-allele-len 160                            --keep /home/jupyter/Mapping/PHENO_AMR_S97.IID                            --recode A --out /home/jupyter/Mapping/AMR_S97\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/Mapping/AMR_S97.log.\n",
      "Options in effect:\n",
      "  --export A\n",
      "  --keep /home/jupyter/Mapping/PHENO_AMR_S97.IID\n",
      "  --new-id-max-allele-len 160\n",
      "  --out /home/jupyter/Mapping/AMR_S97\n",
      "  --set-all-var-ids @:#:$r:$a\n",
      "  --vcf /home/jupyter/Mapping/AMR_S97.vcf.gz\n",
      "\n",
      "Start time: Tue Nov 26 05:25:50 2024\n",
      "628867 MiB RAM detected, ~621596 available; reserving 314433 MiB for main\n",
      "workspace.\n",
      "Using up to 96 threads (change this with --threads).\n",
      "--vcf: 57669 variants scanned.\n",
      "--vcf: /home/jupyter/Mapping/AMR_S97-temporary.pgen +\n",
      "/home/jupyter/Mapping/AMR_S97-temporary.pvar.zst +\n",
      "/home/jupyter/Mapping/AMR_S97-temporary.psam written.\n",
      "2 samples (0 females, 0 males, 2 ambiguous; 2 founders) loaded from\n",
      "/home/jupyter/Mapping/AMR_S97-temporary.psam.\n",
      "57669 variants loaded from /home/jupyter/Mapping/AMR_S97-temporary.pvar.zst.\n",
      "Note: No phenotype data present.\n",
      "--keep: 1 sample remaining.\n",
      "1 sample (0 females, 0 males, 1 ambiguous; 1 founder) remaining after main\n",
      "filters.\n",
      "--export A pass 1/1: loading... 0writing... done.\n",
      "--export A: /home/jupyter/Mapping/AMR_S97.raw written.\n",
      "End time: Tue Nov 26 05:25:51 2024\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: gunzip -c /home/jupyter/Mapping/AMR_S97.vcf.gz > /home/jupyter/Mapping/AMR_S97.vcf\n",
      "Executing: cat /home/jupyter/Mapping/AMR_S97.vcf | grep -v '##' | cut -f1-5 | sed 's/^chr//g' > /home/jupyter/Mapping/AMR_S97_begin\n",
      "Executing: cat /home/jupyter/Mapping/AMR_S97.vcf | grep -v '##' | cut -f8 | sed 's/.*CSQ=//g' | sed 's/|/\t/g' |                      awk -F '\t' 'BEGIN{OFS=FS}{print $18,$4,$2,$22,$3,$53,$66,$126,$96,$108,$76,$78,$39,$40}' > /home/jupyter/Mapping/AMR_S97_end\n",
      "Executing: paste /home/jupyter/Mapping/AMR_S97_begin /home/jupyter/Mapping/AMR_S97_end | sed 1d > /home/jupyter/Mapping/AMR_S97_reduced.tsv\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing pool: /home/jupyter/Mapping/AMR_S122\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: cat /home/jupyter/Mapping/PHENO_AMR_S122.tsv | cut -f1 | sed 1d > /home/jupyter/Mapping/PHENO_AMR_S122.IID\n",
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/AMR_S122.vcf.gz --make-just-psam --out /home/jupyter/Mapping/AMR_S122\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/Mapping/AMR_S122.log.\n",
      "Options in effect:\n",
      "  --make-just-psam\n",
      "  --out /home/jupyter/Mapping/AMR_S122\n",
      "  --vcf /home/jupyter/Mapping/AMR_S122.vcf.gz\n",
      "\n",
      "Start time: Tue Nov 26 05:26:08 2024\n",
      "628867 MiB RAM detected, ~621593 available; reserving 314433 MiB for main\n",
      "workspace.\n",
      "Using up to 96 threads (change this with --threads).\n",
      "--vcf: 14808 variants scanned.\n",
      "--vcf: /home/jupyter/Mapping/AMR_S122-temporary.pgen +\n",
      "/home/jupyter/Mapping/AMR_S122-temporary.pvar.zst +\n",
      "/home/jupyter/Mapping/AMR_S122-temporary.psam written.\n",
      "2 samples (0 females, 0 males, 2 ambiguous; 2 founders) loaded from\n",
      "/home/jupyter/Mapping/AMR_S122-temporary.psam.\n",
      "Note: No phenotype data present.\n",
      "Writing /home/jupyter/Mapping/AMR_S122.psam ... done.\n",
      "End time: Tue Nov 26 05:26:08 2024\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/AMR_S122.vcf.gz                            --set-all-var-ids @:#:\\$r:\\$a                            --new-id-max-allele-len 160                            --keep /home/jupyter/Mapping/PHENO_AMR_S122.IID                            --recode A --out /home/jupyter/Mapping/AMR_S122\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/Mapping/AMR_S122.log.\n",
      "Options in effect:\n",
      "  --export A\n",
      "  --keep /home/jupyter/Mapping/PHENO_AMR_S122.IID\n",
      "  --new-id-max-allele-len 160\n",
      "  --out /home/jupyter/Mapping/AMR_S122\n",
      "  --set-all-var-ids @:#:$r:$a\n",
      "  --vcf /home/jupyter/Mapping/AMR_S122.vcf.gz\n",
      "\n",
      "Start time: Tue Nov 26 05:26:09 2024\n",
      "628867 MiB RAM detected, ~621595 available; reserving 314433 MiB for main\n",
      "workspace.\n",
      "Using up to 96 threads (change this with --threads).\n",
      "--vcf: 14808 variants scanned.\n",
      "--vcf: /home/jupyter/Mapping/AMR_S122-temporary.pgen +\n",
      "/home/jupyter/Mapping/AMR_S122-temporary.pvar.zst +\n",
      "/home/jupyter/Mapping/AMR_S122-temporary.psam written.\n",
      "2 samples (0 females, 0 males, 2 ambiguous; 2 founders) loaded from\n",
      "/home/jupyter/Mapping/AMR_S122-temporary.psam.\n",
      "14808 variants loaded from /home/jupyter/Mapping/AMR_S122-temporary.pvar.zst.\n",
      "Note: No phenotype data present.\n",
      "--keep: 1 sample remaining.\n",
      "1 sample (0 females, 0 males, 1 ambiguous; 1 founder) remaining after main\n",
      "filters.\n",
      "--export A pass 1/1: loading... 0writing... done.\n",
      "--export A: /home/jupyter/Mapping/AMR_S122.raw written.\n",
      "End time: Tue Nov 26 05:26:09 2024\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: gunzip -c /home/jupyter/Mapping/AMR_S122.vcf.gz > /home/jupyter/Mapping/AMR_S122.vcf\n",
      "Executing: cat /home/jupyter/Mapping/AMR_S122.vcf | grep -v '##' | cut -f1-5 | sed 's/^chr//g' > /home/jupyter/Mapping/AMR_S122_begin\n",
      "Executing: cat /home/jupyter/Mapping/AMR_S122.vcf | grep -v '##' | cut -f8 | sed 's/.*CSQ=//g' | sed 's/|/\t/g' |                      awk -F '\t' 'BEGIN{OFS=FS}{print $18,$4,$2,$22,$3,$53,$66,$126,$96,$108,$76,$78,$39,$40}' > /home/jupyter/Mapping/AMR_S122_end\n",
      "Executing: paste /home/jupyter/Mapping/AMR_S122_begin /home/jupyter/Mapping/AMR_S122_end | sed 1d > /home/jupyter/Mapping/AMR_S122_reduced.tsv\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing pool: /home/jupyter/Mapping/AMR_S129\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: cat /home/jupyter/Mapping/PHENO_AMR_S129.tsv | cut -f1 | sed 1d > /home/jupyter/Mapping/PHENO_AMR_S129.IID\n",
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/AMR_S129.vcf.gz --make-just-psam --out /home/jupyter/Mapping/AMR_S129\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/Mapping/AMR_S129.log.\n",
      "Options in effect:\n",
      "  --make-just-psam\n",
      "  --out /home/jupyter/Mapping/AMR_S129\n",
      "  --vcf /home/jupyter/Mapping/AMR_S129.vcf.gz\n",
      "\n",
      "Start time: Tue Nov 26 05:26:19 2024\n",
      "628867 MiB RAM detected, ~621595 available; reserving 314433 MiB for main\n",
      "workspace.\n",
      "Using up to 96 threads (change this with --threads).\n",
      "--vcf: 32245 variants scanned.\n",
      "--vcf: /home/jupyter/Mapping/AMR_S129-temporary.pgen +\n",
      "/home/jupyter/Mapping/AMR_S129-temporary.pvar.zst +\n",
      "/home/jupyter/Mapping/AMR_S129-temporary.psam written.\n",
      "2 samples (0 females, 0 males, 2 ambiguous; 2 founders) loaded from\n",
      "/home/jupyter/Mapping/AMR_S129-temporary.psam.\n",
      "Note: No phenotype data present.\n",
      "Writing /home/jupyter/Mapping/AMR_S129.psam ... done.\n",
      "End time: Tue Nov 26 05:26:19 2024\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/AMR_S129.vcf.gz                            --set-all-var-ids @:#:\\$r:\\$a                            --new-id-max-allele-len 160                            --keep /home/jupyter/Mapping/PHENO_AMR_S129.IID                            --recode A --out /home/jupyter/Mapping/AMR_S129\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/Mapping/AMR_S129.log.\n",
      "Options in effect:\n",
      "  --export A\n",
      "  --keep /home/jupyter/Mapping/PHENO_AMR_S129.IID\n",
      "  --new-id-max-allele-len 160\n",
      "  --out /home/jupyter/Mapping/AMR_S129\n",
      "  --set-all-var-ids @:#:$r:$a\n",
      "  --vcf /home/jupyter/Mapping/AMR_S129.vcf.gz\n",
      "\n",
      "Start time: Tue Nov 26 05:26:20 2024\n",
      "628867 MiB RAM detected, ~621602 available; reserving 314433 MiB for main\n",
      "workspace.\n",
      "Using up to 96 threads (change this with --threads).\n",
      "--vcf: 32245 variants scanned.\n",
      "--vcf: /home/jupyter/Mapping/AMR_S129-temporary.pgen +\n",
      "/home/jupyter/Mapping/AMR_S129-temporary.pvar.zst +\n",
      "/home/jupyter/Mapping/AMR_S129-temporary.psam written.\n",
      "2 samples (0 females, 0 males, 2 ambiguous; 2 founders) loaded from\n",
      "/home/jupyter/Mapping/AMR_S129-temporary.psam.\n",
      "32245 variants loaded from /home/jupyter/Mapping/AMR_S129-temporary.pvar.zst.\n",
      "Note: No phenotype data present.\n",
      "--keep: 1 sample remaining.\n",
      "1 sample (0 females, 0 males, 1 ambiguous; 1 founder) remaining after main\n",
      "filters.\n",
      "--export A pass 1/1: loading... 0writing... done.\n",
      "--export A: /home/jupyter/Mapping/AMR_S129.raw written.\n",
      "End time: Tue Nov 26 05:26:20 2024\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: gunzip -c /home/jupyter/Mapping/AMR_S129.vcf.gz > /home/jupyter/Mapping/AMR_S129.vcf\n",
      "Executing: cat /home/jupyter/Mapping/AMR_S129.vcf | grep -v '##' | cut -f1-5 | sed 's/^chr//g' > /home/jupyter/Mapping/AMR_S129_begin\n",
      "Executing: cat /home/jupyter/Mapping/AMR_S129.vcf | grep -v '##' | cut -f8 | sed 's/.*CSQ=//g' | sed 's/|/\t/g' |                      awk -F '\t' 'BEGIN{OFS=FS}{print $18,$4,$2,$22,$3,$53,$66,$126,$96,$108,$76,$78,$39,$40}' > /home/jupyter/Mapping/AMR_S129_end\n",
      "Executing: paste /home/jupyter/Mapping/AMR_S129_begin /home/jupyter/Mapping/AMR_S129_end | sed 1d > /home/jupyter/Mapping/AMR_S129_reduced.tsv\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing pool: /home/jupyter/Mapping/MDE_S7\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: cat /home/jupyter/Mapping/PHENO_MDE_S7.tsv | cut -f1 | sed 1d > /home/jupyter/Mapping/PHENO_MDE_S7.IID\n",
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/MDE_S7.vcf.gz --make-just-psam --out /home/jupyter/Mapping/MDE_S7\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\r\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\r\n",
      "Logging to /home/jupyter/Mapping/MDE_S7.log.\r\n",
      "Options in effect:\r\n",
      "  --make-just-psam\r\n",
      "  --out /home/jupyter/Mapping/MDE_S7\r\n",
      "  --vcf /home/jupyter/Mapping/MDE_S7.vcf.gz\r\n",
      "\r\n",
      "Start time: Tue Nov 26 05:26:33 2024\r\n",
      "628867 MiB RAM detected, ~621590 available; reserving 314433 MiB for main\r\n",
      "workspace.\r\n",
      "Using up to 96 threads (change this with --threads).\r\n",
      "--vcf: 8263 variants scanned.\r\n",
      "--vcf: /home/jupyter/Mapping/MDE_S7-temporary.pgen +\r\n",
      "/home/jupyter/Mapping/MDE_S7-temporary.pvar.zst +\r\n",
      "/home/jupyter/Mapping/MDE_S7-temporary.psam written.\r\n",
      "5 samples (0 females, 0 males, 5 ambiguous; 5 founders) loaded from\r\n",
      "/home/jupyter/Mapping/MDE_S7-temporary.psam.\r\n",
      "Note: No phenotype data present.\r\n",
      "Writing /home/jupyter/Mapping/MDE_S7.psam ... done.\r\n",
      "End time: Tue Nov 26 05:26:33 2024\r\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/MDE_S7.vcf.gz                            --set-all-var-ids @:#:\\$r:\\$a                            --new-id-max-allele-len 160                            --keep /home/jupyter/Mapping/PHENO_MDE_S7.IID                            --recode A --out /home/jupyter/Mapping/MDE_S7\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/Mapping/MDE_S7.log.\n",
      "Options in effect:\n",
      "  --export A\n",
      "  --keep /home/jupyter/Mapping/PHENO_MDE_S7.IID\n",
      "  --new-id-max-allele-len 160\n",
      "  --out /home/jupyter/Mapping/MDE_S7\n",
      "  --set-all-var-ids @:#:$r:$a\n",
      "  --vcf /home/jupyter/Mapping/MDE_S7.vcf.gz\n",
      "\n",
      "Start time: Tue Nov 26 05:26:34 2024\n",
      "628867 MiB RAM detected, ~621597 available; reserving 314433 MiB for main\n",
      "workspace.\n",
      "Using up to 96 threads (change this with --threads).\n",
      "--vcf: 8263 variants scanned.\n",
      "--vcf: /home/jupyter/Mapping/MDE_S7-temporary.pgen +\n",
      "/home/jupyter/Mapping/MDE_S7-temporary.pvar.zst +\n",
      "/home/jupyter/Mapping/MDE_S7-temporary.psam written.\n",
      "5 samples (0 females, 0 males, 5 ambiguous; 5 founders) loaded from\n",
      "/home/jupyter/Mapping/MDE_S7-temporary.psam.\n",
      "8263 variants loaded from /home/jupyter/Mapping/MDE_S7-temporary.pvar.zst.\n",
      "Note: No phenotype data present.\n",
      "--keep: 2 samples remaining.\n",
      "2 samples (0 females, 0 males, 2 ambiguous; 2 founders) remaining after main\n",
      "filters.\n",
      "--export A pass 1/1: loading... 0writing... 50done.\n",
      "--export A: /home/jupyter/Mapping/MDE_S7.raw written.\n",
      "End time: Tue Nov 26 05:26:34 2024\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: gunzip -c /home/jupyter/Mapping/MDE_S7.vcf.gz > /home/jupyter/Mapping/MDE_S7.vcf\n",
      "Executing: cat /home/jupyter/Mapping/MDE_S7.vcf | grep -v '##' | cut -f1-5 | sed 's/^chr//g' > /home/jupyter/Mapping/MDE_S7_begin\n",
      "Executing: cat /home/jupyter/Mapping/MDE_S7.vcf | grep -v '##' | cut -f8 | sed 's/.*CSQ=//g' | sed 's/|/\t/g' |                      awk -F '\t' 'BEGIN{OFS=FS}{print $18,$4,$2,$22,$3,$53,$66,$126,$96,$108,$76,$78,$39,$40}' > /home/jupyter/Mapping/MDE_S7_end\n",
      "Executing: paste /home/jupyter/Mapping/MDE_S7_begin /home/jupyter/Mapping/MDE_S7_end | sed 1d > /home/jupyter/Mapping/MDE_S7_reduced.tsv\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing pool: /home/jupyter/Mapping/MDE_S51\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: cat /home/jupyter/Mapping/PHENO_MDE_S51.tsv | cut -f1 | sed 1d > /home/jupyter/Mapping/PHENO_MDE_S51.IID\n",
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/MDE_S51.vcf.gz --make-just-psam --out /home/jupyter/Mapping/MDE_S51\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/Mapping/MDE_S51.log.\n",
      "Options in effect:\n",
      "  --make-just-psam\n",
      "  --out /home/jupyter/Mapping/MDE_S51\n",
      "  --vcf /home/jupyter/Mapping/MDE_S51.vcf.gz\n",
      "\n",
      "Start time: Tue Nov 26 05:26:42 2024\n",
      "628867 MiB RAM detected, ~621595 available; reserving 314433 MiB for main\n",
      "workspace.\n",
      "Using up to 96 threads (change this with --threads).\n",
      "--vcf: 53709 variants scanned.\n",
      "--vcf: /home/jupyter/Mapping/MDE_S51-temporary.pgen +\n",
      "/home/jupyter/Mapping/MDE_S51-temporary.pvar.zst +\n",
      "/home/jupyter/Mapping/MDE_S51-temporary.psam written.\n",
      "5 samples (0 females, 0 males, 5 ambiguous; 5 founders) loaded from\n",
      "/home/jupyter/Mapping/MDE_S51-temporary.psam.\n",
      "Note: No phenotype data present.\n",
      "Writing /home/jupyter/Mapping/MDE_S51.psam ... done.\n",
      "End time: Tue Nov 26 05:26:42 2024\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/MDE_S51.vcf.gz                            --set-all-var-ids @:#:\\$r:\\$a                            --new-id-max-allele-len 160                            --keep /home/jupyter/Mapping/PHENO_MDE_S51.IID                            --recode A --out /home/jupyter/Mapping/MDE_S51\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/Mapping/MDE_S51.log.\n",
      "Options in effect:\n",
      "  --export A\n",
      "  --keep /home/jupyter/Mapping/PHENO_MDE_S51.IID\n",
      "  --new-id-max-allele-len 160\n",
      "  --out /home/jupyter/Mapping/MDE_S51\n",
      "  --set-all-var-ids @:#:$r:$a\n",
      "  --vcf /home/jupyter/Mapping/MDE_S51.vcf.gz\n",
      "\n",
      "Start time: Tue Nov 26 05:26:44 2024\n",
      "628867 MiB RAM detected, ~621598 available; reserving 314433 MiB for main\n",
      "workspace.\n",
      "Using up to 96 threads (change this with --threads).\n",
      "--vcf: 53709 variants scanned.\n",
      "--vcf: /home/jupyter/Mapping/MDE_S51-temporary.pgen +\n",
      "/home/jupyter/Mapping/MDE_S51-temporary.pvar.zst +\n",
      "/home/jupyter/Mapping/MDE_S51-temporary.psam written.\n",
      "5 samples (0 females, 0 males, 5 ambiguous; 5 founders) loaded from\n",
      "/home/jupyter/Mapping/MDE_S51-temporary.psam.\n",
      "53709 variants loaded from /home/jupyter/Mapping/MDE_S51-temporary.pvar.zst.\n",
      "Note: No phenotype data present.\n",
      "--keep: 1 sample remaining.\n",
      "1 sample (0 females, 0 males, 1 ambiguous; 1 founder) remaining after main\n",
      "filters.\n",
      "--export A pass 1/1: loading... 0writing... done.\n",
      "--export A: /home/jupyter/Mapping/MDE_S51.raw written.\n",
      "End time: Tue Nov 26 05:26:44 2024\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: gunzip -c /home/jupyter/Mapping/MDE_S51.vcf.gz > /home/jupyter/Mapping/MDE_S51.vcf\n",
      "Executing: cat /home/jupyter/Mapping/MDE_S51.vcf | grep -v '##' | cut -f1-5 | sed 's/^chr//g' > /home/jupyter/Mapping/MDE_S51_begin\n",
      "Executing: cat /home/jupyter/Mapping/MDE_S51.vcf | grep -v '##' | cut -f8 | sed 's/.*CSQ=//g' | sed 's/|/\t/g' |                      awk -F '\t' 'BEGIN{OFS=FS}{print $18,$4,$2,$22,$3,$53,$66,$126,$96,$108,$76,$78,$39,$40}' > /home/jupyter/Mapping/MDE_S51_end\n",
      "Executing: paste /home/jupyter/Mapping/MDE_S51_begin /home/jupyter/Mapping/MDE_S51_end | sed 1d > /home/jupyter/Mapping/MDE_S51_reduced.tsv\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing pool: /home/jupyter/Mapping/MDE_S164\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: cat /home/jupyter/Mapping/PHENO_MDE_S164.tsv | cut -f1 | sed 1d > /home/jupyter/Mapping/PHENO_MDE_S164.IID\n",
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/MDE_S164.vcf.gz --make-just-psam --out /home/jupyter/Mapping/MDE_S164\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/Mapping/MDE_S164.log.\n",
      "Options in effect:\n",
      "  --make-just-psam\n",
      "  --out /home/jupyter/Mapping/MDE_S164\n",
      "  --vcf /home/jupyter/Mapping/MDE_S164.vcf.gz\n",
      "\n",
      "Start time: Tue Nov 26 05:27:02 2024\n",
      "628867 MiB RAM detected, ~621603 available; reserving 314433 MiB for main\n",
      "workspace.\n",
      "Using up to 96 threads (change this with --threads).\n",
      "--vcf: 308278 variants scanned.\n",
      "--vcf: /home/jupyter/Mapping/MDE_S164-temporary.pgen +\n",
      "/home/jupyter/Mapping/MDE_S164-temporary.pvar.zst +\n",
      "/home/jupyter/Mapping/MDE_S164-temporary.psam written.\n",
      "6 samples (0 females, 0 males, 6 ambiguous; 6 founders) loaded from\n",
      "/home/jupyter/Mapping/MDE_S164-temporary.psam.\n",
      "Note: No phenotype data present.\n",
      "Writing /home/jupyter/Mapping/MDE_S164.psam ... done.\n",
      "End time: Tue Nov 26 05:27:03 2024\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/MDE_S164.vcf.gz                            --set-all-var-ids @:#:\\$r:\\$a                            --new-id-max-allele-len 160                            --keep /home/jupyter/Mapping/PHENO_MDE_S164.IID                            --recode A --out /home/jupyter/Mapping/MDE_S164\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/Mapping/MDE_S164.log.\n",
      "Options in effect:\n",
      "  --export A\n",
      "  --keep /home/jupyter/Mapping/PHENO_MDE_S164.IID\n",
      "  --new-id-max-allele-len 160\n",
      "  --out /home/jupyter/Mapping/MDE_S164\n",
      "  --set-all-var-ids @:#:$r:$a\n",
      "  --vcf /home/jupyter/Mapping/MDE_S164.vcf.gz\n",
      "\n",
      "Start time: Tue Nov 26 05:27:04 2024\n",
      "628867 MiB RAM detected, ~621605 available; reserving 314433 MiB for main\n",
      "workspace.\n",
      "Using up to 96 threads (change this with --threads).\n",
      "--vcf: 308278 variants scanned.\n",
      "--vcf: /home/jupyter/Mapping/MDE_S164-temporary.pgen +\n",
      "/home/jupyter/Mapping/MDE_S164-temporary.pvar.zst +\n",
      "/home/jupyter/Mapping/MDE_S164-temporary.psam written.\n",
      "6 samples (0 females, 0 males, 6 ambiguous; 6 founders) loaded from\n",
      "/home/jupyter/Mapping/MDE_S164-temporary.psam.\n",
      "308278 variants loaded from /home/jupyter/Mapping/MDE_S164-temporary.pvar.zst.\n",
      "Note: No phenotype data present.\n",
      "--keep: 4 samples remaining.\n",
      "4 samples (0 females, 0 males, 4 ambiguous; 4 founders) remaining after main\n",
      "filters.\n",
      "--export A pass 1/1: loading... 2142638writing... 255075done.\n",
      "--export A: /home/jupyter/Mapping/MDE_S164.raw written.\n",
      "End time: Tue Nov 26 05:27:06 2024\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: gunzip -c /home/jupyter/Mapping/MDE_S164.vcf.gz > /home/jupyter/Mapping/MDE_S164.vcf\n",
      "Executing: cat /home/jupyter/Mapping/MDE_S164.vcf | grep -v '##' | cut -f1-5 | sed 's/^chr//g' > /home/jupyter/Mapping/MDE_S164_begin\n",
      "Executing: cat /home/jupyter/Mapping/MDE_S164.vcf | grep -v '##' | cut -f8 | sed 's/.*CSQ=//g' | sed 's/|/\t/g' |                      awk -F '\t' 'BEGIN{OFS=FS}{print $18,$4,$2,$22,$3,$53,$66,$126,$96,$108,$76,$78,$39,$40}' > /home/jupyter/Mapping/MDE_S164_end\n",
      "Executing: paste /home/jupyter/Mapping/MDE_S164_begin /home/jupyter/Mapping/MDE_S164_end | sed 1d > /home/jupyter/Mapping/MDE_S164_reduced.tsv\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing pool: /home/jupyter/Mapping/MDE_S186\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: cat /home/jupyter/Mapping/PHENO_MDE_S186.tsv | cut -f1 | sed 1d > /home/jupyter/Mapping/PHENO_MDE_S186.IID\n",
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/MDE_S186.vcf.gz --make-just-psam --out /home/jupyter/Mapping/MDE_S186\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/Mapping/MDE_S186.log.\n",
      "Options in effect:\n",
      "  --make-just-psam\n",
      "  --out /home/jupyter/Mapping/MDE_S186\n",
      "  --vcf /home/jupyter/Mapping/MDE_S186.vcf.gz\n",
      "\n",
      "Start time: Tue Nov 26 05:28:24 2024\n",
      "628867 MiB RAM detected, ~621431 available; reserving 314433 MiB for main\n",
      "workspace.\n",
      "Using up to 96 threads (change this with --threads).\n",
      "--vcf: 332557 variants scanned.\n",
      "--vcf: /home/jupyter/Mapping/MDE_S186-temporary.pgen +\n",
      "/home/jupyter/Mapping/MDE_S186-temporary.pvar.zst +\n",
      "/home/jupyter/Mapping/MDE_S186-temporary.psam written.\n",
      "7 samples (0 females, 0 males, 7 ambiguous; 7 founders) loaded from\n",
      "/home/jupyter/Mapping/MDE_S186-temporary.psam.\n",
      "Note: No phenotype data present.\n",
      "Writing /home/jupyter/Mapping/MDE_S186.psam ... done.\n",
      "End time: Tue Nov 26 05:28:26 2024\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/MDE_S186.vcf.gz                            --set-all-var-ids @:#:\\$r:\\$a                            --new-id-max-allele-len 160                            --keep /home/jupyter/Mapping/PHENO_MDE_S186.IID                            --recode A --out /home/jupyter/Mapping/MDE_S186\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/Mapping/MDE_S186.log.\n",
      "Options in effect:\n",
      "  --export A\n",
      "  --keep /home/jupyter/Mapping/PHENO_MDE_S186.IID\n",
      "  --new-id-max-allele-len 160\n",
      "  --out /home/jupyter/Mapping/MDE_S186\n",
      "  --set-all-var-ids @:#:$r:$a\n",
      "  --vcf /home/jupyter/Mapping/MDE_S186.vcf.gz\n",
      "\n",
      "Start time: Tue Nov 26 05:28:27 2024\n",
      "628867 MiB RAM detected, ~621441 available; reserving 314433 MiB for main\n",
      "workspace.\n",
      "Using up to 96 threads (change this with --threads).\n",
      "--vcf: 332557 variants scanned.\n",
      "--vcf: /home/jupyter/Mapping/MDE_S186-temporary.pgen +\n",
      "/home/jupyter/Mapping/MDE_S186-temporary.pvar.zst +\n",
      "/home/jupyter/Mapping/MDE_S186-temporary.psam written.\n",
      "7 samples (0 females, 0 males, 7 ambiguous; 7 founders) loaded from\n",
      "/home/jupyter/Mapping/MDE_S186-temporary.psam.\n",
      "332557 variants loaded from /home/jupyter/Mapping/MDE_S186-temporary.pvar.zst.\n",
      "Note: No phenotype data present.\n",
      "--keep: 4 samples remaining.\n",
      "4 samples (0 females, 0 males, 4 ambiguous; 4 founders) remaining after main\n",
      "filters.\n",
      "--export A pass 1/1: loading... 193959789writing... 255075done.\n",
      "--export A: /home/jupyter/Mapping/MDE_S186.raw written.\n",
      "End time: Tue Nov 26 05:28:29 2024\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: gunzip -c /home/jupyter/Mapping/MDE_S186.vcf.gz > /home/jupyter/Mapping/MDE_S186.vcf\n",
      "Executing: cat /home/jupyter/Mapping/MDE_S186.vcf | grep -v '##' | cut -f1-5 | sed 's/^chr//g' > /home/jupyter/Mapping/MDE_S186_begin\n",
      "Executing: cat /home/jupyter/Mapping/MDE_S186.vcf | grep -v '##' | cut -f8 | sed 's/.*CSQ=//g' | sed 's/|/\t/g' |                      awk -F '\t' 'BEGIN{OFS=FS}{print $18,$4,$2,$22,$3,$53,$66,$126,$96,$108,$76,$78,$39,$40}' > /home/jupyter/Mapping/MDE_S186_end\n",
      "Executing: paste /home/jupyter/Mapping/MDE_S186_begin /home/jupyter/Mapping/MDE_S186_end | sed 1d > /home/jupyter/Mapping/MDE_S186_reduced.tsv\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing pool: /home/jupyter/Mapping/MDE_S235\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: cat /home/jupyter/Mapping/PHENO_MDE_S235.tsv | cut -f1 | sed 1d > /home/jupyter/Mapping/PHENO_MDE_S235.IID\n",
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/MDE_S235.vcf.gz --make-just-psam --out /home/jupyter/Mapping/MDE_S235\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/Mapping/MDE_S235.log.\n",
      "Options in effect:\n",
      "  --make-just-psam\n",
      "  --out /home/jupyter/Mapping/MDE_S235\n",
      "  --vcf /home/jupyter/Mapping/MDE_S235.vcf.gz\n",
      "\n",
      "Start time: Tue Nov 26 05:29:52 2024\n",
      "628867 MiB RAM detected, ~621330 available; reserving 314433 MiB for main\n",
      "workspace.\n",
      "Using up to 96 threads (change this with --threads).\n",
      "--vcf: 71738 variants scanned.\n",
      "--vcf: /home/jupyter/Mapping/MDE_S235-temporary.pgen +\n",
      "/home/jupyter/Mapping/MDE_S235-temporary.pvar.zst +\n",
      "/home/jupyter/Mapping/MDE_S235-temporary.psam written.\n",
      "3 samples (0 females, 0 males, 3 ambiguous; 3 founders) loaded from\n",
      "/home/jupyter/Mapping/MDE_S235-temporary.psam.\n",
      "Note: No phenotype data present.\n",
      "Writing /home/jupyter/Mapping/MDE_S235.psam ... done.\n",
      "End time: Tue Nov 26 05:29:53 2024\n",
      "Processing pool: /home/jupyter/Mapping/MDE_S288\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: cat /home/jupyter/Mapping/PHENO_MDE_S288.tsv | cut -f1 | sed 1d > /home/jupyter/Mapping/PHENO_MDE_S288.IID\n",
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/MDE_S288.vcf.gz --make-just-psam --out /home/jupyter/Mapping/MDE_S288\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/Mapping/MDE_S288.log.\n",
      "Options in effect:\n",
      "  --make-just-psam\n",
      "  --out /home/jupyter/Mapping/MDE_S288\n",
      "  --vcf /home/jupyter/Mapping/MDE_S288.vcf.gz\n",
      "\n",
      "Start time: Tue Nov 26 05:29:55 2024\n",
      "628867 MiB RAM detected, ~621337 available; reserving 314433 MiB for main\n",
      "workspace.\n",
      "Using up to 96 threads (change this with --threads).\n",
      "--vcf: 38535 variants scanned.\n",
      "--vcf: /home/jupyter/Mapping/MDE_S288-temporary.pgen +\n",
      "/home/jupyter/Mapping/MDE_S288-temporary.pvar.zst +\n",
      "/home/jupyter/Mapping/MDE_S288-temporary.psam written.\n",
      "6 samples (0 females, 0 males, 6 ambiguous; 6 founders) loaded from\n",
      "/home/jupyter/Mapping/MDE_S288-temporary.psam.\n",
      "Note: No phenotype data present.\n",
      "Writing /home/jupyter/Mapping/MDE_S288.psam ... done.\n",
      "End time: Tue Nov 26 05:29:55 2024\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/MDE_S288.vcf.gz                            --set-all-var-ids @:#:\\$r:\\$a                            --new-id-max-allele-len 160                            --keep /home/jupyter/Mapping/PHENO_MDE_S288.IID                            --recode A --out /home/jupyter/Mapping/MDE_S288\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/Mapping/MDE_S288.log.\n",
      "Options in effect:\n",
      "  --export A\n",
      "  --keep /home/jupyter/Mapping/PHENO_MDE_S288.IID\n",
      "  --new-id-max-allele-len 160\n",
      "  --out /home/jupyter/Mapping/MDE_S288\n",
      "  --set-all-var-ids @:#:$r:$a\n",
      "  --vcf /home/jupyter/Mapping/MDE_S288.vcf.gz\n",
      "\n",
      "Start time: Tue Nov 26 05:29:56 2024\n",
      "628867 MiB RAM detected, ~621342 available; reserving 314433 MiB for main\n",
      "workspace.\n",
      "Using up to 96 threads (change this with --threads).\n",
      "--vcf: 38535 variants scanned.\n",
      "--vcf: /home/jupyter/Mapping/MDE_S288-temporary.pgen +\n",
      "/home/jupyter/Mapping/MDE_S288-temporary.pvar.zst +\n",
      "/home/jupyter/Mapping/MDE_S288-temporary.psam written.\n",
      "6 samples (0 females, 0 males, 6 ambiguous; 6 founders) loaded from\n",
      "/home/jupyter/Mapping/MDE_S288-temporary.psam.\n",
      "38535 variants loaded from /home/jupyter/Mapping/MDE_S288-temporary.pvar.zst.\n",
      "Note: No phenotype data present.\n",
      "--keep: 3 samples remaining.\n",
      "3 samples (0 females, 0 males, 3 ambiguous; 3 founders) remaining after main\n",
      "filters.\n",
      "--export A pass 1/1: loading... 0writing... 3366done.\n",
      "--export A: /home/jupyter/Mapping/MDE_S288.raw written.\n",
      "End time: Tue Nov 26 05:29:56 2024\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: gunzip -c /home/jupyter/Mapping/MDE_S288.vcf.gz > /home/jupyter/Mapping/MDE_S288.vcf\n",
      "Executing: cat /home/jupyter/Mapping/MDE_S288.vcf | grep -v '##' | cut -f1-5 | sed 's/^chr//g' > /home/jupyter/Mapping/MDE_S288_begin\n",
      "Executing: cat /home/jupyter/Mapping/MDE_S288.vcf | grep -v '##' | cut -f8 | sed 's/.*CSQ=//g' | sed 's/|/\t/g' |                      awk -F '\t' 'BEGIN{OFS=FS}{print $18,$4,$2,$22,$3,$53,$66,$126,$96,$108,$76,$78,$39,$40}' > /home/jupyter/Mapping/MDE_S288_end\n",
      "Executing: paste /home/jupyter/Mapping/MDE_S288_begin /home/jupyter/Mapping/MDE_S288_end | sed 1d > /home/jupyter/Mapping/MDE_S288_reduced.tsv\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing pool: /home/jupyter/Mapping/MDE_S289\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: cat /home/jupyter/Mapping/PHENO_MDE_S289.tsv | cut -f1 | sed 1d > /home/jupyter/Mapping/PHENO_MDE_S289.IID\n",
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/MDE_S289.vcf.gz --make-just-psam --out /home/jupyter/Mapping/MDE_S289\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/Mapping/MDE_S289.log.\n",
      "Options in effect:\n",
      "  --make-just-psam\n",
      "  --out /home/jupyter/Mapping/MDE_S289\n",
      "  --vcf /home/jupyter/Mapping/MDE_S289.vcf.gz\n",
      "\n",
      "Start time: Tue Nov 26 05:30:12 2024\n",
      "628867 MiB RAM detected, ~621421 available; reserving 314433 MiB for main\n",
      "workspace.\n",
      "Using up to 96 threads (change this with --threads).\n",
      "--vcf: 18969 variants scanned.\n",
      "--vcf: /home/jupyter/Mapping/MDE_S289-temporary.pgen +\n",
      "/home/jupyter/Mapping/MDE_S289-temporary.pvar.zst +\n",
      "/home/jupyter/Mapping/MDE_S289-temporary.psam written.\n",
      "6 samples (0 females, 0 males, 6 ambiguous; 6 founders) loaded from\n",
      "/home/jupyter/Mapping/MDE_S289-temporary.psam.\n",
      "Note: No phenotype data present.\n",
      "Writing /home/jupyter/Mapping/MDE_S289.psam ... done.\n",
      "End time: Tue Nov 26 05:30:12 2024\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/MDE_S289.vcf.gz                            --set-all-var-ids @:#:\\$r:\\$a                            --new-id-max-allele-len 160                            --keep /home/jupyter/Mapping/PHENO_MDE_S289.IID                            --recode A --out /home/jupyter/Mapping/MDE_S289\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/Mapping/MDE_S289.log.\n",
      "Options in effect:\n",
      "  --export A\n",
      "  --keep /home/jupyter/Mapping/PHENO_MDE_S289.IID\n",
      "  --new-id-max-allele-len 160\n",
      "  --out /home/jupyter/Mapping/MDE_S289\n",
      "  --set-all-var-ids @:#:$r:$a\n",
      "  --vcf /home/jupyter/Mapping/MDE_S289.vcf.gz\n",
      "\n",
      "Start time: Tue Nov 26 05:30:13 2024\n",
      "628867 MiB RAM detected, ~621424 available; reserving 314433 MiB for main\n",
      "workspace.\n",
      "Using up to 96 threads (change this with --threads).\n",
      "--vcf: 18969 variants scanned.\n",
      "--vcf: /home/jupyter/Mapping/MDE_S289-temporary.pgen +\n",
      "/home/jupyter/Mapping/MDE_S289-temporary.pvar.zst +\n",
      "/home/jupyter/Mapping/MDE_S289-temporary.psam written.\n",
      "6 samples (0 females, 0 males, 6 ambiguous; 6 founders) loaded from\n",
      "/home/jupyter/Mapping/MDE_S289-temporary.psam.\n",
      "18969 variants loaded from /home/jupyter/Mapping/MDE_S289-temporary.pvar.zst.\n",
      "Note: No phenotype data present.\n",
      "--keep: 3 samples remaining.\n",
      "3 samples (0 females, 0 males, 3 ambiguous; 3 founders) remaining after main\n",
      "filters.\n",
      "--export A pass 1/1: loading... 0writing... 3366done.\n",
      "--export A: /home/jupyter/Mapping/MDE_S289.raw written.\n",
      "End time: Tue Nov 26 05:30:13 2024\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: gunzip -c /home/jupyter/Mapping/MDE_S289.vcf.gz > /home/jupyter/Mapping/MDE_S289.vcf\n",
      "Executing: cat /home/jupyter/Mapping/MDE_S289.vcf | grep -v '##' | cut -f1-5 | sed 's/^chr//g' > /home/jupyter/Mapping/MDE_S289_begin\n",
      "Executing: cat /home/jupyter/Mapping/MDE_S289.vcf | grep -v '##' | cut -f8 | sed 's/.*CSQ=//g' | sed 's/|/\t/g' |                      awk -F '\t' 'BEGIN{OFS=FS}{print $18,$4,$2,$22,$3,$53,$66,$126,$96,$108,$76,$78,$39,$40}' > /home/jupyter/Mapping/MDE_S289_end\n",
      "Executing: paste /home/jupyter/Mapping/MDE_S289_begin /home/jupyter/Mapping/MDE_S289_end | sed 1d > /home/jupyter/Mapping/MDE_S289_reduced.tsv\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing pool: /home/jupyter/Mapping/MDE_S406\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: cat /home/jupyter/Mapping/PHENO_MDE_S406.tsv | cut -f1 | sed 1d > /home/jupyter/Mapping/PHENO_MDE_S406.IID\n",
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/MDE_S406.vcf.gz --make-just-psam --out /home/jupyter/Mapping/MDE_S406\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/Mapping/MDE_S406.log.\n",
      "Options in effect:\n",
      "  --make-just-psam\n",
      "  --out /home/jupyter/Mapping/MDE_S406\n",
      "  --vcf /home/jupyter/Mapping/MDE_S406.vcf.gz\n",
      "\n",
      "Start time: Tue Nov 26 05:30:23 2024\n",
      "628867 MiB RAM detected, ~621434 available; reserving 314433 MiB for main\n",
      "workspace.\n",
      "Using up to 96 threads (change this with --threads).\n",
      "--vcf: 30203 variants scanned.\n",
      "--vcf: /home/jupyter/Mapping/MDE_S406-temporary.pgen +\n",
      "/home/jupyter/Mapping/MDE_S406-temporary.pvar.zst +\n",
      "/home/jupyter/Mapping/MDE_S406-temporary.psam written.\n",
      "4 samples (0 females, 0 males, 4 ambiguous; 4 founders) loaded from\n",
      "/home/jupyter/Mapping/MDE_S406-temporary.psam.\n",
      "Note: No phenotype data present.\n",
      "Writing /home/jupyter/Mapping/MDE_S406.psam ... done.\n",
      "End time: Tue Nov 26 05:30:23 2024\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/MDE_S406.vcf.gz                            --set-all-var-ids @:#:\\$r:\\$a                            --new-id-max-allele-len 160                            --keep /home/jupyter/Mapping/PHENO_MDE_S406.IID                            --recode A --out /home/jupyter/Mapping/MDE_S406\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/Mapping/MDE_S406.log.\n",
      "Options in effect:\n",
      "  --export A\n",
      "  --keep /home/jupyter/Mapping/PHENO_MDE_S406.IID\n",
      "  --new-id-max-allele-len 160\n",
      "  --out /home/jupyter/Mapping/MDE_S406\n",
      "  --set-all-var-ids @:#:$r:$a\n",
      "  --vcf /home/jupyter/Mapping/MDE_S406.vcf.gz\n",
      "\n",
      "Start time: Tue Nov 26 05:30:25 2024\n",
      "628867 MiB RAM detected, ~621432 available; reserving 314433 MiB for main\n",
      "workspace.\n",
      "Using up to 96 threads (change this with --threads).\n",
      "--vcf: 30203 variants scanned.\n",
      "--vcf: /home/jupyter/Mapping/MDE_S406-temporary.pgen +\n",
      "/home/jupyter/Mapping/MDE_S406-temporary.pvar.zst +\n",
      "/home/jupyter/Mapping/MDE_S406-temporary.psam written.\n",
      "4 samples (0 females, 0 males, 4 ambiguous; 4 founders) loaded from\n",
      "/home/jupyter/Mapping/MDE_S406-temporary.psam.\n",
      "30203 variants loaded from /home/jupyter/Mapping/MDE_S406-temporary.pvar.zst.\n",
      "Note: No phenotype data present.\n",
      "--keep: 3 samples remaining.\n",
      "3 samples (0 females, 0 males, 3 ambiguous; 3 founders) remaining after main\n",
      "filters.\n",
      "--export A pass 1/1: loading... 0writing... 3366done.\n",
      "--export A: /home/jupyter/Mapping/MDE_S406.raw written.\n",
      "End time: Tue Nov 26 05:30:25 2024\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: gunzip -c /home/jupyter/Mapping/MDE_S406.vcf.gz > /home/jupyter/Mapping/MDE_S406.vcf\n",
      "Executing: cat /home/jupyter/Mapping/MDE_S406.vcf | grep -v '##' | cut -f1-5 | sed 's/^chr//g' > /home/jupyter/Mapping/MDE_S406_begin\n",
      "Executing: cat /home/jupyter/Mapping/MDE_S406.vcf | grep -v '##' | cut -f8 | sed 's/.*CSQ=//g' | sed 's/|/\t/g' |                      awk -F '\t' 'BEGIN{OFS=FS}{print $18,$4,$2,$22,$3,$53,$66,$126,$96,$108,$76,$78,$39,$40}' > /home/jupyter/Mapping/MDE_S406_end\n",
      "Executing: paste /home/jupyter/Mapping/MDE_S406_begin /home/jupyter/Mapping/MDE_S406_end | sed 1d > /home/jupyter/Mapping/MDE_S406_reduced.tsv\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing pool: /home/jupyter/Mapping/MDE_S419\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: cat /home/jupyter/Mapping/PHENO_MDE_S419.tsv | cut -f1 | sed 1d > /home/jupyter/Mapping/PHENO_MDE_S419.IID\n",
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/MDE_S419.vcf.gz --make-just-psam --out /home/jupyter/Mapping/MDE_S419\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/Mapping/MDE_S419.log.\n",
      "Options in effect:\n",
      "  --make-just-psam\n",
      "  --out /home/jupyter/Mapping/MDE_S419\n",
      "  --vcf /home/jupyter/Mapping/MDE_S419.vcf.gz\n",
      "\n",
      "Start time: Tue Nov 26 05:30:38 2024\n",
      "628867 MiB RAM detected, ~621428 available; reserving 314433 MiB for main\n",
      "workspace.\n",
      "Using up to 96 threads (change this with --threads).\n",
      "--vcf: 191710 variants scanned.\n",
      "--vcf: /home/jupyter/Mapping/MDE_S419-temporary.pgen +\n",
      "/home/jupyter/Mapping/MDE_S419-temporary.pvar.zst +\n",
      "/home/jupyter/Mapping/MDE_S419-temporary.psam written.\n",
      "5 samples (0 females, 0 males, 5 ambiguous; 5 founders) loaded from\n",
      "/home/jupyter/Mapping/MDE_S419-temporary.psam.\n",
      "Note: No phenotype data present.\n",
      "Writing /home/jupyter/Mapping/MDE_S419.psam ... done.\n",
      "End time: Tue Nov 26 05:30:39 2024\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/MDE_S419.vcf.gz                            --set-all-var-ids @:#:\\$r:\\$a                            --new-id-max-allele-len 160                            --keep /home/jupyter/Mapping/PHENO_MDE_S419.IID                            --recode A --out /home/jupyter/Mapping/MDE_S419\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/Mapping/MDE_S419.log.\n",
      "Options in effect:\n",
      "  --export A\n",
      "  --keep /home/jupyter/Mapping/PHENO_MDE_S419.IID\n",
      "  --new-id-max-allele-len 160\n",
      "  --out /home/jupyter/Mapping/MDE_S419\n",
      "  --set-all-var-ids @:#:$r:$a\n",
      "  --vcf /home/jupyter/Mapping/MDE_S419.vcf.gz\n",
      "\n",
      "Start time: Tue Nov 26 05:30:40 2024\n",
      "628867 MiB RAM detected, ~621436 available; reserving 314433 MiB for main\n",
      "workspace.\n",
      "Using up to 96 threads (change this with --threads).\n",
      "--vcf: 191710 variants scanned.\n",
      "--vcf: /home/jupyter/Mapping/MDE_S419-temporary.pgen +\n",
      "/home/jupyter/Mapping/MDE_S419-temporary.pvar.zst +\n",
      "/home/jupyter/Mapping/MDE_S419-temporary.psam written.\n",
      "5 samples (0 females, 0 males, 5 ambiguous; 5 founders) loaded from\n",
      "/home/jupyter/Mapping/MDE_S419-temporary.psam.\n",
      "191710 variants loaded from /home/jupyter/Mapping/MDE_S419-temporary.pvar.zst.\n",
      "Note: No phenotype data present.\n",
      "--keep: 1 sample remaining.\n",
      "1 sample (0 females, 0 males, 1 ambiguous; 1 founder) remaining after main\n",
      "filters.\n",
      "--export A pass 1/1: loading... 346writing... done.\n",
      "--export A: /home/jupyter/Mapping/MDE_S419.raw written.\n",
      "End time: Tue Nov 26 05:30:41 2024\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: gunzip -c /home/jupyter/Mapping/MDE_S419.vcf.gz > /home/jupyter/Mapping/MDE_S419.vcf\n",
      "Executing: cat /home/jupyter/Mapping/MDE_S419.vcf | grep -v '##' | cut -f1-5 | sed 's/^chr//g' > /home/jupyter/Mapping/MDE_S419_begin\n",
      "Executing: cat /home/jupyter/Mapping/MDE_S419.vcf | grep -v '##' | cut -f8 | sed 's/.*CSQ=//g' | sed 's/|/\t/g' |                      awk -F '\t' 'BEGIN{OFS=FS}{print $18,$4,$2,$22,$3,$53,$66,$126,$96,$108,$76,$78,$39,$40}' > /home/jupyter/Mapping/MDE_S419_end\n",
      "Executing: paste /home/jupyter/Mapping/MDE_S419_begin /home/jupyter/Mapping/MDE_S419_end | sed 1d > /home/jupyter/Mapping/MDE_S419_reduced.tsv\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing pool: /home/jupyter/Mapping/MDE_S618\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: cat /home/jupyter/Mapping/PHENO_MDE_S618.tsv | cut -f1 | sed 1d > /home/jupyter/Mapping/PHENO_MDE_S618.IID\n",
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/MDE_S618.vcf.gz --make-just-psam --out /home/jupyter/Mapping/MDE_S618\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\r\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\r\n",
      "Logging to /home/jupyter/Mapping/MDE_S618.log.\r\n",
      "Options in effect:\r\n",
      "  --make-just-psam\r\n",
      "  --out /home/jupyter/Mapping/MDE_S618\r\n",
      "  --vcf /home/jupyter/Mapping/MDE_S618.vcf.gz\r\n",
      "\r\n",
      "Start time: Tue Nov 26 05:31:29 2024\r\n",
      "628867 MiB RAM detected, ~621417 available; reserving 314433 MiB for main\r\n",
      "workspace.\r\n",
      "Using up to 96 threads (change this with --threads).\r\n",
      "--vcf: 7592 variants scanned.\r\n",
      "--vcf: /home/jupyter/Mapping/MDE_S618-temporary.pgen +\r\n",
      "/home/jupyter/Mapping/MDE_S618-temporary.pvar.zst +\r\n",
      "/home/jupyter/Mapping/MDE_S618-temporary.psam written.\r\n",
      "1 sample (0 females, 0 males, 1 ambiguous; 1 founder) loaded from\r\n",
      "/home/jupyter/Mapping/MDE_S618-temporary.psam.\r\n",
      "Note: No phenotype data present.\r\n",
      "Writing /home/jupyter/Mapping/MDE_S618.psam ... done.\r\n",
      "End time: Tue Nov 26 05:31:29 2024\r\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Ancestry</th>\n",
       "      <th>Pool</th>\n",
       "      <th>chr</th>\n",
       "      <th>bp</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>rsid</th>\n",
       "      <th>gene</th>\n",
       "      <th>variant_type</th>\n",
       "      <th>Variant_class</th>\n",
       "      <th>...</th>\n",
       "      <th>GnomADg_AF</th>\n",
       "      <th>clinical_signif</th>\n",
       "      <th>clinvar_signif</th>\n",
       "      <th>cadd_phred</th>\n",
       "      <th>REVEL</th>\n",
       "      <th>eve_class</th>\n",
       "      <th>am_class</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>Polyphen</th>\n",
       "      <th>colname</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>43647</th>\n",
       "      <td>MDE</td>\n",
       "      <td>S7</td>\n",
       "      <td>1</td>\n",
       "      <td>171501065</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>rs6671036</td>\n",
       "      <td>PRRC2C</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>0.9936</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.154</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1:171501065:G:A_G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43648</th>\n",
       "      <td>MDE</td>\n",
       "      <td>S7</td>\n",
       "      <td>1</td>\n",
       "      <td>171521600</td>\n",
       "      <td>G</td>\n",
       "      <td>GTTC</td>\n",
       "      <td>rs10633005</td>\n",
       "      <td>PRRC2C</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>insertion</td>\n",
       "      <td>...</td>\n",
       "      <td>0.9908</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6.044</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1:171521600:G:GTTC_G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43649</th>\n",
       "      <td>MDE</td>\n",
       "      <td>S7</td>\n",
       "      <td>1</td>\n",
       "      <td>171583376</td>\n",
       "      <td>G</td>\n",
       "      <td>GC</td>\n",
       "      <td>rs57171374</td>\n",
       "      <td>PRRC2C</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>insertion</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8.052</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1:171583376:G:GC_G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43650</th>\n",
       "      <td>MDE</td>\n",
       "      <td>S7</td>\n",
       "      <td>1</td>\n",
       "      <td>171606471</td>\n",
       "      <td>T</td>\n",
       "      <td>TG</td>\n",
       "      <td>rs71107328</td>\n",
       "      <td>MYOCOS</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>insertion</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.703</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1:171606471:T:TG_T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43651</th>\n",
       "      <td>MDE</td>\n",
       "      <td>S7</td>\n",
       "      <td>1</td>\n",
       "      <td>171612323</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>rs12024517</td>\n",
       "      <td>MYOCOS</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.094</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1:171612323:A:C_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26945</th>\n",
       "      <td>AMR</td>\n",
       "      <td>S122</td>\n",
       "      <td>10</td>\n",
       "      <td>29735144</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>rs7098589</td>\n",
       "      <td>SVIL</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>0.1510</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7.393</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10:29735144:C:A_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26946</th>\n",
       "      <td>AMR</td>\n",
       "      <td>S122</td>\n",
       "      <td>10</td>\n",
       "      <td>29739723</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>rs12782184</td>\n",
       "      <td>SVIL</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>0.1459</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.750</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10:29739723:A:G_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26947</th>\n",
       "      <td>AMR</td>\n",
       "      <td>S122</td>\n",
       "      <td>10</td>\n",
       "      <td>29741673</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>rs11007724</td>\n",
       "      <td>SVIL</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>0.7994</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6.252</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10:29741673:T:C_T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26948</th>\n",
       "      <td>AMR</td>\n",
       "      <td>S122</td>\n",
       "      <td>10</td>\n",
       "      <td>29742568</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>rs34535212</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intergenic_variant</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>0.1728</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.982</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10:29742568:G:A_G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26949</th>\n",
       "      <td>AMR</td>\n",
       "      <td>S122</td>\n",
       "      <td>10</td>\n",
       "      <td>29743231</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>rs10826686</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intergenic_variant</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>0.6528</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.083</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10:29743231:C:T_C</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>195642 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Ancestry  Pool  chr         bp REF   ALT        rsid    gene  \\\n",
       "43647      MDE    S7    1  171501065   G     A   rs6671036  PRRC2C   \n",
       "43648      MDE    S7    1  171521600   G  GTTC  rs10633005  PRRC2C   \n",
       "43649      MDE    S7    1  171583376   G    GC  rs57171374  PRRC2C   \n",
       "43650      MDE    S7    1  171606471   T    TG  rs71107328  MYOCOS   \n",
       "43651      MDE    S7    1  171612323   A     C  rs12024517  MYOCOS   \n",
       "...        ...   ...  ...        ...  ..   ...         ...     ...   \n",
       "26945      AMR  S122   10   29735144   C     A   rs7098589    SVIL   \n",
       "26946      AMR  S122   10   29739723   A     G  rs12782184    SVIL   \n",
       "26947      AMR  S122   10   29741673   T     C  rs11007724    SVIL   \n",
       "26948      AMR  S122   10   29742568   G     A  rs34535212     NaN   \n",
       "26949      AMR  S122   10   29743231   C     T  rs10826686     NaN   \n",
       "\n",
       "                variant_type Variant_class  ... GnomADg_AF  clinical_signif  \\\n",
       "43647         intron_variant           SNV  ...     0.9936              NaN   \n",
       "43648         intron_variant     insertion  ...     0.9908              NaN   \n",
       "43649         intron_variant     insertion  ...     1.0000              NaN   \n",
       "43650         intron_variant     insertion  ...     1.0000              NaN   \n",
       "43651         intron_variant           SNV  ...     1.0000              NaN   \n",
       "...                      ...           ...  ...        ...              ...   \n",
       "26945         intron_variant           SNV  ...     0.1510              NaN   \n",
       "26946  upstream_gene_variant           SNV  ...     0.1459              NaN   \n",
       "26947  upstream_gene_variant           SNV  ...     0.7994              NaN   \n",
       "26948     intergenic_variant           SNV  ...     0.1728              NaN   \n",
       "26949     intergenic_variant           SNV  ...     0.6528              NaN   \n",
       "\n",
       "      clinvar_signif  cadd_phred  REVEL  eve_class am_class SIFT Polyphen  \\\n",
       "43647            NaN       2.154    NaN        NaN      NaN  NaN      NaN   \n",
       "43648            NaN       6.044    NaN        NaN      NaN  NaN      NaN   \n",
       "43649            NaN       8.052    NaN        NaN      NaN  NaN      NaN   \n",
       "43650            NaN       4.703    NaN        NaN      NaN  NaN      NaN   \n",
       "43651            NaN       2.094    NaN        NaN      NaN  NaN      NaN   \n",
       "...              ...         ...    ...        ...      ...  ...      ...   \n",
       "26945            NaN       7.393    NaN        NaN      NaN  NaN      NaN   \n",
       "26946            NaN       3.750    NaN        NaN      NaN  NaN      NaN   \n",
       "26947            NaN       6.252    NaN        NaN      NaN  NaN      NaN   \n",
       "26948            NaN       0.982    NaN        NaN      NaN  NaN      NaN   \n",
       "26949            NaN       0.083    NaN        NaN      NaN  NaN      NaN   \n",
       "\n",
       "                    colname  \n",
       "43647     1:171501065:G:A_G  \n",
       "43648  1:171521600:G:GTTC_G  \n",
       "43649    1:171583376:G:GC_G  \n",
       "43650    1:171606471:T:TG_T  \n",
       "43651     1:171612323:A:C_A  \n",
       "...                     ...  \n",
       "26945     10:29735144:C:A_C  \n",
       "26946     10:29739723:A:G_A  \n",
       "26947     10:29741673:T:C_T  \n",
       "26948     10:29742568:G:A_G  \n",
       "26949     10:29743231:C:T_C  \n",
       "\n",
       "[195642 rows x 21 columns]"
      ]
     },
     "execution_count": 92,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Initialize an empty DataFrame to store results from all pools\n",
    "df_results = pd.DataFrame()\n",
    "\n",
    "# Open and read the file containing the list of pools\n",
    "with open(f'{WORK_DIR}/list_shared_alleles_pools.txt', 'r') as file:\n",
    "    # Iterate over each pool listed in the file\n",
    "    for line in file:\n",
    "        # Remove trailing newline characters from the line\n",
    "        pool = line.strip()\n",
    "        print(f\"Processing pool: {pool}\")\n",
    "\n",
    "        # Extract the pool name (e.g., ancestry-specific identifier) from the file path\n",
    "        pool_name = pool.split('/')[4]\n",
    "        \n",
    "        # Generate the PHENO file to subset pools based on individual IDs\n",
    "        shell_do(f\"cat {WORK_DIR}/PHENO_{pool_name}.tsv | cut -f1 | sed 1d > {WORK_DIR}/PHENO_{pool_name}.IID\")\n",
    "\n",
    "        # Read the PHENO file to extract genomic positions and chromosome information\n",
    "        positions = pd.read_csv(f\"{WORK_DIR}/PHENO_{pool_name}.tsv\", sep='\\t')\n",
    "        CHR = positions['CHR'].max()  # Extract the maximum chromosome number (optional)\n",
    "\n",
    "        # Check if WGS data contains the samples sharing alleles\n",
    "        shell_do(f'/home/jupyter/tools/plink2 --vcf {pool}.vcf.gz --make-just-psam --out {pool}')\n",
    "        IIDs_WGS = pd.read_csv(f\"{pool}.psam\", sep='\\t')\n",
    "        is_contained = IIDs_WGS['#IID'].isin(positions['IID']).any()\n",
    "\n",
    "        # Proceed only if at least one sample from the pool is found in WGS data\n",
    "        if is_contained:\n",
    "            # Run PLINK2 to set variant IDs and generate a .raw file with sample data\n",
    "            shell_do(f'/home/jupyter/tools/plink2 --vcf {pool}.vcf.gz \\\n",
    "                           --set-all-var-ids @:#:\\$r:\\$a \\\n",
    "                           --new-id-max-allele-len 160 \\\n",
    "                           --keep {WORK_DIR}/PHENO_{pool_name}.IID \\\n",
    "                           --recode A --out {pool}')\n",
    "            \n",
    "            # Read the generated .raw file into a DataFrame\n",
    "            raw = pd.read_csv(f'{pool}.raw', sep='\\t')\n",
    "            \n",
    "            # Identify columns in the .raw file containing only zeros (homozygous variants)\n",
    "            columns_with_only_0 = raw.columns[(raw.eq(0).all())]\n",
    "\n",
    "            # Decompress the VCF file for annotation purposes\n",
    "            shell_do(f'gunzip -c {pool}.vcf.gz > {pool}.vcf')\n",
    "            \n",
    "            # Extract chromosome, position, reference allele, and alternate allele from the VCF file (first part)\n",
    "            shell_do(f\"cat {pool}.vcf | grep -v '##' | cut -f1-5 | sed 's/^chr//g' > {pool}_begin\")\n",
    "            \n",
    "            # Extract annotation information (e.g., functional effects, allele frequencies) from the VCF file (second part)\n",
    "            shell_do(f\"cat {pool}.vcf | grep -v '##' | cut -f8 | sed 's/.*CSQ=//g' | sed 's/|/\\t/g' | \\\n",
    "                     awk -F '\\t' 'BEGIN{{OFS=FS}}{{print $18,$4,$2,$22,$3,$53,$66,$126,$96,$108,$76,$78,$39,$40}}' > {pool}_end\")\n",
    "            \n",
    "            # Combine the extracted information into a reduced annotation file\n",
    "            shell_do(f'paste {pool}_begin {pool}_end | sed 1d > {pool}_reduced.tsv')\n",
    "            \n",
    "            # Define column names for the reduced annotation file\n",
    "            column_names = [\n",
    "                \"chr\", \"bp\", \"ID\", \"REF\", \"ALT\", \"rsid\", \"gene\", \"variant_type\", \"Variant_class\", \n",
    "                \"variant_impact\", \"GnomADg_AF\", \"clinical_signif\", \"clinvar_signif\", \"cadd_phred\", \n",
    "                \"REVEL\", \"eve_class\", \"am_class\", \"SIFT\", \"Polyphen\"\n",
    "            ]\n",
    "            \n",
    "            # Read the reduced annotation file into a DataFrame\n",
    "            vars_anno_reduced = pd.read_csv(\n",
    "                f'{pool}_reduced.tsv', sep='\\t', low_memory=False, header=None, names=column_names\n",
    "            )\n",
    "            \n",
    "            # Generate a unique ID for each variant by combining chromosome, position, REF, and ALT columns\n",
    "            vars_anno_reduced['ID'] = vars_anno_reduced['chr'].astype(str) + ':' + vars_anno_reduced['bp'].astype(str) + ':' + vars_anno_reduced['REF'] + ':' + vars_anno_reduced['ALT']\n",
    "            \n",
    "            # Create a new column to match the format of column names in the .raw file\n",
    "            vars_anno_reduced['colname'] = vars_anno_reduced['ID'] + '_' + vars_anno_reduced['REF']\n",
    "            \n",
    "            # Drop the temporary ID column as it is no longer needed\n",
    "            vars_anno_reduced = vars_anno_reduced.drop('ID', axis=1)\n",
    "            \n",
    "            # Filter the reduced annotation DataFrame to retain only rows with columns containing zeros in the .raw file\n",
    "            results = vars_anno_reduced.loc[vars_anno_reduced['colname'].isin(columns_with_only_0)]\n",
    "            \n",
    "            # Extract ancestry and pool ID from the pool name\n",
    "            ancestry, pool_id = pool_name.split('_')\n",
    "            \n",
    "            # Add the ancestry as the first column in the filtered DataFrame\n",
    "            results.insert(0, 'Ancestry', ancestry)\n",
    "            \n",
    "            # Add the pool ID as the second column in the filtered DataFrame\n",
    "            results.insert(1, 'Pool', pool_id)\n",
    "            \n",
    "            # Append the filtered results to the main DataFrame\n",
    "            df_results = pd.concat([df_results, results], ignore_index=True)\n",
    "\n",
    "# Sort the concatenated results by 'Ancestry' and 'Pool' in descending order\n",
    "df_results = df_results.sort_values(by=['Ancestry', 'Pool'], ascending=[False, False])\n",
    "\n",
    "# Display the final sorted DataFrame\n",
    "df_results"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Prioritization criteria\n",
    "\n",
    "Apply prioritization criteria to select variants most likely to be disease-causing."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Ancestry</th>\n",
       "      <th>Pool</th>\n",
       "      <th>chr</th>\n",
       "      <th>bp</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>rsid</th>\n",
       "      <th>gene</th>\n",
       "      <th>variant_type</th>\n",
       "      <th>Variant_class</th>\n",
       "      <th>...</th>\n",
       "      <th>GnomADg_AF</th>\n",
       "      <th>clinical_signif</th>\n",
       "      <th>clinvar_signif</th>\n",
       "      <th>cadd_phred</th>\n",
       "      <th>REVEL</th>\n",
       "      <th>eve_class</th>\n",
       "      <th>am_class</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>Polyphen</th>\n",
       "      <th>colname</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>158125</th>\n",
       "      <td>MDE</td>\n",
       "      <td>S419</td>\n",
       "      <td>14</td>\n",
       "      <td>60118341</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>rs773266496&amp;COSV58303323</td>\n",
       "      <td>PCNX4</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000033</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>22.000</td>\n",
       "      <td>0.129504</td>\n",
       "      <td>NaN</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>tolerated(0.06)</td>\n",
       "      <td>benign(0.007)</td>\n",
       "      <td>14:60118341:C:T_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>165508</th>\n",
       "      <td>MDE</td>\n",
       "      <td>S419</td>\n",
       "      <td>14</td>\n",
       "      <td>66924264</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>rs41285470</td>\n",
       "      <td>GPHN</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>0.005868</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.120</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>tolerated_low_confidence(1)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>14:66924264:A:G_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>166222</th>\n",
       "      <td>MDE</td>\n",
       "      <td>S419</td>\n",
       "      <td>14</td>\n",
       "      <td>67577308</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>rs369145807</td>\n",
       "      <td>PLEKHH1</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000066</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>15.360</td>\n",
       "      <td>0.067710</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>deleterious_low_confidence(0.03)</td>\n",
       "      <td>benign(0.078)</td>\n",
       "      <td>14:67577308:C:T_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>167862</th>\n",
       "      <td>MDE</td>\n",
       "      <td>S419</td>\n",
       "      <td>14</td>\n",
       "      <td>69352130</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>rs144512254</td>\n",
       "      <td>GALNT16</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000460</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7.087</td>\n",
       "      <td>0.061827</td>\n",
       "      <td>NaN</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>tolerated(0.17)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>14:69352130:G:A_G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170967</th>\n",
       "      <td>MDE</td>\n",
       "      <td>S419</td>\n",
       "      <td>14</td>\n",
       "      <td>72940346</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>rs145062084&amp;COSV105912372</td>\n",
       "      <td>DCAF4</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000388</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8.009</td>\n",
       "      <td>0.017264</td>\n",
       "      <td>NaN</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>tolerated(0.76)</td>\n",
       "      <td>benign(0.005)</td>\n",
       "      <td>14:72940346:G:A_G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>195580</th>\n",
       "      <td>MDE</td>\n",
       "      <td>S419</td>\n",
       "      <td>14</td>\n",
       "      <td>105863242</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>rs2338627</td>\n",
       "      <td>IGHJ6</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>0.008675</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10.070</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>14:105863242:T:C_T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>195581</th>\n",
       "      <td>MDE</td>\n",
       "      <td>S419</td>\n",
       "      <td>14</td>\n",
       "      <td>105863243</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>rs2338626</td>\n",
       "      <td>IGHJ6</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>0.002090</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9.352</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>14:105863243:A:C_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91062</th>\n",
       "      <td>MDE</td>\n",
       "      <td>S186</td>\n",
       "      <td>1</td>\n",
       "      <td>20649109</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>rs45539432&amp;CM052340</td>\n",
       "      <td>PINK1</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000053</td>\n",
       "      <td>pathogenic</td>\n",
       "      <td>NaN</td>\n",
       "      <td>38.000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1:20649109:C:T_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60131</th>\n",
       "      <td>MDE</td>\n",
       "      <td>S164</td>\n",
       "      <td>1</td>\n",
       "      <td>20649109</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>rs45539432&amp;CM052340</td>\n",
       "      <td>PINK1</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000053</td>\n",
       "      <td>pathogenic</td>\n",
       "      <td>NaN</td>\n",
       "      <td>38.000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1:20649109:C:T_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36334</th>\n",
       "      <td>AMR</td>\n",
       "      <td>S129</td>\n",
       "      <td>4</td>\n",
       "      <td>130964119</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>rs78589926</td>\n",
       "      <td>NaN</td>\n",
       "      <td>splice_donor_variant&amp;non_coding_transcript_var...</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>0.005998</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.198</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4:130964119:G:T_G</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Ancestry  Pool  chr         bp REF ALT                       rsid  \\\n",
       "158125      MDE  S419   14   60118341   C   T   rs773266496&COSV58303323   \n",
       "165508      MDE  S419   14   66924264   A   G                 rs41285470   \n",
       "166222      MDE  S419   14   67577308   C   T                rs369145807   \n",
       "167862      MDE  S419   14   69352130   G   A                rs144512254   \n",
       "170967      MDE  S419   14   72940346   G   A  rs145062084&COSV105912372   \n",
       "195580      MDE  S419   14  105863242   T   C                  rs2338627   \n",
       "195581      MDE  S419   14  105863243   A   C                  rs2338626   \n",
       "91062       MDE  S186    1   20649109   C   T        rs45539432&CM052340   \n",
       "60131       MDE  S164    1   20649109   C   T        rs45539432&CM052340   \n",
       "36334       AMR  S129    4  130964119   G   T                 rs78589926   \n",
       "\n",
       "           gene                                       variant_type  \\\n",
       "158125    PCNX4                                   missense_variant   \n",
       "165508     GPHN                                   missense_variant   \n",
       "166222  PLEKHH1                                   missense_variant   \n",
       "167862  GALNT16                                   missense_variant   \n",
       "170967    DCAF4                                   missense_variant   \n",
       "195580    IGHJ6                                   missense_variant   \n",
       "195581    IGHJ6                                   missense_variant   \n",
       "91062     PINK1                                        stop_gained   \n",
       "60131     PINK1                                        stop_gained   \n",
       "36334       NaN  splice_donor_variant&non_coding_transcript_var...   \n",
       "\n",
       "       Variant_class  ... GnomADg_AF  clinical_signif clinvar_signif  \\\n",
       "158125           SNV  ...   0.000033              NaN            NaN   \n",
       "165508           SNV  ...   0.005868    likely_benign            NaN   \n",
       "166222           SNV  ...   0.000066              NaN            NaN   \n",
       "167862           SNV  ...   0.000460              NaN            NaN   \n",
       "170967           SNV  ...   0.000388    likely_benign            NaN   \n",
       "195580           SNV  ...   0.008675              NaN            NaN   \n",
       "195581           SNV  ...   0.002090              NaN            NaN   \n",
       "91062            SNV  ...   0.000053       pathogenic            NaN   \n",
       "60131            SNV  ...   0.000053       pathogenic            NaN   \n",
       "36334            SNV  ...   0.005998              NaN            NaN   \n",
       "\n",
       "        cadd_phred     REVEL  eve_class       am_class  \\\n",
       "158125      22.000  0.129504        NaN  likely_benign   \n",
       "165508       0.120       NaN        NaN            NaN   \n",
       "166222      15.360  0.067710        NaN            NaN   \n",
       "167862       7.087  0.061827        NaN  likely_benign   \n",
       "170967       8.009  0.017264        NaN  likely_benign   \n",
       "195580      10.070       NaN        NaN            NaN   \n",
       "195581       9.352       NaN        NaN            NaN   \n",
       "91062       38.000       NaN        NaN            NaN   \n",
       "60131       38.000       NaN        NaN            NaN   \n",
       "36334        1.198       NaN        NaN            NaN   \n",
       "\n",
       "                                    SIFT       Polyphen             colname  \n",
       "158125                   tolerated(0.06)  benign(0.007)   14:60118341:C:T_C  \n",
       "165508       tolerated_low_confidence(1)      benign(0)   14:66924264:A:G_A  \n",
       "166222  deleterious_low_confidence(0.03)  benign(0.078)   14:67577308:C:T_C  \n",
       "167862                   tolerated(0.17)      benign(0)   14:69352130:G:A_G  \n",
       "170967                   tolerated(0.76)  benign(0.005)   14:72940346:G:A_G  \n",
       "195580                               NaN            NaN  14:105863242:T:C_T  \n",
       "195581                               NaN            NaN  14:105863243:A:C_A  \n",
       "91062                                NaN            NaN    1:20649109:C:T_C  \n",
       "60131                                NaN            NaN    1:20649109:C:T_C  \n",
       "36334                                NaN            NaN   4:130964119:G:T_G  \n",
       "\n",
       "[10 rows x 21 columns]"
      ]
     },
     "execution_count": 93,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Filter candidate variants based on variant impact and GnomADg_AF frequency\n",
    "candidates_df_results = df_results[\n",
    "    (\n",
    "        (df_results['variant_impact'].isin(['HIGH', 'MODERATE']))  # Impact is HIGH or MODERATE\n",
    "    ) & (\n",
    "        (df_results['GnomADg_AF'].isnull()) | (df_results['GnomADg_AF'] <= 0.01)  # GnomADg_AF is null or <= 0.01\n",
    "    )\n",
    "]\n",
    "\n",
    "# Display the filtered candidate results\n",
    "candidates_df_results"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Select homozygous variants and extract CSQ from the VCF file with unique samples\n",
    "\n",
    "For unique samples, extract homozygous variants and retrieve their consequences from the VCF file."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing pool: /home/jupyter/Mapping/AAC_S396\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/AAC_S396.vcf.gz --make-just-psam --out /home/jupyter/Mapping/AAC_S396\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\r\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\r\n",
      "Logging to /home/jupyter/Mapping/AAC_S396.log.\r\n",
      "Options in effect:\r\n",
      "  --make-just-psam\r\n",
      "  --out /home/jupyter/Mapping/AAC_S396\r\n",
      "  --vcf /home/jupyter/Mapping/AAC_S396.vcf.gz\r\n",
      "\r\n",
      "Start time: Tue Nov 26 05:31:31 2024\r\n",
      "628867 MiB RAM detected, ~621420 available; reserving 314433 MiB for main\r\n",
      "workspace.\r\n",
      "Using up to 96 threads (change this with --threads).\r\n",
      "--vcf: 4516 variants scanned.\r\n",
      "--vcf: /home/jupyter/Mapping/AAC_S396-temporary.pgen +\r\n",
      "/home/jupyter/Mapping/AAC_S396-temporary.pvar.zst +\r\n",
      "/home/jupyter/Mapping/AAC_S396-temporary.psam written.\r\n",
      "1 sample (0 females, 0 males, 1 ambiguous; 1 founder) loaded from\r\n",
      "/home/jupyter/Mapping/AAC_S396-temporary.psam.\r\n",
      "Note: No phenotype data present.\r\n",
      "Writing /home/jupyter/Mapping/AAC_S396.psam ... done.\r\n",
      "End time: Tue Nov 26 05:31:31 2024\r\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/AAC_S396.vcf.gz                            --set-all-var-ids @:#:\\$r:\\$a                            --new-id-max-allele-len 160                            --recode A --out /home/jupyter/Mapping/AAC_S396\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\r\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\r\n",
      "Logging to /home/jupyter/Mapping/AAC_S396.log.\r\n",
      "Options in effect:\r\n",
      "  --export A\r\n",
      "  --new-id-max-allele-len 160\r\n",
      "  --out /home/jupyter/Mapping/AAC_S396\r\n",
      "  --set-all-var-ids @:#:$r:$a\r\n",
      "  --vcf /home/jupyter/Mapping/AAC_S396.vcf.gz\r\n",
      "\r\n",
      "Start time: Tue Nov 26 05:31:32 2024\r\n",
      "628867 MiB RAM detected, ~621413 available; reserving 314433 MiB for main\r\n",
      "workspace.\r\n",
      "Using up to 96 threads (change this with --threads).\r\n",
      "--vcf: 4516 variants scanned.\r\n",
      "--vcf: /home/jupyter/Mapping/AAC_S396-temporary.pgen +\r\n",
      "/home/jupyter/Mapping/AAC_S396-temporary.pvar.zst +\r\n",
      "/home/jupyter/Mapping/AAC_S396-temporary.psam written.\r\n",
      "1 sample (0 females, 0 males, 1 ambiguous; 1 founder) loaded from\r\n",
      "/home/jupyter/Mapping/AAC_S396-temporary.psam.\r\n",
      "4516 variants loaded from /home/jupyter/Mapping/AAC_S396-temporary.pvar.zst.\r\n",
      "Note: No phenotype data present.\r\n",
      "--export A pass 1/1: loading... 0%\b\b\b\b\b\b\b\b\b\b\b\b\bwriting... 0%\b\b0%\b\bdone.\r\n",
      "--export A: /home/jupyter/Mapping/AAC_S396.raw written.\r\n",
      "End time: Tue Nov 26 05:31:32 2024\r\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: gunzip -c /home/jupyter/Mapping/AAC_S396.vcf.gz > /home/jupyter/Mapping/AAC_S396.vcf\n",
      "Executing: cat /home/jupyter/Mapping/AAC_S396.vcf | grep -v '##' | cut -f1-5 | sed 's/^chr//g' > /home/jupyter/Mapping/AAC_S396_begin\n",
      "Executing: cat /home/jupyter/Mapping/AAC_S396.vcf | grep -v '##' | cut -f8 | sed 's/.*CSQ=//g' | sed 's/|/\t/g' |                      awk -F '\t' 'BEGIN{OFS=FS}{print $18,$4,$2,$22,$3,$53,$66,$126,$96,$108,$76,$78,$39,$40}' > /home/jupyter/Mapping/AAC_S396_end\n",
      "Executing: paste /home/jupyter/Mapping/AAC_S396_begin /home/jupyter/Mapping/AAC_S396_end | sed 1d > /home/jupyter/Mapping/AAC_S396_reduced.tsv\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing pool: /home/jupyter/Mapping/AAC_S397\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/AAC_S397.vcf.gz --make-just-psam --out /home/jupyter/Mapping/AAC_S397\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\r\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\r\n",
      "Logging to /home/jupyter/Mapping/AAC_S397.log.\r\n",
      "Options in effect:\r\n",
      "  --make-just-psam\r\n",
      "  --out /home/jupyter/Mapping/AAC_S397\r\n",
      "  --vcf /home/jupyter/Mapping/AAC_S397.vcf.gz\r\n",
      "\r\n",
      "Start time: Tue Nov 26 05:31:38 2024\r\n",
      "628867 MiB RAM detected, ~621416 available; reserving 314433 MiB for main\r\n",
      "workspace.\r\n",
      "Using up to 96 threads (change this with --threads).\r\n",
      "--vcf: 4790 variants scanned.\r\n",
      "--vcf: /home/jupyter/Mapping/AAC_S397-temporary.pgen +\r\n",
      "/home/jupyter/Mapping/AAC_S397-temporary.pvar.zst +\r\n",
      "/home/jupyter/Mapping/AAC_S397-temporary.psam written.\r\n",
      "1 sample (0 females, 0 males, 1 ambiguous; 1 founder) loaded from\r\n",
      "/home/jupyter/Mapping/AAC_S397-temporary.psam.\r\n",
      "Note: No phenotype data present.\r\n",
      "Writing /home/jupyter/Mapping/AAC_S397.psam ... done.\r\n",
      "End time: Tue Nov 26 05:31:38 2024\r\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/AAC_S397.vcf.gz                            --set-all-var-ids @:#:\\$r:\\$a                            --new-id-max-allele-len 160                            --recode A --out /home/jupyter/Mapping/AAC_S397\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\r\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\r\n",
      "Logging to /home/jupyter/Mapping/AAC_S397.log.\r\n",
      "Options in effect:\r\n",
      "  --export A\r\n",
      "  --new-id-max-allele-len 160\r\n",
      "  --out /home/jupyter/Mapping/AAC_S397\r\n",
      "  --set-all-var-ids @:#:$r:$a\r\n",
      "  --vcf /home/jupyter/Mapping/AAC_S397.vcf.gz\r\n",
      "\r\n",
      "Start time: Tue Nov 26 05:31:39 2024\r\n",
      "628867 MiB RAM detected, ~621418 available; reserving 314433 MiB for main\r\n",
      "workspace.\r\n",
      "Using up to 96 threads (change this with --threads).\r\n",
      "--vcf: 4790 variants scanned.\r\n",
      "--vcf: /home/jupyter/Mapping/AAC_S397-temporary.pgen +\r\n",
      "/home/jupyter/Mapping/AAC_S397-temporary.pvar.zst +\r\n",
      "/home/jupyter/Mapping/AAC_S397-temporary.psam written.\r\n",
      "1 sample (0 females, 0 males, 1 ambiguous; 1 founder) loaded from\r\n",
      "/home/jupyter/Mapping/AAC_S397-temporary.psam.\r\n",
      "4790 variants loaded from /home/jupyter/Mapping/AAC_S397-temporary.pvar.zst.\r\n",
      "Note: No phenotype data present.\r\n",
      "--export A pass 1/1: loading... 0%\b\b\b\b\b\b\b\b\b\b\b\b\bwriting... 0%\b\b0%\b\bdone.\r\n",
      "--export A: /home/jupyter/Mapping/AAC_S397.raw written.\r\n",
      "End time: Tue Nov 26 05:31:39 2024\r\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: gunzip -c /home/jupyter/Mapping/AAC_S397.vcf.gz > /home/jupyter/Mapping/AAC_S397.vcf\n",
      "Executing: cat /home/jupyter/Mapping/AAC_S397.vcf | grep -v '##' | cut -f1-5 | sed 's/^chr//g' > /home/jupyter/Mapping/AAC_S397_begin\n",
      "Executing: cat /home/jupyter/Mapping/AAC_S397.vcf | grep -v '##' | cut -f8 | sed 's/.*CSQ=//g' | sed 's/|/\t/g' |                      awk -F '\t' 'BEGIN{OFS=FS}{print $18,$4,$2,$22,$3,$53,$66,$126,$96,$108,$76,$78,$39,$40}' > /home/jupyter/Mapping/AAC_S397_end\n",
      "Executing: paste /home/jupyter/Mapping/AAC_S397_begin /home/jupyter/Mapping/AAC_S397_end | sed 1d > /home/jupyter/Mapping/AAC_S397_reduced.tsv\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing pool: /home/jupyter/Mapping/AJ_S1341\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/AJ_S1341.vcf.gz --make-just-psam --out /home/jupyter/Mapping/AJ_S1341\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/Mapping/AJ_S1341.log.\n",
      "Options in effect:\n",
      "  --make-just-psam\n",
      "  --out /home/jupyter/Mapping/AJ_S1341\n",
      "  --vcf /home/jupyter/Mapping/AJ_S1341.vcf.gz\n",
      "\n",
      "Start time: Tue Nov 26 05:31:46 2024\n",
      "628867 MiB RAM detected, ~621401 available; reserving 314433 MiB for main\n",
      "workspace.\n",
      "Using up to 96 threads (change this with --threads).\n",
      "--vcf: 13411 variants scanned.\n",
      "--vcf: /home/jupyter/Mapping/AJ_S1341-temporary.pgen +\n",
      "/home/jupyter/Mapping/AJ_S1341-temporary.pvar.zst +\n",
      "/home/jupyter/Mapping/AJ_S1341-temporary.psam written.\n",
      "1 sample (0 females, 0 males, 1 ambiguous; 1 founder) loaded from\n",
      "/home/jupyter/Mapping/AJ_S1341-temporary.psam.\n",
      "Note: No phenotype data present.\n",
      "Writing /home/jupyter/Mapping/AJ_S1341.psam ... done.\n",
      "End time: Tue Nov 26 05:31:46 2024\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/AJ_S1341.vcf.gz                            --set-all-var-ids @:#:\\$r:\\$a                            --new-id-max-allele-len 160                            --recode A --out /home/jupyter/Mapping/AJ_S1341\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/Mapping/AJ_S1341.log.\n",
      "Options in effect:\n",
      "  --export A\n",
      "  --new-id-max-allele-len 160\n",
      "  --out /home/jupyter/Mapping/AJ_S1341\n",
      "  --set-all-var-ids @:#:$r:$a\n",
      "  --vcf /home/jupyter/Mapping/AJ_S1341.vcf.gz\n",
      "\n",
      "Start time: Tue Nov 26 05:31:47 2024\n",
      "628867 MiB RAM detected, ~621404 available; reserving 314433 MiB for main\n",
      "workspace.\n",
      "Using up to 96 threads (change this with --threads).\n",
      "--vcf: 13411 variants scanned.\n",
      "--vcf: /home/jupyter/Mapping/AJ_S1341-temporary.pgen +\n",
      "/home/jupyter/Mapping/AJ_S1341-temporary.pvar.zst +\n",
      "/home/jupyter/Mapping/AJ_S1341-temporary.psam written.\n",
      "1 sample (0 females, 0 males, 1 ambiguous; 1 founder) loaded from\n",
      "/home/jupyter/Mapping/AJ_S1341-temporary.psam.\n",
      "13411 variants loaded from /home/jupyter/Mapping/AJ_S1341-temporary.pvar.zst.\n",
      "Note: No phenotype data present.\n",
      "--export A pass 1/1: loading... 0writing... done.\n",
      "--export A: /home/jupyter/Mapping/AJ_S1341.raw written.\n",
      "End time: Tue Nov 26 05:31:47 2024\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: gunzip -c /home/jupyter/Mapping/AJ_S1341.vcf.gz > /home/jupyter/Mapping/AJ_S1341.vcf\n",
      "Executing: cat /home/jupyter/Mapping/AJ_S1341.vcf | grep -v '##' | cut -f1-5 | sed 's/^chr//g' > /home/jupyter/Mapping/AJ_S1341_begin\n",
      "Executing: cat /home/jupyter/Mapping/AJ_S1341.vcf | grep -v '##' | cut -f8 | sed 's/.*CSQ=//g' | sed 's/|/\t/g' |                      awk -F '\t' 'BEGIN{OFS=FS}{print $18,$4,$2,$22,$3,$53,$66,$126,$96,$108,$76,$78,$39,$40}' > /home/jupyter/Mapping/AJ_S1341_end\n",
      "Executing: paste /home/jupyter/Mapping/AJ_S1341_begin /home/jupyter/Mapping/AJ_S1341_end | sed 1d > /home/jupyter/Mapping/AJ_S1341_reduced.tsv\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing pool: /home/jupyter/Mapping/AJ_S1443\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/AJ_S1443.vcf.gz --make-just-psam --out /home/jupyter/Mapping/AJ_S1443\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/Mapping/AJ_S1443.log.\n",
      "Options in effect:\n",
      "  --make-just-psam\n",
      "  --out /home/jupyter/Mapping/AJ_S1443\n",
      "  --vcf /home/jupyter/Mapping/AJ_S1443.vcf.gz\n",
      "\n",
      "Start time: Tue Nov 26 05:31:55 2024\n",
      "628867 MiB RAM detected, ~621409 available; reserving 314433 MiB for main\n",
      "workspace.\n",
      "Using up to 96 threads (change this with --threads).\n",
      "--vcf: 54041 variants scanned.\n",
      "--vcf: /home/jupyter/Mapping/AJ_S1443-temporary.pgen +\n",
      "/home/jupyter/Mapping/AJ_S1443-temporary.pvar.zst +\n",
      "/home/jupyter/Mapping/AJ_S1443-temporary.psam written.\n",
      "1 sample (0 females, 0 males, 1 ambiguous; 1 founder) loaded from\n",
      "/home/jupyter/Mapping/AJ_S1443-temporary.psam.\n",
      "Note: No phenotype data present.\n",
      "Writing /home/jupyter/Mapping/AJ_S1443.psam ... done.\n",
      "End time: Tue Nov 26 05:31:55 2024\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/AJ_S1443.vcf.gz                            --set-all-var-ids @:#:\\$r:\\$a                            --new-id-max-allele-len 160                            --recode A --out /home/jupyter/Mapping/AJ_S1443\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/Mapping/AJ_S1443.log.\n",
      "Options in effect:\n",
      "  --export A\n",
      "  --new-id-max-allele-len 160\n",
      "  --out /home/jupyter/Mapping/AJ_S1443\n",
      "  --set-all-var-ids @:#:$r:$a\n",
      "  --vcf /home/jupyter/Mapping/AJ_S1443.vcf.gz\n",
      "\n",
      "Start time: Tue Nov 26 05:31:56 2024\n",
      "628867 MiB RAM detected, ~621414 available; reserving 314433 MiB for main\n",
      "workspace.\n",
      "Using up to 96 threads (change this with --threads).\n",
      "--vcf: 54041 variants scanned.\n",
      "--vcf: /home/jupyter/Mapping/AJ_S1443-temporary.pgen +\n",
      "/home/jupyter/Mapping/AJ_S1443-temporary.pvar.zst +\n",
      "/home/jupyter/Mapping/AJ_S1443-temporary.psam written.\n",
      "1 sample (0 females, 0 males, 1 ambiguous; 1 founder) loaded from\n",
      "/home/jupyter/Mapping/AJ_S1443-temporary.psam.\n",
      "54041 variants loaded from /home/jupyter/Mapping/AJ_S1443-temporary.pvar.zst.\n",
      "Note: No phenotype data present.\n",
      "--export A pass 1/1: loading... 0writing... done.\n",
      "--export A: /home/jupyter/Mapping/AJ_S1443.raw written.\n",
      "End time: Tue Nov 26 05:31:57 2024\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: gunzip -c /home/jupyter/Mapping/AJ_S1443.vcf.gz > /home/jupyter/Mapping/AJ_S1443.vcf\n",
      "Executing: cat /home/jupyter/Mapping/AJ_S1443.vcf | grep -v '##' | cut -f1-5 | sed 's/^chr//g' > /home/jupyter/Mapping/AJ_S1443_begin\n",
      "Executing: cat /home/jupyter/Mapping/AJ_S1443.vcf | grep -v '##' | cut -f8 | sed 's/.*CSQ=//g' | sed 's/|/\t/g' |                      awk -F '\t' 'BEGIN{OFS=FS}{print $18,$4,$2,$22,$3,$53,$66,$126,$96,$108,$76,$78,$39,$40}' > /home/jupyter/Mapping/AJ_S1443_end\n",
      "Executing: paste /home/jupyter/Mapping/AJ_S1443_begin /home/jupyter/Mapping/AJ_S1443_end | sed 1d > /home/jupyter/Mapping/AJ_S1443_reduced.tsv\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing pool: /home/jupyter/Mapping/AJ_S1510\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/AJ_S1510.vcf.gz --make-just-psam --out /home/jupyter/Mapping/AJ_S1510\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\r\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\r\n",
      "Logging to /home/jupyter/Mapping/AJ_S1510.log.\r\n",
      "Options in effect:\r\n",
      "  --make-just-psam\r\n",
      "  --out /home/jupyter/Mapping/AJ_S1510\r\n",
      "  --vcf /home/jupyter/Mapping/AJ_S1510.vcf.gz\r\n",
      "\r\n",
      "Start time: Tue Nov 26 05:32:13 2024\r\n",
      "628867 MiB RAM detected, ~621417 available; reserving 314433 MiB for main\r\n",
      "workspace.\r\n",
      "Using up to 96 threads (change this with --threads).\r\n",
      "--vcf: 4626 variants scanned.\r\n",
      "--vcf: /home/jupyter/Mapping/AJ_S1510-temporary.pgen +\r\n",
      "/home/jupyter/Mapping/AJ_S1510-temporary.pvar.zst +\r\n",
      "/home/jupyter/Mapping/AJ_S1510-temporary.psam written.\r\n",
      "1 sample (0 females, 0 males, 1 ambiguous; 1 founder) loaded from\r\n",
      "/home/jupyter/Mapping/AJ_S1510-temporary.psam.\r\n",
      "Note: No phenotype data present.\r\n",
      "Writing /home/jupyter/Mapping/AJ_S1510.psam ... done.\r\n",
      "End time: Tue Nov 26 05:32:14 2024\r\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/AJ_S1510.vcf.gz                            --set-all-var-ids @:#:\\$r:\\$a                            --new-id-max-allele-len 160                            --recode A --out /home/jupyter/Mapping/AJ_S1510\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\r\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\r\n",
      "Logging to /home/jupyter/Mapping/AJ_S1510.log.\r\n",
      "Options in effect:\r\n",
      "  --export A\r\n",
      "  --new-id-max-allele-len 160\r\n",
      "  --out /home/jupyter/Mapping/AJ_S1510\r\n",
      "  --set-all-var-ids @:#:$r:$a\r\n",
      "  --vcf /home/jupyter/Mapping/AJ_S1510.vcf.gz\r\n",
      "\r\n",
      "Start time: Tue Nov 26 05:32:15 2024\r\n",
      "628867 MiB RAM detected, ~621419 available; reserving 314433 MiB for main\r\n",
      "workspace.\r\n",
      "Using up to 96 threads (change this with --threads).\r\n",
      "--vcf: 4626 variants scanned.\r\n",
      "--vcf: /home/jupyter/Mapping/AJ_S1510-temporary.pgen +\r\n",
      "/home/jupyter/Mapping/AJ_S1510-temporary.pvar.zst +\r\n",
      "/home/jupyter/Mapping/AJ_S1510-temporary.psam written.\r\n",
      "1 sample (0 females, 0 males, 1 ambiguous; 1 founder) loaded from\r\n",
      "/home/jupyter/Mapping/AJ_S1510-temporary.psam.\r\n",
      "4626 variants loaded from /home/jupyter/Mapping/AJ_S1510-temporary.pvar.zst.\r\n",
      "Note: No phenotype data present.\r\n",
      "--export A pass 1/1: loading... 0%\b\b\b\b\b\b\b\b\b\b\b\b\bwriting... 0%\b\b0%\b\bdone.\r\n",
      "--export A: /home/jupyter/Mapping/AJ_S1510.raw written.\r\n",
      "End time: Tue Nov 26 05:32:15 2024\r\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: gunzip -c /home/jupyter/Mapping/AJ_S1510.vcf.gz > /home/jupyter/Mapping/AJ_S1510.vcf\n",
      "Executing: cat /home/jupyter/Mapping/AJ_S1510.vcf | grep -v '##' | cut -f1-5 | sed 's/^chr//g' > /home/jupyter/Mapping/AJ_S1510_begin\n",
      "Executing: cat /home/jupyter/Mapping/AJ_S1510.vcf | grep -v '##' | cut -f8 | sed 's/.*CSQ=//g' | sed 's/|/\t/g' |                      awk -F '\t' 'BEGIN{OFS=FS}{print $18,$4,$2,$22,$3,$53,$66,$126,$96,$108,$76,$78,$39,$40}' > /home/jupyter/Mapping/AJ_S1510_end\n",
      "Executing: paste /home/jupyter/Mapping/AJ_S1510_begin /home/jupyter/Mapping/AJ_S1510_end | sed 1d > /home/jupyter/Mapping/AJ_S1510_reduced.tsv\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing pool: /home/jupyter/Mapping/AJ_S1511\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/AJ_S1511.vcf.gz --make-just-psam --out /home/jupyter/Mapping/AJ_S1511\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\r\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\r\n",
      "Logging to /home/jupyter/Mapping/AJ_S1511.log.\r\n",
      "Options in effect:\r\n",
      "  --make-just-psam\r\n",
      "  --out /home/jupyter/Mapping/AJ_S1511\r\n",
      "  --vcf /home/jupyter/Mapping/AJ_S1511.vcf.gz\r\n",
      "\r\n",
      "Start time: Tue Nov 26 05:32:21 2024\r\n",
      "628867 MiB RAM detected, ~621414 available; reserving 314433 MiB for main\r\n",
      "workspace.\r\n",
      "Using up to 96 threads (change this with --threads).\r\n",
      "--vcf: 6053 variants scanned.\r\n",
      "--vcf: /home/jupyter/Mapping/AJ_S1511-temporary.pgen +\r\n",
      "/home/jupyter/Mapping/AJ_S1511-temporary.pvar.zst +\r\n",
      "/home/jupyter/Mapping/AJ_S1511-temporary.psam written.\r\n",
      "1 sample (0 females, 0 males, 1 ambiguous; 1 founder) loaded from\r\n",
      "/home/jupyter/Mapping/AJ_S1511-temporary.psam.\r\n",
      "Note: No phenotype data present.\r\n",
      "Writing /home/jupyter/Mapping/AJ_S1511.psam ... done.\r\n",
      "End time: Tue Nov 26 05:32:21 2024\r\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/AJ_S1511.vcf.gz                            --set-all-var-ids @:#:\\$r:\\$a                            --new-id-max-allele-len 160                            --recode A --out /home/jupyter/Mapping/AJ_S1511\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\r\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\r\n",
      "Logging to /home/jupyter/Mapping/AJ_S1511.log.\r\n",
      "Options in effect:\r\n",
      "  --export A\r\n",
      "  --new-id-max-allele-len 160\r\n",
      "  --out /home/jupyter/Mapping/AJ_S1511\r\n",
      "  --set-all-var-ids @:#:$r:$a\r\n",
      "  --vcf /home/jupyter/Mapping/AJ_S1511.vcf.gz\r\n",
      "\r\n",
      "Start time: Tue Nov 26 05:32:22 2024\r\n",
      "628867 MiB RAM detected, ~621417 available; reserving 314433 MiB for main\r\n",
      "workspace.\r\n",
      "Using up to 96 threads (change this with --threads).\r\n",
      "--vcf: 6053 variants scanned.\r\n",
      "--vcf: /home/jupyter/Mapping/AJ_S1511-temporary.pgen +\r\n",
      "/home/jupyter/Mapping/AJ_S1511-temporary.pvar.zst +\r\n",
      "/home/jupyter/Mapping/AJ_S1511-temporary.psam written.\r\n",
      "1 sample (0 females, 0 males, 1 ambiguous; 1 founder) loaded from\r\n",
      "/home/jupyter/Mapping/AJ_S1511-temporary.psam.\r\n",
      "6053 variants loaded from /home/jupyter/Mapping/AJ_S1511-temporary.pvar.zst.\r\n",
      "Note: No phenotype data present.\r\n",
      "--export A pass 1/1: loading... 0%\b\b\b\b\b\b\b\b\b\b\b\b\bwriting... 0%\b\b0%\b\bdone.\r\n",
      "--export A: /home/jupyter/Mapping/AJ_S1511.raw written.\r\n",
      "End time: Tue Nov 26 05:32:22 2024\r\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: gunzip -c /home/jupyter/Mapping/AJ_S1511.vcf.gz > /home/jupyter/Mapping/AJ_S1511.vcf\n",
      "Executing: cat /home/jupyter/Mapping/AJ_S1511.vcf | grep -v '##' | cut -f1-5 | sed 's/^chr//g' > /home/jupyter/Mapping/AJ_S1511_begin\n",
      "Executing: cat /home/jupyter/Mapping/AJ_S1511.vcf | grep -v '##' | cut -f8 | sed 's/.*CSQ=//g' | sed 's/|/\t/g' |                      awk -F '\t' 'BEGIN{OFS=FS}{print $18,$4,$2,$22,$3,$53,$66,$126,$96,$108,$76,$78,$39,$40}' > /home/jupyter/Mapping/AJ_S1511_end\n",
      "Executing: paste /home/jupyter/Mapping/AJ_S1511_begin /home/jupyter/Mapping/AJ_S1511_end | sed 1d > /home/jupyter/Mapping/AJ_S1511_reduced.tsv\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing pool: /home/jupyter/Mapping/AJ_S1515\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/AJ_S1515.vcf.gz --make-just-psam --out /home/jupyter/Mapping/AJ_S1515\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/Mapping/AJ_S1515.log.\n",
      "Options in effect:\n",
      "  --make-just-psam\n",
      "  --out /home/jupyter/Mapping/AJ_S1515\n",
      "  --vcf /home/jupyter/Mapping/AJ_S1515.vcf.gz\n",
      "\n",
      "Start time: Tue Nov 26 05:32:29 2024\n",
      "628867 MiB RAM detected, ~621424 available; reserving 314433 MiB for main\n",
      "workspace.\n",
      "Using up to 96 threads (change this with --threads).\n",
      "--vcf: 12764 variants scanned.\n",
      "--vcf: /home/jupyter/Mapping/AJ_S1515-temporary.pgen +\n",
      "/home/jupyter/Mapping/AJ_S1515-temporary.pvar.zst +\n",
      "/home/jupyter/Mapping/AJ_S1515-temporary.psam written.\n",
      "2 samples (0 females, 0 males, 2 ambiguous; 2 founders) loaded from\n",
      "/home/jupyter/Mapping/AJ_S1515-temporary.psam.\n",
      "Note: No phenotype data present.\n",
      "Writing /home/jupyter/Mapping/AJ_S1515.psam ... done.\n",
      "End time: Tue Nov 26 05:32:29 2024\n",
      "Processing pool: /home/jupyter/Mapping/AJ_S1623\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/AJ_S1623.vcf.gz --make-just-psam --out /home/jupyter/Mapping/AJ_S1623\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/Mapping/AJ_S1623.log.\n",
      "Options in effect:\n",
      "  --make-just-psam\n",
      "  --out /home/jupyter/Mapping/AJ_S1623\n",
      "  --vcf /home/jupyter/Mapping/AJ_S1623.vcf.gz\n",
      "\n",
      "Start time: Tue Nov 26 05:32:30 2024\n",
      "628867 MiB RAM detected, ~621420 available; reserving 314433 MiB for main\n",
      "workspace.\n",
      "Using up to 96 threads (change this with --threads).\n",
      "--vcf: 12708 variants scanned.\n",
      "--vcf: /home/jupyter/Mapping/AJ_S1623-temporary.pgen +\n",
      "/home/jupyter/Mapping/AJ_S1623-temporary.pvar.zst +\n",
      "/home/jupyter/Mapping/AJ_S1623-temporary.psam written.\n",
      "1 sample (0 females, 0 males, 1 ambiguous; 1 founder) loaded from\n",
      "/home/jupyter/Mapping/AJ_S1623-temporary.psam.\n",
      "Note: No phenotype data present.\n",
      "Writing /home/jupyter/Mapping/AJ_S1623.psam ... done.\n",
      "End time: Tue Nov 26 05:32:30 2024\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/AJ_S1623.vcf.gz                            --set-all-var-ids @:#:\\$r:\\$a                            --new-id-max-allele-len 160                            --recode A --out /home/jupyter/Mapping/AJ_S1623\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/Mapping/AJ_S1623.log.\n",
      "Options in effect:\n",
      "  --export A\n",
      "  --new-id-max-allele-len 160\n",
      "  --out /home/jupyter/Mapping/AJ_S1623\n",
      "  --set-all-var-ids @:#:$r:$a\n",
      "  --vcf /home/jupyter/Mapping/AJ_S1623.vcf.gz\n",
      "\n",
      "Start time: Tue Nov 26 05:32:31 2024\n",
      "628867 MiB RAM detected, ~621423 available; reserving 314433 MiB for main\n",
      "workspace.\n",
      "Using up to 96 threads (change this with --threads).\n",
      "--vcf: 12708 variants scanned.\n",
      "--vcf: /home/jupyter/Mapping/AJ_S1623-temporary.pgen +\n",
      "/home/jupyter/Mapping/AJ_S1623-temporary.pvar.zst +\n",
      "/home/jupyter/Mapping/AJ_S1623-temporary.psam written.\n",
      "1 sample (0 females, 0 males, 1 ambiguous; 1 founder) loaded from\n",
      "/home/jupyter/Mapping/AJ_S1623-temporary.psam.\n",
      "12708 variants loaded from /home/jupyter/Mapping/AJ_S1623-temporary.pvar.zst.\n",
      "Note: No phenotype data present.\n",
      "--export A pass 1/1: loading... 0writing... done.\n",
      "--export A: /home/jupyter/Mapping/AJ_S1623.raw written.\n",
      "End time: Tue Nov 26 05:32:31 2024\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: gunzip -c /home/jupyter/Mapping/AJ_S1623.vcf.gz > /home/jupyter/Mapping/AJ_S1623.vcf\n",
      "Executing: cat /home/jupyter/Mapping/AJ_S1623.vcf | grep -v '##' | cut -f1-5 | sed 's/^chr//g' > /home/jupyter/Mapping/AJ_S1623_begin\n",
      "Executing: cat /home/jupyter/Mapping/AJ_S1623.vcf | grep -v '##' | cut -f8 | sed 's/.*CSQ=//g' | sed 's/|/\t/g' |                      awk -F '\t' 'BEGIN{OFS=FS}{print $18,$4,$2,$22,$3,$53,$66,$126,$96,$108,$76,$78,$39,$40}' > /home/jupyter/Mapping/AJ_S1623_end\n",
      "Executing: paste /home/jupyter/Mapping/AJ_S1623_begin /home/jupyter/Mapping/AJ_S1623_end | sed 1d > /home/jupyter/Mapping/AJ_S1623_reduced.tsv\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing pool: /home/jupyter/Mapping/AJ_S1650\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/AJ_S1650.vcf.gz --make-just-psam --out /home/jupyter/Mapping/AJ_S1650\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\r\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\r\n",
      "Logging to /home/jupyter/Mapping/AJ_S1650.log.\r\n",
      "Options in effect:\r\n",
      "  --make-just-psam\r\n",
      "  --out /home/jupyter/Mapping/AJ_S1650\r\n",
      "  --vcf /home/jupyter/Mapping/AJ_S1650.vcf.gz\r\n",
      "\r\n",
      "Start time: Tue Nov 26 05:32:39 2024\r\n",
      "628867 MiB RAM detected, ~621430 available; reserving 314433 MiB for main\r\n",
      "workspace.\r\n",
      "Using up to 96 threads (change this with --threads).\r\n",
      "--vcf: 8451 variants scanned.\r\n",
      "--vcf: /home/jupyter/Mapping/AJ_S1650-temporary.pgen +\r\n",
      "/home/jupyter/Mapping/AJ_S1650-temporary.pvar.zst +\r\n",
      "/home/jupyter/Mapping/AJ_S1650-temporary.psam written.\r\n",
      "1 sample (0 females, 0 males, 1 ambiguous; 1 founder) loaded from\r\n",
      "/home/jupyter/Mapping/AJ_S1650-temporary.psam.\r\n",
      "Note: No phenotype data present.\r\n",
      "Writing /home/jupyter/Mapping/AJ_S1650.psam ... done.\r\n",
      "End time: Tue Nov 26 05:32:39 2024\r\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/AJ_S1650.vcf.gz                            --set-all-var-ids @:#:\\$r:\\$a                            --new-id-max-allele-len 160                            --recode A --out /home/jupyter/Mapping/AJ_S1650\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/Mapping/AJ_S1650.log.\n",
      "Options in effect:\n",
      "  --export A\n",
      "  --new-id-max-allele-len 160\n",
      "  --out /home/jupyter/Mapping/AJ_S1650\n",
      "  --set-all-var-ids @:#:$r:$a\n",
      "  --vcf /home/jupyter/Mapping/AJ_S1650.vcf.gz\n",
      "\n",
      "Start time: Tue Nov 26 05:32:40 2024\n",
      "628867 MiB RAM detected, ~621431 available; reserving 314433 MiB for main\n",
      "workspace.\n",
      "Using up to 96 threads (change this with --threads).\n",
      "--vcf: 8451 variants scanned.\n",
      "--vcf: /home/jupyter/Mapping/AJ_S1650-temporary.pgen +\n",
      "/home/jupyter/Mapping/AJ_S1650-temporary.pvar.zst +\n",
      "/home/jupyter/Mapping/AJ_S1650-temporary.psam written.\n",
      "1 sample (0 females, 0 males, 1 ambiguous; 1 founder) loaded from\n",
      "/home/jupyter/Mapping/AJ_S1650-temporary.psam.\n",
      "8451 variants loaded from /home/jupyter/Mapping/AJ_S1650-temporary.pvar.zst.\n",
      "Note: No phenotype data present.\n",
      "--export A pass 1/1: loading... 0writing... done.\n",
      "--export A: /home/jupyter/Mapping/AJ_S1650.raw written.\n",
      "End time: Tue Nov 26 05:32:40 2024\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: gunzip -c /home/jupyter/Mapping/AJ_S1650.vcf.gz > /home/jupyter/Mapping/AJ_S1650.vcf\n",
      "Executing: cat /home/jupyter/Mapping/AJ_S1650.vcf | grep -v '##' | cut -f1-5 | sed 's/^chr//g' > /home/jupyter/Mapping/AJ_S1650_begin\n",
      "Executing: cat /home/jupyter/Mapping/AJ_S1650.vcf | grep -v '##' | cut -f8 | sed 's/.*CSQ=//g' | sed 's/|/\t/g' |                      awk -F '\t' 'BEGIN{OFS=FS}{print $18,$4,$2,$22,$3,$53,$66,$126,$96,$108,$76,$78,$39,$40}' > /home/jupyter/Mapping/AJ_S1650_end\n",
      "Executing: paste /home/jupyter/Mapping/AJ_S1650_begin /home/jupyter/Mapping/AJ_S1650_end | sed 1d > /home/jupyter/Mapping/AJ_S1650_reduced.tsv\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing pool: /home/jupyter/Mapping/AJ_S1651\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/AJ_S1651.vcf.gz --make-just-psam --out /home/jupyter/Mapping/AJ_S1651\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/Mapping/AJ_S1651.log.\n",
      "Options in effect:\n",
      "  --make-just-psam\n",
      "  --out /home/jupyter/Mapping/AJ_S1651\n",
      "  --vcf /home/jupyter/Mapping/AJ_S1651.vcf.gz\n",
      "\n",
      "Start time: Tue Nov 26 05:32:48 2024\n",
      "628867 MiB RAM detected, ~621430 available; reserving 314433 MiB for main\n",
      "workspace.\n",
      "Using up to 96 threads (change this with --threads).\n",
      "--vcf: 18394 variants scanned.\n",
      "--vcf: /home/jupyter/Mapping/AJ_S1651-temporary.pgen +\n",
      "/home/jupyter/Mapping/AJ_S1651-temporary.pvar.zst +\n",
      "/home/jupyter/Mapping/AJ_S1651-temporary.psam written.\n",
      "1 sample (0 females, 0 males, 1 ambiguous; 1 founder) loaded from\n",
      "/home/jupyter/Mapping/AJ_S1651-temporary.psam.\n",
      "Note: No phenotype data present.\n",
      "Writing /home/jupyter/Mapping/AJ_S1651.psam ... done.\n",
      "End time: Tue Nov 26 05:32:48 2024\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/AJ_S1651.vcf.gz                            --set-all-var-ids @:#:\\$r:\\$a                            --new-id-max-allele-len 160                            --recode A --out /home/jupyter/Mapping/AJ_S1651\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/Mapping/AJ_S1651.log.\n",
      "Options in effect:\n",
      "  --export A\n",
      "  --new-id-max-allele-len 160\n",
      "  --out /home/jupyter/Mapping/AJ_S1651\n",
      "  --set-all-var-ids @:#:$r:$a\n",
      "  --vcf /home/jupyter/Mapping/AJ_S1651.vcf.gz\n",
      "\n",
      "Start time: Tue Nov 26 05:32:49 2024\n",
      "628867 MiB RAM detected, ~621434 available; reserving 314433 MiB for main\n",
      "workspace.\n",
      "Using up to 96 threads (change this with --threads).\n",
      "--vcf: 18394 variants scanned.\n",
      "--vcf: /home/jupyter/Mapping/AJ_S1651-temporary.pgen +\n",
      "/home/jupyter/Mapping/AJ_S1651-temporary.pvar.zst +\n",
      "/home/jupyter/Mapping/AJ_S1651-temporary.psam written.\n",
      "1 sample (0 females, 0 males, 1 ambiguous; 1 founder) loaded from\n",
      "/home/jupyter/Mapping/AJ_S1651-temporary.psam.\n",
      "18394 variants loaded from /home/jupyter/Mapping/AJ_S1651-temporary.pvar.zst.\n",
      "Note: No phenotype data present.\n",
      "--export A pass 1/1: loading... 0writing... done.\n",
      "--export A: /home/jupyter/Mapping/AJ_S1651.raw written.\n",
      "End time: Tue Nov 26 05:32:49 2024\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: gunzip -c /home/jupyter/Mapping/AJ_S1651.vcf.gz > /home/jupyter/Mapping/AJ_S1651.vcf\n",
      "Executing: cat /home/jupyter/Mapping/AJ_S1651.vcf | grep -v '##' | cut -f1-5 | sed 's/^chr//g' > /home/jupyter/Mapping/AJ_S1651_begin\n",
      "Executing: cat /home/jupyter/Mapping/AJ_S1651.vcf | grep -v '##' | cut -f8 | sed 's/.*CSQ=//g' | sed 's/|/\t/g' |                      awk -F '\t' 'BEGIN{OFS=FS}{print $18,$4,$2,$22,$3,$53,$66,$126,$96,$108,$76,$78,$39,$40}' > /home/jupyter/Mapping/AJ_S1651_end\n",
      "Executing: paste /home/jupyter/Mapping/AJ_S1651_begin /home/jupyter/Mapping/AJ_S1651_end | sed 1d > /home/jupyter/Mapping/AJ_S1651_reduced.tsv\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing pool: /home/jupyter/Mapping/AJ_S1665\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/AJ_S1665.vcf.gz --make-just-psam --out /home/jupyter/Mapping/AJ_S1665\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/Mapping/AJ_S1665.log.\n",
      "Options in effect:\n",
      "  --make-just-psam\n",
      "  --out /home/jupyter/Mapping/AJ_S1665\n",
      "  --vcf /home/jupyter/Mapping/AJ_S1665.vcf.gz\n",
      "\n",
      "Start time: Tue Nov 26 05:32:58 2024\n",
      "628867 MiB RAM detected, ~621435 available; reserving 314433 MiB for main\n",
      "workspace.\n",
      "Using up to 96 threads (change this with --threads).\n",
      "--vcf: 12032 variants scanned.\n",
      "--vcf: /home/jupyter/Mapping/AJ_S1665-temporary.pgen +\n",
      "/home/jupyter/Mapping/AJ_S1665-temporary.pvar.zst +\n",
      "/home/jupyter/Mapping/AJ_S1665-temporary.psam written.\n",
      "1 sample (0 females, 0 males, 1 ambiguous; 1 founder) loaded from\n",
      "/home/jupyter/Mapping/AJ_S1665-temporary.psam.\n",
      "Note: No phenotype data present.\n",
      "Writing /home/jupyter/Mapping/AJ_S1665.psam ... done.\n",
      "End time: Tue Nov 26 05:32:58 2024\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/AJ_S1665.vcf.gz                            --set-all-var-ids @:#:\\$r:\\$a                            --new-id-max-allele-len 160                            --recode A --out /home/jupyter/Mapping/AJ_S1665\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/Mapping/AJ_S1665.log.\n",
      "Options in effect:\n",
      "  --export A\n",
      "  --new-id-max-allele-len 160\n",
      "  --out /home/jupyter/Mapping/AJ_S1665\n",
      "  --set-all-var-ids @:#:$r:$a\n",
      "  --vcf /home/jupyter/Mapping/AJ_S1665.vcf.gz\n",
      "\n",
      "Start time: Tue Nov 26 05:32:59 2024\n",
      "628867 MiB RAM detected, ~621437 available; reserving 314433 MiB for main\n",
      "workspace.\n",
      "Using up to 96 threads (change this with --threads).\n",
      "--vcf: 12032 variants scanned.\n",
      "--vcf: /home/jupyter/Mapping/AJ_S1665-temporary.pgen +\n",
      "/home/jupyter/Mapping/AJ_S1665-temporary.pvar.zst +\n",
      "/home/jupyter/Mapping/AJ_S1665-temporary.psam written.\n",
      "1 sample (0 females, 0 males, 1 ambiguous; 1 founder) loaded from\n",
      "/home/jupyter/Mapping/AJ_S1665-temporary.psam.\n",
      "12032 variants loaded from /home/jupyter/Mapping/AJ_S1665-temporary.pvar.zst.\n",
      "Note: No phenotype data present.\n",
      "--export A pass 1/1: loading... 0writing... done.\n",
      "--export A: /home/jupyter/Mapping/AJ_S1665.raw written.\n",
      "End time: Tue Nov 26 05:33:00 2024\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: gunzip -c /home/jupyter/Mapping/AJ_S1665.vcf.gz > /home/jupyter/Mapping/AJ_S1665.vcf\n",
      "Executing: cat /home/jupyter/Mapping/AJ_S1665.vcf | grep -v '##' | cut -f1-5 | sed 's/^chr//g' > /home/jupyter/Mapping/AJ_S1665_begin\n",
      "Executing: cat /home/jupyter/Mapping/AJ_S1665.vcf | grep -v '##' | cut -f8 | sed 's/.*CSQ=//g' | sed 's/|/\t/g' |                      awk -F '\t' 'BEGIN{OFS=FS}{print $18,$4,$2,$22,$3,$53,$66,$126,$96,$108,$76,$78,$39,$40}' > /home/jupyter/Mapping/AJ_S1665_end\n",
      "Executing: paste /home/jupyter/Mapping/AJ_S1665_begin /home/jupyter/Mapping/AJ_S1665_end | sed 1d > /home/jupyter/Mapping/AJ_S1665_reduced.tsv\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing pool: /home/jupyter/Mapping/AMR_S82\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/AMR_S82.vcf.gz --make-just-psam --out /home/jupyter/Mapping/AMR_S82\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/Mapping/AMR_S82.log.\n",
      "Options in effect:\n",
      "  --make-just-psam\n",
      "  --out /home/jupyter/Mapping/AMR_S82\n",
      "  --vcf /home/jupyter/Mapping/AMR_S82.vcf.gz\n",
      "\n",
      "Start time: Tue Nov 26 05:33:07 2024\n",
      "628867 MiB RAM detected, ~621440 available; reserving 314433 MiB for main\n",
      "workspace.\n",
      "Using up to 96 threads (change this with --threads).\n",
      "--vcf: 27524 variants scanned.\n",
      "--vcf: /home/jupyter/Mapping/AMR_S82-temporary.pgen +\n",
      "/home/jupyter/Mapping/AMR_S82-temporary.pvar.zst +\n",
      "/home/jupyter/Mapping/AMR_S82-temporary.psam written.\n",
      "1 sample (0 females, 0 males, 1 ambiguous; 1 founder) loaded from\n",
      "/home/jupyter/Mapping/AMR_S82-temporary.psam.\n",
      "Note: No phenotype data present.\n",
      "Writing /home/jupyter/Mapping/AMR_S82.psam ... done.\n",
      "End time: Tue Nov 26 05:33:08 2024\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/AMR_S82.vcf.gz                            --set-all-var-ids @:#:\\$r:\\$a                            --new-id-max-allele-len 160                            --recode A --out /home/jupyter/Mapping/AMR_S82\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/Mapping/AMR_S82.log.\n",
      "Options in effect:\n",
      "  --export A\n",
      "  --new-id-max-allele-len 160\n",
      "  --out /home/jupyter/Mapping/AMR_S82\n",
      "  --set-all-var-ids @:#:$r:$a\n",
      "  --vcf /home/jupyter/Mapping/AMR_S82.vcf.gz\n",
      "\n",
      "Start time: Tue Nov 26 05:33:09 2024\n",
      "628867 MiB RAM detected, ~621445 available; reserving 314433 MiB for main\n",
      "workspace.\n",
      "Using up to 96 threads (change this with --threads).\n",
      "--vcf: 27524 variants scanned.\n",
      "--vcf: /home/jupyter/Mapping/AMR_S82-temporary.pgen +\n",
      "/home/jupyter/Mapping/AMR_S82-temporary.pvar.zst +\n",
      "/home/jupyter/Mapping/AMR_S82-temporary.psam written.\n",
      "1 sample (0 females, 0 males, 1 ambiguous; 1 founder) loaded from\n",
      "/home/jupyter/Mapping/AMR_S82-temporary.psam.\n",
      "27524 variants loaded from /home/jupyter/Mapping/AMR_S82-temporary.pvar.zst.\n",
      "Note: No phenotype data present.\n",
      "--export A pass 1/1: loading... 0writing... done.\n",
      "--export A: /home/jupyter/Mapping/AMR_S82.raw written.\n",
      "End time: Tue Nov 26 05:33:09 2024\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: gunzip -c /home/jupyter/Mapping/AMR_S82.vcf.gz > /home/jupyter/Mapping/AMR_S82.vcf\n",
      "Executing: cat /home/jupyter/Mapping/AMR_S82.vcf | grep -v '##' | cut -f1-5 | sed 's/^chr//g' > /home/jupyter/Mapping/AMR_S82_begin\n",
      "Executing: cat /home/jupyter/Mapping/AMR_S82.vcf | grep -v '##' | cut -f8 | sed 's/.*CSQ=//g' | sed 's/|/\t/g' |                      awk -F '\t' 'BEGIN{OFS=FS}{print $18,$4,$2,$22,$3,$53,$66,$126,$96,$108,$76,$78,$39,$40}' > /home/jupyter/Mapping/AMR_S82_end\n",
      "Executing: paste /home/jupyter/Mapping/AMR_S82_begin /home/jupyter/Mapping/AMR_S82_end | sed 1d > /home/jupyter/Mapping/AMR_S82_reduced.tsv\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing pool: /home/jupyter/Mapping/AMR_S97\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/AMR_S97.vcf.gz --make-just-psam --out /home/jupyter/Mapping/AMR_S97\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/Mapping/AMR_S97.log.\n",
      "Options in effect:\n",
      "  --make-just-psam\n",
      "  --out /home/jupyter/Mapping/AMR_S97\n",
      "  --vcf /home/jupyter/Mapping/AMR_S97.vcf.gz\n",
      "\n",
      "Start time: Tue Nov 26 05:33:20 2024\n",
      "628867 MiB RAM detected, ~621427 available; reserving 314433 MiB for main\n",
      "workspace.\n",
      "Using up to 96 threads (change this with --threads).\n",
      "--vcf: 57669 variants scanned.\n",
      "--vcf: /home/jupyter/Mapping/AMR_S97-temporary.pgen +\n",
      "/home/jupyter/Mapping/AMR_S97-temporary.pvar.zst +\n",
      "/home/jupyter/Mapping/AMR_S97-temporary.psam written.\n",
      "2 samples (0 females, 0 males, 2 ambiguous; 2 founders) loaded from\n",
      "/home/jupyter/Mapping/AMR_S97-temporary.psam.\n",
      "Note: No phenotype data present.\n",
      "Writing /home/jupyter/Mapping/AMR_S97.psam ... done.\n",
      "End time: Tue Nov 26 05:33:20 2024\n",
      "Processing pool: /home/jupyter/Mapping/AMR_S121\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/AMR_S121.vcf.gz --make-just-psam --out /home/jupyter/Mapping/AMR_S121\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/Mapping/AMR_S121.log.\n",
      "Options in effect:\n",
      "  --make-just-psam\n",
      "  --out /home/jupyter/Mapping/AMR_S121\n",
      "  --vcf /home/jupyter/Mapping/AMR_S121.vcf.gz\n",
      "\n",
      "Start time: Tue Nov 26 05:33:21 2024\n",
      "628867 MiB RAM detected, ~621423 available; reserving 314433 MiB for main\n",
      "workspace.\n",
      "Using up to 96 threads (change this with --threads).\n",
      "--vcf: 13090 variants scanned.\n",
      "--vcf: /home/jupyter/Mapping/AMR_S121-temporary.pgen +\n",
      "/home/jupyter/Mapping/AMR_S121-temporary.pvar.zst +\n",
      "/home/jupyter/Mapping/AMR_S121-temporary.psam written.\n",
      "2 samples (0 females, 0 males, 2 ambiguous; 2 founders) loaded from\n",
      "/home/jupyter/Mapping/AMR_S121-temporary.psam.\n",
      "Note: No phenotype data present.\n",
      "Writing /home/jupyter/Mapping/AMR_S121.psam ... done.\n",
      "End time: Tue Nov 26 05:33:21 2024\n",
      "Processing pool: /home/jupyter/Mapping/AMR_S122\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/AMR_S122.vcf.gz --make-just-psam --out /home/jupyter/Mapping/AMR_S122\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/Mapping/AMR_S122.log.\n",
      "Options in effect:\n",
      "  --make-just-psam\n",
      "  --out /home/jupyter/Mapping/AMR_S122\n",
      "  --vcf /home/jupyter/Mapping/AMR_S122.vcf.gz\n",
      "\n",
      "Start time: Tue Nov 26 05:33:22 2024\n",
      "628867 MiB RAM detected, ~621424 available; reserving 314433 MiB for main\n",
      "workspace.\n",
      "Using up to 96 threads (change this with --threads).\n",
      "--vcf: 14808 variants scanned.\n",
      "--vcf: /home/jupyter/Mapping/AMR_S122-temporary.pgen +\n",
      "/home/jupyter/Mapping/AMR_S122-temporary.pvar.zst +\n",
      "/home/jupyter/Mapping/AMR_S122-temporary.psam written.\n",
      "2 samples (0 females, 0 males, 2 ambiguous; 2 founders) loaded from\n",
      "/home/jupyter/Mapping/AMR_S122-temporary.psam.\n",
      "Note: No phenotype data present.\n",
      "Writing /home/jupyter/Mapping/AMR_S122.psam ... done.\n",
      "End time: Tue Nov 26 05:33:22 2024\n",
      "Processing pool: /home/jupyter/Mapping/AMR_S127\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/AMR_S127.vcf.gz --make-just-psam --out /home/jupyter/Mapping/AMR_S127\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/Mapping/AMR_S127.log.\n",
      "Options in effect:\n",
      "  --make-just-psam\n",
      "  --out /home/jupyter/Mapping/AMR_S127\n",
      "  --vcf /home/jupyter/Mapping/AMR_S127.vcf.gz\n",
      "\n",
      "Start time: Tue Nov 26 05:33:23 2024\n",
      "628867 MiB RAM detected, ~621432 available; reserving 314433 MiB for main\n",
      "workspace.\n",
      "Using up to 96 threads (change this with --threads).\n",
      "--vcf: 33935 variants scanned.\n",
      "--vcf: /home/jupyter/Mapping/AMR_S127-temporary.pgen +\n",
      "/home/jupyter/Mapping/AMR_S127-temporary.pvar.zst +\n",
      "/home/jupyter/Mapping/AMR_S127-temporary.psam written.\n",
      "1 sample (0 females, 0 males, 1 ambiguous; 1 founder) loaded from\n",
      "/home/jupyter/Mapping/AMR_S127-temporary.psam.\n",
      "Note: No phenotype data present.\n",
      "Writing /home/jupyter/Mapping/AMR_S127.psam ... done.\n",
      "End time: Tue Nov 26 05:33:23 2024\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/AMR_S127.vcf.gz                            --set-all-var-ids @:#:\\$r:\\$a                            --new-id-max-allele-len 160                            --recode A --out /home/jupyter/Mapping/AMR_S127\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/Mapping/AMR_S127.log.\n",
      "Options in effect:\n",
      "  --export A\n",
      "  --new-id-max-allele-len 160\n",
      "  --out /home/jupyter/Mapping/AMR_S127\n",
      "  --set-all-var-ids @:#:$r:$a\n",
      "  --vcf /home/jupyter/Mapping/AMR_S127.vcf.gz\n",
      "\n",
      "Start time: Tue Nov 26 05:33:24 2024\n",
      "628867 MiB RAM detected, ~621433 available; reserving 314433 MiB for main\n",
      "workspace.\n",
      "Using up to 96 threads (change this with --threads).\n",
      "--vcf: 33935 variants scanned.\n",
      "--vcf: /home/jupyter/Mapping/AMR_S127-temporary.pgen +\n",
      "/home/jupyter/Mapping/AMR_S127-temporary.pvar.zst +\n",
      "/home/jupyter/Mapping/AMR_S127-temporary.psam written.\n",
      "1 sample (0 females, 0 males, 1 ambiguous; 1 founder) loaded from\n",
      "/home/jupyter/Mapping/AMR_S127-temporary.psam.\n",
      "33935 variants loaded from /home/jupyter/Mapping/AMR_S127-temporary.pvar.zst.\n",
      "Note: No phenotype data present.\n",
      "--export A pass 1/1: loading... 0writing... done.\n",
      "--export A: /home/jupyter/Mapping/AMR_S127.raw written.\n",
      "End time: Tue Nov 26 05:33:25 2024\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: gunzip -c /home/jupyter/Mapping/AMR_S127.vcf.gz > /home/jupyter/Mapping/AMR_S127.vcf\n",
      "Executing: cat /home/jupyter/Mapping/AMR_S127.vcf | grep -v '##' | cut -f1-5 | sed 's/^chr//g' > /home/jupyter/Mapping/AMR_S127_begin\n",
      "Executing: cat /home/jupyter/Mapping/AMR_S127.vcf | grep -v '##' | cut -f8 | sed 's/.*CSQ=//g' | sed 's/|/\t/g' |                      awk -F '\t' 'BEGIN{OFS=FS}{print $18,$4,$2,$22,$3,$53,$66,$126,$96,$108,$76,$78,$39,$40}' > /home/jupyter/Mapping/AMR_S127_end\n",
      "Executing: paste /home/jupyter/Mapping/AMR_S127_begin /home/jupyter/Mapping/AMR_S127_end | sed 1d > /home/jupyter/Mapping/AMR_S127_reduced.tsv\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing pool: /home/jupyter/Mapping/AMR_S128\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/AMR_S128.vcf.gz --make-just-psam --out /home/jupyter/Mapping/AMR_S128\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/Mapping/AMR_S128.log.\n",
      "Options in effect:\n",
      "  --make-just-psam\n",
      "  --out /home/jupyter/Mapping/AMR_S128\n",
      "  --vcf /home/jupyter/Mapping/AMR_S128.vcf.gz\n",
      "\n",
      "Start time: Tue Nov 26 05:33:37 2024\n",
      "628867 MiB RAM detected, ~621439 available; reserving 314433 MiB for main\n",
      "workspace.\n",
      "Using up to 96 threads (change this with --threads).\n",
      "--vcf: 35627 variants scanned.\n",
      "--vcf: /home/jupyter/Mapping/AMR_S128-temporary.pgen +\n",
      "/home/jupyter/Mapping/AMR_S128-temporary.pvar.zst +\n",
      "/home/jupyter/Mapping/AMR_S128-temporary.psam written.\n",
      "2 samples (0 females, 0 males, 2 ambiguous; 2 founders) loaded from\n",
      "/home/jupyter/Mapping/AMR_S128-temporary.psam.\n",
      "Note: No phenotype data present.\n",
      "Writing /home/jupyter/Mapping/AMR_S128.psam ... done.\n",
      "End time: Tue Nov 26 05:33:37 2024\n",
      "Processing pool: /home/jupyter/Mapping/AMR_S129\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/AMR_S129.vcf.gz --make-just-psam --out /home/jupyter/Mapping/AMR_S129\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/Mapping/AMR_S129.log.\n",
      "Options in effect:\n",
      "  --make-just-psam\n",
      "  --out /home/jupyter/Mapping/AMR_S129\n",
      "  --vcf /home/jupyter/Mapping/AMR_S129.vcf.gz\n",
      "\n",
      "Start time: Tue Nov 26 05:33:38 2024\n",
      "628867 MiB RAM detected, ~621442 available; reserving 314433 MiB for main\n",
      "workspace.\n",
      "Using up to 96 threads (change this with --threads).\n",
      "--vcf: 32245 variants scanned.\n",
      "--vcf: /home/jupyter/Mapping/AMR_S129-temporary.pgen +\n",
      "/home/jupyter/Mapping/AMR_S129-temporary.pvar.zst +\n",
      "/home/jupyter/Mapping/AMR_S129-temporary.psam written.\n",
      "2 samples (0 females, 0 males, 2 ambiguous; 2 founders) loaded from\n",
      "/home/jupyter/Mapping/AMR_S129-temporary.psam.\n",
      "Note: No phenotype data present.\n",
      "Writing /home/jupyter/Mapping/AMR_S129.psam ... done.\n",
      "End time: Tue Nov 26 05:33:38 2024\n",
      "Processing pool: /home/jupyter/Mapping/EUR_S5327\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/EUR_S5327.vcf.gz --make-just-psam --out /home/jupyter/Mapping/EUR_S5327\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/Mapping/EUR_S5327.log.\n",
      "Options in effect:\n",
      "  --make-just-psam\n",
      "  --out /home/jupyter/Mapping/EUR_S5327\n",
      "  --vcf /home/jupyter/Mapping/EUR_S5327.vcf.gz\n",
      "\n",
      "Start time: Tue Nov 26 05:33:39 2024\n",
      "628867 MiB RAM detected, ~621441 available; reserving 314433 MiB for main\n",
      "workspace.\n",
      "Using up to 96 threads (change this with --threads).\n",
      "--vcf: 30157 variants scanned.\n",
      "--vcf: /home/jupyter/Mapping/EUR_S5327-temporary.pgen +\n",
      "/home/jupyter/Mapping/EUR_S5327-temporary.pvar.zst +\n",
      "/home/jupyter/Mapping/EUR_S5327-temporary.psam written.\n",
      "3 samples (0 females, 0 males, 3 ambiguous; 3 founders) loaded from\n",
      "/home/jupyter/Mapping/EUR_S5327-temporary.psam.\n",
      "Note: No phenotype data present.\n",
      "Writing /home/jupyter/Mapping/EUR_S5327.psam ... done.\n",
      "End time: Tue Nov 26 05:33:39 2024\n",
      "Processing pool: /home/jupyter/Mapping/EUR_S5365\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/EUR_S5365.vcf.gz --make-just-psam --out /home/jupyter/Mapping/EUR_S5365\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/Mapping/EUR_S5365.log.\n",
      "Options in effect:\n",
      "  --make-just-psam\n",
      "  --out /home/jupyter/Mapping/EUR_S5365\n",
      "  --vcf /home/jupyter/Mapping/EUR_S5365.vcf.gz\n",
      "\n",
      "Start time: Tue Nov 26 05:33:41 2024\n",
      "628867 MiB RAM detected, ~621444 available; reserving 314433 MiB for main\n",
      "workspace.\n",
      "Using up to 96 threads (change this with --threads).\n",
      "--vcf: 9944 variants scanned.\n",
      "--vcf: /home/jupyter/Mapping/EUR_S5365-temporary.pgen +\n",
      "/home/jupyter/Mapping/EUR_S5365-temporary.pvar.zst +\n",
      "/home/jupyter/Mapping/EUR_S5365-temporary.psam written.\n",
      "2 samples (0 females, 0 males, 2 ambiguous; 2 founders) loaded from\n",
      "/home/jupyter/Mapping/EUR_S5365-temporary.psam.\n",
      "Note: No phenotype data present.\n",
      "Writing /home/jupyter/Mapping/EUR_S5365.psam ... done.\n",
      "End time: Tue Nov 26 05:33:41 2024\n",
      "Processing pool: /home/jupyter/Mapping/EUR_S5386\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/EUR_S5386.vcf.gz --make-just-psam --out /home/jupyter/Mapping/EUR_S5386\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\r\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\r\n",
      "Logging to /home/jupyter/Mapping/EUR_S5386.log.\r\n",
      "Options in effect:\r\n",
      "  --make-just-psam\r\n",
      "  --out /home/jupyter/Mapping/EUR_S5386\r\n",
      "  --vcf /home/jupyter/Mapping/EUR_S5386.vcf.gz\r\n",
      "\r\n",
      "Start time: Tue Nov 26 05:33:42 2024\r\n",
      "628867 MiB RAM detected, ~621444 available; reserving 314433 MiB for main\r\n",
      "workspace.\r\n",
      "Using up to 96 threads (change this with --threads).\r\n",
      "--vcf: 4294 variants scanned.\r\n",
      "--vcf: /home/jupyter/Mapping/EUR_S5386-temporary.pgen +\r\n",
      "/home/jupyter/Mapping/EUR_S5386-temporary.pvar.zst +\r\n",
      "/home/jupyter/Mapping/EUR_S5386-temporary.psam written.\r\n",
      "1 sample (0 females, 0 males, 1 ambiguous; 1 founder) loaded from\r\n",
      "/home/jupyter/Mapping/EUR_S5386-temporary.psam.\r\n",
      "Note: No phenotype data present.\r\n",
      "Writing /home/jupyter/Mapping/EUR_S5386.psam ... done.\r\n",
      "End time: Tue Nov 26 05:33:42 2024\r\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/EUR_S5386.vcf.gz                            --set-all-var-ids @:#:\\$r:\\$a                            --new-id-max-allele-len 160                            --recode A --out /home/jupyter/Mapping/EUR_S5386\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\r\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\r\n",
      "Logging to /home/jupyter/Mapping/EUR_S5386.log.\r\n",
      "Options in effect:\r\n",
      "  --export A\r\n",
      "  --new-id-max-allele-len 160\r\n",
      "  --out /home/jupyter/Mapping/EUR_S5386\r\n",
      "  --set-all-var-ids @:#:$r:$a\r\n",
      "  --vcf /home/jupyter/Mapping/EUR_S5386.vcf.gz\r\n",
      "\r\n",
      "Start time: Tue Nov 26 05:33:43 2024\r\n",
      "628867 MiB RAM detected, ~621443 available; reserving 314433 MiB for main\r\n",
      "workspace.\r\n",
      "Using up to 96 threads (change this with --threads).\r\n",
      "--vcf: 4294 variants scanned.\r\n",
      "--vcf: /home/jupyter/Mapping/EUR_S5386-temporary.pgen +\r\n",
      "/home/jupyter/Mapping/EUR_S5386-temporary.pvar.zst +\r\n",
      "/home/jupyter/Mapping/EUR_S5386-temporary.psam written.\r\n",
      "1 sample (0 females, 0 males, 1 ambiguous; 1 founder) loaded from\r\n",
      "/home/jupyter/Mapping/EUR_S5386-temporary.psam.\r\n",
      "4294 variants loaded from /home/jupyter/Mapping/EUR_S5386-temporary.pvar.zst.\r\n",
      "Note: No phenotype data present.\r\n",
      "--export A pass 1/1: loading... 0%\b\b\b\b\b\b\b\b\b\b\b\b\bwriting... 0%\b\b0%\b\bdone.\r\n",
      "--export A: /home/jupyter/Mapping/EUR_S5386.raw written.\r\n",
      "End time: Tue Nov 26 05:33:43 2024\r\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: gunzip -c /home/jupyter/Mapping/EUR_S5386.vcf.gz > /home/jupyter/Mapping/EUR_S5386.vcf\n",
      "Executing: cat /home/jupyter/Mapping/EUR_S5386.vcf | grep -v '##' | cut -f1-5 | sed 's/^chr//g' > /home/jupyter/Mapping/EUR_S5386_begin\n",
      "Executing: cat /home/jupyter/Mapping/EUR_S5386.vcf | grep -v '##' | cut -f8 | sed 's/.*CSQ=//g' | sed 's/|/\t/g' |                      awk -F '\t' 'BEGIN{OFS=FS}{print $18,$4,$2,$22,$3,$53,$66,$126,$96,$108,$76,$78,$39,$40}' > /home/jupyter/Mapping/EUR_S5386_end\n",
      "Executing: paste /home/jupyter/Mapping/EUR_S5386_begin /home/jupyter/Mapping/EUR_S5386_end | sed 1d > /home/jupyter/Mapping/EUR_S5386_reduced.tsv\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing pool: /home/jupyter/Mapping/EUR_S5387\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/EUR_S5387.vcf.gz --make-just-psam --out /home/jupyter/Mapping/EUR_S5387\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\r\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\r\n",
      "Logging to /home/jupyter/Mapping/EUR_S5387.log.\r\n",
      "Options in effect:\r\n",
      "  --make-just-psam\r\n",
      "  --out /home/jupyter/Mapping/EUR_S5387\r\n",
      "  --vcf /home/jupyter/Mapping/EUR_S5387.vcf.gz\r\n",
      "\r\n",
      "Start time: Tue Nov 26 05:33:49 2024\r\n",
      "628867 MiB RAM detected, ~621434 available; reserving 314433 MiB for main\r\n",
      "workspace.\r\n",
      "Using up to 96 threads (change this with --threads).\r\n",
      "--vcf: 2490 variants scanned.\r\n",
      "--vcf: /home/jupyter/Mapping/EUR_S5387-temporary.pgen +\r\n",
      "/home/jupyter/Mapping/EUR_S5387-temporary.pvar.zst +\r\n",
      "/home/jupyter/Mapping/EUR_S5387-temporary.psam written.\r\n",
      "1 sample (0 females, 0 males, 1 ambiguous; 1 founder) loaded from\r\n",
      "/home/jupyter/Mapping/EUR_S5387-temporary.psam.\r\n",
      "Note: No phenotype data present.\r\n",
      "Writing /home/jupyter/Mapping/EUR_S5387.psam ... done.\r\n",
      "End time: Tue Nov 26 05:33:49 2024\r\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/EUR_S5387.vcf.gz                            --set-all-var-ids @:#:\\$r:\\$a                            --new-id-max-allele-len 160                            --recode A --out /home/jupyter/Mapping/EUR_S5387\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\r\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\r\n",
      "Logging to /home/jupyter/Mapping/EUR_S5387.log.\r\n",
      "Options in effect:\r\n",
      "  --export A\r\n",
      "  --new-id-max-allele-len 160\r\n",
      "  --out /home/jupyter/Mapping/EUR_S5387\r\n",
      "  --set-all-var-ids @:#:$r:$a\r\n",
      "  --vcf /home/jupyter/Mapping/EUR_S5387.vcf.gz\r\n",
      "\r\n",
      "Start time: Tue Nov 26 05:33:50 2024\r\n",
      "628867 MiB RAM detected, ~621436 available; reserving 314433 MiB for main\r\n",
      "workspace.\r\n",
      "Using up to 96 threads (change this with --threads).\r\n",
      "--vcf: 2490 variants scanned.\r\n",
      "--vcf: /home/jupyter/Mapping/EUR_S5387-temporary.pgen +\r\n",
      "/home/jupyter/Mapping/EUR_S5387-temporary.pvar.zst +\r\n",
      "/home/jupyter/Mapping/EUR_S5387-temporary.psam written.\r\n",
      "1 sample (0 females, 0 males, 1 ambiguous; 1 founder) loaded from\r\n",
      "/home/jupyter/Mapping/EUR_S5387-temporary.psam.\r\n",
      "2490 variants loaded from /home/jupyter/Mapping/EUR_S5387-temporary.pvar.zst.\r\n",
      "Note: No phenotype data present.\r\n",
      "--export A pass 1/1: loading... 0%\b\b\b\b\b\b\b\b\b\b\b\b\bwriting... 0%\b\b0%\b\bdone.\r\n",
      "--export A: /home/jupyter/Mapping/EUR_S5387.raw written.\r\n",
      "End time: Tue Nov 26 05:33:50 2024\r\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: gunzip -c /home/jupyter/Mapping/EUR_S5387.vcf.gz > /home/jupyter/Mapping/EUR_S5387.vcf\n",
      "Executing: cat /home/jupyter/Mapping/EUR_S5387.vcf | grep -v '##' | cut -f1-5 | sed 's/^chr//g' > /home/jupyter/Mapping/EUR_S5387_begin\n",
      "Executing: cat /home/jupyter/Mapping/EUR_S5387.vcf | grep -v '##' | cut -f8 | sed 's/.*CSQ=//g' | sed 's/|/\t/g' |                      awk -F '\t' 'BEGIN{OFS=FS}{print $18,$4,$2,$22,$3,$53,$66,$126,$96,$108,$76,$78,$39,$40}' > /home/jupyter/Mapping/EUR_S5387_end\n",
      "Executing: paste /home/jupyter/Mapping/EUR_S5387_begin /home/jupyter/Mapping/EUR_S5387_end | sed 1d > /home/jupyter/Mapping/EUR_S5387_reduced.tsv\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing pool: /home/jupyter/Mapping/MDE_S7\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/MDE_S7.vcf.gz --make-just-psam --out /home/jupyter/Mapping/MDE_S7\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/Mapping/MDE_S7.log.\n",
      "Options in effect:\n",
      "  --make-just-psam\n",
      "  --out /home/jupyter/Mapping/MDE_S7\n",
      "  --vcf /home/jupyter/Mapping/MDE_S7.vcf.gz\n",
      "\n",
      "Start time: Tue Nov 26 05:33:56 2024\n",
      "628867 MiB RAM detected, ~621445 available; reserving 314433 MiB for main\n",
      "workspace.\n",
      "Using up to 96 threads (change this with --threads).\n",
      "--vcf: 8263 variants scanned.\n",
      "--vcf: /home/jupyter/Mapping/MDE_S7-temporary.pgen +\n",
      "/home/jupyter/Mapping/MDE_S7-temporary.pvar.zst +\n",
      "/home/jupyter/Mapping/MDE_S7-temporary.psam written.\n",
      "5 samples (0 females, 0 males, 5 ambiguous; 5 founders) loaded from\n",
      "/home/jupyter/Mapping/MDE_S7-temporary.psam.\n",
      "Note: No phenotype data present.\n",
      "Writing /home/jupyter/Mapping/MDE_S7.psam ... done.\n",
      "End time: Tue Nov 26 05:33:56 2024\n",
      "Processing pool: /home/jupyter/Mapping/MDE_S14\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/MDE_S14.vcf.gz --make-just-psam --out /home/jupyter/Mapping/MDE_S14\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/Mapping/MDE_S14.log.\n",
      "Options in effect:\n",
      "  --make-just-psam\n",
      "  --out /home/jupyter/Mapping/MDE_S14\n",
      "  --vcf /home/jupyter/Mapping/MDE_S14.vcf.gz\n",
      "\n",
      "Start time: Tue Nov 26 05:33:57 2024\n",
      "628867 MiB RAM detected, ~621445 available; reserving 314433 MiB for main\n",
      "workspace.\n",
      "Using up to 96 threads (change this with --threads).\n",
      "--vcf: 6673 variants scanned.\n",
      "--vcf: /home/jupyter/Mapping/MDE_S14-temporary.pgen +\n",
      "/home/jupyter/Mapping/MDE_S14-temporary.pvar.zst +\n",
      "/home/jupyter/Mapping/MDE_S14-temporary.psam written.\n",
      "3 samples (0 females, 0 males, 3 ambiguous; 3 founders) loaded from\n",
      "/home/jupyter/Mapping/MDE_S14-temporary.psam.\n",
      "Note: No phenotype data present.\n",
      "Writing /home/jupyter/Mapping/MDE_S14.psam ... done.\n",
      "End time: Tue Nov 26 05:33:57 2024\n",
      "Processing pool: /home/jupyter/Mapping/MDE_S15\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/MDE_S15.vcf.gz --make-just-psam --out /home/jupyter/Mapping/MDE_S15\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/Mapping/MDE_S15.log.\n",
      "Options in effect:\n",
      "  --make-just-psam\n",
      "  --out /home/jupyter/Mapping/MDE_S15\n",
      "  --vcf /home/jupyter/Mapping/MDE_S15.vcf.gz\n",
      "\n",
      "Start time: Tue Nov 26 05:33:58 2024\n",
      "628867 MiB RAM detected, ~621448 available; reserving 314433 MiB for main\n",
      "workspace.\n",
      "Using up to 96 threads (change this with --threads).\n",
      "--vcf: 30628 variants scanned.\n",
      "--vcf: /home/jupyter/Mapping/MDE_S15-temporary.pgen +\n",
      "/home/jupyter/Mapping/MDE_S15-temporary.pvar.zst +\n",
      "/home/jupyter/Mapping/MDE_S15-temporary.psam written.\n",
      "3 samples (0 females, 0 males, 3 ambiguous; 3 founders) loaded from\n",
      "/home/jupyter/Mapping/MDE_S15-temporary.psam.\n",
      "Note: No phenotype data present.\n",
      "Writing /home/jupyter/Mapping/MDE_S15.psam ... done.\n",
      "End time: Tue Nov 26 05:33:58 2024\n",
      "Processing pool: /home/jupyter/Mapping/MDE_S16\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/MDE_S16.vcf.gz --make-just-psam --out /home/jupyter/Mapping/MDE_S16\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/Mapping/MDE_S16.log.\n",
      "Options in effect:\n",
      "  --make-just-psam\n",
      "  --out /home/jupyter/Mapping/MDE_S16\n",
      "  --vcf /home/jupyter/Mapping/MDE_S16.vcf.gz\n",
      "\n",
      "Start time: Tue Nov 26 05:33:59 2024\n",
      "628867 MiB RAM detected, ~621451 available; reserving 314433 MiB for main\n",
      "workspace.\n",
      "Using up to 96 threads (change this with --threads).\n",
      "--vcf: 23568 variants scanned.\n",
      "--vcf: /home/jupyter/Mapping/MDE_S16-temporary.pgen +\n",
      "/home/jupyter/Mapping/MDE_S16-temporary.pvar.zst +\n",
      "/home/jupyter/Mapping/MDE_S16-temporary.psam written.\n",
      "2 samples (0 females, 0 males, 2 ambiguous; 2 founders) loaded from\n",
      "/home/jupyter/Mapping/MDE_S16-temporary.psam.\n",
      "Note: No phenotype data present.\n",
      "Writing /home/jupyter/Mapping/MDE_S16.psam ... done.\n",
      "End time: Tue Nov 26 05:33:59 2024\n",
      "Processing pool: /home/jupyter/Mapping/MDE_S17\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/MDE_S17.vcf.gz --make-just-psam --out /home/jupyter/Mapping/MDE_S17\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/Mapping/MDE_S17.log.\n",
      "Options in effect:\n",
      "  --make-just-psam\n",
      "  --out /home/jupyter/Mapping/MDE_S17\n",
      "  --vcf /home/jupyter/Mapping/MDE_S17.vcf.gz\n",
      "\n",
      "Start time: Tue Nov 26 05:34:01 2024\n",
      "628867 MiB RAM detected, ~621453 available; reserving 314433 MiB for main\n",
      "workspace.\n",
      "Using up to 96 threads (change this with --threads).\n",
      "--vcf: 24540 variants scanned.\n",
      "--vcf: /home/jupyter/Mapping/MDE_S17-temporary.pgen +\n",
      "/home/jupyter/Mapping/MDE_S17-temporary.pvar.zst +\n",
      "/home/jupyter/Mapping/MDE_S17-temporary.psam written.\n",
      "2 samples (0 females, 0 males, 2 ambiguous; 2 founders) loaded from\n",
      "/home/jupyter/Mapping/MDE_S17-temporary.psam.\n",
      "Note: No phenotype data present.\n",
      "Writing /home/jupyter/Mapping/MDE_S17.psam ... done.\n",
      "End time: Tue Nov 26 05:34:01 2024\n",
      "Processing pool: /home/jupyter/Mapping/MDE_S51\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/MDE_S51.vcf.gz --make-just-psam --out /home/jupyter/Mapping/MDE_S51\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/Mapping/MDE_S51.log.\n",
      "Options in effect:\n",
      "  --make-just-psam\n",
      "  --out /home/jupyter/Mapping/MDE_S51\n",
      "  --vcf /home/jupyter/Mapping/MDE_S51.vcf.gz\n",
      "\n",
      "Start time: Tue Nov 26 05:34:02 2024\n",
      "628867 MiB RAM detected, ~621454 available; reserving 314433 MiB for main\n",
      "workspace.\n",
      "Using up to 96 threads (change this with --threads).\n",
      "--vcf: 53709 variants scanned.\n",
      "--vcf: /home/jupyter/Mapping/MDE_S51-temporary.pgen +\n",
      "/home/jupyter/Mapping/MDE_S51-temporary.pvar.zst +\n",
      "/home/jupyter/Mapping/MDE_S51-temporary.psam written.\n",
      "5 samples (0 females, 0 males, 5 ambiguous; 5 founders) loaded from\n",
      "/home/jupyter/Mapping/MDE_S51-temporary.psam.\n",
      "Note: No phenotype data present.\n",
      "Writing /home/jupyter/Mapping/MDE_S51.psam ... done.\n",
      "End time: Tue Nov 26 05:34:02 2024\n",
      "Processing pool: /home/jupyter/Mapping/MDE_S65\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/MDE_S65.vcf.gz --make-just-psam --out /home/jupyter/Mapping/MDE_S65\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/Mapping/MDE_S65.log.\n",
      "Options in effect:\n",
      "  --make-just-psam\n",
      "  --out /home/jupyter/Mapping/MDE_S65\n",
      "  --vcf /home/jupyter/Mapping/MDE_S65.vcf.gz\n",
      "\n",
      "Start time: Tue Nov 26 05:34:03 2024\n",
      "628867 MiB RAM detected, ~621454 available; reserving 314433 MiB for main\n",
      "workspace.\n",
      "Using up to 96 threads (change this with --threads).\n",
      "--vcf: 298387 variants scanned.\n",
      "--vcf: /home/jupyter/Mapping/MDE_S65-temporary.pgen +\n",
      "/home/jupyter/Mapping/MDE_S65-temporary.pvar.zst +\n",
      "/home/jupyter/Mapping/MDE_S65-temporary.psam written.\n",
      "7 samples (0 females, 0 males, 7 ambiguous; 7 founders) loaded from\n",
      "/home/jupyter/Mapping/MDE_S65-temporary.psam.\n",
      "Note: No phenotype data present.\n",
      "Writing /home/jupyter/Mapping/MDE_S65.psam ... done.\n",
      "End time: Tue Nov 26 05:34:04 2024\n",
      "Processing pool: /home/jupyter/Mapping/MDE_S83\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/MDE_S83.vcf.gz --make-just-psam --out /home/jupyter/Mapping/MDE_S83\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/Mapping/MDE_S83.log.\n",
      "Options in effect:\n",
      "  --make-just-psam\n",
      "  --out /home/jupyter/Mapping/MDE_S83\n",
      "  --vcf /home/jupyter/Mapping/MDE_S83.vcf.gz\n",
      "\n",
      "Start time: Tue Nov 26 05:34:05 2024\n",
      "628867 MiB RAM detected, ~621455 available; reserving 314433 MiB for main\n",
      "workspace.\n",
      "Using up to 96 threads (change this with --threads).\n",
      "--vcf: 237673 variants scanned.\n",
      "--vcf: /home/jupyter/Mapping/MDE_S83-temporary.pgen +\n",
      "/home/jupyter/Mapping/MDE_S83-temporary.pvar.zst +\n",
      "/home/jupyter/Mapping/MDE_S83-temporary.psam written.\n",
      "5 samples (0 females, 0 males, 5 ambiguous; 5 founders) loaded from\n",
      "/home/jupyter/Mapping/MDE_S83-temporary.psam.\n",
      "Note: No phenotype data present.\n",
      "Writing /home/jupyter/Mapping/MDE_S83.psam ... done.\n",
      "End time: Tue Nov 26 05:34:06 2024\n",
      "Processing pool: /home/jupyter/Mapping/MDE_S87\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/MDE_S87.vcf.gz --make-just-psam --out /home/jupyter/Mapping/MDE_S87\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/Mapping/MDE_S87.log.\n",
      "Options in effect:\n",
      "  --make-just-psam\n",
      "  --out /home/jupyter/Mapping/MDE_S87\n",
      "  --vcf /home/jupyter/Mapping/MDE_S87.vcf.gz\n",
      "\n",
      "Start time: Tue Nov 26 05:34:07 2024\n",
      "628867 MiB RAM detected, ~621457 available; reserving 314433 MiB for main\n",
      "workspace.\n",
      "Using up to 96 threads (change this with --threads).\n",
      "--vcf: 78597 variants scanned.\n",
      "--vcf: /home/jupyter/Mapping/MDE_S87-temporary.pgen +\n",
      "/home/jupyter/Mapping/MDE_S87-temporary.pvar.zst +\n",
      "/home/jupyter/Mapping/MDE_S87-temporary.psam written.\n",
      "4 samples (0 females, 0 males, 4 ambiguous; 4 founders) loaded from\n",
      "/home/jupyter/Mapping/MDE_S87-temporary.psam.\n",
      "Note: No phenotype data present.\n",
      "Writing /home/jupyter/Mapping/MDE_S87.psam ... done.\n",
      "End time: Tue Nov 26 05:34:08 2024\n",
      "Processing pool: /home/jupyter/Mapping/MDE_S90\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/MDE_S90.vcf.gz --make-just-psam --out /home/jupyter/Mapping/MDE_S90\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/Mapping/MDE_S90.log.\n",
      "Options in effect:\n",
      "  --make-just-psam\n",
      "  --out /home/jupyter/Mapping/MDE_S90\n",
      "  --vcf /home/jupyter/Mapping/MDE_S90.vcf.gz\n",
      "\n",
      "Start time: Tue Nov 26 05:34:09 2024\n",
      "628867 MiB RAM detected, ~621458 available; reserving 314433 MiB for main\n",
      "workspace.\n",
      "Using up to 96 threads (change this with --threads).\n",
      "--vcf: 83919 variants scanned.\n",
      "--vcf: /home/jupyter/Mapping/MDE_S90-temporary.pgen +\n",
      "/home/jupyter/Mapping/MDE_S90-temporary.pvar.zst +\n",
      "/home/jupyter/Mapping/MDE_S90-temporary.psam written.\n",
      "3 samples (0 females, 0 males, 3 ambiguous; 3 founders) loaded from\n",
      "/home/jupyter/Mapping/MDE_S90-temporary.psam.\n",
      "Note: No phenotype data present.\n",
      "Writing /home/jupyter/Mapping/MDE_S90.psam ... done.\n",
      "End time: Tue Nov 26 05:34:09 2024\n",
      "Processing pool: /home/jupyter/Mapping/MDE_S164\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/MDE_S164.vcf.gz --make-just-psam --out /home/jupyter/Mapping/MDE_S164\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/Mapping/MDE_S164.log.\n",
      "Options in effect:\n",
      "  --make-just-psam\n",
      "  --out /home/jupyter/Mapping/MDE_S164\n",
      "  --vcf /home/jupyter/Mapping/MDE_S164.vcf.gz\n",
      "\n",
      "Start time: Tue Nov 26 05:34:10 2024\n",
      "628867 MiB RAM detected, ~621459 available; reserving 314433 MiB for main\n",
      "workspace.\n",
      "Using up to 96 threads (change this with --threads).\n",
      "--vcf: 308278 variants scanned.\n",
      "--vcf: /home/jupyter/Mapping/MDE_S164-temporary.pgen +\n",
      "/home/jupyter/Mapping/MDE_S164-temporary.pvar.zst +\n",
      "/home/jupyter/Mapping/MDE_S164-temporary.psam written.\n",
      "6 samples (0 females, 0 males, 6 ambiguous; 6 founders) loaded from\n",
      "/home/jupyter/Mapping/MDE_S164-temporary.psam.\n",
      "Note: No phenotype data present.\n",
      "Writing /home/jupyter/Mapping/MDE_S164.psam ... done.\n",
      "End time: Tue Nov 26 05:34:11 2024\n",
      "Processing pool: /home/jupyter/Mapping/MDE_S186\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/MDE_S186.vcf.gz --make-just-psam --out /home/jupyter/Mapping/MDE_S186\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/Mapping/MDE_S186.log.\n",
      "Options in effect:\n",
      "  --make-just-psam\n",
      "  --out /home/jupyter/Mapping/MDE_S186\n",
      "  --vcf /home/jupyter/Mapping/MDE_S186.vcf.gz\n",
      "\n",
      "Start time: Tue Nov 26 05:34:13 2024\n",
      "628867 MiB RAM detected, ~621460 available; reserving 314433 MiB for main\n",
      "workspace.\n",
      "Using up to 96 threads (change this with --threads).\n",
      "--vcf: 332557 variants scanned.\n",
      "--vcf: /home/jupyter/Mapping/MDE_S186-temporary.pgen +\n",
      "/home/jupyter/Mapping/MDE_S186-temporary.pvar.zst +\n",
      "/home/jupyter/Mapping/MDE_S186-temporary.psam written.\n",
      "7 samples (0 females, 0 males, 7 ambiguous; 7 founders) loaded from\n",
      "/home/jupyter/Mapping/MDE_S186-temporary.psam.\n",
      "Note: No phenotype data present.\n",
      "Writing /home/jupyter/Mapping/MDE_S186.psam ... done.\n",
      "End time: Tue Nov 26 05:34:14 2024\n",
      "Processing pool: /home/jupyter/Mapping/MDE_S235\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/MDE_S235.vcf.gz --make-just-psam --out /home/jupyter/Mapping/MDE_S235\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/Mapping/MDE_S235.log.\n",
      "Options in effect:\n",
      "  --make-just-psam\n",
      "  --out /home/jupyter/Mapping/MDE_S235\n",
      "  --vcf /home/jupyter/Mapping/MDE_S235.vcf.gz\n",
      "\n",
      "Start time: Tue Nov 26 05:34:15 2024\n",
      "628867 MiB RAM detected, ~621463 available; reserving 314433 MiB for main\n",
      "workspace.\n",
      "Using up to 96 threads (change this with --threads).\n",
      "--vcf: 71738 variants scanned.\n",
      "--vcf: /home/jupyter/Mapping/MDE_S235-temporary.pgen +\n",
      "/home/jupyter/Mapping/MDE_S235-temporary.pvar.zst +\n",
      "/home/jupyter/Mapping/MDE_S235-temporary.psam written.\n",
      "3 samples (0 females, 0 males, 3 ambiguous; 3 founders) loaded from\n",
      "/home/jupyter/Mapping/MDE_S235-temporary.psam.\n",
      "Note: No phenotype data present.\n",
      "Writing /home/jupyter/Mapping/MDE_S235.psam ... done.\n",
      "End time: Tue Nov 26 05:34:15 2024\n",
      "Processing pool: /home/jupyter/Mapping/MDE_S259\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/MDE_S259.vcf.gz --make-just-psam --out /home/jupyter/Mapping/MDE_S259\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\r\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\r\n",
      "Logging to /home/jupyter/Mapping/MDE_S259.log.\r\n",
      "Options in effect:\r\n",
      "  --make-just-psam\r\n",
      "  --out /home/jupyter/Mapping/MDE_S259\r\n",
      "  --vcf /home/jupyter/Mapping/MDE_S259.vcf.gz\r\n",
      "\r\n",
      "Start time: Tue Nov 26 05:34:16 2024\r\n",
      "628867 MiB RAM detected, ~621462 available; reserving 314433 MiB for main\r\n",
      "workspace.\r\n",
      "Using up to 96 threads (change this with --threads).\r\n",
      "--vcf: 2592 variants scanned.\r\n",
      "--vcf: /home/jupyter/Mapping/MDE_S259-temporary.pgen +\r\n",
      "/home/jupyter/Mapping/MDE_S259-temporary.pvar.zst +\r\n",
      "/home/jupyter/Mapping/MDE_S259-temporary.psam written.\r\n",
      "1 sample (0 females, 0 males, 1 ambiguous; 1 founder) loaded from\r\n",
      "/home/jupyter/Mapping/MDE_S259-temporary.psam.\r\n",
      "Note: No phenotype data present.\r\n",
      "Writing /home/jupyter/Mapping/MDE_S259.psam ... done.\r\n",
      "End time: Tue Nov 26 05:34:16 2024\r\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/MDE_S259.vcf.gz                            --set-all-var-ids @:#:\\$r:\\$a                            --new-id-max-allele-len 160                            --recode A --out /home/jupyter/Mapping/MDE_S259\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\r\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\r\n",
      "Logging to /home/jupyter/Mapping/MDE_S259.log.\r\n",
      "Options in effect:\r\n",
      "  --export A\r\n",
      "  --new-id-max-allele-len 160\r\n",
      "  --out /home/jupyter/Mapping/MDE_S259\r\n",
      "  --set-all-var-ids @:#:$r:$a\r\n",
      "  --vcf /home/jupyter/Mapping/MDE_S259.vcf.gz\r\n",
      "\r\n",
      "Start time: Tue Nov 26 05:34:17 2024\r\n",
      "628867 MiB RAM detected, ~621461 available; reserving 314433 MiB for main\r\n",
      "workspace.\r\n",
      "Using up to 96 threads (change this with --threads).\r\n",
      "--vcf: 2592 variants scanned.\r\n",
      "--vcf: /home/jupyter/Mapping/MDE_S259-temporary.pgen +\r\n",
      "/home/jupyter/Mapping/MDE_S259-temporary.pvar.zst +\r\n",
      "/home/jupyter/Mapping/MDE_S259-temporary.psam written.\r\n",
      "1 sample (0 females, 0 males, 1 ambiguous; 1 founder) loaded from\r\n",
      "/home/jupyter/Mapping/MDE_S259-temporary.psam.\r\n",
      "2592 variants loaded from /home/jupyter/Mapping/MDE_S259-temporary.pvar.zst.\r\n",
      "Note: No phenotype data present.\r\n",
      "--export A pass 1/1: loading... 0%\b\b\b\b\b\b\b\b\b\b\b\b\bwriting... 0%\b\b0%\b\bdone.\r\n",
      "--export A: /home/jupyter/Mapping/MDE_S259.raw written.\r\n",
      "End time: Tue Nov 26 05:34:17 2024\r\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: gunzip -c /home/jupyter/Mapping/MDE_S259.vcf.gz > /home/jupyter/Mapping/MDE_S259.vcf\n",
      "Executing: cat /home/jupyter/Mapping/MDE_S259.vcf | grep -v '##' | cut -f1-5 | sed 's/^chr//g' > /home/jupyter/Mapping/MDE_S259_begin\n",
      "Executing: cat /home/jupyter/Mapping/MDE_S259.vcf | grep -v '##' | cut -f8 | sed 's/.*CSQ=//g' | sed 's/|/\t/g' |                      awk -F '\t' 'BEGIN{OFS=FS}{print $18,$4,$2,$22,$3,$53,$66,$126,$96,$108,$76,$78,$39,$40}' > /home/jupyter/Mapping/MDE_S259_end\n",
      "Executing: paste /home/jupyter/Mapping/MDE_S259_begin /home/jupyter/Mapping/MDE_S259_end | sed 1d > /home/jupyter/Mapping/MDE_S259_reduced.tsv\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing pool: /home/jupyter/Mapping/MDE_S288\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/MDE_S288.vcf.gz --make-just-psam --out /home/jupyter/Mapping/MDE_S288\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/Mapping/MDE_S288.log.\n",
      "Options in effect:\n",
      "  --make-just-psam\n",
      "  --out /home/jupyter/Mapping/MDE_S288\n",
      "  --vcf /home/jupyter/Mapping/MDE_S288.vcf.gz\n",
      "\n",
      "Start time: Tue Nov 26 05:34:23 2024\n",
      "628867 MiB RAM detected, ~621455 available; reserving 314433 MiB for main\n",
      "workspace.\n",
      "Using up to 96 threads (change this with --threads).\n",
      "--vcf: 38535 variants scanned.\n",
      "--vcf: /home/jupyter/Mapping/MDE_S288-temporary.pgen +\n",
      "/home/jupyter/Mapping/MDE_S288-temporary.pvar.zst +\n",
      "/home/jupyter/Mapping/MDE_S288-temporary.psam written.\n",
      "6 samples (0 females, 0 males, 6 ambiguous; 6 founders) loaded from\n",
      "/home/jupyter/Mapping/MDE_S288-temporary.psam.\n",
      "Note: No phenotype data present.\n",
      "Writing /home/jupyter/Mapping/MDE_S288.psam ... done.\n",
      "End time: Tue Nov 26 05:34:23 2024\n",
      "Processing pool: /home/jupyter/Mapping/MDE_S289\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/MDE_S289.vcf.gz --make-just-psam --out /home/jupyter/Mapping/MDE_S289\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/Mapping/MDE_S289.log.\n",
      "Options in effect:\n",
      "  --make-just-psam\n",
      "  --out /home/jupyter/Mapping/MDE_S289\n",
      "  --vcf /home/jupyter/Mapping/MDE_S289.vcf.gz\n",
      "\n",
      "Start time: Tue Nov 26 05:34:25 2024\n",
      "628867 MiB RAM detected, ~621456 available; reserving 314433 MiB for main\n",
      "workspace.\n",
      "Using up to 96 threads (change this with --threads).\n",
      "--vcf: 18969 variants scanned.\n",
      "--vcf: /home/jupyter/Mapping/MDE_S289-temporary.pgen +\n",
      "/home/jupyter/Mapping/MDE_S289-temporary.pvar.zst +\n",
      "/home/jupyter/Mapping/MDE_S289-temporary.psam written.\n",
      "6 samples (0 females, 0 males, 6 ambiguous; 6 founders) loaded from\n",
      "/home/jupyter/Mapping/MDE_S289-temporary.psam.\n",
      "Note: No phenotype data present.\n",
      "Writing /home/jupyter/Mapping/MDE_S289.psam ... done.\n",
      "End time: Tue Nov 26 05:34:25 2024\n",
      "Processing pool: /home/jupyter/Mapping/MDE_S406\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/MDE_S406.vcf.gz --make-just-psam --out /home/jupyter/Mapping/MDE_S406\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/Mapping/MDE_S406.log.\n",
      "Options in effect:\n",
      "  --make-just-psam\n",
      "  --out /home/jupyter/Mapping/MDE_S406\n",
      "  --vcf /home/jupyter/Mapping/MDE_S406.vcf.gz\n",
      "\n",
      "Start time: Tue Nov 26 05:34:26 2024\n",
      "628867 MiB RAM detected, ~621459 available; reserving 314433 MiB for main\n",
      "workspace.\n",
      "Using up to 96 threads (change this with --threads).\n",
      "--vcf: 30203 variants scanned.\n",
      "--vcf: /home/jupyter/Mapping/MDE_S406-temporary.pgen +\n",
      "/home/jupyter/Mapping/MDE_S406-temporary.pvar.zst +\n",
      "/home/jupyter/Mapping/MDE_S406-temporary.psam written.\n",
      "4 samples (0 females, 0 males, 4 ambiguous; 4 founders) loaded from\n",
      "/home/jupyter/Mapping/MDE_S406-temporary.psam.\n",
      "Note: No phenotype data present.\n",
      "Writing /home/jupyter/Mapping/MDE_S406.psam ... done.\n",
      "End time: Tue Nov 26 05:34:26 2024\n",
      "Processing pool: /home/jupyter/Mapping/MDE_S419\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/MDE_S419.vcf.gz --make-just-psam --out /home/jupyter/Mapping/MDE_S419\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/Mapping/MDE_S419.log.\n",
      "Options in effect:\n",
      "  --make-just-psam\n",
      "  --out /home/jupyter/Mapping/MDE_S419\n",
      "  --vcf /home/jupyter/Mapping/MDE_S419.vcf.gz\n",
      "\n",
      "Start time: Tue Nov 26 05:34:27 2024\n",
      "628867 MiB RAM detected, ~621457 available; reserving 314433 MiB for main\n",
      "workspace.\n",
      "Using up to 96 threads (change this with --threads).\n",
      "--vcf: 191710 variants scanned.\n",
      "--vcf: /home/jupyter/Mapping/MDE_S419-temporary.pgen +\n",
      "/home/jupyter/Mapping/MDE_S419-temporary.pvar.zst +\n",
      "/home/jupyter/Mapping/MDE_S419-temporary.psam written.\n",
      "5 samples (0 females, 0 males, 5 ambiguous; 5 founders) loaded from\n",
      "/home/jupyter/Mapping/MDE_S419-temporary.psam.\n",
      "Note: No phenotype data present.\n",
      "Writing /home/jupyter/Mapping/MDE_S419.psam ... done.\n",
      "End time: Tue Nov 26 05:34:28 2024\n",
      "Processing pool: /home/jupyter/Mapping/MDE_S606\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/MDE_S606.vcf.gz --make-just-psam --out /home/jupyter/Mapping/MDE_S606\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/Mapping/MDE_S606.log.\n",
      "Options in effect:\n",
      "  --make-just-psam\n",
      "  --out /home/jupyter/Mapping/MDE_S606\n",
      "  --vcf /home/jupyter/Mapping/MDE_S606.vcf.gz\n",
      "\n",
      "Start time: Tue Nov 26 05:34:29 2024\n",
      "628867 MiB RAM detected, ~621452 available; reserving 314433 MiB for main\n",
      "workspace.\n",
      "Using up to 96 threads (change this with --threads).\n",
      "--vcf: 23950 variants scanned.\n",
      "--vcf: /home/jupyter/Mapping/MDE_S606-temporary.pgen +\n",
      "/home/jupyter/Mapping/MDE_S606-temporary.pvar.zst +\n",
      "/home/jupyter/Mapping/MDE_S606-temporary.psam written.\n",
      "1 sample (0 females, 0 males, 1 ambiguous; 1 founder) loaded from\n",
      "/home/jupyter/Mapping/MDE_S606-temporary.psam.\n",
      "Note: No phenotype data present.\n",
      "Writing /home/jupyter/Mapping/MDE_S606.psam ... done.\n",
      "End time: Tue Nov 26 05:34:29 2024\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/MDE_S606.vcf.gz                            --set-all-var-ids @:#:\\$r:\\$a                            --new-id-max-allele-len 160                            --recode A --out /home/jupyter/Mapping/MDE_S606\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/Mapping/MDE_S606.log.\n",
      "Options in effect:\n",
      "  --export A\n",
      "  --new-id-max-allele-len 160\n",
      "  --out /home/jupyter/Mapping/MDE_S606\n",
      "  --set-all-var-ids @:#:$r:$a\n",
      "  --vcf /home/jupyter/Mapping/MDE_S606.vcf.gz\n",
      "\n",
      "Start time: Tue Nov 26 05:34:30 2024\n",
      "628867 MiB RAM detected, ~621452 available; reserving 314433 MiB for main\n",
      "workspace.\n",
      "Using up to 96 threads (change this with --threads).\n",
      "--vcf: 23950 variants scanned.\n",
      "--vcf: /home/jupyter/Mapping/MDE_S606-temporary.pgen +\n",
      "/home/jupyter/Mapping/MDE_S606-temporary.pvar.zst +\n",
      "/home/jupyter/Mapping/MDE_S606-temporary.psam written.\n",
      "1 sample (0 females, 0 males, 1 ambiguous; 1 founder) loaded from\n",
      "/home/jupyter/Mapping/MDE_S606-temporary.psam.\n",
      "23950 variants loaded from /home/jupyter/Mapping/MDE_S606-temporary.pvar.zst.\n",
      "Note: No phenotype data present.\n",
      "--export A pass 1/1: loading... 0writing... done.\n",
      "--export A: /home/jupyter/Mapping/MDE_S606.raw written.\n",
      "End time: Tue Nov 26 05:34:30 2024\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: gunzip -c /home/jupyter/Mapping/MDE_S606.vcf.gz > /home/jupyter/Mapping/MDE_S606.vcf\n",
      "Executing: cat /home/jupyter/Mapping/MDE_S606.vcf | grep -v '##' | cut -f1-5 | sed 's/^chr//g' > /home/jupyter/Mapping/MDE_S606_begin\n",
      "Executing: cat /home/jupyter/Mapping/MDE_S606.vcf | grep -v '##' | cut -f8 | sed 's/.*CSQ=//g' | sed 's/|/\t/g' |                      awk -F '\t' 'BEGIN{OFS=FS}{print $18,$4,$2,$22,$3,$53,$66,$126,$96,$108,$76,$78,$39,$40}' > /home/jupyter/Mapping/MDE_S606_end\n",
      "Executing: paste /home/jupyter/Mapping/MDE_S606_begin /home/jupyter/Mapping/MDE_S606_end | sed 1d > /home/jupyter/Mapping/MDE_S606_reduced.tsv\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing pool: /home/jupyter/Mapping/MDE_S618\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/MDE_S618.vcf.gz --make-just-psam --out /home/jupyter/Mapping/MDE_S618\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\r\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\r\n",
      "Logging to /home/jupyter/Mapping/MDE_S618.log.\r\n",
      "Options in effect:\r\n",
      "  --make-just-psam\r\n",
      "  --out /home/jupyter/Mapping/MDE_S618\r\n",
      "  --vcf /home/jupyter/Mapping/MDE_S618.vcf.gz\r\n",
      "\r\n",
      "Start time: Tue Nov 26 05:34:41 2024\r\n",
      "628867 MiB RAM detected, ~621454 available; reserving 314433 MiB for main\r\n",
      "workspace.\r\n",
      "Using up to 96 threads (change this with --threads).\r\n",
      "--vcf: 7592 variants scanned.\r\n",
      "--vcf: /home/jupyter/Mapping/MDE_S618-temporary.pgen +\r\n",
      "/home/jupyter/Mapping/MDE_S618-temporary.pvar.zst +\r\n",
      "/home/jupyter/Mapping/MDE_S618-temporary.psam written.\r\n",
      "1 sample (0 females, 0 males, 1 ambiguous; 1 founder) loaded from\r\n",
      "/home/jupyter/Mapping/MDE_S618-temporary.psam.\r\n",
      "Note: No phenotype data present.\r\n",
      "Writing /home/jupyter/Mapping/MDE_S618.psam ... done.\r\n",
      "End time: Tue Nov 26 05:34:41 2024\r\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/MDE_S618.vcf.gz                            --set-all-var-ids @:#:\\$r:\\$a                            --new-id-max-allele-len 160                            --recode A --out /home/jupyter/Mapping/MDE_S618\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/Mapping/MDE_S618.log.\n",
      "Options in effect:\n",
      "  --export A\n",
      "  --new-id-max-allele-len 160\n",
      "  --out /home/jupyter/Mapping/MDE_S618\n",
      "  --set-all-var-ids @:#:$r:$a\n",
      "  --vcf /home/jupyter/Mapping/MDE_S618.vcf.gz\n",
      "\n",
      "Start time: Tue Nov 26 05:34:42 2024\n",
      "628867 MiB RAM detected, ~621453 available; reserving 314433 MiB for main\n",
      "workspace.\n",
      "Using up to 96 threads (change this with --threads).\n",
      "--vcf: 7592 variants scanned.\n",
      "--vcf: /home/jupyter/Mapping/MDE_S618-temporary.pgen +\n",
      "/home/jupyter/Mapping/MDE_S618-temporary.pvar.zst +\n",
      "/home/jupyter/Mapping/MDE_S618-temporary.psam written.\n",
      "1 sample (0 females, 0 males, 1 ambiguous; 1 founder) loaded from\n",
      "/home/jupyter/Mapping/MDE_S618-temporary.psam.\n",
      "7592 variants loaded from /home/jupyter/Mapping/MDE_S618-temporary.pvar.zst.\n",
      "Note: No phenotype data present.\n",
      "--export A pass 1/1: loading... 0writing... done.\n",
      "--export A: /home/jupyter/Mapping/MDE_S618.raw written.\n",
      "End time: Tue Nov 26 05:34:42 2024\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: gunzip -c /home/jupyter/Mapping/MDE_S618.vcf.gz > /home/jupyter/Mapping/MDE_S618.vcf\n",
      "Executing: cat /home/jupyter/Mapping/MDE_S618.vcf | grep -v '##' | cut -f1-5 | sed 's/^chr//g' > /home/jupyter/Mapping/MDE_S618_begin\n",
      "Executing: cat /home/jupyter/Mapping/MDE_S618.vcf | grep -v '##' | cut -f8 | sed 's/.*CSQ=//g' | sed 's/|/\t/g' |                      awk -F '\t' 'BEGIN{OFS=FS}{print $18,$4,$2,$22,$3,$53,$66,$126,$96,$108,$76,$78,$39,$40}' > /home/jupyter/Mapping/MDE_S618_end\n",
      "Executing: paste /home/jupyter/Mapping/MDE_S618_begin /home/jupyter/Mapping/MDE_S618_end | sed 1d > /home/jupyter/Mapping/MDE_S618_reduced.tsv\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing pool: /home/jupyter/Mapping/SAS_S892\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/SAS_S892.vcf.gz --make-just-psam --out /home/jupyter/Mapping/SAS_S892\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/Mapping/SAS_S892.log.\n",
      "Options in effect:\n",
      "  --make-just-psam\n",
      "  --out /home/jupyter/Mapping/SAS_S892\n",
      "  --vcf /home/jupyter/Mapping/SAS_S892.vcf.gz\n",
      "\n",
      "Start time: Tue Nov 26 05:34:49 2024\n",
      "628867 MiB RAM detected, ~621452 available; reserving 314433 MiB for main\n",
      "workspace.\n",
      "Using up to 96 threads (change this with --threads).\n",
      "--vcf: 423437 variants scanned.\n",
      "--vcf: /home/jupyter/Mapping/SAS_S892-temporary.pgen +\n",
      "/home/jupyter/Mapping/SAS_S892-temporary.pvar.zst +\n",
      "/home/jupyter/Mapping/SAS_S892-temporary.psam written.\n",
      "1 sample (0 females, 0 males, 1 ambiguous; 1 founder) loaded from\n",
      "/home/jupyter/Mapping/SAS_S892-temporary.psam.\n",
      "Note: No phenotype data present.\n",
      "Writing /home/jupyter/Mapping/SAS_S892.psam ... done.\n",
      "End time: Tue Nov 26 05:34:50 2024\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: /home/jupyter/tools/plink2 --vcf /home/jupyter/Mapping/SAS_S892.vcf.gz                            --set-all-var-ids @:#:\\$r:\\$a                            --new-id-max-allele-len 160                            --recode A --out /home/jupyter/Mapping/SAS_S892\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 Intel (6 Aug 2024)        www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/Mapping/SAS_S892.log.\n",
      "Options in effect:\n",
      "  --export A\n",
      "  --new-id-max-allele-len 160\n",
      "  --out /home/jupyter/Mapping/SAS_S892\n",
      "  --set-all-var-ids @:#:$r:$a\n",
      "  --vcf /home/jupyter/Mapping/SAS_S892.vcf.gz\n",
      "\n",
      "Start time: Tue Nov 26 05:34:51 2024\n",
      "628867 MiB RAM detected, ~621459 available; reserving 314433 MiB for main\n",
      "workspace.\n",
      "Using up to 96 threads (change this with --threads).\n",
      "--vcf: 423437 variants scanned.\n",
      "--vcf: /home/jupyter/Mapping/SAS_S892-temporary.pgen +\n",
      "/home/jupyter/Mapping/SAS_S892-temporary.pvar.zst +\n",
      "/home/jupyter/Mapping/SAS_S892-temporary.psam written.\n",
      "1 sample (0 females, 0 males, 1 ambiguous; 1 founder) loaded from\n",
      "/home/jupyter/Mapping/SAS_S892-temporary.psam.\n",
      "423437 variants loaded from /home/jupyter/Mapping/SAS_S892-temporary.pvar.zst.\n",
      "Note: No phenotype data present.\n",
      "--export A pass 1/1: loading... 15304661779writing... done.\n",
      "--export A: /home/jupyter/Mapping/SAS_S892.raw written.\n",
      "End time: Tue Nov 26 05:34:53 2024\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: gunzip -c /home/jupyter/Mapping/SAS_S892.vcf.gz > /home/jupyter/Mapping/SAS_S892.vcf\n",
      "Executing: cat /home/jupyter/Mapping/SAS_S892.vcf | grep -v '##' | cut -f1-5 | sed 's/^chr//g' > /home/jupyter/Mapping/SAS_S892_begin\n",
      "Executing: cat /home/jupyter/Mapping/SAS_S892.vcf | grep -v '##' | cut -f8 | sed 's/.*CSQ=//g' | sed 's/|/\t/g' |                      awk -F '\t' 'BEGIN{OFS=FS}{print $18,$4,$2,$22,$3,$53,$66,$126,$96,$108,$76,$78,$39,$40}' > /home/jupyter/Mapping/SAS_S892_end\n",
      "Executing: paste /home/jupyter/Mapping/SAS_S892_begin /home/jupyter/Mapping/SAS_S892_end | sed 1d > /home/jupyter/Mapping/SAS_S892_reduced.tsv\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Ancestry</th>\n",
       "      <th>Pool</th>\n",
       "      <th>chr</th>\n",
       "      <th>bp</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>rsid</th>\n",
       "      <th>gene</th>\n",
       "      <th>variant_type</th>\n",
       "      <th>Variant_class</th>\n",
       "      <th>...</th>\n",
       "      <th>GnomADg_AF</th>\n",
       "      <th>clinical_signif</th>\n",
       "      <th>clinvar_signif</th>\n",
       "      <th>cadd_phred</th>\n",
       "      <th>REVEL</th>\n",
       "      <th>eve_class</th>\n",
       "      <th>am_class</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>Polyphen</th>\n",
       "      <th>colname</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>157756</th>\n",
       "      <td>SAS</td>\n",
       "      <td>S892</td>\n",
       "      <td>8</td>\n",
       "      <td>124557139</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>rs16899688</td>\n",
       "      <td>MTSS1</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>0.3105</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.250</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8:124557139:C:A_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>157757</th>\n",
       "      <td>SAS</td>\n",
       "      <td>S892</td>\n",
       "      <td>8</td>\n",
       "      <td>124557442</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>rs7841224</td>\n",
       "      <td>MTSS1</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>0.3602</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.125</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8:124557442:A:G_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>157758</th>\n",
       "      <td>SAS</td>\n",
       "      <td>S892</td>\n",
       "      <td>8</td>\n",
       "      <td>124557903</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>rs4870903&amp;COSV61499448</td>\n",
       "      <td>MTSS1</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>0.7082</td>\n",
       "      <td>benign</td>\n",
       "      <td>Benign</td>\n",
       "      <td>6.763</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8:124557903:G:A_G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>157759</th>\n",
       "      <td>SAS</td>\n",
       "      <td>S892</td>\n",
       "      <td>8</td>\n",
       "      <td>124558287</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>rs3750232</td>\n",
       "      <td>MTSS1</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>0.4729</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7.564</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8:124558287:A:G_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>157760</th>\n",
       "      <td>SAS</td>\n",
       "      <td>S892</td>\n",
       "      <td>8</td>\n",
       "      <td>124558559</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>rs7846270</td>\n",
       "      <td>MTSS1</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>0.4260</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>16.060</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8:124558559:C:T_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4505</th>\n",
       "      <td>AAC</td>\n",
       "      <td>S396</td>\n",
       "      <td>6</td>\n",
       "      <td>42299036</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>rs4711713</td>\n",
       "      <td>TRERF1</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>0.1336</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.253</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6:42299036:T:A_T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4506</th>\n",
       "      <td>AAC</td>\n",
       "      <td>S396</td>\n",
       "      <td>6</td>\n",
       "      <td>42299293</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>rs4711714</td>\n",
       "      <td>TRERF1</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>0.1447</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.110</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6:42299293:A:G_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4507</th>\n",
       "      <td>AAC</td>\n",
       "      <td>S396</td>\n",
       "      <td>6</td>\n",
       "      <td>42299322</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>rs4711715</td>\n",
       "      <td>TRERF1</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>0.1454</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.098</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6:42299322:T:C_T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4508</th>\n",
       "      <td>AAC</td>\n",
       "      <td>S396</td>\n",
       "      <td>6</td>\n",
       "      <td>42299485</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>rs4711716</td>\n",
       "      <td>TRERF1</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>0.1352</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.567</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6:42299485:A:G_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4509</th>\n",
       "      <td>AAC</td>\n",
       "      <td>S396</td>\n",
       "      <td>6</td>\n",
       "      <td>42299583</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>rs9462797</td>\n",
       "      <td>TRERF1</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>0.1353</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.965</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6:42299583:A:G_A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>162437 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Ancestry  Pool  chr         bp REF ALT                    rsid    gene  \\\n",
       "157756      SAS  S892    8  124557139   C   A              rs16899688   MTSS1   \n",
       "157757      SAS  S892    8  124557442   A   G               rs7841224   MTSS1   \n",
       "157758      SAS  S892    8  124557903   G   A  rs4870903&COSV61499448   MTSS1   \n",
       "157759      SAS  S892    8  124558287   A   G               rs3750232   MTSS1   \n",
       "157760      SAS  S892    8  124558559   C   T               rs7846270   MTSS1   \n",
       "...         ...   ...  ...        ...  ..  ..                     ...     ...   \n",
       "4505        AAC  S396    6   42299036   T   A               rs4711713  TRERF1   \n",
       "4506        AAC  S396    6   42299293   A   G               rs4711714  TRERF1   \n",
       "4507        AAC  S396    6   42299322   T   C               rs4711715  TRERF1   \n",
       "4508        AAC  S396    6   42299485   A   G               rs4711716  TRERF1   \n",
       "4509        AAC  S396    6   42299583   A   G               rs9462797  TRERF1   \n",
       "\n",
       "          variant_type Variant_class  ... GnomADg_AF  clinical_signif  \\\n",
       "157756  intron_variant           SNV  ...     0.3105              NaN   \n",
       "157757  intron_variant           SNV  ...     0.3602              NaN   \n",
       "157758  intron_variant           SNV  ...     0.7082           benign   \n",
       "157759  intron_variant           SNV  ...     0.4729              NaN   \n",
       "157760  intron_variant           SNV  ...     0.4260              NaN   \n",
       "...                ...           ...  ...        ...              ...   \n",
       "4505    intron_variant           SNV  ...     0.1336              NaN   \n",
       "4506    intron_variant           SNV  ...     0.1447              NaN   \n",
       "4507    intron_variant           SNV  ...     0.1454              NaN   \n",
       "4508    intron_variant           SNV  ...     0.1352              NaN   \n",
       "4509    intron_variant           SNV  ...     0.1353              NaN   \n",
       "\n",
       "       clinvar_signif cadd_phred  REVEL  eve_class am_class SIFT Polyphen  \\\n",
       "157756            NaN      3.250    NaN        NaN      NaN  NaN      NaN   \n",
       "157757            NaN      0.125    NaN        NaN      NaN  NaN      NaN   \n",
       "157758         Benign      6.763    NaN        NaN      NaN  NaN      NaN   \n",
       "157759            NaN      7.564    NaN        NaN      NaN  NaN      NaN   \n",
       "157760            NaN     16.060    NaN        NaN      NaN  NaN      NaN   \n",
       "...               ...        ...    ...        ...      ...  ...      ...   \n",
       "4505              NaN      1.253    NaN        NaN      NaN  NaN      NaN   \n",
       "4506              NaN      0.110    NaN        NaN      NaN  NaN      NaN   \n",
       "4507              NaN      0.098    NaN        NaN      NaN  NaN      NaN   \n",
       "4508              NaN      2.567    NaN        NaN      NaN  NaN      NaN   \n",
       "4509              NaN      4.965    NaN        NaN      NaN  NaN      NaN   \n",
       "\n",
       "                  colname  \n",
       "157756  8:124557139:C:A_C  \n",
       "157757  8:124557442:A:G_A  \n",
       "157758  8:124557903:G:A_G  \n",
       "157759  8:124558287:A:G_A  \n",
       "157760  8:124558559:C:T_C  \n",
       "...                   ...  \n",
       "4505     6:42299036:T:A_T  \n",
       "4506     6:42299293:A:G_A  \n",
       "4507     6:42299322:T:C_T  \n",
       "4508     6:42299485:A:G_A  \n",
       "4509     6:42299583:A:G_A  \n",
       "\n",
       "[162437 rows x 21 columns]"
      ]
     },
     "execution_count": 94,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Initialize an empty DataFrame to store unique samples across pools\n",
    "uniq_samples = pd.DataFrame()\n",
    "\n",
    "# Open and read the file containing the list of renamed pools\n",
    "with open(f'{WORK_DIR}/list_renamed_pools.txt', 'r') as file:\n",
    "    # Iterate through each pool listed in the file\n",
    "    for line in file:\n",
    "        # Remove trailing newline characters from the line\n",
    "        pool = line.strip()\n",
    "        print(f\"Processing pool: {pool}\")\n",
    "\n",
    "        # Extract the pool name (e.g., ancestry-specific identifier) from the file path\n",
    "        pool_name = pool.split('/')[4]\n",
    "\n",
    "        # Check if the pool contains a unique sample in WGS data\n",
    "        shell_do(f'/home/jupyter/tools/plink2 --vcf {pool}.vcf.gz --make-just-psam --out {pool}')\n",
    "        IIDs_WGS = pd.read_csv(f\"{pool}.psam\", sep='\\t')\n",
    "\n",
    "        # Proceed only if the pool contains a single unique sample\n",
    "        if IIDs_WGS['#IID'].nunique() == 1:\n",
    "            # Generate a .raw file using PLINK2 with proper variant IDs\n",
    "            shell_do(f'/home/jupyter/tools/plink2 --vcf {pool}.vcf.gz \\\n",
    "                           --set-all-var-ids @:#:\\$r:\\$a \\\n",
    "                           --new-id-max-allele-len 160 \\\n",
    "                           --recode A --out {pool}')\n",
    "            \n",
    "            # Read the generated .raw file into a DataFrame\n",
    "            raw = pd.read_csv(f'{pool}.raw', sep='\\t')\n",
    "\n",
    "            # Identify columns in the .raw file containing only zeros (homozygous variants)\n",
    "            columns_with_only_0 = raw.columns[(raw.eq(0).all())]\n",
    "\n",
    "            # Decompress the VCF file for annotation purposes\n",
    "            shell_do(f'gunzip -c {pool}.vcf.gz > {pool}.vcf')\n",
    "\n",
    "            # Extract chromosome, position, reference allele, and alternate allele from the VCF file (first part)\n",
    "            shell_do(f\"cat {pool}.vcf | grep -v '##' | cut -f1-5 | sed 's/^chr//g' > {pool}_begin\")\n",
    "            \n",
    "            # Extract annotation fields (e.g., functional effects, allele frequencies) from the VCF file (second part)\n",
    "            shell_do(f\"cat {pool}.vcf | grep -v '##' | cut -f8 | sed 's/.*CSQ=//g' | sed 's/|/\\t/g' | \\\n",
    "                     awk -F '\\t' 'BEGIN{{OFS=FS}}{{print $18,$4,$2,$22,$3,$53,$66,$126,$96,$108,$76,$78,$39,$40}}' > {pool}_end\")\n",
    "            \n",
    "            # Combine the extracted fields into a reduced annotation file\n",
    "            shell_do(f'paste {pool}_begin {pool}_end | sed 1d > {pool}_reduced.tsv')\n",
    "\n",
    "            # Define column names for the reduced annotation file\n",
    "            column_names = [\n",
    "                \"chr\", \"bp\", \"ID\", \"REF\", \"ALT\", \"rsid\", \"gene\", \"variant_type\", \"Variant_class\", \n",
    "                \"variant_impact\", \"GnomADg_AF\", \"clinical_signif\", \"clinvar_signif\", \"cadd_phred\", \n",
    "                \"REVEL\", \"eve_class\", \"am_class\", \"SIFT\", \"Polyphen\"\n",
    "            ]\n",
    "\n",
    "            # Read the reduced annotation file into a DataFrame\n",
    "            vars_anno_reduced = pd.read_csv(\n",
    "                f'{pool}_reduced.tsv', sep='\\t', low_memory=False, header=None, names=column_names\n",
    "            )\n",
    "\n",
    "            # Create a unique ID for each variant by combining chromosome, position, REF, and ALT\n",
    "            vars_anno_reduced['ID'] = vars_anno_reduced['chr'].astype(str) + ':' + vars_anno_reduced['bp'].astype(str) + ':' + vars_anno_reduced['REF'] + ':' + vars_anno_reduced['ALT']\n",
    "            \n",
    "            # Generate a new column to match the column format in the .raw file\n",
    "            vars_anno_reduced['colname'] = vars_anno_reduced['ID'] + '_' + vars_anno_reduced['REF']\n",
    "            \n",
    "            # Drop the temporary ID column as it is no longer needed\n",
    "            vars_anno_reduced = vars_anno_reduced.drop('ID', axis=1)\n",
    "\n",
    "            # Filter the annotation DataFrame based on columns that contain only zeros in the .raw file\n",
    "            results = vars_anno_reduced.loc[vars_anno_reduced['colname'].isin(columns_with_only_0)]\n",
    "\n",
    "            # Extract ancestry and pool ID from the pool name\n",
    "            ancestry, pool_id = pool_name.split('_')\n",
    "\n",
    "            # Add the ancestry information as the first column\n",
    "            results.insert(0, 'Ancestry', ancestry)\n",
    "\n",
    "            # Add the pool ID as the second column\n",
    "            results.insert(1, 'Pool', pool_id)\n",
    "\n",
    "            # Append the filtered results to the main DataFrame\n",
    "            uniq_samples = pd.concat([uniq_samples, results], ignore_index=True)\n",
    "\n",
    "# Sort the concatenated results by 'Ancestry' and 'Pool' in descending order\n",
    "uniq_samples = uniq_samples.sort_values(by=['Ancestry', 'Pool'], ascending=[False, False])\n",
    "\n",
    "# Display the final sorted DataFrame\n",
    "uniq_samples"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Prioritization criteria\n",
    "\n",
    "Apply criteria to prioritize variants for further analysis."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Ancestry</th>\n",
       "      <th>Pool</th>\n",
       "      <th>chr</th>\n",
       "      <th>bp</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>rsid</th>\n",
       "      <th>gene</th>\n",
       "      <th>variant_type</th>\n",
       "      <th>Variant_class</th>\n",
       "      <th>...</th>\n",
       "      <th>GnomADg_AF</th>\n",
       "      <th>clinical_signif</th>\n",
       "      <th>clinvar_signif</th>\n",
       "      <th>cadd_phred</th>\n",
       "      <th>REVEL</th>\n",
       "      <th>eve_class</th>\n",
       "      <th>am_class</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>Polyphen</th>\n",
       "      <th>colname</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>161981</th>\n",
       "      <td>SAS</td>\n",
       "      <td>S892</td>\n",
       "      <td>8</td>\n",
       "      <td>128916918</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>rs369143754</td>\n",
       "      <td>CCDC26</td>\n",
       "      <td>splice_acceptor_variant&amp;non_coding_transcript_...</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000624</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5.673</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8:128916918:T:A_T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>143437</th>\n",
       "      <td>MDE</td>\n",
       "      <td>S606</td>\n",
       "      <td>15</td>\n",
       "      <td>78766976</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>rs142824118&amp;COSV66308197</td>\n",
       "      <td>ADAMTS7</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>0.004863</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>15.070</td>\n",
       "      <td>0.034523</td>\n",
       "      <td>NaN</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>tolerated(0.05)</td>\n",
       "      <td>benign(0.079)</td>\n",
       "      <td>15:78766976:C:T_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>107935</th>\n",
       "      <td>AMR</td>\n",
       "      <td>S82</td>\n",
       "      <td>9</td>\n",
       "      <td>84723600</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>rs139913267&amp;COSV52853311&amp;COSV52860967</td>\n",
       "      <td>NTRK2</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>0.001032</td>\n",
       "      <td>benign/likely_benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>16.340</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>tolerated_low_confidence(0.52)</td>\n",
       "      <td>benign(0.007)</td>\n",
       "      <td>9:84723600:C:A_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>92767</th>\n",
       "      <td>AJ</td>\n",
       "      <td>S1665</td>\n",
       "      <td>13</td>\n",
       "      <td>23340893</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>rs142967124</td>\n",
       "      <td>SACS</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>0.001722</td>\n",
       "      <td>benign&amp;benign/likely_benign&amp;likely_benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9.213</td>\n",
       "      <td>0.121609</td>\n",
       "      <td>Benign</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>deleterious(0.01)</td>\n",
       "      <td>benign(0.003)</td>\n",
       "      <td>13:23340893:C:A_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>81060</th>\n",
       "      <td>AJ</td>\n",
       "      <td>S1651</td>\n",
       "      <td>11</td>\n",
       "      <td>8985795</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>rs61756060&amp;COSV58469013</td>\n",
       "      <td>NRIP3</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>0.002735</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>25.100</td>\n",
       "      <td>0.218651</td>\n",
       "      <td>NaN</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>deleterious(0.04)</td>\n",
       "      <td>possibly_damaging(0.825)</td>\n",
       "      <td>11:8985795:G:A_G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62263</th>\n",
       "      <td>AJ</td>\n",
       "      <td>S1650</td>\n",
       "      <td>11</td>\n",
       "      <td>551741</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>rs139348192</td>\n",
       "      <td>LRRC56</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>0.003883</td>\n",
       "      <td>benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>22.600</td>\n",
       "      <td>0.101013</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>deleterious(0)</td>\n",
       "      <td>possibly_damaging(0.815)</td>\n",
       "      <td>11:551741:C:T_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51970</th>\n",
       "      <td>AJ</td>\n",
       "      <td>S1623</td>\n",
       "      <td>9</td>\n",
       "      <td>2823786</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>rs775393848</td>\n",
       "      <td>PUM3</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000125</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>22.600</td>\n",
       "      <td>0.516913</td>\n",
       "      <td>NaN</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>deleterious(0.01)</td>\n",
       "      <td>benign(0.184)</td>\n",
       "      <td>9:2823786:C:T_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33224</th>\n",
       "      <td>AJ</td>\n",
       "      <td>S1443</td>\n",
       "      <td>8</td>\n",
       "      <td>120255342</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>rs141272095</td>\n",
       "      <td>COL14A1</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000565</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>22.700</td>\n",
       "      <td>0.245616</td>\n",
       "      <td>NaN</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>deleterious_low_confidence(0)</td>\n",
       "      <td>benign(0.092)</td>\n",
       "      <td>8:120255342:T:C_T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7321</th>\n",
       "      <td>AAC</td>\n",
       "      <td>S397</td>\n",
       "      <td>6</td>\n",
       "      <td>42188686</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>rs137853903&amp;CM045052&amp;COSV57835221</td>\n",
       "      <td>GUCA1B</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>0.004838</td>\n",
       "      <td>benign&amp;likely_benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>22.700</td>\n",
       "      <td>0.239087</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>deleterious(0.01)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6:42188686:C:T_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7867</th>\n",
       "      <td>AAC</td>\n",
       "      <td>S397</td>\n",
       "      <td>6</td>\n",
       "      <td>42745604</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>rs7742995</td>\n",
       "      <td>TBCC</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>0.009927</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.136</td>\n",
       "      <td>0.041897</td>\n",
       "      <td>NaN</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>tolerated(0.38)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6:42745604:C:T_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8191</th>\n",
       "      <td>AAC</td>\n",
       "      <td>S397</td>\n",
       "      <td>6</td>\n",
       "      <td>42978919</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>rs1420377361</td>\n",
       "      <td>PEX6</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000013</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>27.700</td>\n",
       "      <td>0.387918</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>deleterious(0)</td>\n",
       "      <td>possibly_damaging(0.874)</td>\n",
       "      <td>6:42978919:G:A_G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4385</th>\n",
       "      <td>AAC</td>\n",
       "      <td>S396</td>\n",
       "      <td>6</td>\n",
       "      <td>42188686</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>rs137853903&amp;CM045052&amp;COSV57835221</td>\n",
       "      <td>GUCA1B</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>0.004838</td>\n",
       "      <td>benign&amp;likely_benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>22.700</td>\n",
       "      <td>0.239087</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>deleterious(0.01)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6:42188686:C:T_C</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>12 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Ancestry   Pool  chr         bp REF ALT  \\\n",
       "161981      SAS   S892    8  128916918   T   A   \n",
       "143437      MDE   S606   15   78766976   C   T   \n",
       "107935      AMR    S82    9   84723600   C   A   \n",
       "92767        AJ  S1665   13   23340893   C   A   \n",
       "81060        AJ  S1651   11    8985795   G   A   \n",
       "62263        AJ  S1650   11     551741   C   T   \n",
       "51970        AJ  S1623    9    2823786   C   T   \n",
       "33224        AJ  S1443    8  120255342   T   C   \n",
       "7321        AAC   S397    6   42188686   C   T   \n",
       "7867        AAC   S397    6   42745604   C   T   \n",
       "8191        AAC   S397    6   42978919   G   A   \n",
       "4385        AAC   S396    6   42188686   C   T   \n",
       "\n",
       "                                         rsid     gene  \\\n",
       "161981                            rs369143754   CCDC26   \n",
       "143437               rs142824118&COSV66308197  ADAMTS7   \n",
       "107935  rs139913267&COSV52853311&COSV52860967    NTRK2   \n",
       "92767                             rs142967124     SACS   \n",
       "81060                 rs61756060&COSV58469013    NRIP3   \n",
       "62263                             rs139348192   LRRC56   \n",
       "51970                             rs775393848     PUM3   \n",
       "33224                             rs141272095  COL14A1   \n",
       "7321        rs137853903&CM045052&COSV57835221   GUCA1B   \n",
       "7867                                rs7742995     TBCC   \n",
       "8191                             rs1420377361     PEX6   \n",
       "4385        rs137853903&CM045052&COSV57835221   GUCA1B   \n",
       "\n",
       "                                             variant_type Variant_class  ...  \\\n",
       "161981  splice_acceptor_variant&non_coding_transcript_...           SNV  ...   \n",
       "143437                                   missense_variant           SNV  ...   \n",
       "107935                                   missense_variant           SNV  ...   \n",
       "92767                                    missense_variant           SNV  ...   \n",
       "81060                                    missense_variant           SNV  ...   \n",
       "62263                                    missense_variant           SNV  ...   \n",
       "51970                                    missense_variant           SNV  ...   \n",
       "33224                                    missense_variant           SNV  ...   \n",
       "7321                                     missense_variant           SNV  ...   \n",
       "7867                                     missense_variant           SNV  ...   \n",
       "8191                                     missense_variant           SNV  ...   \n",
       "4385                                     missense_variant           SNV  ...   \n",
       "\n",
       "       GnomADg_AF                            clinical_signif clinvar_signif  \\\n",
       "161981   0.000624                                        NaN            NaN   \n",
       "143437   0.004863                                        NaN            NaN   \n",
       "107935   0.001032                       benign/likely_benign            NaN   \n",
       "92767    0.001722  benign&benign/likely_benign&likely_benign            NaN   \n",
       "81060    0.002735                                        NaN            NaN   \n",
       "62263    0.003883                                     benign            NaN   \n",
       "51970    0.000125                                        NaN            NaN   \n",
       "33224    0.000565                                        NaN            NaN   \n",
       "7321     0.004838                       benign&likely_benign            NaN   \n",
       "7867     0.009927                                        NaN            NaN   \n",
       "8191     0.000013                                        NaN            NaN   \n",
       "4385     0.004838                       benign&likely_benign            NaN   \n",
       "\n",
       "       cadd_phred     REVEL   eve_class       am_class  \\\n",
       "161981      5.673       NaN         NaN            NaN   \n",
       "143437     15.070  0.034523         NaN  likely_benign   \n",
       "107935     16.340       NaN         NaN            NaN   \n",
       "92767       9.213  0.121609      Benign  likely_benign   \n",
       "81060      25.100  0.218651         NaN  likely_benign   \n",
       "62263      22.600  0.101013  Pathogenic  likely_benign   \n",
       "51970      22.600  0.516913         NaN  likely_benign   \n",
       "33224      22.700  0.245616         NaN  likely_benign   \n",
       "7321       22.700  0.239087         NaN      ambiguous   \n",
       "7867        3.136  0.041897         NaN  likely_benign   \n",
       "8191       27.700  0.387918         NaN            NaN   \n",
       "4385       22.700  0.239087         NaN      ambiguous   \n",
       "\n",
       "                                  SIFT                  Polyphen  \\\n",
       "161981                             NaN                       NaN   \n",
       "143437                 tolerated(0.05)             benign(0.079)   \n",
       "107935  tolerated_low_confidence(0.52)             benign(0.007)   \n",
       "92767                deleterious(0.01)             benign(0.003)   \n",
       "81060                deleterious(0.04)  possibly_damaging(0.825)   \n",
       "62263                   deleterious(0)  possibly_damaging(0.815)   \n",
       "51970                deleterious(0.01)             benign(0.184)   \n",
       "33224    deleterious_low_confidence(0)             benign(0.092)   \n",
       "7321                 deleterious(0.01)                       NaN   \n",
       "7867                   tolerated(0.38)                       NaN   \n",
       "8191                    deleterious(0)  possibly_damaging(0.874)   \n",
       "4385                 deleterious(0.01)                       NaN   \n",
       "\n",
       "                  colname  \n",
       "161981  8:128916918:T:A_T  \n",
       "143437  15:78766976:C:T_C  \n",
       "107935   9:84723600:C:A_C  \n",
       "92767   13:23340893:C:A_C  \n",
       "81060    11:8985795:G:A_G  \n",
       "62263     11:551741:C:T_C  \n",
       "51970     9:2823786:C:T_C  \n",
       "33224   8:120255342:T:C_T  \n",
       "7321     6:42188686:C:T_C  \n",
       "7867     6:42745604:C:T_C  \n",
       "8191     6:42978919:G:A_G  \n",
       "4385     6:42188686:C:T_C  \n",
       "\n",
       "[12 rows x 21 columns]"
      ]
     },
     "execution_count": 95,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Filter unique sample variants with HIGH or MODERATE impact and GnomADg_AF <= 0.01 or null\n",
    "candidates_uniq_samples = uniq_samples[\n",
    "    (uniq_samples['variant_impact'].isin(['HIGH', 'MODERATE']))  # HIGH or MODERATE impact\n",
    "    & ((uniq_samples['GnomADg_AF'].isnull()) | (uniq_samples['GnomADg_AF'] <= 0.01))  # GnomADg_AF is null or <= 0.01\n",
    "]\n",
    "\n",
    "# Display the filtered candidate variants\n",
    "candidates_uniq_samples"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.12"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {
    "height": "calc(100% - 180px)",
    "left": "10px",
    "top": "150px",
    "width": "598.203px"
   },
   "toc_section_display": true,
   "toc_window_display": true
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
